{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import re\n",
    "import pandas as pd\n",
    "import collections\n",
    "from collections import Counter\n",
    "import compress_pickle\n",
    "import copy\n",
    "import fnmatch\n",
    "import itertools\n",
    "import libsbml\n",
    "import numpy as np\n",
    "import time\n",
    "import matplotlib.pyplot as plt\n",
    "import string\n",
    "\n",
    "# Import my local AMAS package\n",
    "sys.path.insert(0, '/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2')\n",
    "from AMAS import annotation_maker as am\n",
    "from AMAS import constants as cn\n",
    "from AMAS import iterator as it\n",
    "from AMAS import tools\n",
    "from AMAS import species_annotation as sa\n",
    "from AMAS import reaction_annotation as ra \n",
    "from AMAS import gene_annotation as ga\n",
    "from AMAS import recommender\n",
    "\n",
    "# Directory\n",
    "biomd_model_dir = '/Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/'\n",
    "bigg_model_dir = '/Users/luna/Desktop/CRBM/AMAS_proj/Models/BiggModels/'\n",
    "biodivine_model_dir = '/Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine/'\n",
    "output_dir = '/Users/luna/Desktop/CRBM/AMAS_proj/Results/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "ecoli_mg1655_models = ['e_coli_core.xml', 'iML1515.xml', 'iAF1260.xml','iJR904.xml','iAF1260b.xml','iJO1366.xml']\n",
    "human_models = ['Recon3D.xml', 'iAB_RBC_283.xml','iAT_PLT_636.xml','RECON1.xml']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "137"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "one_fpath = \"/Users/luna/Desktop/CRBM/AMAS_proj/Models/BiggModels/e_coli_core.xml\"\n",
    "reader = libsbml.SBMLReader()\n",
    "document = reader.readSBML(one_fpath)\n",
    "model = document.getModel().getPlugin(\"fbc\")\n",
    "model.getNumGeneProducts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_product = fbc_plugin.getGeneProduct(3)\n",
    "print(gene_product)\n",
    "print(gene_product.getIdAttribute())\n",
    "print(gene_product.getLabel())\n",
    "print(gene_product.getName())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>predictions</th>\n",
       "      <th>predictions_labels</th>\n",
       "      <th>num_candidates</th>\n",
       "      <th>match_score</th>\n",
       "      <th>recall</th>\n",
       "      <th>precision</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>G_PF3D7_0507200</td>\n",
       "      <td>PF3D7_0507200</td>\n",
       "      <td>[(812667, 1.0), (812844, 1.0), (812882, 1.0), ...</td>\n",
       "      <td>[PF3D7_0207500, PF3D7_0502700, PF3D7_0507200, ...</td>\n",
       "      <td>9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>G_PF3D7_0932300</td>\n",
       "      <td>PF3D7_0932300</td>\n",
       "      <td>[(813509, 1.0), (813594, 1.0), (813602, 1.0), ...</td>\n",
       "      <td>[PF3D7_0923300, PF3D7_0932300, PF3D7_0933200, ...</td>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>G_PF3D7_1360800</td>\n",
       "      <td>PF3D7_1360800</td>\n",
       "      <td>[(810517, 1.0), (810533, 1.0), (813907, 1.0), ...</td>\n",
       "      <td>[PF3D7_1036800, PF3D7_1038600, PF3D7_1306800, ...</td>\n",
       "      <td>15</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.067</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G_PF3D7_0409500</td>\n",
       "      <td>PF3D7_0409500</td>\n",
       "      <td>[(812389, 1.0), (812414, 1.0), (812845, 1.0), ...</td>\n",
       "      <td>[PF3D7_0409500, PF3D7_0405900, PF3D7_0509400, ...</td>\n",
       "      <td>6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>G_PF3D7_1446200</td>\n",
       "      <td>PF3D7_1446200</td>\n",
       "      <td>[(811480, 1.0), (811497, 1.0), (811986, 1.0), ...</td>\n",
       "      <td>[PF3D7_1244600, PF3D7_1246400, PF3D7_1442600, ...</td>\n",
       "      <td>8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>455</th>\n",
       "      <td>G_PF3D7_1115300</td>\n",
       "      <td>PF3D7_1115300</td>\n",
       "      <td>[(810690, 1.0), (810708, 1.0), (810872, 1.0), ...</td>\n",
       "      <td>[PF3D7_1113500, PF3D7_1115300, PF3D7_1131500, ...</td>\n",
       "      <td>7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>G_PF3D7_1115700</td>\n",
       "      <td>PF3D7_1115700</td>\n",
       "      <td>[(810712, 1.0), (810728, 1.0)]</td>\n",
       "      <td>[PF3D7_1115700, PF3D7_1117500]</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>G_PF3D7_1407800</td>\n",
       "      <td>PF3D7_1407800</td>\n",
       "      <td>[(811657, 1.0), (811665, 1.0), (812257, 1.0), ...</td>\n",
       "      <td>[PF3D7_1407800, PF3D7_1408700, PF3D7_1470800, ...</td>\n",
       "      <td>10</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458</th>\n",
       "      <td>G_PF3D7_1311800</td>\n",
       "      <td>PF3D7_1311800</td>\n",
       "      <td>[(810693, 1.0), (810736, 1.0), (810874, 1.0), ...</td>\n",
       "      <td>[PF3D7_1113800, PF3D7_1118300, PF3D7_1131800, ...</td>\n",
       "      <td>6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>459</th>\n",
       "      <td>G_PF3D7_1136900</td>\n",
       "      <td>PF3D7_1136900</td>\n",
       "      <td>[(810927, 1.0), (810952, 1.0), (813979, 1.0), ...</td>\n",
       "      <td>[PF3D7_1136900, PF3D7_1139600, PF3D7_1316900, ...</td>\n",
       "      <td>6</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.167</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>460 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_id      gene_name  \\\n",
       "0    G_PF3D7_0507200  PF3D7_0507200   \n",
       "1    G_PF3D7_0932300  PF3D7_0932300   \n",
       "2    G_PF3D7_1360800  PF3D7_1360800   \n",
       "3    G_PF3D7_0409500  PF3D7_0409500   \n",
       "4    G_PF3D7_1446200  PF3D7_1446200   \n",
       "..               ...            ...   \n",
       "455  G_PF3D7_1115300  PF3D7_1115300   \n",
       "456  G_PF3D7_1115700  PF3D7_1115700   \n",
       "457  G_PF3D7_1407800  PF3D7_1407800   \n",
       "458  G_PF3D7_1311800  PF3D7_1311800   \n",
       "459  G_PF3D7_1136900  PF3D7_1136900   \n",
       "\n",
       "                                           predictions  \\\n",
       "0    [(812667, 1.0), (812844, 1.0), (812882, 1.0), ...   \n",
       "1    [(813509, 1.0), (813594, 1.0), (813602, 1.0), ...   \n",
       "2    [(810517, 1.0), (810533, 1.0), (813907, 1.0), ...   \n",
       "3    [(812389, 1.0), (812414, 1.0), (812845, 1.0), ...   \n",
       "4    [(811480, 1.0), (811497, 1.0), (811986, 1.0), ...   \n",
       "..                                                 ...   \n",
       "455  [(810690, 1.0), (810708, 1.0), (810872, 1.0), ...   \n",
       "456                     [(810712, 1.0), (810728, 1.0)]   \n",
       "457  [(811657, 1.0), (811665, 1.0), (812257, 1.0), ...   \n",
       "458  [(810693, 1.0), (810736, 1.0), (810874, 1.0), ...   \n",
       "459  [(810927, 1.0), (810952, 1.0), (813979, 1.0), ...   \n",
       "\n",
       "                                    predictions_labels  num_candidates  \\\n",
       "0    [PF3D7_0207500, PF3D7_0502700, PF3D7_0507200, ...               9   \n",
       "1    [PF3D7_0923300, PF3D7_0932300, PF3D7_0933200, ...               4   \n",
       "2    [PF3D7_1036800, PF3D7_1038600, PF3D7_1306800, ...              15   \n",
       "3    [PF3D7_0409500, PF3D7_0405900, PF3D7_0509400, ...               6   \n",
       "4    [PF3D7_1244600, PF3D7_1246400, PF3D7_1442600, ...               8   \n",
       "..                                                 ...             ...   \n",
       "455  [PF3D7_1113500, PF3D7_1115300, PF3D7_1131500, ...               7   \n",
       "456                     [PF3D7_1115700, PF3D7_1117500]               2   \n",
       "457  [PF3D7_1407800, PF3D7_1408700, PF3D7_1470800, ...              10   \n",
       "458  [PF3D7_1113800, PF3D7_1118300, PF3D7_1131800, ...               6   \n",
       "459  [PF3D7_1136900, PF3D7_1139600, PF3D7_1316900, ...               6   \n",
       "\n",
       "     match_score  recall  precision  \n",
       "0            1.0     1.0      0.111  \n",
       "1            1.0     1.0      0.250  \n",
       "2            1.0     1.0      0.067  \n",
       "3            1.0     1.0      0.167  \n",
       "4            1.0     1.0      0.125  \n",
       "..           ...     ...        ...  \n",
       "455          1.0     1.0      0.143  \n",
       "456          1.0     1.0      0.500  \n",
       "457          1.0     1.0      0.100  \n",
       "458          1.0     1.0      0.167  \n",
       "459          1.0     1.0      0.167  \n",
       "\n",
       "[460 rows x 8 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model_file = \"iAM_Pf480.xml\"\n",
    "one_fpath = bigg_model_dir + model_file\n",
    "recom = recommender.Recommender(libsbml_fpath=one_fpath)\n",
    "geneids2evaluate = list(recom.genes.exist_annotation.keys())\n",
    "gene_id = []\n",
    "gene_name = []\n",
    "predictions = []\n",
    "predictions_labels = []\n",
    "num_cands = []\n",
    "match_scores = []\n",
    "recall = []\n",
    "precision = []\n",
    "tax_dict_file = bigg_model_dir + \"bigg_organisms.csv\"\n",
    "tax_id, tax_name = get_model_taxonomy(model_file, tax_dict_file)\n",
    "recoms = recom.getGeneListRecommendation(pred_ids=geneids2evaluate,tax_id=tax_id)\n",
    "recalls = recom.getGeneStatistics(tax_id=tax_id,model_mean=False)['recall']\n",
    "precisions = recom.getGeneStatistics(tax_id=tax_id,model_mean=False)['precision']\n",
    "for one_recom in recoms:\n",
    "    one_spec = one_recom.id\n",
    "    gene_id.append(one_spec)\n",
    "    gene_name.append(recom.genes.getNameToUse(one_spec))\n",
    "    predictions.append(one_recom.candidates)\n",
    "    predictions_labels.append(one_recom.labels)\n",
    "    num_cands.append(len(one_recom.candidates))\n",
    "    match_scores.append(one_recom.candidates[0][1])\n",
    "    recall.append(recalls[one_spec])\n",
    "    precision.append(precisions[one_spec])\n",
    "spec_df = pd.DataFrame({\n",
    "                    'gene_id': gene_id,\n",
    "                    'gene_name': gene_name,\n",
    "                    'predictions': predictions,\n",
    "                    'predictions_labels': predictions_labels,\n",
    "                    'num_candidates': num_cands,\n",
    "                    'match_score': match_scores,\n",
    "                    'recall': recall,\n",
    "                    'precision': precision})\n",
    "spec_df.to_csv(output_dir + model_file + \".csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Test on Gene Predictions  \n",
    "### Bigg Models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We are at 0: Processing iCHOv1.xml\n",
      "We are at 1: Processing iYL1228.xml\n",
      "We are at 2: Processing iIS312.xml\n",
      "We are at 3: Processing iSDY_1059.xml\n",
      "We are at 4: Processing iEC1356_Bl21DE3.xml\n",
      "We are at 5: Processing iECP_1309.xml\n",
      "We are at 6: Processing iJN746.xml\n",
      "We are at 7: Processing iJR904.xml\n",
      "We are at 8: Processing iJB785.xml\n",
      "We are at 9: Processing iY75_1357.xml\n",
      "We are at 10: Processing iEcHS_1320.xml\n",
      "We are at 11: Processing iUTI89_1310.xml\n",
      "We are at 12: Processing iB21_1397.xml\n",
      "We are at 13: Processing iSbBS512_1146.xml\n",
      "We are at 14: Processing iNF517.xml\n",
      "We are at 15: Processing iNJ661.xml\n",
      "We are at 16: Processing iECIAI39_1322.xml\n",
      "Skipping iECIAI39_1322.xml (invalid model)\n",
      "We are at 17: Processing iECO103_1326.xml\n",
      "We are at 18: Processing iSSON_1240.xml\n",
      "We are at 19: Processing iCHOv1_DG44.xml\n",
      "We are at 20: Processing iAB_RBC_283.xml\n",
      "We are at 21: Processing iAM_Pk459.xml\n",
      "We are at 22: Processing iPC815.xml\n",
      "We are at 23: Processing iAPECO1_1312.xml\n",
      "We are at 24: Processing iMM1415.xml\n",
      "We are at 25: Processing iECDH1ME8569_1439.xml\n",
      "We are at 26: Processing iJO1366.xml\n",
      "We are at 27: Processing iEcE24377_1341.xml\n",
      "We are at 28: Processing iEC042_1314.xml\n",
      "We are at 29: Processing iRC1080.xml\n",
      "We are at 30: Processing iCN900.xml\n",
      "We are at 31: Processing iEC1364_W.xml\n",
      "We are at 32: Processing iJN678.xml\n",
      "We are at 33: Processing iEcSMS35_1347.xml\n",
      "We are at 34: Processing iSBO_1134.xml\n",
      "We are at 35: Processing STM_v1_0.xml\n",
      "We are at 36: Processing iAF1260.xml\n",
      "We are at 37: Processing iJN1463.xml\n",
      "We are at 38: Processing iS_1188.xml\n",
      "We are at 39: Processing iML1515.xml\n",
      "We are at 40: Processing iCN718.xml\n",
      "We are at 41: Processing iAF692.xml\n",
      "We are at 42: Processing iETEC_1333.xml\n",
      "We are at 43: Processing iECD_1391.xml\n",
      "We are at 44: Processing iHN637.xml\n",
      "We are at 45: Processing iG2583_1286.xml\n",
      "We are at 46: Processing iMM904.xml\n",
      "We are at 47: Processing Recon3D.xml\n",
      "We are at 48: Processing iAM_Pf480.xml\n",
      "We are at 49: Processing e_coli_core.xml\n",
      "We are at 50: Processing iAT_PLT_636.xml\n",
      "We are at 51: Processing iEC1344_C.xml\n",
      "We are at 52: Processing iWFL_1372.xml\n",
      "We are at 53: Processing iECABU_c1320.xml\n",
      "We are at 54: Processing iBWG_1329.xml\n",
      "We are at 55: Processing iIT341.xml\n",
      "We are at 56: Processing iEcolC_1368.xml\n",
      "We are at 57: Processing iECs_1301.xml\n",
      "We are at 58: Processing iZ_1308.xml\n",
      "We are at 59: Processing iSF_1195.xml\n",
      "We are at 60: Processing iND750.xml\n",
      "We are at 61: Processing iECB_1328.xml\n",
      "We are at 62: Processing iYO844.xml\n",
      "We are at 63: Processing iECIAI1_1343.xml\n",
      "We are at 64: Processing iLJ478.xml\n",
      "We are at 65: Processing iIS312_Amastigote.xml\n",
      "We are at 66: Processing RECON1.xml\n",
      "We are at 67: Processing iECBD_1354.xml\n",
      "We are at 68: Processing iEC1349_Crooks.xml\n",
      "We are at 69: Processing iSFxv_1172.xml\n",
      "We are at 70: Processing iECUMN_1333.xml\n",
      "We are at 71: Processing iECS88_1305.xml\n",
      "We are at 72: Processing iEC1368_DH5a.xml\n",
      "We are at 73: Processing iEcDH1_1363.xml\n",
      "We are at 74: Processing iAM_Pc455.xml\n",
      "We are at 75: Processing iUMN146_1321.xml\n",
      "We are at 76: Processing iECSP_1301.xml\n",
      "We are at 77: Processing iIS312_Epimastigote.xml\n",
      "We are at 78: Processing iLF82_1304.xml\n",
      "We are at 79: Processing iLB1027_lipid.xml\n",
      "We are at 80: Processing iECH74115_1262.xml\n",
      "We are at 81: Processing iE2348C_1286.xml\n",
      "We are at 82: Processing iECOK1_1307.xml\n",
      "We are at 83: Processing iUMNK88_1353.xml\n",
      "We are at 84: Processing iECSE_1348.xml\n",
      "We are at 85: Processing iEK1008.xml\n",
      "We are at 86: Processing iECSF_1327.xml\n",
      "We are at 87: Processing iYS854.xml\n",
      "We are at 88: Processing iSFV_1184.xml\n",
      "We are at 89: Processing iEC1372_W3110.xml\n",
      "We are at 90: Processing ic_1306.xml\n",
      "We are at 91: Processing iECED1_1282.xml\n",
      "We are at 92: Processing iECW_1372.xml\n",
      "We are at 93: Processing iSynCJ816.xml\n",
      "We are at 94: Processing iIS312_Trypomastigote.xml\n",
      "We are at 95: Processing iECDH10B_1368.xml\n",
      "We are at 96: Processing iYS1720.xml\n",
      "We are at 97: Processing iECNA114_1301.xml\n",
      "We are at 98: Processing iAM_Pb448.xml\n",
      "We are at 99: Processing iAM_Pv461.xml\n",
      "We are at 100: Processing iAF987.xml\n",
      "We are at 101: Processing iECO111_1330.xml\n",
      "We are at 102: Processing iSB619.xml\n",
      "We are at 103: Processing iNRG857_1313.xml\n",
      "We are at 104: Processing iECO26_1355.xml\n",
      "We are at 105: Processing iAF1260b.xml\n",
      "We are at 106: Processing iEC55989_1330.xml\n",
      "We are at 107: Processing iEKO11_1354.xml\n"
     ]
    }
   ],
   "source": [
    "def generate_model_genes_statistics(model_dir, output_csv):\n",
    "    model_statistics_data = []\n",
    "\n",
    "    # All SBML files in the model directory\n",
    "    model_files = [f for f in os.listdir(model_dir) if f.endswith('.xml')]\n",
    "\n",
    "    for idx, model_file in enumerate(model_files):\n",
    "        print(f\"We are at {idx}: Processing {model_file}\")\n",
    "        one_fpath = os.path.join(model_dir, model_file)\n",
    "        reader = libsbml.SBMLReader()\n",
    "        document = reader.readSBML(one_fpath)\n",
    "        model = document.getModel()\n",
    "        if model is None:\n",
    "            print(f\"Skipping {model_file} (invalid model)\")\n",
    "            continue\n",
    "        fbc_plugin = model.getPlugin(\"fbc\")\n",
    "        if fbc_plugin is None:\n",
    "            print(f\"Skipping {model_file} (no fbc package)\")\n",
    "            continue\n",
    "        recom = recommender.Recommender(libsbml_fpath=one_fpath)\n",
    "\n",
    "        # Get species statistics\n",
    "        num_genes = fbc_plugin.getNumGeneProducts()\n",
    "        genes2evaluate = list(recom.genes.exist_annotation.keys())\n",
    "        num_annotated_genes = len(genes2evaluate)\n",
    "        annotated = num_annotated_genes > 0\n",
    "        \n",
    "        # Append the results for this species\n",
    "        model_statistics_data.append({\n",
    "            'model': model_file,\n",
    "            'annotated': annotated,\n",
    "            'num_genes': num_genes,\n",
    "            'num_annotated_genes': num_annotated_genes\n",
    "        })\n",
    "\n",
    "\n",
    "    # Create a DataFrame from the collected data\n",
    "    df = pd.DataFrame(model_statistics_data)\n",
    "\n",
    "    # Save the DataFrame to a CSV file\n",
    "    df.to_csv(output_csv, index=False)\n",
    "\n",
    "output_csv = output_dir + \"bigg_model_gene_statistics.csv\"\n",
    "generate_model_genes_statistics(bigg_model_dir, output_csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Average number of genes: 987.6\n",
      "Average number of annotated genes: 904.9\n",
      "There are 28 models with annotated genes:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model</th>\n",
       "      <th>annotated</th>\n",
       "      <th>num_genes</th>\n",
       "      <th>num_annotated_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>iCHOv1.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1766</td>\n",
       "      <td>1743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>iJN746.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>746</td>\n",
       "      <td>729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>iJR904.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>904</td>\n",
       "      <td>892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>iNJ661.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>661</td>\n",
       "      <td>655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>iCHOv1_DG44.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1184</td>\n",
       "      <td>1168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>iAB_RBC_283.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>346</td>\n",
       "      <td>311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>iPC815.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>815</td>\n",
       "      <td>814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>iMM1415.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1375</td>\n",
       "      <td>1344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>iJO1366.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1367</td>\n",
       "      <td>1366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>iAF1260.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1261</td>\n",
       "      <td>1260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>iJN1463.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1462</td>\n",
       "      <td>1442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>iML1515.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1516</td>\n",
       "      <td>1515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>iMM904.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>905</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Recon3D.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>2248</td>\n",
       "      <td>2229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>iAM_Pf480.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>480</td>\n",
       "      <td>460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>e_coli_core.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>137</td>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>iAT_PLT_636.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>636</td>\n",
       "      <td>632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>iIT341.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>339</td>\n",
       "      <td>335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>iND750.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>750</td>\n",
       "      <td>746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>iLJ478.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>482</td>\n",
       "      <td>478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>RECON1.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1905</td>\n",
       "      <td>1880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>iAM_Pc455.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>455</td>\n",
       "      <td>452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>iLB1027_lipid.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1027</td>\n",
       "      <td>699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>iEK1008.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1008</td>\n",
       "      <td>996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>iYS1720.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1707</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>iAM_Pb448.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>448</td>\n",
       "      <td>441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>iAM_Pv461.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>461</td>\n",
       "      <td>450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>iAF1260b.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1261</td>\n",
       "      <td>1260</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 model  annotated  num_genes  num_annotated_genes\n",
       "0           iCHOv1.xml       True       1766                 1743\n",
       "6           iJN746.xml       True        746                  729\n",
       "7           iJR904.xml       True        904                  892\n",
       "15          iNJ661.xml       True        661                  655\n",
       "18     iCHOv1_DG44.xml       True       1184                 1168\n",
       "19     iAB_RBC_283.xml       True        346                  311\n",
       "21          iPC815.xml       True        815                  814\n",
       "23         iMM1415.xml       True       1375                 1344\n",
       "25         iJO1366.xml       True       1367                 1366\n",
       "35         iAF1260.xml       True       1261                 1260\n",
       "36         iJN1463.xml       True       1462                 1442\n",
       "38         iML1515.xml       True       1516                 1515\n",
       "45          iMM904.xml       True        905                  900\n",
       "46         Recon3D.xml       True       2248                 2229\n",
       "47       iAM_Pf480.xml       True        480                  460\n",
       "48     e_coli_core.xml       True        137                  136\n",
       "49     iAT_PLT_636.xml       True        636                  632\n",
       "54          iIT341.xml       True        339                  335\n",
       "59          iND750.xml       True        750                  746\n",
       "63          iLJ478.xml       True        482                  478\n",
       "65          RECON1.xml       True       1905                 1880\n",
       "73       iAM_Pc455.xml       True        455                  452\n",
       "78   iLB1027_lipid.xml       True       1027                  699\n",
       "84         iEK1008.xml       True       1008                  996\n",
       "95         iYS1720.xml       True       1707                    3\n",
       "97       iAM_Pb448.xml       True        448                  441\n",
       "98       iAM_Pv461.xml       True        461                  450\n",
       "104       iAF1260b.xml       True       1261                 1260"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bigg_model_gene_statistics = pd.read_csv(output_dir + \"bigg_model_gene_statistics.csv\")\n",
    "annotated_models = bigg_model_gene_statistics[bigg_model_gene_statistics['num_annotated_genes'] != 0]\n",
    "print(f\"Average number of genes: {annotated_models['num_genes'].mean():.1f}\")\n",
    "print(f\"Average number of annotated genes: {annotated_models['num_annotated_genes'].mean():.1f}\")\n",
    "print(f\"There are {len(annotated_models)} models with annotated genes:\")\n",
    "annotated_models\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We are at 0/108: Processing iCHOv1.xml\n",
      "We are at 1/108: Processing iYL1228.xml\n",
      "We are at 2/108: Processing iIS312.xml\n",
      "We are at 3/108: Processing iSDY_1059.xml\n",
      "We are at 4/108: Processing iEC1356_Bl21DE3.xml\n",
      "We are at 5/108: Processing iECP_1309.xml\n",
      "We are at 6/108: Processing iJN746.xml\n",
      "Error processing iJN746.xml: Reference data not found for tax_id 160488\n",
      "We are at 7/108: Processing iJR904.xml\n",
      "We are at 8/108: Processing iJB785.xml\n",
      "We are at 9/108: Processing iY75_1357.xml\n",
      "We are at 10/108: Processing iEcHS_1320.xml\n",
      "We are at 11/108: Processing iUTI89_1310.xml\n",
      "We are at 12/108: Processing iB21_1397.xml\n",
      "We are at 13/108: Processing iSbBS512_1146.xml\n",
      "We are at 14/108: Processing iNF517.xml\n",
      "We are at 15/108: Processing iNJ661.xml\n",
      "We are at 16/108: Processing iECIAI39_1322.xml\n",
      "Error processing iECIAI39_1322.xml: 'NoneType' object has no attribute 'keys'\n",
      "We are at 17/108: Processing iECO103_1326.xml\n",
      "We are at 18/108: Processing iSSON_1240.xml\n",
      "We are at 19/108: Processing iCHOv1_DG44.xml\n",
      "We are at 20/108: Processing iAB_RBC_283.xml\n",
      "We are at 21/108: Processing iAM_Pk459.xml\n",
      "We are at 22/108: Processing iPC815.xml\n",
      "Error processing iPC815.xml: Reference data not found for tax_id 214092\n",
      "We are at 23/108: Processing iAPECO1_1312.xml\n",
      "We are at 24/108: Processing iMM1415.xml\n",
      "We are at 25/108: Processing iECDH1ME8569_1439.xml\n",
      "We are at 26/108: Processing iJO1366.xml\n",
      "We are at 27/108: Processing iEcE24377_1341.xml\n",
      "We are at 28/108: Processing iEC042_1314.xml\n",
      "We are at 29/108: Processing iRC1080.xml\n",
      "We are at 30/108: Processing iCN900.xml\n",
      "We are at 31/108: Processing iEC1364_W.xml\n",
      "We are at 32/108: Processing iJN678.xml\n",
      "We are at 33/108: Processing iEcSMS35_1347.xml\n",
      "We are at 34/108: Processing iSBO_1134.xml\n",
      "We are at 35/108: Processing STM_v1_0.xml\n",
      "We are at 36/108: Processing iAF1260.xml\n",
      "We are at 37/108: Processing iJN1463.xml\n",
      "Error processing iJN1463.xml: Reference data not found for tax_id 160488\n",
      "We are at 38/108: Processing iS_1188.xml\n",
      "We are at 39/108: Processing iML1515.xml\n",
      "We are at 40/108: Processing iCN718.xml\n",
      "We are at 41/108: Processing iAF692.xml\n",
      "We are at 42/108: Processing iETEC_1333.xml\n",
      "We are at 43/108: Processing iECD_1391.xml\n",
      "We are at 44/108: Processing iHN637.xml\n",
      "We are at 45/108: Processing iG2583_1286.xml\n",
      "We are at 46/108: Processing iMM904.xml\n",
      "We are at 47/108: Processing Recon3D.xml\n",
      "We are at 48/108: Processing iAM_Pf480.xml\n",
      "We are at 49/108: Processing e_coli_core.xml\n",
      "We are at 50/108: Processing iAT_PLT_636.xml\n",
      "We are at 51/108: Processing iEC1344_C.xml\n",
      "We are at 52/108: Processing iWFL_1372.xml\n",
      "We are at 53/108: Processing iECABU_c1320.xml\n",
      "We are at 54/108: Processing iBWG_1329.xml\n",
      "We are at 55/108: Processing iIT341.xml\n",
      "Error processing iIT341.xml: Reference data not found for tax_id 85962\n",
      "We are at 56/108: Processing iEcolC_1368.xml\n",
      "We are at 57/108: Processing iECs_1301.xml\n",
      "We are at 58/108: Processing iZ_1308.xml\n",
      "We are at 59/108: Processing iSF_1195.xml\n",
      "We are at 60/108: Processing iND750.xml\n",
      "We are at 61/108: Processing iECB_1328.xml\n",
      "We are at 62/108: Processing iYO844.xml\n",
      "We are at 63/108: Processing iECIAI1_1343.xml\n",
      "We are at 64/108: Processing iLJ478.xml\n",
      "Error processing iLJ478.xml: Reference data not found for tax_id 243274\n",
      "We are at 65/108: Processing iIS312_Amastigote.xml\n",
      "We are at 66/108: Processing RECON1.xml\n",
      "We are at 67/108: Processing iECBD_1354.xml\n",
      "We are at 68/108: Processing iEC1349_Crooks.xml\n",
      "We are at 69/108: Processing iSFxv_1172.xml\n",
      "We are at 70/108: Processing iECUMN_1333.xml\n",
      "We are at 71/108: Processing iECS88_1305.xml\n",
      "We are at 72/108: Processing iEC1368_DH5a.xml\n",
      "We are at 73/108: Processing iEcDH1_1363.xml\n",
      "We are at 74/108: Processing iAM_Pc455.xml\n",
      "We are at 75/108: Processing iUMN146_1321.xml\n",
      "We are at 76/108: Processing iECSP_1301.xml\n",
      "We are at 77/108: Processing iIS312_Epimastigote.xml\n",
      "We are at 78/108: Processing iLF82_1304.xml\n",
      "We are at 79/108: Processing iLB1027_lipid.xml\n",
      "We are at 80/108: Processing iECH74115_1262.xml\n",
      "We are at 81/108: Processing iE2348C_1286.xml\n",
      "We are at 82/108: Processing iECOK1_1307.xml\n",
      "We are at 83/108: Processing iUMNK88_1353.xml\n",
      "We are at 84/108: Processing iECSE_1348.xml\n",
      "We are at 85/108: Processing iEK1008.xml\n",
      "We are at 86/108: Processing iECSF_1327.xml\n",
      "We are at 87/108: Processing iYS854.xml\n",
      "We are at 88/108: Processing iSFV_1184.xml\n",
      "We are at 89/108: Processing iEC1372_W3110.xml\n",
      "We are at 90/108: Processing ic_1306.xml\n",
      "We are at 91/108: Processing iECED1_1282.xml\n",
      "We are at 92/108: Processing iECW_1372.xml\n",
      "We are at 93/108: Processing iSynCJ816.xml\n",
      "We are at 94/108: Processing iIS312_Trypomastigote.xml\n",
      "We are at 95/108: Processing iECDH10B_1368.xml\n",
      "We are at 96/108: Processing iYS1720.xml\n",
      "Error processing iYS1720.xml: Reference data not found for tax_id  \n",
      "We are at 97/108: Processing iECNA114_1301.xml\n",
      "We are at 98/108: Processing iAM_Pb448.xml\n",
      "We are at 99/108: Processing iAM_Pv461.xml\n",
      "We are at 100/108: Processing iAF987.xml\n",
      "We are at 101/108: Processing iECO111_1330.xml\n",
      "We are at 102/108: Processing iSB619.xml\n",
      "We are at 103/108: Processing iNRG857_1313.xml\n",
      "We are at 104/108: Processing iECO26_1355.xml\n",
      "We are at 105/108: Processing iAF1260b.xml\n",
      "We are at 106/108: Processing iEC55989_1330.xml\n",
      "We are at 107/108: Processing iEKO11_1354.xml\n"
     ]
    }
   ],
   "source": [
    "def get_model_taxonomy(model_file, tax_dict_file):\n",
    "    tax_dict_df = pd.read_csv(tax_dict_file)\n",
    "    model_file_id = model_file.replace('.xml', '')\n",
    "    tax_id = tax_dict_df[tax_dict_df['id'] == model_file_id]['tax_id'].values[0]\n",
    "    tax_name = tax_dict_df[tax_dict_df['id'] == model_file_id]['organism'].values[0]\n",
    "    return tax_id, tax_name\n",
    "\n",
    "def generate_gene_accuracy(model_dir, output_csv, tax_dict_file):\n",
    "    model_files = [f for f in os.listdir(model_dir) if f.endswith('.xml')]\n",
    "    model_names = []\n",
    "    taxonomy_id = []\n",
    "    taxonomy_name = []\n",
    "    gene_id = []\n",
    "    gene_name = []\n",
    "    len_names = []\n",
    "    predictions = []\n",
    "    predictions_labels = []\n",
    "    num_cands = []\n",
    "    match_scores = []\n",
    "    recall = []\n",
    "    precision = []\n",
    "    is_correct_included = []\n",
    "    processing_time = []\n",
    "\n",
    "    for idx,model_file in enumerate(model_files):\n",
    "        print(f\"We are at {idx}/{len(model_files)}: Processing {model_file}\")\n",
    "        one_fpath = os.path.join(model_dir, model_file)\n",
    "        tax_id, tax_name = get_model_taxonomy(model_file, tax_dict_file)\n",
    "        try:\n",
    "            recom = recommender.Recommender(libsbml_fpath=one_fpath)\n",
    "            if recom is None or recom.genes is None:\n",
    "                print(f\"Skipping {model_file} (invalid recommender or genes)\")\n",
    "                continue\n",
    "            genes2evaluate = list(recom.genes.exist_annotation.keys())\n",
    "            # run algorithm if there is at least one species to evaluate\n",
    "            if genes2evaluate:\n",
    "                time_start = time.time()\n",
    "                recoms = recom.getGeneListRecommendation(tax_id=tax_id,pred_ids=genes2evaluate)\n",
    "                time_end = time.time()\n",
    "                recalls = recom.getGeneStatistics(tax_id=tax_id,model_mean=False)['recall']\n",
    "                precisions = recom.getGeneStatistics(tax_id=tax_id,model_mean=False)['precision']\n",
    "                for one_recom in recoms:\n",
    "                    model_names.append(model_file)\n",
    "                    taxonomy_id.append(tax_id)\n",
    "                    taxonomy_name.append(tax_name)\n",
    "                    one_spec = one_recom.id\n",
    "                    gene_id.append(one_spec)\n",
    "                    gene_name.append(recom.genes.getNameToUse(one_spec))\n",
    "                    len_names.append(len(gene_name[-1]))\n",
    "                    predictions.append(one_recom.candidates)\n",
    "                    predictions_labels.append(one_recom.labels)\n",
    "                    num_cands.append(len(one_recom.candidates))\n",
    "                    match_scores.append(one_recom.candidates[0][1])\n",
    "                    recall.append(recalls[one_spec])\n",
    "                    precision.append(precisions[one_spec])\n",
    "                    processing_time.append(time_end - time_start)\n",
    "                    if recall[-1] > 0:\n",
    "                        is_correct_included.append(1)\n",
    "                    else:\n",
    "                        is_correct_included.append(0)\n",
    "        except Exception as e:\n",
    "            print(f\"Error processing {model_file}: {str(e)}\")\n",
    "            continue\n",
    "\n",
    "    # Create a DataFrame from the collected data\n",
    "    spec_df = pd.DataFrame({'model': model_names,\n",
    "                        'taxonomy_id': taxonomy_id,\n",
    "                        'taxonomy_name': taxonomy_name,\n",
    "                        'gene_id': gene_id,\n",
    "                        'gene_name': gene_name,\n",
    "                        'name_length': len_names,\n",
    "                        'predictions': predictions,\n",
    "                        'predictions_labels': predictions_labels,\n",
    "                        'num_candidates': num_cands,\n",
    "                        'match_score': match_scores,\n",
    "                        'recall': recall,\n",
    "                        'precision': precision,\n",
    "                        'accuracy': is_correct_included,\n",
    "                        'processing_time': processing_time})\n",
    "\n",
    "    # Save the DataFrame to a CSV file\n",
    "    spec_df.to_csv(output_csv, index=False)\n",
    "\n",
    "output_csv = output_dir + \"bigg_model_gene_accuracy_protein_coding.csv\"\n",
    "tax_dict_file = bigg_model_dir + \"bigg_organisms.csv\"\n",
    "generate_gene_accuracy(bigg_model_dir, output_csv, tax_dict_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "<Bigg Genes (All)>\n",
      "Number of models assessed: 22\n",
      "Number of genes assessed: 12322\n",
      "Average recall (per model) : 0.906\n",
      "Average precision (per model) : 0.604\n",
      "Average match score (per model): 0.928\n",
      "Average accuracy (per model): 0.906\n",
      "Ave. processing time (per model): 2.794\n",
      "Ave. processing time (per element, per model): 0.003\n",
      "Average number of predictions per row: 2.345\n"
     ]
    }
   ],
   "source": [
    "def safe_eval_with_eval(x):\n",
    "    try:\n",
    "        return eval(x) if isinstance(x, str) else []\n",
    "    except Exception as e:\n",
    "        print(f\"Failed to parse: {x} with error: {e}\")\n",
    "        return []\n",
    "def print_model_accuracy(df, show_organism=False):\n",
    "    if show_organism:\n",
    "        print(\"Number of organisms assessed: %d\" % df['taxonomy_name'].nunique())\n",
    "        for one_organism in df['taxonomy_name'].unique():\n",
    "            print(f\"  - {one_organism}\")\n",
    "            df_organism = df[df['taxonomy_name'] == one_organism]\n",
    "            print(\"Number of models assessed: %d\" % df_organism['model'].nunique())\n",
    "            print(\"Number of genes assessed: %d\" % df_organism['gene_id'].nunique())\n",
    "            print(\"Average recall (per model) : %.003f\" % df_organism.groupby('model')['recall'].mean().mean())\n",
    "            print(\"Average precision (per model) : %.003f\" % df_organism.groupby('model')['precision'].mean().mean())\n",
    "            print(\"Average match score (per model): %.003f\" % df_organism.groupby('model')['match_score'].mean().mean())\n",
    "            print(\"Average accuracy (per model): %.003f\" % df_organism.groupby('model')['accuracy'].mean().mean())\n",
    "            df_copy = df_organism.copy()\n",
    "            df_copy['parsed_predictions'] = df_copy['predictions'].apply(safe_eval_with_eval)\n",
    "            df_copy['num_predictions'] = df_copy['parsed_predictions'].apply(len)\n",
    "            average_predictions = df_copy['num_predictions'].mean()\n",
    "            print(f\"Average number of predictions per row: %.003f\" % average_predictions)\n",
    "            print(\"__________________________________________________________\")\n",
    "    else:\n",
    "        print(\"Number of models assessed: %d\" % df['model'].nunique())\n",
    "        print(\"Number of genes assessed: %d\" % df['gene_id'].nunique())\n",
    "        print(\"Average recall (per model) : %.003f\" % df.groupby('model')['recall'].mean().mean())\n",
    "        print(\"Average precision (per model) : %.003f\" % df.groupby('model')['precision'].mean().mean())\n",
    "        print(\"Average match score (per model): %.003f\" % df.groupby('model')['match_score'].mean().mean())\n",
    "        print(\"Average accuracy (per model): %.003f\" % df.groupby('model')['accuracy'].mean().mean())\n",
    "    # Fixed processing time calculations\n",
    "    mean_processing_time = df.groupby('model')['processing_time'].first().mean()\n",
    "    print(\"Ave. processing time (per model): %.003f\" % mean_processing_time)\n",
    "    # Calculate per element processing time\n",
    "    num_elements = df.groupby('model').size().mean()\n",
    "    mean_processing_time_per_element = mean_processing_time / num_elements\n",
    "    print(\"Ave. processing time (per element, per model): %.003f\" % mean_processing_time_per_element)\n",
    "    # Average number of predictions per species\n",
    "    df = df.copy()\n",
    "    df['parsed_predictions'] = df['predictions'].apply(safe_eval_with_eval)\n",
    "    df['num_predictions'] = df['parsed_predictions'].apply(len)\n",
    "    average_predictions = df['num_predictions'].mean()\n",
    "    print(f\"Average number of predictions per row: %.003f\" % average_predictions)\n",
    "    \n",
    "bigg_model_gene_accuracy = pd.read_csv(output_dir + \"bigg_model_gene_accuracy_protein_coding.csv\")\n",
    "print(\"\\n\\n<Bigg Genes (All)>\")\n",
    "print_model_accuracy(bigg_model_gene_accuracy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of models assessed: 18\n",
      "Number of genes assessed: 10519\n",
      "Average recall (per model) : 0.996\n",
      "Average precision (per model) : 0.720\n",
      "Average match score (per model): 1.000\n",
      "Average accuracy (per model): 0.996\n",
      "Ave. processing time (per model): 3.327\n",
      "Ave. processing time (per element, per model): 0.003\n",
      "Average number of predictions per row: 2.050\n"
     ]
    }
   ],
   "source": [
    "discontinued = ['Plasmodium falciparum 3D7','Plasmodium cynomolgi strain B','Pseudomonas putida KT2440','Yersinia pestis CO92','Helicobacter pylori 26695','Thermotoga maritima MSB8','Plasmodium vivax Sal-1','Plasmodium berghei']\n",
    "bigg_model_gene_accuracy = pd.read_csv(output_dir+\"bigg_model_gene_accuracy_protein_coding.csv\")\n",
    "bigg_model_gene_accuracy = bigg_model_gene_accuracy[~bigg_model_gene_accuracy['taxonomy_name'].isin(discontinued)]\n",
    "print_model_accuracy(bigg_model_gene_accuracy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of organisms assessed: 7\n",
      "  - Cricetulus griseus\n",
      "Number of models assessed: 2\n",
      "Number of genes assessed: 1743\n",
      "Average recall (per model) : 0.984\n",
      "Average precision (per model) : 0.776\n",
      "Average match score (per model): 1.000\n",
      "Average accuracy (per model): 0.984\n",
      "Average number of predictions per row: 1.962\n",
      "__________________________________________________________\n",
      "  - Escherichia coli str. K-12 substr. MG1655\n",
      "Number of models assessed: 6\n",
      "Number of genes assessed: 1553\n",
      "Average recall (per model) : 0.999\n",
      "Average precision (per model) : 0.727\n",
      "Average match score (per model): 1.000\n",
      "Average accuracy (per model): 0.999\n",
      "Average number of predictions per row: 1.867\n",
      "__________________________________________________________\n",
      "  - Mycobacterium tuberculosis H37Rv\n",
      "Number of models assessed: 2\n",
      "Number of genes assessed: 1014\n",
      "Average recall (per model) : 1.000\n",
      "Average precision (per model) : 0.856\n",
      "Average match score (per model): 1.000\n",
      "Average accuracy (per model): 1.000\n",
      "Average number of predictions per row: 1.435\n",
      "__________________________________________________________\n",
      "  - Homo sapiens\n",
      "Number of models assessed: 4\n",
      "Number of genes assessed: 3266\n",
      "Average recall (per model) : 0.999\n",
      "Average precision (per model) : 0.713\n",
      "Average match score (per model): 1.000\n",
      "Average accuracy (per model): 0.999\n",
      "Average number of predictions per row: 2.019\n",
      "__________________________________________________________\n",
      "  - Mus musculus\n",
      "Number of models assessed: 1\n",
      "Number of genes assessed: 1344\n",
      "Average recall (per model) : 0.986\n",
      "Average precision (per model) : 0.745\n",
      "Average match score (per model): 0.999\n",
      "Average accuracy (per model): 0.986\n",
      "Average number of predictions per row: 1.795\n",
      "__________________________________________________________\n",
      "  - Saccharomyces cerevisiae S288C\n",
      "Number of models assessed: 2\n",
      "Number of genes assessed: 909\n",
      "Average recall (per model) : 0.995\n",
      "Average precision (per model) : 0.653\n",
      "Average match score (per model): 0.999\n",
      "Average accuracy (per model): 0.995\n",
      "Average number of predictions per row: 2.180\n",
      "__________________________________________________________\n",
      "  - Phaeodactylum tricornutum CCAP 1055/1\n",
      "Number of models assessed: 1\n",
      "Number of genes assessed: 699\n",
      "Average recall (per model) : 1.000\n",
      "Average precision (per model) : 0.437\n",
      "Average match score (per model): 1.000\n",
      "Average accuracy (per model): 1.000\n",
      "Average number of predictions per row: 5.963\n",
      "__________________________________________________________\n",
      "Ave. processing time (per model): 3.327\n",
      "Ave. processing time (per element, per model): 0.003\n",
      "Average number of predictions per row: 2.050\n"
     ]
    }
   ],
   "source": [
    "print_model_accuracy(bigg_model_gene_accuracy, show_organism=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model</th>\n",
       "      <th>taxonomy_id</th>\n",
       "      <th>taxonomy_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>name_length</th>\n",
       "      <th>predictions</th>\n",
       "      <th>predictions_labels</th>\n",
       "      <th>num_candidates</th>\n",
       "      <th>match_score</th>\n",
       "      <th>recall</th>\n",
       "      <th>precision</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>processing_time</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12430</th>\n",
       "      <td>Recon3D.xml</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G_55350_AT3</td>\n",
       "      <td>VNN3</td>\n",
       "      <td>4</td>\n",
       "      <td>[('7448', np.float64(0.866))]</td>\n",
       "      <td>['VTN']</td>\n",
       "      <td>1</td>\n",
       "      <td>0.866</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>11.421148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12441</th>\n",
       "      <td>Recon3D.xml</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G_55350_AT1</td>\n",
       "      <td>VNN3</td>\n",
       "      <td>4</td>\n",
       "      <td>[('7448', np.float64(0.866))]</td>\n",
       "      <td>['VTN']</td>\n",
       "      <td>1</td>\n",
       "      <td>0.866</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>11.421148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12446</th>\n",
       "      <td>Recon3D.xml</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G_55350_AT2</td>\n",
       "      <td>VNN3</td>\n",
       "      <td>4</td>\n",
       "      <td>[('7448', np.float64(0.866))]</td>\n",
       "      <td>['VTN']</td>\n",
       "      <td>1</td>\n",
       "      <td>0.866</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>11.421148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12695</th>\n",
       "      <td>Recon3D.xml</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G_728441_AT1</td>\n",
       "      <td>GGT2</td>\n",
       "      <td>4</td>\n",
       "      <td>[('102724197', np.float64(1.0))]</td>\n",
       "      <td>['LOC102724197']</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>11.421148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13243</th>\n",
       "      <td>Recon3D.xml</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G_200810_AT1</td>\n",
       "      <td>ALG1L</td>\n",
       "      <td>5</td>\n",
       "      <td>[('2657', np.float64(0.945)), ('3956', np.floa...</td>\n",
       "      <td>['GDF1', 'LGALS1', 'S100A8', 'S100A9', 'CCL4',...</td>\n",
       "      <td>8</td>\n",
       "      <td>0.945</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>11.421148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17029</th>\n",
       "      <td>RECON1.xml</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G_55350_AT3</td>\n",
       "      <td>VNN3</td>\n",
       "      <td>4</td>\n",
       "      <td>[('7448', np.float64(0.866))]</td>\n",
       "      <td>['VTN']</td>\n",
       "      <td>1</td>\n",
       "      <td>0.866</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>9.244529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17030</th>\n",
       "      <td>RECON1.xml</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G_55350_AT2</td>\n",
       "      <td>VNN3</td>\n",
       "      <td>4</td>\n",
       "      <td>[('7448', np.float64(0.866))]</td>\n",
       "      <td>['VTN']</td>\n",
       "      <td>1</td>\n",
       "      <td>0.866</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>9.244529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17038</th>\n",
       "      <td>RECON1.xml</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>G_55350_AT1</td>\n",
       "      <td>VNN3</td>\n",
       "      <td>4</td>\n",
       "      <td>[('7448', np.float64(0.866))]</td>\n",
       "      <td>['VTN']</td>\n",
       "      <td>1</td>\n",
       "      <td>0.866</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>9.244529</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             model  taxonomy_id taxonomy_name       gene_id gene_name  \\\n",
       "12430  Recon3D.xml         9606  Homo sapiens   G_55350_AT3      VNN3   \n",
       "12441  Recon3D.xml         9606  Homo sapiens   G_55350_AT1      VNN3   \n",
       "12446  Recon3D.xml         9606  Homo sapiens   G_55350_AT2      VNN3   \n",
       "12695  Recon3D.xml         9606  Homo sapiens  G_728441_AT1      GGT2   \n",
       "13243  Recon3D.xml         9606  Homo sapiens  G_200810_AT1     ALG1L   \n",
       "17029   RECON1.xml         9606  Homo sapiens   G_55350_AT3      VNN3   \n",
       "17030   RECON1.xml         9606  Homo sapiens   G_55350_AT2      VNN3   \n",
       "17038   RECON1.xml         9606  Homo sapiens   G_55350_AT1      VNN3   \n",
       "\n",
       "       name_length                                        predictions  \\\n",
       "12430            4                      [('7448', np.float64(0.866))]   \n",
       "12441            4                      [('7448', np.float64(0.866))]   \n",
       "12446            4                      [('7448', np.float64(0.866))]   \n",
       "12695            4                   [('102724197', np.float64(1.0))]   \n",
       "13243            5  [('2657', np.float64(0.945)), ('3956', np.floa...   \n",
       "17029            4                      [('7448', np.float64(0.866))]   \n",
       "17030            4                      [('7448', np.float64(0.866))]   \n",
       "17038            4                      [('7448', np.float64(0.866))]   \n",
       "\n",
       "                                      predictions_labels  num_candidates  \\\n",
       "12430                                            ['VTN']               1   \n",
       "12441                                            ['VTN']               1   \n",
       "12446                                            ['VTN']               1   \n",
       "12695                                   ['LOC102724197']               1   \n",
       "13243  ['GDF1', 'LGALS1', 'S100A8', 'S100A9', 'CCL4',...               8   \n",
       "17029                                            ['VTN']               1   \n",
       "17030                                            ['VTN']               1   \n",
       "17038                                            ['VTN']               1   \n",
       "\n",
       "       match_score  recall  precision  accuracy  processing_time  \n",
       "12430        0.866     0.0        0.0         0        11.421148  \n",
       "12441        0.866     0.0        0.0         0        11.421148  \n",
       "12446        0.866     0.0        0.0         0        11.421148  \n",
       "12695        1.000     0.0        0.0         0        11.421148  \n",
       "13243        0.945     0.0        0.0         0        11.421148  \n",
       "17029        0.866     0.0        0.0         0         9.244529  \n",
       "17030        0.866     0.0        0.0         0         9.244529  \n",
       "17038        0.866     0.0        0.0         0         9.244529  "
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human = bigg_model_gene_accuracy[bigg_model_gene_accuracy['taxonomy_name'] == 'Homo sapiens']\n",
    "human[human['recall'] != 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Number of genes with zero recall: 3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model</th>\n",
       "      <th>taxonomy</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>name_length</th>\n",
       "      <th>predictions</th>\n",
       "      <th>num_candidates</th>\n",
       "      <th>match_score</th>\n",
       "      <th>recall</th>\n",
       "      <th>precision</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>processing_time</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5117</th>\n",
       "      <td>iML1515.xml</td>\n",
       "      <td>ecoli_mg1655</td>\n",
       "      <td>G_b0624</td>\n",
       "      <td>flc</td>\n",
       "      <td>3</td>\n",
       "      <td>[(946563, np.float64(1.0))]</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.941852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5159</th>\n",
       "      <td>iML1515.xml</td>\n",
       "      <td>ecoli_mg1655</td>\n",
       "      <td>G_b2245</td>\n",
       "      <td>rhmA</td>\n",
       "      <td>4</td>\n",
       "      <td>[(948402, np.float64(1.0))]</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.941852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5177</th>\n",
       "      <td>iML1515.xml</td>\n",
       "      <td>ecoli_mg1655</td>\n",
       "      <td>G_b4104</td>\n",
       "      <td>phnE</td>\n",
       "      <td>4</td>\n",
       "      <td>[(38094977, np.float64(1.0))]</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.941852</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            model      taxonomy  gene_id gene_name  name_length  \\\n",
       "5117  iML1515.xml  ecoli_mg1655  G_b0624       flc            3   \n",
       "5159  iML1515.xml  ecoli_mg1655  G_b2245      rhmA            4   \n",
       "5177  iML1515.xml  ecoli_mg1655  G_b4104      phnE            4   \n",
       "\n",
       "                        predictions  num_candidates  match_score  recall  \\\n",
       "5117    [(946563, np.float64(1.0))]               1          1.0     0.0   \n",
       "5159    [(948402, np.float64(1.0))]               1          1.0     0.0   \n",
       "5177  [(38094977, np.float64(1.0))]               1          1.0     0.0   \n",
       "\n",
       "      precision  accuracy  processing_time  \n",
       "5117        0.0         0         1.941852  \n",
       "5159        0.0         0         1.941852  \n",
       "5177        0.0         0         1.941852  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Find rows where recall is 0\n",
    "zero_recall_df = bigg_model_gene_accuracy[bigg_model_gene_accuracy['recall'] == 0]\n",
    "print(\"\\n\\nNumber of genes with zero recall: %d\" % len(zero_recall_df))\n",
    "zero_recall_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Biomodels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['BIOMD0000001062.xml']\n",
      "We are at 0/1: Processing BIOMD0000001062.xml\n"
     ]
    }
   ],
   "source": [
    "dir = \"/Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/ecoli_mg1655\"\n",
    "ecoli_mg1655_models = [f for f in os.listdir(dir) if f.endswith('.xml')]\n",
    "output_csv = output_dir + \"biomd_model_gene_accuracy_ecoli.csv\"\n",
    "generate_gene_accuracy(dir, output_csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of models assessed: 1\n",
      "Number of genes assessed: 1371\n",
      "Average recall (per model) : 0.998\n",
      "Average precision (per model) : 0.706\n",
      "Average match score (per model): 1.000\n",
      "Average accuracy (per model): 0.998\n",
      "Ave. processing time (per model): 1.596\n",
      "Ave. processing time (per element, per model): 0.001\n",
      "Average number of predictions per row: 1.9438366156090445\n"
     ]
    }
   ],
   "source": [
    "biomd_model_gene_accuracy_ecoli = pd.read_csv(output_dir + \"biomd_model_gene_accuracy_ecoli.csv\")\n",
    "print_model_accuracy(biomd_model_gene_accuracy_ecoli)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Number of genes with zero recall: 3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model</th>\n",
       "      <th>taxonomy</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>name_length</th>\n",
       "      <th>predictions</th>\n",
       "      <th>predictions_labels</th>\n",
       "      <th>num_candidates</th>\n",
       "      <th>match_score</th>\n",
       "      <th>recall</th>\n",
       "      <th>precision</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>processing_time</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1278</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>ecoli_mg1655</td>\n",
       "      <td>G_ECOLIN_RS03405</td>\n",
       "      <td>flc</td>\n",
       "      <td>3</td>\n",
       "      <td>[(946563, np.float64(1.0))]</td>\n",
       "      <td>['fcl']</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.596149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1308</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>ecoli_mg1655</td>\n",
       "      <td>G_ECOLIN_RS13085</td>\n",
       "      <td>rhmA</td>\n",
       "      <td>4</td>\n",
       "      <td>[(948402, np.float64(1.0))]</td>\n",
       "      <td>['rhaM']</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.596149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1355</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>ecoli_mg1655</td>\n",
       "      <td>G_ECOLIN_RS23755</td>\n",
       "      <td>phnE</td>\n",
       "      <td>4</td>\n",
       "      <td>[(38094977, np.float64(1.0))]</td>\n",
       "      <td>['phnE']</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.596149</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    model      taxonomy           gene_id gene_name  \\\n",
       "1278  BIOMD0000001062.xml  ecoli_mg1655  G_ECOLIN_RS03405       flc   \n",
       "1308  BIOMD0000001062.xml  ecoli_mg1655  G_ECOLIN_RS13085      rhmA   \n",
       "1355  BIOMD0000001062.xml  ecoli_mg1655  G_ECOLIN_RS23755      phnE   \n",
       "\n",
       "      name_length                    predictions predictions_labels  \\\n",
       "1278            3    [(946563, np.float64(1.0))]            ['fcl']   \n",
       "1308            4    [(948402, np.float64(1.0))]           ['rhaM']   \n",
       "1355            4  [(38094977, np.float64(1.0))]           ['phnE']   \n",
       "\n",
       "      num_candidates  match_score  recall  precision  accuracy  \\\n",
       "1278               1          1.0     0.0        0.0         0   \n",
       "1308               1          1.0     0.0        0.0         0   \n",
       "1355               1          1.0     0.0        0.0         0   \n",
       "\n",
       "      processing_time  \n",
       "1278         1.596149  \n",
       "1308         1.596149  \n",
       "1355         1.596149  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Find rows where recall is 0\n",
    "zero_recall_df = biomd_model_gene_accuracy_ecoli[biomd_model_gene_accuracy_ecoli['recall'] == 0]\n",
    "print(\"\\n\\nNumber of genes with zero recall: %d\" % len(zero_recall_df))\n",
    "zero_recall_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Statistics of Gene annotations\n",
    "### Bigg Models "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getOntologyFromString(string_annotation,\n",
    "                          bqbiol_qualifiers=['is', 'isVersionOf']):\n",
    "  \"\"\"\n",
    "  Parse string and return string annotation,\n",
    "  marked as <bqbiol:is> or <bqbiol:isVersionOf>.\n",
    "  If neither exists, return None.\n",
    "\n",
    "  Parameters\n",
    "  ----------\n",
    "  string_annotation: str\n",
    "  bqbiol_qualifiers: str-list\n",
    "      Use 'is' and 'isVersionOf' by default\n",
    "  \n",
    "\n",
    "  Returns\n",
    "  -------\n",
    "  list-tuple (ontology type, ontology id)\n",
    "       Return [] if none is provided\n",
    "  \"\"\"\n",
    "  combined_str = ''\n",
    "  for one_qualifier in bqbiol_qualifiers:\n",
    "    one_match = '<bqbiol:' + one_qualifier + \\\n",
    "                '[^a-zA-Z].*?<\\/bqbiol:' + \\\n",
    "                one_qualifier + '>'\n",
    "    one_matched = re.findall(one_match,\n",
    "                  string_annotation,\n",
    "                  flags=re.DOTALL)\n",
    "    if len(one_matched)>0:\n",
    "      matched_filt = [s.replace(\"      \", \"\") for s in one_matched]\n",
    "      one_str = '\\n'.join(matched_filt) \n",
    "    else:\n",
    "      one_str = ''\n",
    "    combined_str = combined_str + one_str\n",
    "  identifiers_list = re.findall('identifiers\\.org/.*/', combined_str)\n",
    "  result_identifiers = [(r.split('/')[1],r.split('/')[2].replace('\\\"', '')) \\\n",
    "                        for r in identifiers_list]\n",
    "  return result_identifiers\n",
    "\n",
    "def collect_metabolites_statistics(model_dir, output_csv, model_ontology_csv):\n",
    "    \"\"\"\n",
    "    Collect statistics of ontology types from SBML models in a folder.\n",
    "\n",
    "    Parameters:\n",
    "    - folder_path (str): Path to the folder containing SBML models.\n",
    "    - output_csv (str): Path to save overall ontology statistics as a CSV.\n",
    "    - model_ontology_csv (str): Path to save per-model ontology statistics as a CSV.\n",
    "    \"\"\"\n",
    "    overall_counter = Counter()\n",
    "    model_data = []\n",
    "\n",
    "    # Iterate over all SBML files in the folder\n",
    "    model_files = [f for f in os.listdir(model_dir) if f.endswith('.xml')]\n",
    "\n",
    "    for idx, file_name in enumerate(model_files):\n",
    "      print(f\"We are at {idx}: Processing {file_name}\")\n",
    "\n",
    "      # Read the SBML model\n",
    "      file_path = os.path.join(model_dir, file_name)\n",
    "      reader = libsbml.SBMLReader()\n",
    "      document = reader.readSBML(file_path)\n",
    "      model = document.getModel()\n",
    "\n",
    "      if model is None:\n",
    "          print(f\"Skipping {file_name} (invalid model)\")\n",
    "          continue\n",
    "\n",
    "      # Count ontology types for the current model\n",
    "      model_counter = Counter()\n",
    "\n",
    "      # Iterate over all species in the model\n",
    "      for species in model.getListOfSpecies():\n",
    "          annotation_string = species.getAnnotationString()\n",
    "          resources = getOntologyFromString(annotation_string)\n",
    "\n",
    "          # Count ontology types\n",
    "          for ontology_type, ontology_id in resources:\n",
    "              overall_counter[ontology_type] += 1\n",
    "              model_counter[ontology_type] += 1\n",
    "\n",
    "      # Store model-level data\n",
    "      for ontology_type, count in model_counter.items():\n",
    "          model_data.append({\n",
    "              'model': file_name,\n",
    "              'ontology_type': ontology_type,\n",
    "              'count': count\n",
    "          })\n",
    "\n",
    "    # Convert overall statistics to a DataFrame\n",
    "    overall_stats_df = pd.DataFrame.from_dict(overall_counter, orient='index', columns=['count'])\n",
    "    overall_stats_df.reset_index(inplace=True)\n",
    "    overall_stats_df.columns = ['ontology_type', 'count']\n",
    "\n",
    "    # Convert model-specific data to a DataFrame\n",
    "    model_stats_df = pd.DataFrame(model_data)\n",
    "\n",
    "    # Save the dataframes to CSV\n",
    "    overall_stats_df.to_csv(output_csv, index=False)\n",
    "    model_stats_df.to_csv(model_ontology_csv, index=False)\n",
    "    print(f\"Overall ontology statistics saved to {output_csv}\")\n",
    "    print(f\"Model-specific ontology statistics saved to {model_ontology_csv}\")\n",
    "\n",
    "\n",
    "def get_gene_annotation_string(sbml_file):\n",
    "    \"\"\"\n",
    "    Extract GeneProducts annotatino string from an SBML model.\n",
    "\n",
    "    Parameters:\n",
    "    - sbml_file (str): Path to the SBML file.\n",
    "\n",
    "    Returns:\n",
    "    - list: List of GeneProduct annotation strings.\n",
    "    \"\"\"\n",
    "    # Read the SBML model\n",
    "    reader = libsbml.SBMLReader()\n",
    "    document = reader.readSBML(sbml_file)\n",
    "    model = document.getModel()\n",
    "    if model is None:\n",
    "        print(f\"Failed to read the model: {sbml_file}\")\n",
    "        return []\n",
    "        \n",
    "    # Check if the model uses the FBC package\n",
    "    fbc_plugin = model.getPlugin(\"fbc\")\n",
    "    if not fbc_plugin:\n",
    "        print(f\"The model does not use the FBC package: {sbml_file}\")\n",
    "        return []\n",
    "\n",
    "    # Get GeneProducts from the FBC plugin\n",
    "    gene_annotations = []\n",
    "    for i in range(fbc_plugin.getNumGeneProducts()):\n",
    "        gene_product = fbc_plugin.getGeneProduct(i)\n",
    "        annotation_string = gene_product.getAnnotationString()\n",
    "        gene_annotations.append(annotation_string)\n",
    "    return gene_annotations\n",
    "\n",
    "def collect_gene_product_statistics(model_dir, overall_output_csv, model_ontology_csv):\n",
    "    \"\"\"\n",
    "    Collect statistics of GeneProducts from SBML models in a folder.\n",
    "\n",
    "    Parameters:\n",
    "    - folder_path (str): Path to the folder containing SBML models.\n",
    "    - overall_output_csv (str): Path to save overall GeneProduct statistics as a CSV.\n",
    "    - model_ontology_csv (str): Path to save per-model GeneProduct statistics as a CSV.\n",
    "    \"\"\"\n",
    "    overall_counter = Counter()\n",
    "    model_data = []\n",
    "\n",
    "    # Iterate over all SBML files in the folder\n",
    "    model_files = [f for f in os.listdir(model_dir) if f.endswith('.xml')]\n",
    "    for idx, file_name in enumerate(model_files):\n",
    "      print(f\"We are at {idx}: Processing {file_name}\")\n",
    "      file_path = os.path.join(model_dir, file_name)\n",
    "      gene_annotations = get_gene_annotation_string(file_path)\n",
    "\n",
    "      model_counter = Counter()\n",
    "      for gene_annotation in gene_annotations:\n",
    "        resources = getOntologyFromString(gene_annotation)\n",
    "\n",
    "        # Count ontology types\n",
    "        for ontology_type, ontology_id in resources:\n",
    "            overall_counter[ontology_type] += 1\n",
    "            model_counter[ontology_type] += 1\n",
    "\n",
    "      # Store model-level data\n",
    "      for ontology_type, count in model_counter.items():\n",
    "        model_data.append({\n",
    "            'model': file_name,\n",
    "            'ontology_type': ontology_type,\n",
    "            'count': count\n",
    "        })\n",
    "\n",
    "    # Convert overall statistics to a DataFrame\n",
    "    overall_stats_df = pd.DataFrame.from_dict(overall_counter, orient='index', columns=['count'])\n",
    "    overall_stats_df.reset_index(inplace=True)\n",
    "    overall_stats_df.columns = ['ontology_type', 'count']\n",
    "\n",
    "    # Convert model-specific data to a DataFrame\n",
    "    model_stats_df = pd.DataFrame(model_data)\n",
    "\n",
    "    # Save the dataframes to CSV\n",
    "    overall_stats_df.to_csv(output_csv, index=False)\n",
    "    model_stats_df.to_csv(model_ontology_csv, index=False)\n",
    "    print(f\"Overall GeneProduct statistics saved to {overall_output_csv}\")\n",
    "    print(f\"Model-specific GeneProduct statistics saved to {model_ontology_csv}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We are at 0\n",
      "We are at 1\n",
      "We are at 2\n",
      "We are at 3\n",
      "We are at 4\n",
      "We are at 5\n",
      "We are at 6\n",
      "We are at 7\n",
      "We are at 8\n",
      "We are at 9\n",
      "We are at 10\n",
      "We are at 11\n",
      "We are at 12\n",
      "We are at 13\n",
      "We are at 14\n",
      "We are at 15\n",
      "We are at 16\n",
      "We are at 17\n",
      "We are at 18\n",
      "We are at 19\n",
      "We are at 20\n",
      "We are at 21\n",
      "We are at 22\n",
      "We are at 23\n",
      "We are at 24\n",
      "We are at 25\n",
      "We are at 26\n",
      "We are at 27\n",
      "We are at 28\n",
      "We are at 29\n",
      "We are at 30\n",
      "We are at 31\n",
      "We are at 32\n",
      "We are at 33\n",
      "We are at 34\n",
      "We are at 35\n",
      "We are at 36\n",
      "We are at 37\n",
      "We are at 38\n",
      "We are at 39\n",
      "We are at 40\n",
      "We are at 41\n",
      "We are at 42\n",
      "We are at 43\n",
      "We are at 44\n",
      "We are at 45\n",
      "We are at 46\n",
      "We are at 47\n",
      "We are at 48\n",
      "We are at 49\n",
      "We are at 50\n",
      "We are at 51\n",
      "We are at 52\n",
      "We are at 53\n",
      "We are at 54\n",
      "We are at 55\n",
      "We are at 56\n",
      "We are at 57\n",
      "We are at 58\n",
      "We are at 59\n",
      "We are at 60\n",
      "We are at 61\n",
      "We are at 62\n",
      "We are at 63\n",
      "We are at 64\n",
      "We are at 65\n",
      "We are at 66\n",
      "We are at 67\n",
      "We are at 68\n",
      "We are at 69\n",
      "We are at 70\n",
      "We are at 71\n",
      "We are at 72\n",
      "We are at 73\n",
      "We are at 74\n",
      "We are at 75\n",
      "We are at 76\n",
      "We are at 77\n",
      "We are at 78\n",
      "We are at 79\n",
      "We are at 80\n",
      "We are at 81\n",
      "We are at 82\n",
      "We are at 83\n",
      "We are at 84\n",
      "We are at 85\n",
      "We are at 86\n",
      "We are at 87\n",
      "We are at 88\n",
      "We are at 89\n",
      "We are at 90\n",
      "We are at 91\n",
      "We are at 92\n",
      "We are at 93\n",
      "We are at 94\n",
      "We are at 95\n",
      "We are at 96\n",
      "We are at 97\n",
      "We are at 98\n",
      "We are at 99\n",
      "We are at 100\n",
      "We are at 101\n",
      "We are at 102\n",
      "We are at 103\n",
      "We are at 104\n",
      "We are at 105\n",
      "We are at 106\n",
      "We are at 107\n",
      "Overall ontology statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_ontology_statistics.csv\n",
      "Model-specific ontology statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_model_ontology_statistics.csv\n"
     ]
    }
   ],
   "source": [
    "output_csv = \"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_ontology_statistics.csv\"\n",
    "model_ontology_csv = \"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_model_ontology_statistics.csv\"\n",
    "collect_metabolites_statistics(bigg_model_dir, output_csv, model_ontology_csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ontology_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chebi</td>\n",
       "      <td>881182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>bigg.metabolite</td>\n",
       "      <td>180802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>seed.compound</td>\n",
       "      <td>177255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>reactome.compound</td>\n",
       "      <td>176773</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>metanetx.chemical</td>\n",
       "      <td>173772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>biocyc</td>\n",
       "      <td>151278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>hmdb</td>\n",
       "      <td>142708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>kegg.compound</td>\n",
       "      <td>140296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>inchi_key</td>\n",
       "      <td>134440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>sabiork</td>\n",
       "      <td>124143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>envipath</td>\n",
       "      <td>49436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>kegg.drug</td>\n",
       "      <td>30210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>lipidmaps</td>\n",
       "      <td>25793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>slm</td>\n",
       "      <td>10172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>kegg.glycan</td>\n",
       "      <td>6042</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        ontology_type   count\n",
       "3               chebi  881182\n",
       "0     bigg.metabolite  180802\n",
       "10      seed.compound  177255\n",
       "11  reactome.compound  176773\n",
       "1   metanetx.chemical  173772\n",
       "2              biocyc  151278\n",
       "5                hmdb  142708\n",
       "7       kegg.compound  140296\n",
       "6           inchi_key  134440\n",
       "9             sabiork  124143\n",
       "4            envipath   49436\n",
       "13          kegg.drug   30210\n",
       "8           lipidmaps   25793\n",
       "12                slm   10172\n",
       "14        kegg.glycan    6042"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bigg_metabolites_ontology_statistics = pd.read_csv(\"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_metabolites_ontology_statistics.csv\")\n",
    "bigg_metabolites_ontology_statistics.sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We are at 0: Processing iCHOv1.xml\n",
      "We are at 1: Processing iYL1228.xml\n",
      "We are at 2: Processing iIS312.xml\n",
      "We are at 3: Processing iSDY_1059.xml\n",
      "We are at 4: Processing iEC1356_Bl21DE3.xml\n",
      "We are at 5: Processing iECP_1309.xml\n",
      "We are at 6: Processing iJN746.xml\n",
      "We are at 7: Processing iJR904.xml\n",
      "We are at 8: Processing iJB785.xml\n",
      "We are at 9: Processing iY75_1357.xml\n",
      "We are at 10: Processing iEcHS_1320.xml\n",
      "We are at 11: Processing iUTI89_1310.xml\n",
      "We are at 12: Processing iB21_1397.xml\n",
      "We are at 13: Processing iSbBS512_1146.xml\n",
      "We are at 14: Processing iNF517.xml\n",
      "We are at 15: Processing iNJ661.xml\n",
      "We are at 16: Processing iECIAI39_1322.xml\n",
      "Failed to read the model: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BiggModels/iECIAI39_1322.xml\n",
      "We are at 17: Processing iECO103_1326.xml\n",
      "We are at 18: Processing iSSON_1240.xml\n",
      "We are at 19: Processing iCHOv1_DG44.xml\n",
      "We are at 20: Processing iAB_RBC_283.xml\n",
      "We are at 21: Processing iAM_Pk459.xml\n",
      "We are at 22: Processing iPC815.xml\n",
      "We are at 23: Processing iAPECO1_1312.xml\n",
      "We are at 24: Processing iMM1415.xml\n",
      "We are at 25: Processing iECDH1ME8569_1439.xml\n",
      "We are at 26: Processing iJO1366.xml\n",
      "We are at 27: Processing iEcE24377_1341.xml\n",
      "We are at 28: Processing iEC042_1314.xml\n",
      "We are at 29: Processing iRC1080.xml\n",
      "We are at 30: Processing iCN900.xml\n",
      "We are at 31: Processing iEC1364_W.xml\n",
      "We are at 32: Processing iJN678.xml\n",
      "We are at 33: Processing iEcSMS35_1347.xml\n",
      "We are at 34: Processing iSBO_1134.xml\n",
      "We are at 35: Processing STM_v1_0.xml\n",
      "We are at 36: Processing iAF1260.xml\n",
      "We are at 37: Processing iJN1463.xml\n",
      "We are at 38: Processing iS_1188.xml\n",
      "We are at 39: Processing iML1515.xml\n",
      "We are at 40: Processing iCN718.xml\n",
      "We are at 41: Processing iAF692.xml\n",
      "We are at 42: Processing iETEC_1333.xml\n",
      "We are at 43: Processing iECD_1391.xml\n",
      "We are at 44: Processing iHN637.xml\n",
      "We are at 45: Processing iG2583_1286.xml\n",
      "We are at 46: Processing iMM904.xml\n",
      "We are at 47: Processing Recon3D.xml\n",
      "We are at 48: Processing iAM_Pf480.xml\n",
      "We are at 49: Processing e_coli_core.xml\n",
      "We are at 50: Processing iAT_PLT_636.xml\n",
      "We are at 51: Processing iEC1344_C.xml\n",
      "We are at 52: Processing iWFL_1372.xml\n",
      "We are at 53: Processing iECABU_c1320.xml\n",
      "We are at 54: Processing iBWG_1329.xml\n",
      "We are at 55: Processing iIT341.xml\n",
      "We are at 56: Processing iEcolC_1368.xml\n",
      "We are at 57: Processing iECs_1301.xml\n",
      "We are at 58: Processing iZ_1308.xml\n",
      "We are at 59: Processing iSF_1195.xml\n",
      "We are at 60: Processing iND750.xml\n",
      "We are at 61: Processing iECB_1328.xml\n",
      "We are at 62: Processing iYO844.xml\n",
      "We are at 63: Processing iECIAI1_1343.xml\n",
      "We are at 64: Processing iLJ478.xml\n",
      "We are at 65: Processing iIS312_Amastigote.xml\n",
      "We are at 66: Processing RECON1.xml\n",
      "We are at 67: Processing iECBD_1354.xml\n",
      "We are at 68: Processing iEC1349_Crooks.xml\n",
      "We are at 69: Processing iSFxv_1172.xml\n",
      "We are at 70: Processing iECUMN_1333.xml\n",
      "We are at 71: Processing iECS88_1305.xml\n",
      "We are at 72: Processing iEC1368_DH5a.xml\n",
      "We are at 73: Processing iEcDH1_1363.xml\n",
      "We are at 74: Processing iAM_Pc455.xml\n",
      "We are at 75: Processing iUMN146_1321.xml\n",
      "We are at 76: Processing iECSP_1301.xml\n",
      "We are at 77: Processing iIS312_Epimastigote.xml\n",
      "We are at 78: Processing iLF82_1304.xml\n",
      "We are at 79: Processing iLB1027_lipid.xml\n",
      "We are at 80: Processing iECH74115_1262.xml\n",
      "We are at 81: Processing iE2348C_1286.xml\n",
      "We are at 82: Processing iECOK1_1307.xml\n",
      "We are at 83: Processing iUMNK88_1353.xml\n",
      "We are at 84: Processing iECSE_1348.xml\n",
      "We are at 85: Processing iEK1008.xml\n",
      "We are at 86: Processing iECSF_1327.xml\n",
      "We are at 87: Processing iYS854.xml\n",
      "We are at 88: Processing iSFV_1184.xml\n",
      "We are at 89: Processing iEC1372_W3110.xml\n",
      "We are at 90: Processing ic_1306.xml\n",
      "We are at 91: Processing iECED1_1282.xml\n",
      "We are at 92: Processing iECW_1372.xml\n",
      "We are at 93: Processing iSynCJ816.xml\n",
      "We are at 94: Processing iIS312_Trypomastigote.xml\n",
      "We are at 95: Processing iECDH10B_1368.xml\n",
      "We are at 96: Processing iYS1720.xml\n",
      "We are at 97: Processing iECNA114_1301.xml\n",
      "We are at 98: Processing iAM_Pb448.xml\n",
      "We are at 99: Processing iAM_Pv461.xml\n",
      "We are at 100: Processing iAF987.xml\n",
      "We are at 101: Processing iECO111_1330.xml\n",
      "We are at 102: Processing iSB619.xml\n",
      "We are at 103: Processing iNRG857_1313.xml\n",
      "We are at 104: Processing iECO26_1355.xml\n",
      "We are at 105: Processing iAF1260b.xml\n",
      "We are at 106: Processing iEC55989_1330.xml\n",
      "We are at 107: Processing iEKO11_1354.xml\n",
      "Overall GeneProduct statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_genes_ontology_statistics.csv\n",
      "Model-specific GeneProduct statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_genes_model_ontology_statistics.csv\n"
     ]
    }
   ],
   "source": [
    "output_csv = \"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_genes_ontology_statistics.csv\"\n",
    "model_ontology_csv = \"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_genes_model_ontology_statistics.csv\"\n",
    "collect_gene_product_statistics(bigg_model_dir, output_csv, model_ontology_csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ontology_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ncbigi</td>\n",
       "      <td>102867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>refseq_locus_tag</td>\n",
       "      <td>101721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>refseq_name</td>\n",
       "      <td>77574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>interpro</td>\n",
       "      <td>70094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>refseq_synonym</td>\n",
       "      <td>45041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ncbigene</td>\n",
       "      <td>25336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>refseq_old_locus_tag</td>\n",
       "      <td>23578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>uniprot</td>\n",
       "      <td>18857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>goa</td>\n",
       "      <td>14134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>EnsemblGenomes-Gn</td>\n",
       "      <td>12622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>EnsemblGenomes-Tr</td>\n",
       "      <td>12605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ccds</td>\n",
       "      <td>11416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>asap</td>\n",
       "      <td>7728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ecogene</td>\n",
       "      <td>6432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>omim</td>\n",
       "      <td>4903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>hprd</td>\n",
       "      <td>2776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>sgd</td>\n",
       "      <td>1646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>go</td>\n",
       "      <td>1335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>subtilist</td>\n",
       "      <td>843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>refseq_orf_id</td>\n",
       "      <td>743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>pdb</td>\n",
       "      <td>550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PSEUDO</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           ontology_type   count\n",
       "1                 ncbigi  102867\n",
       "4       refseq_locus_tag  101721\n",
       "2            refseq_name   77574\n",
       "13              interpro   70094\n",
       "3         refseq_synonym   45041\n",
       "0               ncbigene   25336\n",
       "5   refseq_old_locus_tag   23578\n",
       "9                uniprot   18857\n",
       "12                   goa   14134\n",
       "10     EnsemblGenomes-Gn   12622\n",
       "11     EnsemblGenomes-Tr   12605\n",
       "15                  ccds   11416\n",
       "7                   asap    7728\n",
       "8                ecogene    6432\n",
       "17                  omim    4903\n",
       "16                  hprd    2776\n",
       "18                   sgd    1646\n",
       "21                    go    1335\n",
       "19             subtilist     843\n",
       "6          refseq_orf_id     743\n",
       "14                   pdb     550\n",
       "20                PSEUDO       1"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bigg_genes_ontology_statistics = pd.read_csv(\"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_genes_ontology_statistics.csv\")\n",
    "bigg_genes_ontology_statistics.sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### BioModels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We are at 0: Processing BIOMD0000000634.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000634.xml\n",
      "We are at 1: Processing BIOMD0000000152.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000152.xml\n",
      "We are at 2: Processing BIOMD0000000146.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000146.xml\n",
      "We are at 3: Processing BIOMD0000000620.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000620.xml\n",
      "We are at 4: Processing BIOMD0000000608.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000608.xml\n",
      "We are at 5: Processing BIOMD0000000191.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000191.xml\n",
      "We are at 6: Processing BIOMD0000000185.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000185.xml\n",
      "We are at 7: Processing BIOMD0000000813.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000813.xml\n",
      "We are at 8: Processing BIOMD0000000807.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000807.xml\n",
      "We are at 9: Processing BIOMD0000000350.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000350.xml\n",
      "We are at 10: Processing BIOMD0000000436.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000436.xml\n",
      "We are at 11: Processing BIOMD0000000422.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000422.xml\n",
      "We are at 12: Processing BIOMD0000000344.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000344.xml\n",
      "We are at 13: Processing BIOMD0000001072.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001072.xml\n",
      "We are at 14: Processing BIOMD0000000378.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000378.xml\n",
      "We are at 15: Processing BIOMD0000000393.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000393.xml\n",
      "We are at 16: Processing BIOMD0000000387.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000387.xml\n",
      "We are at 17: Processing BIOMD0000000224.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000224.xml\n",
      "We are at 18: Processing BIOMD0000000542.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000542.xml\n",
      "We are at 19: Processing BIOMD0000000556.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000556.xml\n",
      "We are at 20: Processing BIOMD0000000230.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000230.xml\n",
      "We are at 21: Processing BIOMD0000000218.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000218.xml\n",
      "We are at 22: Processing BIOMD0000000581.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000581.xml\n",
      "We are at 23: Processing BIOMD0000000595.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000595.xml\n",
      "We are at 24: Processing BIOMD0000000740.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000740.xml\n",
      "We are at 25: Processing BIOMD0000000998.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000998.xml\n",
      "We are at 26: Processing BIOMD0000000026.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000026.xml\n",
      "We are at 27: Processing BIOMD0000000032.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000032.xml\n",
      "We are at 28: Processing BIOMD0000000754.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000754.xml\n",
      "We are at 29: Processing BIOMD0000000768.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000768.xml\n",
      "We are at 30: Processing BIOMD0000000783.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000783.xml\n",
      "We are at 31: Processing BIOMD0000000797.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000797.xml\n",
      "We are at 32: Processing BIOMD0000000967.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000967.xml\n",
      "We are at 33: Processing BIOMD0000000973.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000973.xml\n",
      "We are at 34: Processing BIOMD0000000972.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000972.xml\n",
      "We are at 35: Processing BIOMD0000000966.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000966.xml\n",
      "We are at 36: Processing BIOMD0000000796.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000796.xml\n",
      "We are at 37: Processing BIOMD0000000782.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000782.xml\n",
      "We are at 38: Processing BIOMD0000000769.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000769.xml\n",
      "We are at 39: Processing BIOMD0000000033.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000033.xml\n",
      "We are at 40: Processing BIOMD0000000755.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000755.xml\n",
      "We are at 41: Processing BIOMD0000000999.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000999.xml\n",
      "We are at 42: Processing BIOMD0000000741.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000741.xml\n",
      "We are at 43: Processing BIOMD0000000027.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000027.xml\n",
      "We are at 44: Processing BIOMD0000000594.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000594.xml\n",
      "We are at 45: Processing BIOMD0000000580.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000580.xml\n",
      "We are at 46: Processing BIOMD0000000219.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000219.xml\n",
      "We are at 47: Processing BIOMD0000000557.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000557.xml\n",
      "We are at 48: Processing BIOMD0000000231.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000231.xml\n",
      "We are at 49: Processing BIOMD0000000225.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000225.xml\n",
      "We are at 50: Processing BIOMD0000000543.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000543.xml\n",
      "We are at 51: Processing BIOMD0000000386.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000386.xml\n",
      "We are at 52: Processing BIOMD0000000392.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000392.xml\n",
      "We are at 53: Processing BIOMD0000000379.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000379.xml\n",
      "We are at 54: Processing BIOMD0000000423.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000423.xml\n",
      "We are at 55: Processing BIOMD0000000345.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000345.xml\n",
      "We are at 56: Processing BIOMD0000000351.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000351.xml\n",
      "We are at 57: Processing BIOMD0000000437.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000437.xml\n",
      "We are at 58: Processing BIOMD0000000806.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000806.xml\n",
      "We are at 59: Processing BIOMD0000000812.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000812.xml\n",
      "We are at 60: Processing BIOMD0000000184.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000184.xml\n",
      "We are at 61: Processing BIOMD0000000190.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000190.xml\n",
      "We are at 62: Processing BIOMD0000000609.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000609.xml\n",
      "We are at 63: Processing BIOMD0000000147.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000147.xml\n",
      "We are at 64: Processing BIOMD0000000621.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000621.xml\n",
      "We are at 65: Processing BIOMD0000000635.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000635.xml\n",
      "We are at 66: Processing BIOMD0000000153.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000153.xml\n",
      "We are at 67: Processing BIOMD0000000623.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000623.xml\n",
      "We are at 68: Processing BIOMD0000000145.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000145.xml\n",
      "We are at 69: Processing BIOMD0000000151.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000151.xml\n",
      "We are at 70: Processing BIOMD0000000637.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000637.xml\n",
      "We are at 71: Processing BIOMD0000000179.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000179.xml\n",
      "We are at 72: Processing BIOMD0000000838.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000838.xml\n",
      "We are at 73: Processing BIOMD0000000186.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000186.xml\n",
      "We are at 74: Processing BIOMD0000000192.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000192.xml\n",
      "We are at 75: Processing BIOMD0000000804.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000804.xml\n",
      "We are at 76: Processing BIOMD0000000810.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000810.xml\n",
      "We are at 77: Processing BIOMD0000001059.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001059.xml\n",
      "We are at 78: Processing BIOMD0000000347.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000347.xml\n",
      "We are at 79: Processing BIOMD0000000421.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000421.xml\n",
      "We are at 80: Processing BIOMD0000000435.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000435.xml\n",
      "We are at 81: Processing BIOMD0000000353.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000353.xml\n",
      "We are at 82: Processing BIOMD0000001065.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001065.xml\n",
      "We are at 83: Processing BIOMD0000000409.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000409.xml\n",
      "We are at 84: Processing BIOMD0000000384.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000384.xml\n",
      "We are at 85: Processing BIOMD0000000390.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000390.xml\n",
      "We are at 86: Processing BIOMD0000000233.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000233.xml\n",
      "We are at 87: Processing BIOMD0000000555.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000555.xml\n",
      "We are at 88: Processing BIOMD0000000541.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000541.xml\n",
      "We are at 89: Processing BIOMD0000000227.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000227.xml\n",
      "We are at 90: Processing BIOMD0000000569.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000569.xml\n",
      "We are at 91: Processing BIOMD0000000596.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000596.xml\n",
      "We are at 92: Processing BIOMD0000000582.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000582.xml\n",
      "We are at 93: Processing BIOMD0000000757.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000757.xml\n",
      "We are at 94: Processing BIOMD0000000031.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000031.xml\n",
      "We are at 95: Processing BIOMD0000000025.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000025.xml\n",
      "We are at 96: Processing BIOMD0000000743.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000743.xml\n",
      "We are at 97: Processing BIOMD0000000019.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000019.xml\n",
      "We are at 98: Processing BIOMD0000000794.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000794.xml\n",
      "We are at 99: Processing BIOMD0000000958.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000958.xml\n",
      "We are at 100: Processing BIOMD0000000780.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000780.xml\n",
      "We are at 101: Processing BIOMD0000000970.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000970.xml\n",
      "We are at 102: Processing BIOMD0000000964.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000964.xml\n",
      "We are at 103: Processing BIOMD0000000965.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000965.xml\n",
      "We are at 104: Processing BIOMD0000000971.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000971.xml\n",
      "We are at 105: Processing BIOMD0000000781.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000781.xml\n",
      "We are at 106: Processing BIOMD0000000959.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000959.xml\n",
      "We are at 107: Processing BIOMD0000000795.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000795.xml\n",
      "We are at 108: Processing BIOMD0000000018.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000018.xml\n",
      "We are at 109: Processing BIOMD0000000024.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000024.xml\n",
      "We are at 110: Processing BIOMD0000000742.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000742.xml\n",
      "We are at 111: Processing BIOMD0000000756.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000756.xml\n",
      "We are at 112: Processing BIOMD0000000030.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000030.xml\n",
      "We are at 113: Processing BIOMD0000000583.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000583.xml\n",
      "We are at 114: Processing BIOMD0000000597.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000597.xml\n",
      "We are at 115: Processing BIOMD0000000568.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000568.xml\n",
      "We are at 116: Processing BIOMD0000000540.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000540.xml\n",
      "We are at 117: Processing BIOMD0000000226.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000226.xml\n",
      "We are at 118: Processing BIOMD0000000232.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000232.xml\n",
      "We are at 119: Processing BIOMD0000000554.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000554.xml\n",
      "We are at 120: Processing BIOMD0000000391.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000391.xml\n",
      "We are at 121: Processing BIOMD0000000385.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000385.xml\n",
      "We are at 122: Processing BIOMD0000000408.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000408.xml\n",
      "We are at 123: Processing BIOMD0000001064.xml\n",
      "We are at 124: Processing BIOMD0000000434.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000434.xml\n",
      "We are at 125: Processing BIOMD0000000352.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000352.xml\n",
      "We are at 126: Processing BIOMD0000000346.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000346.xml\n",
      "We are at 127: Processing BIOMD0000001058.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001058.xml\n",
      "We are at 128: Processing BIOMD0000000420.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000420.xml\n",
      "We are at 129: Processing BIOMD0000000811.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000811.xml\n",
      "We are at 130: Processing BIOMD0000000805.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000805.xml\n",
      "We are at 131: Processing BIOMD0000000193.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000193.xml\n",
      "We are at 132: Processing BIOMD0000000839.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000839.xml\n",
      "We are at 133: Processing BIOMD0000000187.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000187.xml\n",
      "We are at 134: Processing BIOMD0000000178.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000178.xml\n",
      "We are at 135: Processing BIOMD0000000150.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000150.xml\n",
      "We are at 136: Processing BIOMD0000000636.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000636.xml\n",
      "We are at 137: Processing BIOMD0000000622.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000622.xml\n",
      "We are at 138: Processing BIOMD0000000144.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000144.xml\n",
      "We are at 139: Processing BIOMD0000000168.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000168.xml\n",
      "We are at 140: Processing BIOMD0000000140.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000140.xml\n",
      "We are at 141: Processing BIOMD0000000626.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000626.xml\n",
      "We are at 142: Processing BIOMD0000000632.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000632.xml\n",
      "We are at 143: Processing BIOMD0000000154.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000154.xml\n",
      "We are at 144: Processing BIOMD0000000801.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000801.xml\n",
      "We are at 145: Processing BIOMD0000000815.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000815.xml\n",
      "We are at 146: Processing BIOMD0000000183.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000183.xml\n",
      "We are at 147: Processing BIOMD0000000829.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000829.xml\n",
      "We are at 148: Processing BIOMD0000000197.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000197.xml\n",
      "We are at 149: Processing BIOMD0000000418.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000418.xml\n",
      "We are at 150: Processing BIOMD0000001060.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001060.xml\n",
      "We are at 151: Processing BIOMD0000000424.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000424.xml\n",
      "We are at 152: Processing BIOMD0000000342.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000342.xml\n",
      "We are at 153: Processing BIOMD0000001048.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001048.xml\n",
      "We are at 154: Processing BIOMD0000000356.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000356.xml\n",
      "We are at 155: Processing BIOMD0000000430.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000430.xml\n",
      "We are at 156: Processing BIOMD0000000381.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000381.xml\n",
      "We are at 157: Processing BIOMD0000000395.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000395.xml\n",
      "We are at 158: Processing BIOMD0000000578.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000578.xml\n",
      "We are at 159: Processing BIOMD0000000550.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000550.xml\n",
      "We are at 160: Processing BIOMD0000000236.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000236.xml\n",
      "We are at 161: Processing BIOMD0000000222.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000222.xml\n",
      "We are at 162: Processing BIOMD0000000544.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000544.xml\n",
      "We are at 163: Processing BIOMD0000000593.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000593.xml\n",
      "We are at 164: Processing BIOMD0000000587.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000587.xml\n",
      "We are at 165: Processing BIOMD0000000008.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000008.xml\n",
      "We are at 166: Processing BIOMD0000000034.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000034.xml\n",
      "We are at 167: Processing BIOMD0000000752.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000752.xml\n",
      "We are at 168: Processing BIOMD0000000746.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000746.xml\n",
      "We are at 169: Processing BIOMD0000000020.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000020.xml\n",
      "We are at 170: Processing BIOMD0000000975.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000975.xml\n",
      "We are at 171: Processing BIOMD0000000961.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000961.xml\n",
      "We are at 172: Processing BIOMD0000000949.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000949.xml\n",
      "We are at 173: Processing BIOMD0000000791.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000791.xml\n",
      "We are at 174: Processing BIOMD0000000785.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000785.xml\n",
      "We are at 175: Processing BIOMD0000000784.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000784.xml\n",
      "We are at 176: Processing BIOMD0000000790.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000790.xml\n",
      "We are at 177: Processing BIOMD0000000948.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000948.xml\n",
      "We are at 178: Processing BIOMD0000000960.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000960.xml\n",
      "We are at 179: Processing BIOMD0000000974.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000974.xml\n",
      "We are at 180: Processing BIOMD0000000747.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000747.xml\n",
      "We are at 181: Processing BIOMD0000000021.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000021.xml\n",
      "We are at 182: Processing BIOMD0000000035.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000035.xml\n",
      "We are at 183: Processing BIOMD0000000753.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000753.xml\n",
      "We are at 184: Processing BIOMD0000000009.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000009.xml\n",
      "We are at 185: Processing BIOMD0000000586.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000586.xml\n",
      "We are at 186: Processing BIOMD0000000592.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000592.xml\n",
      "We are at 187: Processing BIOMD0000000223.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000223.xml\n",
      "We are at 188: Processing BIOMD0000000545.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000545.xml\n",
      "We are at 189: Processing BIOMD0000000551.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000551.xml\n",
      "We are at 190: Processing BIOMD0000000237.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000237.xml\n",
      "We are at 191: Processing BIOMD0000000579.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000579.xml\n",
      "We are at 192: Processing BIOMD0000000394.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000394.xml\n",
      "We are at 193: Processing BIOMD0000000380.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000380.xml\n",
      "We are at 194: Processing BIOMD0000000357.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000357.xml\n",
      "We are at 195: Processing BIOMD0000000431.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000431.xml\n",
      "We are at 196: Processing BIOMD0000000425.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000425.xml\n",
      "We are at 197: Processing BIOMD0000000343.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000343.xml\n",
      "We are at 198: Processing BIOMD0000000419.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000419.xml\n",
      "We are at 199: Processing BIOMD0000001061.xml\n",
      "We are at 200: Processing BIOMD0000000828.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000828.xml\n",
      "We are at 201: Processing BIOMD0000000196.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000196.xml\n",
      "We are at 202: Processing BIOMD0000000182.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000182.xml\n",
      "We are at 203: Processing BIOMD0000000814.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000814.xml\n",
      "We are at 204: Processing BIOMD0000000800.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000800.xml\n",
      "We are at 205: Processing BIOMD0000000633.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000633.xml\n",
      "We are at 206: Processing BIOMD0000000155.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000155.xml\n",
      "We are at 207: Processing BIOMD0000000141.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000141.xml\n",
      "We are at 208: Processing BIOMD0000000627.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000627.xml\n",
      "We are at 209: Processing BIOMD0000000169.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000169.xml\n",
      "We are at 210: Processing BIOMD0000000619.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000619.xml\n",
      "We are at 211: Processing BIOMD0000000157.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000157.xml\n",
      "We are at 212: Processing BIOMD0000000631.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000631.xml\n",
      "We are at 213: Processing BIOMD0000000625.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000625.xml\n",
      "We are at 214: Processing BIOMD0000000143.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000143.xml\n",
      "We are at 215: Processing BIOMD0000000816.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000816.xml\n",
      "We are at 216: Processing BIOMD0000000802.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000802.xml\n",
      "We are at 217: Processing BIOMD0000000194.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000194.xml\n",
      "We are at 218: Processing BIOMD0000000180.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000180.xml\n",
      "We are at 219: Processing BIOMD0000001077.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001077.xml\n",
      "We are at 220: Processing BIOMD0000000369.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000369.xml\n",
      "We are at 221: Processing BIOMD0000001063.xml\n",
      "We are at 222: Processing BIOMD0000000433.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000433.xml\n",
      "We are at 223: Processing BIOMD0000000355.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000355.xml\n",
      "We are at 224: Processing BIOMD0000000341.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000341.xml\n",
      "We are at 225: Processing BIOMD0000000427.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000427.xml\n",
      "We are at 226: Processing BIOMD0000000396.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000396.xml\n",
      "We are at 227: Processing BIOMD0000000382.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000382.xml\n",
      "We are at 228: Processing BIOMD0000000209.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000209.xml\n",
      "We are at 229: Processing BIOMD0000000547.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000547.xml\n",
      "We are at 230: Processing BIOMD0000000221.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000221.xml\n",
      "We are at 231: Processing BIOMD0000000235.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000235.xml\n",
      "We are at 232: Processing BIOMD0000000553.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000553.xml\n",
      "We are at 233: Processing BIOMD0000000584.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000584.xml\n",
      "We are at 234: Processing BIOMD0000000590.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000590.xml\n",
      "We are at 235: Processing BIOMD0000000779.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000779.xml\n",
      "We are at 236: Processing BIOMD0000000023.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000023.xml\n",
      "We are at 237: Processing BIOMD0000000745.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000745.xml\n",
      "We are at 238: Processing BIOMD0000000751.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000751.xml\n",
      "We are at 239: Processing BIOMD0000000989.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000989.xml\n",
      "We are at 240: Processing BIOMD0000000037.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000037.xml\n",
      "We are at 241: Processing BIOMD0000000962.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000962.xml\n",
      "We are at 242: Processing BIOMD0000000976.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000976.xml\n",
      "We are at 243: Processing BIOMD0000000786.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000786.xml\n",
      "We are at 244: Processing BIOMD0000000792.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000792.xml\n",
      "We are at 245: Processing BIOMD0000000793.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000793.xml\n",
      "We are at 246: Processing BIOMD0000000787.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000787.xml\n",
      "We are at 247: Processing BIOMD0000000977.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000977.xml\n",
      "We are at 248: Processing BIOMD0000000963.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000963.xml\n",
      "We are at 249: Processing BIOMD0000000988.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000988.xml\n",
      "We are at 250: Processing BIOMD0000000750.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000750.xml\n",
      "We are at 251: Processing BIOMD0000000036.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000036.xml\n",
      "We are at 252: Processing BIOMD0000000022.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000022.xml\n",
      "We are at 253: Processing BIOMD0000000744.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000744.xml\n",
      "We are at 254: Processing BIOMD0000000778.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000778.xml\n",
      "We are at 255: Processing BIOMD0000000591.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000591.xml\n",
      "We are at 256: Processing BIOMD0000000585.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000585.xml\n",
      "We are at 257: Processing BIOMD0000000234.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000234.xml\n",
      "We are at 258: Processing BIOMD0000000552.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000552.xml\n",
      "We are at 259: Processing BIOMD0000000546.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000546.xml\n",
      "We are at 260: Processing BIOMD0000000220.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000220.xml\n",
      "We are at 261: Processing BIOMD0000000208.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000208.xml\n",
      "We are at 262: Processing BIOMD0000000383.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000383.xml\n",
      "We are at 263: Processing BIOMD0000000397.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000397.xml\n",
      "We are at 264: Processing BIOMD0000000340.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000340.xml\n",
      "We are at 265: Processing BIOMD0000000426.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000426.xml\n",
      "We are at 266: Processing BIOMD0000000432.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000432.xml\n",
      "We are at 267: Processing BIOMD0000000354.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000354.xml\n",
      "We are at 268: Processing BIOMD0000001062.xml\n",
      "We are at 269: Processing BIOMD0000000368.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000368.xml\n",
      "We are at 270: Processing BIOMD0000000181.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000181.xml\n",
      "We are at 271: Processing BIOMD0000000195.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000195.xml\n",
      "We are at 272: Processing BIOMD0000000803.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000803.xml\n",
      "We are at 273: Processing BIOMD0000000817.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000817.xml\n",
      "We are at 274: Processing BIOMD0000000624.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000624.xml\n",
      "We are at 275: Processing BIOMD0000000142.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000142.xml\n",
      "We are at 276: Processing BIOMD0000000156.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000156.xml\n",
      "We are at 277: Processing BIOMD0000000630.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000630.xml\n",
      "We are at 278: Processing BIOMD0000000618.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000618.xml\n",
      "We are at 279: Processing BIOMD0000000131.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000131.xml\n",
      "We are at 280: Processing BIOMD0000000657.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000657.xml\n",
      "We are at 281: Processing BIOMD0000000643.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000643.xml\n",
      "We are at 282: Processing BIOMD0000000125.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000125.xml\n",
      "We are at 283: Processing BIOMD0000000119.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000119.xml\n",
      "We are at 284: Processing BIOMD0000000680.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000680.xml\n",
      "We are at 285: Processing BIOMD0000000858.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000858.xml\n",
      "We are at 286: Processing BIOMD0000000870.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000870.xml\n",
      "We are at 287: Processing BIOMD0000000864.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000864.xml\n",
      "We are at 288: Processing BIOMD0000000455.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000455.xml\n",
      "We are at 289: Processing BIOMD0000000333.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000333.xml\n",
      "We are at 290: Processing BIOMD0000001039.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001039.xml\n",
      "We are at 291: Processing BIOMD0000000327.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000327.xml\n",
      "We are at 292: Processing BIOMD0000000441.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000441.xml\n",
      "We are at 293: Processing BIOMD0000000469.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000469.xml\n",
      "We are at 294: Processing BIOMD0000001011.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001011.xml\n",
      "We are at 295: Processing BIOMD0000001005.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001005.xml\n",
      "We are at 296: Processing BIOMD0000000496.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000496.xml\n",
      "We are at 297: Processing BIOMD0000000482.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000482.xml\n",
      "We are at 298: Processing BIOMD0000000521.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000521.xml\n",
      "We are at 299: Processing BIOMD0000000247.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000247.xml\n",
      "We are at 300: Processing BIOMD0000000253.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000253.xml\n",
      "We are at 301: Processing BIOMD0000000535.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000535.xml\n",
      "We are at 302: Processing BIOMD0000000509.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000509.xml\n",
      "We are at 303: Processing BIOMD0000000284.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000284.xml\n",
      "We are at 304: Processing BIOMD0000000290.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000290.xml\n",
      "We are at 305: Processing BIOMD0000000045.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000045.xml\n",
      "We are at 306: Processing BIOMD0000000723.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000723.xml\n",
      "We are at 307: Processing BIOMD0000000737.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000737.xml\n",
      "We are at 308: Processing BIOMD0000000051.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000051.xml\n",
      "We are at 309: Processing BIOMD0000000079.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000079.xml\n",
      "We are at 310: Processing BIOMD0000000086.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000086.xml\n",
      "We are at 311: Processing BIOMD0000000938.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000938.xml\n",
      "We are at 312: Processing BIOMD0000000092.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000092.xml\n",
      "We are at 313: Processing BIOMD0000000904.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000904.xml\n",
      "We are at 314: Processing BIOMD0000000910.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000910.xml\n",
      "We are at 315: Processing BIOMD0000000911.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000911.xml\n",
      "We are at 316: Processing BIOMD0000000905.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000905.xml\n",
      "We are at 317: Processing BIOMD0000000093.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000093.xml\n",
      "We are at 318: Processing BIOMD0000000087.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000087.xml\n",
      "We are at 319: Processing BIOMD0000000939.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000939.xml\n",
      "We are at 320: Processing BIOMD0000000078.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000078.xml\n",
      "We are at 321: Processing BIOMD0000000736.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000736.xml\n",
      "We are at 322: Processing BIOMD0000000050.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000050.xml\n",
      "We are at 323: Processing BIOMD0000000044.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000044.xml\n",
      "We are at 324: Processing BIOMD0000000722.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000722.xml\n",
      "We are at 325: Processing BIOMD0000000291.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000291.xml\n",
      "We are at 326: Processing BIOMD0000000285.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000285.xml\n",
      "We are at 327: Processing BIOMD0000000508.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000508.xml\n",
      "We are at 328: Processing BIOMD0000000252.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000252.xml\n",
      "We are at 329: Processing BIOMD0000000534.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000534.xml\n",
      "We are at 330: Processing BIOMD0000000520.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000520.xml\n",
      "We are at 331: Processing BIOMD0000000246.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000246.xml\n",
      "We are at 332: Processing BIOMD0000000483.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000483.xml\n",
      "We are at 333: Processing BIOMD0000000497.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000497.xml\n",
      "We are at 334: Processing BIOMD0000001004.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001004.xml\n",
      "We are at 335: Processing BIOMD0000000468.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000468.xml\n",
      "We are at 336: Processing BIOMD0000001010.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001010.xml\n",
      "We are at 337: Processing BIOMD0000000326.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000326.xml\n",
      "We are at 338: Processing BIOMD0000001038.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001038.xml\n",
      "We are at 339: Processing BIOMD0000000440.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000440.xml\n",
      "We are at 340: Processing BIOMD0000000454.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000454.xml\n",
      "We are at 341: Processing BIOMD0000000332.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000332.xml\n",
      "We are at 342: Processing BIOMD0000000865.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000865.xml\n",
      "We are at 343: Processing BIOMD0000000871.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000871.xml\n",
      "We are at 344: Processing BIOMD0000000859.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000859.xml\n",
      "We are at 345: Processing BIOMD0000000681.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000681.xml\n",
      "We are at 346: Processing BIOMD0000000695.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000695.xml\n",
      "We are at 347: Processing BIOMD0000000118.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000118.xml\n",
      "We are at 348: Processing BIOMD0000000642.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000642.xml\n",
      "We are at 349: Processing BIOMD0000000124.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000124.xml\n",
      "We are at 350: Processing BIOMD0000000130.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000130.xml\n",
      "We are at 351: Processing BIOMD0000000656.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000656.xml\n",
      "We are at 352: Processing BIOMD0000000126.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000126.xml\n",
      "We are at 353: Processing BIOMD0000000898.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000898.xml\n",
      "We are at 354: Processing BIOMD0000000640.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000640.xml\n",
      "We are at 355: Processing BIOMD0000000654.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000654.xml\n",
      "We are at 356: Processing BIOMD0000000132.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000132.xml\n",
      "We are at 357: Processing BIOMD0000000668.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000668.xml\n",
      "We are at 358: Processing BIOMD0000000683.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000683.xml\n",
      "We are at 359: Processing BIOMD0000000697.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000697.xml\n",
      "We are at 360: Processing BIOMD0000000867.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000867.xml\n",
      "We are at 361: Processing BIOMD0000000873.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000873.xml\n",
      "We are at 362: Processing BIOMD0000000442.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000442.xml\n",
      "We are at 363: Processing BIOMD0000000324.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000324.xml\n",
      "We are at 364: Processing BIOMD0000000330.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000330.xml\n",
      "We are at 365: Processing BIOMD0000000456.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000456.xml\n",
      "We are at 366: Processing BIOMD0000001006.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001006.xml\n",
      "We are at 367: Processing BIOMD0000000318.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000318.xml\n",
      "We are at 368: Processing BIOMD0000001012.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001012.xml\n",
      "We are at 369: Processing BIOMD0000000481.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000481.xml\n",
      "We are at 370: Processing BIOMD0000000495.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000495.xml\n",
      "We are at 371: Processing BIOMD0000000536.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000536.xml\n",
      "We are at 372: Processing BIOMD0000000250.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000250.xml\n",
      "We are at 373: Processing BIOMD0000000244.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000244.xml\n",
      "We are at 374: Processing BIOMD0000000522.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000522.xml\n",
      "We are at 375: Processing BIOMD0000000278.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000278.xml\n",
      "We are at 376: Processing BIOMD0000000293.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000293.xml\n",
      "We are at 377: Processing BIOMD0000000287.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000287.xml\n",
      "We are at 378: Processing BIOMD0000000052.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000052.xml\n",
      "We are at 379: Processing BIOMD0000000734.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000734.xml\n",
      "We are at 380: Processing BIOMD0000000720.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000720.xml\n",
      "We are at 381: Processing BIOMD0000000046.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000046.xml\n",
      "We are at 382: Processing BIOMD0000000708.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000708.xml\n",
      "We are at 383: Processing BIOMD0000000091.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000091.xml\n",
      "We are at 384: Processing BIOMD0000000085.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000085.xml\n",
      "We are at 385: Processing BIOMD0000000913.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000913.xml\n",
      "We are at 386: Processing BIOMD0000000907.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000907.xml\n",
      "We are at 387: Processing BIOMD0000000906.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000906.xml\n",
      "We are at 388: Processing BIOMD0000000912.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000912.xml\n",
      "We are at 389: Processing BIOMD0000000084.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000084.xml\n",
      "We are at 390: Processing BIOMD0000000090.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000090.xml\n",
      "We are at 391: Processing BIOMD0000000709.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000709.xml\n",
      "We are at 392: Processing BIOMD0000000721.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000721.xml\n",
      "We are at 393: Processing BIOMD0000000047.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000047.xml\n",
      "We are at 394: Processing BIOMD0000000053.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000053.xml\n",
      "We are at 395: Processing BIOMD0000000735.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000735.xml\n",
      "We are at 396: Processing BIOMD0000000286.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000286.xml\n",
      "We are at 397: Processing BIOMD0000000292.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000292.xml\n",
      "We are at 398: Processing BIOMD0000000279.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000279.xml\n",
      "We are at 399: Processing BIOMD0000000245.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000245.xml\n",
      "We are at 400: Processing BIOMD0000000523.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000523.xml\n",
      "We are at 401: Processing BIOMD0000000537.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000537.xml\n",
      "We are at 402: Processing BIOMD0000000251.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000251.xml\n",
      "We are at 403: Processing BIOMD0000000494.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000494.xml\n",
      "We are at 404: Processing BIOMD0000000480.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000480.xml\n",
      "We are at 405: Processing BIOMD0000001013.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001013.xml\n",
      "We are at 406: Processing BIOMD0000000319.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000319.xml\n",
      "We are at 407: Processing BIOMD0000001007.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001007.xml\n",
      "We are at 408: Processing BIOMD0000000331.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000331.xml\n",
      "We are at 409: Processing BIOMD0000000457.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000457.xml\n",
      "We are at 410: Processing BIOMD0000000443.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000443.xml\n",
      "We are at 411: Processing BIOMD0000000325.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000325.xml\n",
      "We are at 412: Processing BIOMD0000000872.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000872.xml\n",
      "We are at 413: Processing BIOMD0000000866.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000866.xml\n",
      "We are at 414: Processing BIOMD0000000696.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000696.xml\n",
      "We are at 415: Processing BIOMD0000000682.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000682.xml\n",
      "We are at 416: Processing BIOMD0000000669.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000669.xml\n",
      "We are at 417: Processing BIOMD0000000655.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000655.xml\n",
      "We are at 418: Processing BIOMD0000000133.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000133.xml\n",
      "We are at 419: Processing BIOMD0000000127.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000127.xml\n",
      "We are at 420: Processing BIOMD0000000641.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000641.xml\n",
      "We are at 421: Processing BIOMD0000000899.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000899.xml\n",
      "We are at 422: Processing BIOMD0000000679.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000679.xml\n",
      "We are at 423: Processing BIOMD0000000645.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000645.xml\n",
      "We are at 424: Processing BIOMD0000000123.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000123.xml\n",
      "We are at 425: Processing BIOMD0000000137.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000137.xml\n",
      "We are at 426: Processing BIOMD0000000889.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000889.xml\n",
      "We are at 427: Processing BIOMD0000000651.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000651.xml\n",
      "We are at 428: Processing BIOMD0000000862.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000862.xml\n",
      "We are at 429: Processing BIOMD0000000876.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000876.xml\n",
      "We are at 430: Processing BIOMD0000000686.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000686.xml\n",
      "We are at 431: Processing BIOMD0000000692.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000692.xml\n",
      "We are at 432: Processing BIOMD0000001003.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001003.xml\n",
      "We are at 433: Processing BIOMD0000001017.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001017.xml\n",
      "We are at 434: Processing BIOMD0000000309.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000309.xml\n",
      "We are at 435: Processing BIOMD0000000321.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000321.xml\n",
      "We are at 436: Processing BIOMD0000000447.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000447.xml\n",
      "We are at 437: Processing BIOMD0000000453.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000453.xml\n",
      "We are at 438: Processing BIOMD0000000335.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000335.xml\n",
      "We are at 439: Processing BIOMD0000000484.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000484.xml\n",
      "We are at 440: Processing BIOMD0000000490.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000490.xml\n",
      "We are at 441: Processing BIOMD0000000269.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000269.xml\n",
      "We are at 442: Processing BIOMD0000000255.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000255.xml\n",
      "We are at 443: Processing BIOMD0000000533.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000533.xml\n",
      "We are at 444: Processing BIOMD0000000527.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000527.xml\n",
      "We are at 445: Processing BIOMD0000000241.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000241.xml\n",
      "We are at 446: Processing BIOMD0000000296.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000296.xml\n",
      "We are at 447: Processing BIOMD0000000282.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000282.xml\n",
      "We are at 448: Processing BIOMD0000000719.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000719.xml\n",
      "We are at 449: Processing BIOMD0000000731.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000731.xml\n",
      "We are at 450: Processing BIOMD0000000057.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000057.xml\n",
      "We are at 451: Processing BIOMD0000000043.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000043.xml\n",
      "We are at 452: Processing BIOMD0000000725.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000725.xml\n",
      "We are at 453: Processing BIOMD0000000916.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000916.xml\n",
      "We are at 454: Processing BIOMD0000000902.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000902.xml\n",
      "We are at 455: Processing BIOMD0000000094.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000094.xml\n",
      "We are at 456: Processing BIOMD0000000080.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000080.xml\n",
      "We are at 457: Processing BIOMD0000000081.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000081.xml\n",
      "We are at 458: Processing BIOMD0000000095.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000095.xml\n",
      "We are at 459: Processing BIOMD0000000903.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000903.xml\n",
      "We are at 460: Processing BIOMD0000000917.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000917.xml\n",
      "We are at 461: Processing BIOMD0000000042.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000042.xml\n",
      "We are at 462: Processing BIOMD0000000724.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000724.xml\n",
      "We are at 463: Processing BIOMD0000000730.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000730.xml\n",
      "We are at 464: Processing BIOMD0000000056.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000056.xml\n",
      "We are at 465: Processing BIOMD0000000718.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000718.xml\n",
      "We are at 466: Processing BIOMD0000000283.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000283.xml\n",
      "We are at 467: Processing BIOMD0000000297.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000297.xml\n",
      "We are at 468: Processing BIOMD0000000526.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000526.xml\n",
      "We are at 469: Processing BIOMD0000000240.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000240.xml\n",
      "We are at 470: Processing BIOMD0000000254.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000254.xml\n",
      "We are at 471: Processing BIOMD0000000532.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000532.xml\n",
      "We are at 472: Processing BIOMD0000000268.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000268.xml\n",
      "We are at 473: Processing BIOMD0000000491.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000491.xml\n",
      "We are at 474: Processing BIOMD0000000485.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000485.xml\n",
      "We are at 475: Processing BIOMD0000000452.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000452.xml\n",
      "We are at 476: Processing BIOMD0000000334.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000334.xml\n",
      "We are at 477: Processing BIOMD0000000320.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000320.xml\n",
      "We are at 478: Processing BIOMD0000000446.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000446.xml\n",
      "We are at 479: Processing BIOMD0000000308.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000308.xml\n",
      "We are at 480: Processing BIOMD0000001016.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001016.xml\n",
      "We are at 481: Processing BIOMD0000001002.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001002.xml\n",
      "We are at 482: Processing BIOMD0000000693.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000693.xml\n",
      "We are at 483: Processing BIOMD0000000687.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000687.xml\n",
      "We are at 484: Processing BIOMD0000000877.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000877.xml\n",
      "We are at 485: Processing BIOMD0000000863.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000863.xml\n",
      "We are at 486: Processing BIOMD0000000136.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000136.xml\n",
      "We are at 487: Processing BIOMD0000000650.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000650.xml\n",
      "We are at 488: Processing BIOMD0000000888.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000888.xml\n",
      "We are at 489: Processing BIOMD0000000644.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000644.xml\n",
      "We are at 490: Processing BIOMD0000000122.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000122.xml\n",
      "We are at 491: Processing BIOMD0000000678.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000678.xml\n",
      "We are at 492: Processing BIOMD0000000108.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000108.xml\n",
      "We are at 493: Processing BIOMD0000000652.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000652.xml\n",
      "We are at 494: Processing BIOMD0000000134.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000134.xml\n",
      "We are at 495: Processing BIOMD0000000120.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000120.xml\n",
      "We are at 496: Processing BIOMD0000000646.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000646.xml\n",
      "We are at 497: Processing BIOMD0000000875.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000875.xml\n",
      "We are at 498: Processing BIOMD0000000861.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000861.xml\n",
      "We are at 499: Processing BIOMD0000000691.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000691.xml\n",
      "We are at 500: Processing BIOMD0000000849.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000849.xml\n",
      "We are at 501: Processing BIOMD0000000685.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000685.xml\n",
      "We are at 502: Processing BIOMD0000001014.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001014.xml\n",
      "We are at 503: Processing BIOMD0000000478.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000478.xml\n",
      "We are at 504: Processing BIOMD0000001000.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001000.xml\n",
      "We are at 505: Processing BIOMD0000001028.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001028.xml\n",
      "We are at 506: Processing BIOMD0000000336.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000336.xml\n",
      "We are at 507: Processing BIOMD0000000450.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000450.xml\n",
      "We are at 508: Processing BIOMD0000000444.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000444.xml\n",
      "We are at 509: Processing BIOMD0000000322.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000322.xml\n",
      "We are at 510: Processing BIOMD0000000493.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000493.xml\n",
      "We are at 511: Processing BIOMD0000000487.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000487.xml\n",
      "We are at 512: Processing BIOMD0000000518.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000518.xml\n",
      "We are at 513: Processing BIOMD0000000242.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000242.xml\n",
      "We are at 514: Processing BIOMD0000000524.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000524.xml\n",
      "We are at 515: Processing BIOMD0000000530.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000530.xml\n",
      "We are at 516: Processing BIOMD0000000256.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000256.xml\n",
      "We are at 517: Processing BIOMD0000000281.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000281.xml\n",
      "We are at 518: Processing BIOMD0000000295.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000295.xml\n",
      "We are at 519: Processing BIOMD0000000068.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000068.xml\n",
      "We are at 520: Processing BIOMD0000000726.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000726.xml\n",
      "We are at 521: Processing BIOMD0000000040.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000040.xml\n",
      "We are at 522: Processing BIOMD0000000054.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000054.xml\n",
      "We are at 523: Processing BIOMD0000000732.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000732.xml\n",
      "We are at 524: Processing BIOMD0000000901.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000901.xml\n",
      "We are at 525: Processing BIOMD0000000915.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000915.xml\n",
      "We are at 526: Processing BIOMD0000000083.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000083.xml\n",
      "We are at 527: Processing BIOMD0000000097.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000097.xml\n",
      "We are at 528: Processing BIOMD0000000929.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000929.xml\n",
      "We are at 529: Processing BIOMD0000000096.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000096.xml\n",
      "We are at 530: Processing BIOMD0000000928.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000928.xml\n",
      "We are at 531: Processing BIOMD0000000082.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000082.xml\n",
      "We are at 532: Processing BIOMD0000000914.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000914.xml\n",
      "We are at 533: Processing BIOMD0000000900.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000900.xml\n",
      "We are at 534: Processing BIOMD0000000055.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000055.xml\n",
      "We are at 535: Processing BIOMD0000000733.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000733.xml\n",
      "We are at 536: Processing BIOMD0000000727.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000727.xml\n",
      "We are at 537: Processing BIOMD0000000041.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000041.xml\n",
      "We are at 538: Processing BIOMD0000000069.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000069.xml\n",
      "We are at 539: Processing BIOMD0000000294.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000294.xml\n",
      "We are at 540: Processing BIOMD0000000280.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000280.xml\n",
      "We are at 541: Processing BIOMD0000000531.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000531.xml\n",
      "We are at 542: Processing BIOMD0000000257.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000257.xml\n",
      "We are at 543: Processing BIOMD0000000243.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000243.xml\n",
      "We are at 544: Processing BIOMD0000000525.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000525.xml\n",
      "We are at 545: Processing BIOMD0000000519.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000519.xml\n",
      "We are at 546: Processing BIOMD0000000486.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000486.xml\n",
      "We are at 547: Processing BIOMD0000000492.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000492.xml\n",
      "We are at 548: Processing BIOMD0000000445.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000445.xml\n",
      "We are at 549: Processing BIOMD0000000323.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000323.xml\n",
      "We are at 550: Processing BIOMD0000000337.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000337.xml\n",
      "We are at 551: Processing BIOMD0000001029.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001029.xml\n",
      "We are at 552: Processing BIOMD0000000451.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000451.xml\n",
      "We are at 553: Processing BIOMD0000000479.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000479.xml\n",
      "We are at 554: Processing BIOMD0000001001.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001001.xml\n",
      "We are at 555: Processing BIOMD0000001015.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001015.xml\n",
      "We are at 556: Processing BIOMD0000000684.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000684.xml\n",
      "We are at 557: Processing BIOMD0000000848.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000848.xml\n",
      "We are at 558: Processing BIOMD0000000690.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000690.xml\n",
      "We are at 559: Processing BIOMD0000000860.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000860.xml\n",
      "We are at 560: Processing BIOMD0000000874.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000874.xml\n",
      "We are at 561: Processing BIOMD0000000121.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000121.xml\n",
      "We are at 562: Processing BIOMD0000000647.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000647.xml\n",
      "We are at 563: Processing BIOMD0000000653.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000653.xml\n",
      "We are at 564: Processing BIOMD0000000135.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000135.xml\n",
      "We are at 565: Processing BIOMD0000000109.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000109.xml\n",
      "We are at 566: Processing BIOMD0000000110.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000110.xml\n",
      "We are at 567: Processing BIOMD0000000676.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000676.xml\n",
      "We are at 568: Processing BIOMD0000000662.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000662.xml\n",
      "We are at 569: Processing BIOMD0000000104.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000104.xml\n",
      "We are at 570: Processing BIOMD0000000892.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000892.xml\n",
      "We are at 571: Processing BIOMD0000000886.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000886.xml\n",
      "We are at 572: Processing BIOMD0000000138.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000138.xml\n",
      "We are at 573: Processing BIOMD0000000879.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000879.xml\n",
      "We are at 574: Processing BIOMD0000000689.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000689.xml\n",
      "We are at 575: Processing BIOMD0000000851.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000851.xml\n",
      "We are at 576: Processing BIOMD0000000845.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000845.xml\n",
      "We are at 577: Processing BIOMD0000000474.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000474.xml\n",
      "We are at 578: Processing BIOMD0000000312.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000312.xml\n",
      "We are at 579: Processing BIOMD0000001018.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001018.xml\n",
      "We are at 580: Processing BIOMD0000000306.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000306.xml\n",
      "We are at 581: Processing BIOMD0000000460.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000460.xml\n",
      "We are at 582: Processing BIOMD0000000448.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000448.xml\n",
      "We are at 583: Processing BIOMD0000001030.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001030.xml\n",
      "We are at 584: Processing BIOMD0000001024.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001024.xml\n",
      "We are at 585: Processing BIOMD0000000500.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000500.xml\n",
      "We are at 586: Processing BIOMD0000000266.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000266.xml\n",
      "We are at 587: Processing BIOMD0000000272.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000272.xml\n",
      "We are at 588: Processing BIOMD0000000514.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000514.xml\n",
      "We are at 589: Processing BIOMD0000000528.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000528.xml\n",
      "We are at 590: Processing BIOMD0000000299.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000299.xml\n",
      "We are at 591: Processing BIOMD0000000064.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000064.xml\n",
      "We are at 592: Processing BIOMD0000000702.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000702.xml\n",
      "We are at 593: Processing BIOMD0000000716.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000716.xml\n",
      "We are at 594: Processing BIOMD0000000070.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000070.xml\n",
      "We are at 595: Processing BIOMD0000000058.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000058.xml\n",
      "We are at 596: Processing BIOMD0000000919.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000919.xml\n",
      "We are at 597: Processing BIOMD0000000925.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000925.xml\n",
      "We are at 598: Processing BIOMD0000000931.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000931.xml\n",
      "We are at 599: Processing BIOMD0000000930.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000930.xml\n",
      "We are at 600: Processing BIOMD0000000924.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000924.xml\n",
      "We are at 601: Processing BIOMD0000000918.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000918.xml\n",
      "We are at 602: Processing BIOMD0000000059.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000059.xml\n",
      "We are at 603: Processing BIOMD0000000717.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000717.xml\n",
      "We are at 604: Processing BIOMD0000000071.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000071.xml\n",
      "We are at 605: Processing BIOMD0000000065.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000065.xml\n",
      "We are at 606: Processing BIOMD0000000703.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000703.xml\n",
      "We are at 607: Processing BIOMD0000000298.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000298.xml\n",
      "We are at 608: Processing BIOMD0000000529.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000529.xml\n",
      "We are at 609: Processing BIOMD0000000273.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000273.xml\n",
      "We are at 610: Processing BIOMD0000000515.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000515.xml\n",
      "We are at 611: Processing BIOMD0000000501.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000501.xml\n",
      "We are at 612: Processing BIOMD0000000267.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000267.xml\n",
      "We are at 613: Processing BIOMD0000001025.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001025.xml\n",
      "We are at 614: Processing BIOMD0000000449.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000449.xml\n",
      "We are at 615: Processing BIOMD0000001031.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001031.xml\n",
      "We are at 616: Processing BIOMD0000000307.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000307.xml\n",
      "We are at 617: Processing BIOMD0000001019.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001019.xml\n",
      "We are at 618: Processing BIOMD0000000461.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000461.xml\n",
      "We are at 619: Processing BIOMD0000000475.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000475.xml\n",
      "We are at 620: Processing BIOMD0000000313.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000313.xml\n",
      "We are at 621: Processing BIOMD0000000844.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000844.xml\n",
      "We are at 622: Processing BIOMD0000000850.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000850.xml\n",
      "We are at 623: Processing BIOMD0000000688.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000688.xml\n",
      "We are at 624: Processing BIOMD0000000878.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000878.xml\n",
      "We are at 625: Processing BIOMD0000000887.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000887.xml\n",
      "We are at 626: Processing BIOMD0000000139.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000139.xml\n",
      "We are at 627: Processing BIOMD0000000893.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000893.xml\n",
      "We are at 628: Processing BIOMD0000000663.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000663.xml\n",
      "We are at 629: Processing BIOMD0000000105.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000105.xml\n",
      "We are at 630: Processing BIOMD0000000111.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000111.xml\n",
      "We are at 631: Processing BIOMD0000000677.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000677.xml\n",
      "We are at 632: Processing BIOMD0000000107.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000107.xml\n",
      "We are at 633: Processing BIOMD0000000661.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000661.xml\n",
      "We are at 634: Processing BIOMD0000000675.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000675.xml\n",
      "We are at 635: Processing BIOMD0000000113.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000113.xml\n",
      "We are at 636: Processing BIOMD0000000885.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000885.xml\n",
      "We are at 637: Processing BIOMD0000000891.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000891.xml\n",
      "We are at 638: Processing BIOMD0000000846.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000846.xml\n",
      "We are at 639: Processing BIOMD0000000852.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000852.xml\n",
      "We are at 640: Processing BIOMD0000000463.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000463.xml\n",
      "We are at 641: Processing BIOMD0000000305.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000305.xml\n",
      "We are at 642: Processing BIOMD0000000311.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000311.xml\n",
      "We are at 643: Processing BIOMD0000000477.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000477.xml\n",
      "We are at 644: Processing BIOMD0000001027.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001027.xml\n",
      "We are at 645: Processing BIOMD0000000339.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000339.xml\n",
      "We are at 646: Processing BIOMD0000001033.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001033.xml\n",
      "We are at 647: Processing BIOMD0000000488.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000488.xml\n",
      "We are at 648: Processing BIOMD0000000517.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000517.xml\n",
      "We are at 649: Processing BIOMD0000000271.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000271.xml\n",
      "We are at 650: Processing BIOMD0000000265.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000265.xml\n",
      "We are at 651: Processing BIOMD0000000503.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000503.xml\n",
      "We are at 652: Processing BIOMD0000000259.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000259.xml\n",
      "We are at 653: Processing BIOMD0000000073.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000073.xml\n",
      "We are at 654: Processing BIOMD0000000715.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000715.xml\n",
      "We are at 655: Processing BIOMD0000000701.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000701.xml\n",
      "We are at 656: Processing BIOMD0000000067.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000067.xml\n",
      "We are at 657: Processing BIOMD0000000729.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000729.xml\n",
      "We are at 658: Processing BIOMD0000000932.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000932.xml\n",
      "We are at 659: Processing BIOMD0000000926.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000926.xml\n",
      "We are at 660: Processing BIOMD0000000098.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000098.xml\n",
      "We are at 661: Processing BIOMD0000000927.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000927.xml\n",
      "We are at 662: Processing BIOMD0000000099.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000099.xml\n",
      "We are at 663: Processing BIOMD0000000933.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000933.xml\n",
      "We are at 664: Processing BIOMD0000000728.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000728.xml\n",
      "We are at 665: Processing BIOMD0000000700.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000700.xml\n",
      "We are at 666: Processing BIOMD0000000066.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000066.xml\n",
      "We are at 667: Processing BIOMD0000000072.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000072.xml\n",
      "We are at 668: Processing BIOMD0000000714.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000714.xml\n",
      "We are at 669: Processing BIOMD0000000258.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000258.xml\n",
      "We are at 670: Processing BIOMD0000000264.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000264.xml\n",
      "We are at 671: Processing BIOMD0000000502.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000502.xml\n",
      "We are at 672: Processing BIOMD0000000516.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000516.xml\n",
      "We are at 673: Processing BIOMD0000000270.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000270.xml\n",
      "We are at 674: Processing BIOMD0000000489.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000489.xml\n",
      "We are at 675: Processing BIOMD0000001032.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001032.xml\n",
      "We are at 676: Processing BIOMD0000000338.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000338.xml\n",
      "We are at 677: Processing BIOMD0000001026.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001026.xml\n",
      "We are at 678: Processing BIOMD0000000310.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000310.xml\n",
      "We are at 679: Processing BIOMD0000000476.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000476.xml\n",
      "We are at 680: Processing BIOMD0000000462.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000462.xml\n",
      "We are at 681: Processing BIOMD0000000304.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000304.xml\n",
      "We are at 682: Processing BIOMD0000000853.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000853.xml\n",
      "We are at 683: Processing BIOMD0000000847.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000847.xml\n",
      "We are at 684: Processing BIOMD0000000648.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000648.xml\n",
      "We are at 685: Processing BIOMD0000000890.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000890.xml\n",
      "We are at 686: Processing BIOMD0000000884.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000884.xml\n",
      "We are at 687: Processing BIOMD0000000674.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000674.xml\n",
      "We are at 688: Processing BIOMD0000000112.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000112.xml\n",
      "We are at 689: Processing BIOMD0000000106.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000106.xml\n",
      "We are at 690: Processing BIOMD0000000660.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000660.xml\n",
      "We are at 691: Processing BIOMD0000000880.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000880.xml\n",
      "We are at 692: Processing BIOMD0000000658.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000658.xml\n",
      "We are at 693: Processing BIOMD0000000894.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000894.xml\n",
      "We are at 694: Processing BIOMD0000000664.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000664.xml\n",
      "We are at 695: Processing BIOMD0000000102.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000102.xml\n",
      "We are at 696: Processing BIOMD0000000116.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000116.xml\n",
      "We are at 697: Processing BIOMD0000000670.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000670.xml\n",
      "We are at 698: Processing BIOMD0000000843.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000843.xml\n",
      "We are at 699: Processing BIOMD0000000857.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000857.xml\n",
      "We are at 700: Processing BIOMD0000001022.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001022.xml\n",
      "We are at 701: Processing BIOMD0000001036.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001036.xml\n",
      "We are at 702: Processing BIOMD0000000328.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000328.xml\n",
      "We are at 703: Processing BIOMD0000000300.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000300.xml\n",
      "We are at 704: Processing BIOMD0000000466.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000466.xml\n",
      "We are at 705: Processing BIOMD0000000472.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000472.xml\n",
      "We are at 706: Processing BIOMD0000000314.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000314.xml\n",
      "We are at 707: Processing BIOMD0000000499.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000499.xml\n",
      "We are at 708: Processing BIOMD0000000248.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000248.xml\n",
      "We are at 709: Processing BIOMD0000000274.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000274.xml\n",
      "We are at 710: Processing BIOMD0000000512.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000512.xml\n",
      "We are at 711: Processing BIOMD0000000506.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000506.xml\n",
      "We are at 712: Processing BIOMD0000000260.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000260.xml\n",
      "We are at 713: Processing BIOMD0000000738.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000738.xml\n",
      "We are at 714: Processing BIOMD0000000710.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000710.xml\n",
      "We are at 715: Processing BIOMD0000000076.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000076.xml\n",
      "We are at 716: Processing BIOMD0000000062.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000062.xml\n",
      "We are at 717: Processing BIOMD0000000704.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000704.xml\n",
      "We are at 718: Processing BIOMD0000000937.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000937.xml\n",
      "We are at 719: Processing BIOMD0000000089.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000089.xml\n",
      "We are at 720: Processing BIOMD0000000923.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000923.xml\n",
      "We are at 721: Processing BIOMD0000000922.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000922.xml\n",
      "We are at 722: Processing BIOMD0000000936.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000936.xml\n",
      "We are at 723: Processing BIOMD0000000088.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000088.xml\n",
      "We are at 724: Processing BIOMD0000000063.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000063.xml\n",
      "We are at 725: Processing BIOMD0000000705.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000705.xml\n",
      "We are at 726: Processing BIOMD0000000711.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000711.xml\n",
      "We are at 727: Processing BIOMD0000000077.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000077.xml\n",
      "We are at 728: Processing BIOMD0000000739.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000739.xml\n",
      "We are at 729: Processing BIOMD0000000507.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000507.xml\n",
      "We are at 730: Processing BIOMD0000000261.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000261.xml\n",
      "We are at 731: Processing BIOMD0000000275.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000275.xml\n",
      "We are at 732: Processing BIOMD0000000513.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000513.xml\n",
      "We are at 733: Processing BIOMD0000000249.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000249.xml\n",
      "We are at 734: Processing BIOMD0000000498.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000498.xml\n",
      "We are at 735: Processing BIOMD0000000473.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000473.xml\n",
      "We are at 736: Processing BIOMD0000000315.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000315.xml\n",
      "We are at 737: Processing BIOMD0000000301.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000301.xml\n",
      "We are at 738: Processing BIOMD0000000467.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000467.xml\n",
      "We are at 739: Processing BIOMD0000000329.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000329.xml\n",
      "We are at 740: Processing BIOMD0000001037.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001037.xml\n",
      "We are at 741: Processing BIOMD0000001023.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001023.xml\n",
      "We are at 742: Processing BIOMD0000000856.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000856.xml\n",
      "We are at 743: Processing BIOMD0000000842.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000842.xml\n",
      "We are at 744: Processing BIOMD0000000117.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000117.xml\n",
      "We are at 745: Processing BIOMD0000000671.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000671.xml\n",
      "We are at 746: Processing BIOMD0000000665.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000665.xml\n",
      "We are at 747: Processing BIOMD0000000103.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000103.xml\n",
      "We are at 748: Processing BIOMD0000000895.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000895.xml\n",
      "We are at 749: Processing BIOMD0000000659.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000659.xml\n",
      "We are at 750: Processing BIOMD0000000881.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000881.xml\n",
      "We are at 751: Processing BIOMD0000000897.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000897.xml\n",
      "We are at 752: Processing BIOMD0000000129.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000129.xml\n",
      "We are at 753: Processing BIOMD0000000883.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000883.xml\n",
      "We are at 754: Processing BIOMD0000000673.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000673.xml\n",
      "We are at 755: Processing BIOMD0000000115.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000115.xml\n",
      "We are at 756: Processing BIOMD0000000101.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000101.xml\n",
      "We are at 757: Processing BIOMD0000000667.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000667.xml\n",
      "We are at 758: Processing BIOMD0000000854.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000854.xml\n",
      "We are at 759: Processing BIOMD0000000698.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000698.xml\n",
      "We are at 760: Processing BIOMD0000000840.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000840.xml\n",
      "We are at 761: Processing BIOMD0000000868.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000868.xml\n",
      "We are at 762: Processing BIOMD0000001035.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001035.xml\n",
      "We are at 763: Processing BIOMD0000000459.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000459.xml\n",
      "We are at 764: Processing BIOMD0000001021.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001021.xml\n",
      "We are at 765: Processing BIOMD0000001009.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001009.xml\n",
      "We are at 766: Processing BIOMD0000000317.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000317.xml\n",
      "We are at 767: Processing BIOMD0000000471.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000471.xml\n",
      "We are at 768: Processing BIOMD0000000465.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000465.xml\n",
      "We are at 769: Processing BIOMD0000000303.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000303.xml\n",
      "We are at 770: Processing BIOMD0000000539.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000539.xml\n",
      "We are at 771: Processing BIOMD0000000263.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000263.xml\n",
      "We are at 772: Processing BIOMD0000000505.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000505.xml\n",
      "We are at 773: Processing BIOMD0000000511.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000511.xml\n",
      "We are at 774: Processing BIOMD0000000277.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000277.xml\n",
      "We are at 775: Processing BIOMD0000000288.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000288.xml\n",
      "We are at 776: Processing BIOMD0000000049.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000049.xml\n",
      "We are at 777: Processing BIOMD0000000707.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000707.xml\n",
      "We are at 778: Processing BIOMD0000000061.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000061.xml\n",
      "We are at 779: Processing BIOMD0000000075.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000075.xml\n",
      "We are at 780: Processing BIOMD0000000713.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000713.xml\n",
      "We are at 781: Processing BIOMD0000000920.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000920.xml\n",
      "We are at 782: Processing BIOMD0000000934.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000934.xml\n",
      "We are at 783: Processing BIOMD0000000908.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000908.xml\n",
      "We are at 784: Processing BIOMD0000000909.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000909.xml\n",
      "We are at 785: Processing BIOMD0000000935.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000935.xml\n",
      "We are at 786: Processing BIOMD0000000921.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000921.xml\n",
      "We are at 787: Processing BIOMD0000000074.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000074.xml\n",
      "We are at 788: Processing BIOMD0000000712.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000712.xml\n",
      "We are at 789: Processing BIOMD0000000706.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000706.xml\n",
      "We are at 790: Processing BIOMD0000000060.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000060.xml\n",
      "We are at 791: Processing BIOMD0000000048.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000048.xml\n",
      "We are at 792: Processing BIOMD0000000289.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000289.xml\n",
      "We are at 793: Processing BIOMD0000000510.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000510.xml\n",
      "We are at 794: Processing BIOMD0000000276.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000276.xml\n",
      "We are at 795: Processing BIOMD0000000262.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000262.xml\n",
      "We are at 796: Processing BIOMD0000000504.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000504.xml\n",
      "We are at 797: Processing BIOMD0000000538.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000538.xml\n",
      "We are at 798: Processing BIOMD0000000464.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000464.xml\n",
      "We are at 799: Processing BIOMD0000000302.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000302.xml\n",
      "We are at 800: Processing BIOMD0000000316.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000316.xml\n",
      "We are at 801: Processing BIOMD0000001008.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001008.xml\n",
      "We are at 802: Processing BIOMD0000000470.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000470.xml\n",
      "We are at 803: Processing BIOMD0000000458.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000458.xml\n",
      "We are at 804: Processing BIOMD0000001020.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001020.xml\n",
      "We are at 805: Processing BIOMD0000001034.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001034.xml\n",
      "We are at 806: Processing BIOMD0000000869.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000869.xml\n",
      "We are at 807: Processing BIOMD0000000841.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000841.xml\n",
      "We are at 808: Processing BIOMD0000000699.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000699.xml\n",
      "We are at 809: Processing BIOMD0000000855.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000855.xml\n",
      "We are at 810: Processing BIOMD0000000100.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000100.xml\n",
      "We are at 811: Processing BIOMD0000000666.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000666.xml\n",
      "We are at 812: Processing BIOMD0000000672.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000672.xml\n",
      "We are at 813: Processing BIOMD0000000114.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000114.xml\n",
      "We are at 814: Processing BIOMD0000000882.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000882.xml\n",
      "We are at 815: Processing BIOMD0000000896.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000896.xml\n",
      "We are at 816: Processing BIOMD0000000128.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000128.xml\n",
      "We are at 817: Processing BIOMD0000000615.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000615.xml\n",
      "We are at 818: Processing BIOMD0000000173.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000173.xml\n",
      "We are at 819: Processing BIOMD0000000167.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000167.xml\n",
      "We are at 820: Processing BIOMD0000000601.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000601.xml\n",
      "We are at 821: Processing BIOMD0000000629.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000629.xml\n",
      "We are at 822: Processing BIOMD0000000832.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000832.xml\n",
      "We are at 823: Processing BIOMD0000000198.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000198.xml\n",
      "We are at 824: Processing BIOMD0000000826.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000826.xml\n",
      "We are at 825: Processing BIOMD0000000371.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000371.xml\n",
      "We are at 826: Processing BIOMD0000000417.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000417.xml\n",
      "We are at 827: Processing BIOMD0000000403.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000403.xml\n",
      "We are at 828: Processing BIOMD0000000365.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000365.xml\n",
      "We are at 829: Processing BIOMD0000001053.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001053.xml\n",
      "We are at 830: Processing BIOMD0000001047.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001047.xml\n",
      "We are at 831: Processing BIOMD0000000359.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000359.xml\n",
      "We are at 832: Processing BIOMD0000000205.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000205.xml\n",
      "We are at 833: Processing BIOMD0000000563.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000563.xml\n",
      "We are at 834: Processing BIOMD0000000577.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000577.xml\n",
      "We are at 835: Processing BIOMD0000000211.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000211.xml\n",
      "We are at 836: Processing BIOMD0000000239.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000239.xml\n",
      "We are at 837: Processing BIOMD0000000588.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000588.xml\n",
      "We are at 838: Processing BIOMD0000000761.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000761.xml\n",
      "We are at 839: Processing BIOMD0000000007.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000007.xml\n",
      "We are at 840: Processing BIOMD0000000013.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000013.xml\n",
      "We are at 841: Processing BIOMD0000000775.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000775.xml\n",
      "We are at 842: Processing BIOMD0000000985.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000985.xml\n",
      "We are at 843: Processing BIOMD0000000749.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000749.xml\n",
      "We are at 844: Processing BIOMD0000000991.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000991.xml\n",
      "We are at 845: Processing BIOMD0000000946.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000946.xml\n",
      "We are at 846: Processing BIOMD0000000952.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000952.xml\n",
      "We are at 847: Processing BIOMD0000000953.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000953.xml\n",
      "We are at 848: Processing BIOMD0000000947.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000947.xml\n",
      "We are at 849: Processing BIOMD0000000990.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000990.xml\n",
      "We are at 850: Processing BIOMD0000000748.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000748.xml\n",
      "We are at 851: Processing BIOMD0000000984.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000984.xml\n",
      "We are at 852: Processing BIOMD0000000012.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000012.xml\n",
      "We are at 853: Processing BIOMD0000000774.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000774.xml\n",
      "We are at 854: Processing BIOMD0000000760.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000760.xml\n",
      "We are at 855: Processing BIOMD0000000006.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000006.xml\n",
      "We are at 856: Processing BIOMD0000000589.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000589.xml\n",
      "We are at 857: Processing BIOMD0000000238.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000238.xml\n",
      "We are at 858: Processing BIOMD0000000576.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000576.xml\n",
      "We are at 859: Processing BIOMD0000000210.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000210.xml\n",
      "We are at 860: Processing BIOMD0000000204.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000204.xml\n",
      "We are at 861: Processing BIOMD0000000562.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000562.xml\n",
      "We are at 862: Processing BIOMD0000000358.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000358.xml\n",
      "We are at 863: Processing BIOMD0000001046.xml\n",
      "We are at 864: Processing BIOMD0000001052.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001052.xml\n",
      "We are at 865: Processing BIOMD0000000402.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000402.xml\n",
      "We are at 866: Processing BIOMD0000000364.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000364.xml\n",
      "We are at 867: Processing BIOMD0000000370.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000370.xml\n",
      "We are at 868: Processing BIOMD0000000416.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000416.xml\n",
      "We are at 869: Processing BIOMD0000000199.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000199.xml\n",
      "We are at 870: Processing BIOMD0000000827.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000827.xml\n",
      "We are at 871: Processing BIOMD0000000833.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000833.xml\n",
      "We are at 872: Processing BIOMD0000000628.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000628.xml\n",
      "We are at 873: Processing BIOMD0000000166.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000166.xml\n",
      "We are at 874: Processing BIOMD0000000600.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000600.xml\n",
      "We are at 875: Processing BIOMD0000000614.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000614.xml\n",
      "We are at 876: Processing BIOMD0000000172.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000172.xml\n",
      "We are at 877: Processing BIOMD0000000602.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000602.xml\n",
      "We are at 878: Processing BIOMD0000000164.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000164.xml\n",
      "We are at 879: Processing BIOMD0000000170.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000170.xml\n",
      "We are at 880: Processing BIOMD0000000616.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000616.xml\n",
      "We are at 881: Processing BIOMD0000000158.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000158.xml\n",
      "We are at 882: Processing BIOMD0000000819.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000819.xml\n",
      "We are at 883: Processing BIOMD0000000825.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000825.xml\n",
      "We are at 884: Processing BIOMD0000000831.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000831.xml\n",
      "We are at 885: Processing BIOMD0000001078.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001078.xml\n",
      "We are at 886: Processing BIOMD0000000366.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000366.xml\n",
      "We are at 887: Processing BIOMD0000000400.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000400.xml\n",
      "We are at 888: Processing BIOMD0000000414.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000414.xml\n",
      "We are at 889: Processing BIOMD0000000372.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000372.xml\n",
      "We are at 890: Processing BIOMD0000001044.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001044.xml\n",
      "We are at 891: Processing BIOMD0000000428.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000428.xml\n",
      "We are at 892: Processing BIOMD0000000399.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000399.xml\n",
      "We are at 893: Processing BIOMD0000000212.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000212.xml\n",
      "We are at 894: Processing BIOMD0000000574.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000574.xml\n",
      "We are at 895: Processing BIOMD0000000560.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000560.xml\n",
      "We are at 896: Processing BIOMD0000000206.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000206.xml\n",
      "We are at 897: Processing BIOMD0000000548.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000548.xml\n",
      "We are at 898: Processing BIOMD0000000776.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000776.xml\n",
      "We are at 899: Processing BIOMD0000000010.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000010.xml\n",
      "We are at 900: Processing BIOMD0000000004.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000004.xml\n",
      "We are at 901: Processing BIOMD0000000762.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000762.xml\n",
      "We are at 902: Processing BIOMD0000000038.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000038.xml\n",
      "We are at 903: Processing BIOMD0000000986.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000986.xml\n",
      "We are at 904: Processing BIOMD0000000979.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000979.xml\n",
      "We are at 905: Processing BIOMD0000000951.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000951.xml\n",
      "We are at 906: Processing BIOMD0000000789.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000789.xml\n",
      "We are at 907: Processing BIOMD0000000945.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000945.xml\n",
      "We are at 908: Processing BIOMD0000000944.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000944.xml\n",
      "We are at 909: Processing BIOMD0000000788.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000788.xml\n",
      "We are at 910: Processing BIOMD0000000950.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000950.xml\n",
      "We are at 911: Processing BIOMD0000000978.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000978.xml\n",
      "We are at 912: Processing BIOMD0000000039.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000039.xml\n",
      "We are at 913: Processing BIOMD0000000987.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000987.xml\n",
      "We are at 914: Processing BIOMD0000000005.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000005.xml\n",
      "We are at 915: Processing BIOMD0000000763.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000763.xml\n",
      "We are at 916: Processing BIOMD0000000777.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000777.xml\n",
      "We are at 917: Processing BIOMD0000000011.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000011.xml\n",
      "We are at 918: Processing BIOMD0000000549.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000549.xml\n",
      "We are at 919: Processing BIOMD0000000561.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000561.xml\n",
      "We are at 920: Processing BIOMD0000000207.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000207.xml\n",
      "We are at 921: Processing BIOMD0000000213.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000213.xml\n",
      "We are at 922: Processing BIOMD0000000575.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000575.xml\n",
      "We are at 923: Processing BIOMD0000000398.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000398.xml\n",
      "We are at 924: Processing BIOMD0000000429.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000429.xml\n",
      "We are at 925: Processing BIOMD0000001045.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001045.xml\n",
      "We are at 926: Processing BIOMD0000000415.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000415.xml\n",
      "We are at 927: Processing BIOMD0000000373.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000373.xml\n",
      "We are at 928: Processing BIOMD0000000367.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000367.xml\n",
      "We are at 929: Processing BIOMD0000001079.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001079.xml\n",
      "We are at 930: Processing BIOMD0000000401.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000401.xml\n",
      "We are at 931: Processing BIOMD0000000830.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000830.xml\n",
      "We are at 932: Processing BIOMD0000000824.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000824.xml\n",
      "We are at 933: Processing BIOMD0000000818.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000818.xml\n",
      "We are at 934: Processing BIOMD0000000159.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000159.xml\n",
      "We are at 935: Processing BIOMD0000000171.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000171.xml\n",
      "We are at 936: Processing BIOMD0000000617.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000617.xml\n",
      "We are at 937: Processing BIOMD0000000603.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000603.xml\n",
      "We are at 938: Processing BIOMD0000000165.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000165.xml\n",
      "We are at 939: Processing BIOMD0000000149.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000149.xml\n",
      "We are at 940: Processing BIOMD0000000161.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000161.xml\n",
      "We are at 941: Processing BIOMD0000000607.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000607.xml\n",
      "We are at 942: Processing BIOMD0000000613.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000613.xml\n",
      "We are at 943: Processing BIOMD0000000175.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000175.xml\n",
      "We are at 944: Processing BIOMD0000000820.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000820.xml\n",
      "We are at 945: Processing BIOMD0000000834.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000834.xml\n",
      "We are at 946: Processing BIOMD0000000808.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000808.xml\n",
      "We are at 947: Processing BIOMD0000000439.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000439.xml\n",
      "We are at 948: Processing BIOMD0000001041.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001041.xml\n",
      "We are at 949: Processing BIOMD0000001055.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001055.xml\n",
      "We are at 950: Processing BIOMD0000000405.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000405.xml\n",
      "We are at 951: Processing BIOMD0000000363.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000363.xml\n",
      "We are at 952: Processing BIOMD0000000377.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000377.xml\n",
      "We are at 953: Processing BIOMD0000000411.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000411.xml\n",
      "We are at 954: Processing BIOMD0000000388.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000388.xml\n",
      "We are at 955: Processing BIOMD0000000559.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000559.xml\n",
      "We are at 956: Processing BIOMD0000000571.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000571.xml\n",
      "We are at 957: Processing BIOMD0000000217.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000217.xml\n",
      "We are at 958: Processing BIOMD0000000203.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000203.xml\n",
      "We are at 959: Processing BIOMD0000000565.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000565.xml\n",
      "We are at 960: Processing BIOMD0000000029.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000029.xml\n",
      "We are at 961: Processing BIOMD0000000997.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000997.xml\n",
      "We are at 962: Processing BIOMD0000000983.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000983.xml\n",
      "We are at 963: Processing BIOMD0000000015.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000015.xml\n",
      "We are at 964: Processing BIOMD0000000773.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000773.xml\n",
      "We are at 965: Processing BIOMD0000000767.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000767.xml\n",
      "We are at 966: Processing BIOMD0000000001.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000001.xml\n",
      "We are at 967: Processing BIOMD0000000954.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000954.xml\n",
      "We are at 968: Processing BIOMD0000000940.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000940.xml\n",
      "We are at 969: Processing BIOMD0000000798.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000798.xml\n",
      "We are at 970: Processing BIOMD0000000968.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000968.xml\n",
      "We are at 971: Processing BIOMD0000000969.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000969.xml\n",
      "We are at 972: Processing BIOMD0000000799.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000799.xml\n",
      "We are at 973: Processing BIOMD0000000941.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000941.xml\n",
      "We are at 974: Processing BIOMD0000000955.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000955.xml\n",
      "We are at 975: Processing BIOMD0000000766.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000766.xml\n",
      "We are at 976: Processing BIOMD0000000014.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000014.xml\n",
      "We are at 977: Processing BIOMD0000000772.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000772.xml\n",
      "We are at 978: Processing BIOMD0000000982.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000982.xml\n",
      "We are at 979: Processing BIOMD0000000028.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000028.xml\n",
      "We are at 980: Processing BIOMD0000000996.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000996.xml\n",
      "We are at 981: Processing BIOMD0000000202.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000202.xml\n",
      "We are at 982: Processing BIOMD0000000564.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000564.xml\n",
      "We are at 983: Processing BIOMD0000000570.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000570.xml\n",
      "We are at 984: Processing BIOMD0000000216.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000216.xml\n",
      "We are at 985: Processing BIOMD0000000558.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000558.xml\n",
      "We are at 986: Processing BIOMD0000000389.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000389.xml\n",
      "We are at 987: Processing BIOMD0000000376.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000376.xml\n",
      "We are at 988: Processing BIOMD0000000410.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000410.xml\n",
      "We are at 989: Processing BIOMD0000000404.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000404.xml\n",
      "We are at 990: Processing BIOMD0000000362.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000362.xml\n",
      "We are at 991: Processing BIOMD0000001054.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001054.xml\n",
      "We are at 992: Processing BIOMD0000000438.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000438.xml\n",
      "We are at 993: Processing BIOMD0000001040.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001040.xml\n",
      "We are at 994: Processing BIOMD0000000809.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000809.xml\n",
      "We are at 995: Processing BIOMD0000000835.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000835.xml\n",
      "We are at 996: Processing BIOMD0000000821.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000821.xml\n",
      "We are at 997: Processing BIOMD0000000612.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000612.xml\n",
      "We are at 998: Processing BIOMD0000000174.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000174.xml\n",
      "We are at 999: Processing BIOMD0000000160.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000160.xml\n",
      "We are at 1000: Processing BIOMD0000000606.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000606.xml\n",
      "We are at 1001: Processing BIOMD0000000148.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000148.xml\n",
      "We are at 1002: Processing BIOMD0000000638.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000638.xml\n",
      "We are at 1003: Processing BIOMD0000000176.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000176.xml\n",
      "We are at 1004: Processing BIOMD0000000610.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000610.xml\n",
      "We are at 1005: Processing BIOMD0000000604.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000604.xml\n",
      "We are at 1006: Processing BIOMD0000000162.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000162.xml\n",
      "We are at 1007: Processing BIOMD0000000189.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000189.xml\n",
      "We are at 1008: Processing BIOMD0000000837.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000837.xml\n",
      "We are at 1009: Processing BIOMD0000000823.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000823.xml\n",
      "We are at 1010: Processing BIOMD0000001056.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001056.xml\n",
      "We are at 1011: Processing BIOMD0000000348.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000348.xml\n",
      "We are at 1012: Processing BIOMD0000001042.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001042.xml\n",
      "We are at 1013: Processing BIOMD0000000412.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000412.xml\n",
      "We are at 1014: Processing BIOMD0000000374.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000374.xml\n",
      "We are at 1015: Processing BIOMD0000000360.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000360.xml\n",
      "We are at 1016: Processing BIOMD0000000406.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000406.xml\n",
      "We are at 1017: Processing BIOMD0000000228.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000228.xml\n",
      "We are at 1018: Processing BIOMD0000000566.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000566.xml\n",
      "We are at 1019: Processing BIOMD0000000200.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000200.xml\n",
      "We are at 1020: Processing BIOMD0000000214.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000214.xml\n",
      "We are at 1021: Processing BIOMD0000000572.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000572.xml\n",
      "We are at 1022: Processing BIOMD0000000599.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000599.xml\n",
      "We are at 1023: Processing BIOMD0000000758.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000758.xml\n",
      "We are at 1024: Processing BIOMD0000000980.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000980.xml\n",
      "We are at 1025: Processing BIOMD0000000994.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000994.xml\n",
      "We are at 1026: Processing BIOMD0000000002.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000002.xml\n",
      "We are at 1027: Processing BIOMD0000000764.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000764.xml\n",
      "We are at 1028: Processing BIOMD0000000770.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000770.xml\n",
      "We are at 1029: Processing BIOMD0000000016.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000016.xml\n",
      "We are at 1030: Processing BIOMD0000000943.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000943.xml\n",
      "We are at 1031: Processing BIOMD0000000957.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000957.xml\n",
      "We are at 1032: Processing BIOMD0000000956.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000956.xml\n",
      "We are at 1033: Processing BIOMD0000000942.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000942.xml\n",
      "We are at 1034: Processing BIOMD0000000771.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000771.xml\n",
      "We are at 1035: Processing BIOMD0000000017.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000017.xml\n",
      "We are at 1036: Processing BIOMD0000000003.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000003.xml\n",
      "We are at 1037: Processing BIOMD0000000765.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000765.xml\n",
      "We are at 1038: Processing BIOMD0000000995.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000995.xml\n",
      "We are at 1039: Processing BIOMD0000000981.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000981.xml\n",
      "We are at 1040: Processing BIOMD0000000759.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000759.xml\n",
      "We are at 1041: Processing BIOMD0000000598.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000598.xml\n",
      "We are at 1042: Processing BIOMD0000000215.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000215.xml\n",
      "We are at 1043: Processing BIOMD0000000573.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000573.xml\n",
      "We are at 1044: Processing BIOMD0000000567.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000567.xml\n",
      "We are at 1045: Processing BIOMD0000000201.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000201.xml\n",
      "We are at 1046: Processing BIOMD0000000229.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000229.xml\n",
      "We are at 1047: Processing BIOMD0000001080.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001080.xml\n",
      "We are at 1048: Processing BIOMD0000000361.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000361.xml\n",
      "We are at 1049: Processing BIOMD0000000407.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000407.xml\n",
      "We are at 1050: Processing BIOMD0000000413.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000413.xml\n",
      "We are at 1051: Processing BIOMD0000000375.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000375.xml\n",
      "We are at 1052: Processing BIOMD0000001043.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001043.xml\n",
      "We are at 1053: Processing BIOMD0000000349.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000349.xml\n",
      "We are at 1054: Processing BIOMD0000001057.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000001057.xml\n",
      "We are at 1055: Processing BIOMD0000000822.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000822.xml\n",
      "We are at 1056: Processing BIOMD0000000188.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000188.xml\n",
      "We are at 1057: Processing BIOMD0000000836.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000836.xml\n",
      "We are at 1058: Processing BIOMD0000000605.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000605.xml\n",
      "We are at 1059: Processing BIOMD0000000163.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000163.xml\n",
      "We are at 1060: Processing BIOMD0000000177.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000177.xml\n",
      "We are at 1061: Processing BIOMD0000000611.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000611.xml\n",
      "We are at 1062: Processing BIOMD0000000639.xml\n",
      "The model does not use the FBC package: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000639.xml\n",
      "Overall GeneProduct statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Results/biomd_genes_ontology_statistics.csv\n",
      "Model-specific GeneProduct statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Results/biomd_genes_model_ontology_statistics.csv\n"
     ]
    }
   ],
   "source": [
    "output_csv = \"/Users/luna/Desktop/CRBM/AMAS_proj/Results/biomd_genes_ontology_statistics.csv\"\n",
    "model_ontology_csv = \"/Users/luna/Desktop/CRBM/AMAS_proj/Results/biomd_genes_model_ontology_statistics.csv\"\n",
    "collect_gene_product_statistics(biomd_model_dir, output_csv, model_ontology_csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model</th>\n",
       "      <th>ontology_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>refseq_synonym</td>\n",
       "      <td>4081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>refseq_locus_tag</td>\n",
       "      <td>1433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>ncbigi</td>\n",
       "      <td>1431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>refseq_name</td>\n",
       "      <td>1373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>asap</td>\n",
       "      <td>1372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>ecogene</td>\n",
       "      <td>1371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>ncbigene</td>\n",
       "      <td>1371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>uniprot</td>\n",
       "      <td>1371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>BIOMD0000001062.xml</td>\n",
       "      <td>refseq_old_locus_tag</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>BIOMD0000001046.xml</td>\n",
       "      <td>kegg.genes</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BIOMD0000001046.xml</td>\n",
       "      <td>ncbigene</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>BIOMD0000001046.xml</td>\n",
       "      <td>ncbiprotein</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>BIOMD0000001046.xml</td>\n",
       "      <td>uniprot</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  model         ontology_type  count\n",
       "6   BIOMD0000001062.xml        refseq_synonym   4081\n",
       "4   BIOMD0000001062.xml      refseq_locus_tag   1433\n",
       "3   BIOMD0000001062.xml                ncbigi   1431\n",
       "5   BIOMD0000001062.xml           refseq_name   1373\n",
       "0   BIOMD0000001062.xml                  asap   1372\n",
       "1   BIOMD0000001062.xml               ecogene   1371\n",
       "2   BIOMD0000001062.xml              ncbigene   1371\n",
       "7   BIOMD0000001062.xml               uniprot   1371\n",
       "8   BIOMD0000001062.xml  refseq_old_locus_tag     62\n",
       "9   BIOMD0000001046.xml            kegg.genes     25\n",
       "10  BIOMD0000001046.xml              ncbigene     25\n",
       "11  BIOMD0000001046.xml           ncbiprotein     25\n",
       "12  BIOMD0000001046.xml               uniprot     25"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bigg_genes_ontology_statistics = pd.read_csv(\"/Users/luna/Desktop/CRBM/AMAS_proj/Results/biomd_genes_model_ontology_statistics.csv\")\n",
    "bigg_genes_ontology_statistics.sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create reference files for NCBI Gene\n",
    "### For all organisms\n",
    "\n",
    "Data are obtained from the NCBI gene FTP site: https://ftp.ncbi.nih.gov/gene/DATA/GENE_INFO/."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tax_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Synonyms</th>\n",
       "      <th>type_of_gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7</td>\n",
       "      <td>5692769</td>\n",
       "      <td>NEWENTRY</td>\n",
       "      <td>-</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>2827857</td>\n",
       "      <td>NEWENTRY</td>\n",
       "      <td>-</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>10823747</td>\n",
       "      <td>NEWENTRY</td>\n",
       "      <td>-</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14</td>\n",
       "      <td>6951813</td>\n",
       "      <td>NEWENTRY</td>\n",
       "      <td>-</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>19</td>\n",
       "      <td>3758873</td>\n",
       "      <td>NEWENTRY</td>\n",
       "      <td>-</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #tax_id    GeneID    Symbol Synonyms type_of_gene\n",
       "0        7   5692769  NEWENTRY        -        other\n",
       "1        9   2827857  NEWENTRY        -        other\n",
       "2       11  10823747  NEWENTRY        -        other\n",
       "3       14   6951813  NEWENTRY        -        other\n",
       "4       19   3758873  NEWENTRY        -        other"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# All gene data\n",
    "input_file = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/All_Data.gene_info.gz\"\n",
    "columns_to_keep = [\"#tax_id\", \"GeneID\", \"Symbol\", \"Synonyms\",\"type_of_gene\"]\n",
    "df = pd.read_csv(input_file, compression='gzip', sep='\\t',usecols=columns_to_keep)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type_of_gene\n",
      "protein-coding       46235835\n",
      "ncRNA                 3480011\n",
      "tRNA                  3033345\n",
      "pseudo                2737966\n",
      "rRNA                   995495\n",
      "snRNA                  586439\n",
      "snoRNA                 505390\n",
      "biological-region      168474\n",
      "other                  108372\n",
      "unknown                 61023\n",
      "miscRNA                  5128\n",
      "scRNA                      21\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# Exclude certain types of genes\n",
    "print(df['type_of_gene'].value_counts())\n",
    "# exclude_types = ['biological-region', 'unknown', 'other']\n",
    "# df = df[~df['type_of_gene'].isin(exclude_types) & (df['Symbol'] != 'NEWENTRY')]\n",
    "\n",
    "# Only include protein-coding genes\n",
    "include_types = ['protein-coding']\n",
    "df = df[df['type_of_gene'].isin(include_types)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GeneID to Symbol saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/ncbigene2label_all_protein_coding.lzma\n"
     ]
    }
   ],
   "source": [
    "output_dir = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/\"\n",
    "output_gene2label = \"ncbigene2label_all_protein_coding.lzma\"\n",
    "gene_id_to_symbol = dict(zip(df['GeneID'], df['Symbol']))\n",
    "with open(os.path.join(output_dir,output_gene2label), 'wb') as handle:\n",
    "    compress_pickle.dump(gene_id_to_symbol, handle, compression=\"lzma\", set_default_extension=False)\n",
    "print(f\"GeneID to Symbol saved to {output_dir+output_gene2label}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This file is too large, only the dictionary takes 99.4MB. I'll go with the organisms in Bigg models first."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of organisms contained in NCBI gene info: 18\n",
      "Number of rows: 146842\n",
      "All gene info of bigg organisms saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/ncbi_gene_info_bigg_organisms_protein_coding.csv\n"
     ]
    }
   ],
   "source": [
    "bigg_organisms = pd.read_csv(bigg_model_dir + \"bigg_organisms.csv\")\n",
    "taxids = bigg_organisms['tax_id'].unique().tolist()\n",
    "taxids = [x for x in taxids if x and str(x).strip()] # Remove empty strings\n",
    "\n",
    "# Only include rows where tax_id is in taxids\n",
    "df['#tax_id'] = df['#tax_id'].astype(str)\n",
    "bigg_df = df[df['#tax_id'].isin(taxids)]\n",
    "bigg_df = bigg_df.map(str)\n",
    "output_geneinfo_bigg = \"ncbi_gene_info_bigg_organisms_protein_coding.csv\"\n",
    "bigg_df.to_csv(output_dir + output_geneinfo_bigg, index=False)\n",
    "\n",
    "print(f\"Number of organisms contained in NCBI gene info: {bigg_df['#tax_id'].nunique()}\")\n",
    "print(f\"Number of rows: {len(bigg_df)}\")\n",
    "bigg_df.head()\n",
    "print(f\"All gene info of bigg organisms saved to {output_dir + output_geneinfo_bigg}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GeneID to Symbol saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/ncbigene2label_bigg_organisms_protein_coding.lzma\n"
     ]
    }
   ],
   "source": [
    "output_gene2label = \"ncbigene2label_bigg_organisms_protein_coding.lzma\"\n",
    "gene_id_to_symbol = dict(zip(bigg_df['GeneID'], bigg_df['Symbol']))\n",
    "with open(os.path.join(output_dir,output_gene2label), 'wb') as handle:\n",
    "    compress_pickle.dump(gene_id_to_symbol, handle, compression=\"lzma\", set_default_extension=False)\n",
    "print(f\"GeneID to Symbol saved to {output_dir+output_gene2label}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The GeneID to Symbol dictionary for bigg organisms takes 502KB, which is fine.  \n",
    "But the character count takes significant memory and casues kernel to crash.  \n",
    "  \n",
    "I'll save the character count for each organism separately.   \n",
    "They will be located at the `files/charcount_gene_df_scaled` folder."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes for 511145 (Escherichia coli str. K-12 substr. MG1655): 4288\n",
      "Number of genes for 9606 (Homo sapiens): 20595\n",
      "No genes found for tax_id 269797(Methanosarcina barkeri str. Fusaro)\n",
      "No genes found for tax_id 269799(Geobacter metallireducens GS-15)\n",
      "Number of genes for 5821 (Plasmodium berghei): 3\n",
      "Number of genes for 1120755 (Plasmodium cynomolgi strain B): 5716\n",
      "Number of genes for 36329 (Plasmodium falciparum 3D7): 5282\n",
      "Number of genes for 5851 (Plasmodium knowlesi strain H): 5321\n",
      "Number of genes for 126793 (Plasmodium vivax Sal-1): 3\n",
      "No genes found for tax_id 405955(Escherichia coli APEC O1)\n",
      "No genes found for tax_id 469008(Escherichia coli BL21(DE3))\n",
      "No genes found for tax_id 595496(Escherichia coli BW2952)\n",
      "No genes found for tax_id 199310(Escherichia coli CFT073)\n",
      "Number of genes for 10029 (Cricetulus griseus): 22956\n",
      "No genes found for tax_id 509173(Acinetobacter baumannii AYE)\n",
      "No genes found for tax_id 272563(Clostridioides difficile 630)\n",
      "No genes found for tax_id 574521(Escherichia coli O127:H6 str. E2348/69)\n",
      "No genes found for tax_id 216592(Escherichia coli 042)\n",
      "No genes found for tax_id 498388(Escherichia coli C)\n",
      "No genes found for tax_id 481805(Escherichia coli ATCC 8739)\n",
      "No genes found for tax_id 566546(Escherichia coli W)\n",
      "No genes found for tax_id 668369(Escherichia coli DH5[alpha])\n",
      "No genes found for tax_id 316407(Escherichia coli str. K-12 substr. W3110)\n",
      "No genes found for tax_id 585055(Escherichia coli 55989)\n",
      "No genes found for tax_id 655817(Escherichia coli ABU 83972)\n",
      "No genes found for tax_id 413997(Escherichia coli B str. REL606)\n",
      "No genes found for tax_id 866768(Escherichia coli 'BL21-Gold(DE3)pLysS AG')\n",
      "No genes found for tax_id 316385(Escherichia coli str. K-12 substr. DH10B)\n",
      "No genes found for tax_id 536056(Escherichia coli DH1)\n",
      "No genes found for tax_id 331111(Escherichia coli O139:H28 str. E24377A)\n",
      "No genes found for tax_id 585397(Escherichia coli ED1a)\n",
      "No genes found for tax_id 444450(Escherichia coli O157:H7 str. EC4115)\n",
      "No genes found for tax_id 331112(Escherichia coli HS)\n",
      "No genes found for tax_id 585034(Escherichia coli IAI1)\n",
      "No genes found for tax_id 585057(Escherichia coli IAI39)\n",
      "No genes found for tax_id 1033813(Escherichia coli NA114)\n",
      "No genes found for tax_id 585395(Escherichia coli O103:H2 str. 12009)\n",
      "No genes found for tax_id 585396(Escherichia coli O111:H- str. 11128)\n",
      "No genes found for tax_id 573235(Escherichia coli O26:H11 str. 11368)\n",
      "No genes found for tax_id 714962(Escherichia coli IHE3034)\n",
      "No genes found for tax_id 362663(Escherichia coli 536)\n",
      "Number of genes for 386585 (Escherichia coli O157:H7 str. Sakai): 5155\n",
      "No genes found for tax_id 585035(Escherichia coli S88)\n",
      "No genes found for tax_id 409438(Escherichia coli SE11)\n",
      "No genes found for tax_id 431946(Escherichia coli SE15)\n",
      "No genes found for tax_id 439855(Escherichia coli SMS-3-5)\n",
      "No genes found for tax_id 544404(Escherichia coli O157:H7 str. TW14359)\n",
      "No genes found for tax_id 585056(Escherichia coli UMN026)\n",
      "Number of genes for 83332 (Mycobacterium tuberculosis H37Rv): 3906\n",
      "No genes found for tax_id 595495(Escherichia coli KO11FL)\n",
      "No genes found for tax_id 316401(Escherichia coli ETEC H10407)\n",
      "No genes found for tax_id 701177(Escherichia coli O55:H7 str. CB9615)\n",
      "No genes found for tax_id 748727(Clostridium ljungdahlii DSM 13528)\n",
      "No genes found for tax_id 1416333(Trypanosoma cruzi Dm28c)\n",
      "No genes found for tax_id 85962(Helicobacter pylori 26695)\n",
      "No genes found for tax_id 1140(Synechococcus elongatus PCC 7942)\n",
      "No genes found for tax_id 160488(Pseudomonas putida KT2440)\n",
      "No genes found for tax_id 1148(Synechocystis sp. PCC 6803)\n",
      "Number of genes for 556484 (Phaeodactylum tricornutum CCAP 1055/1): 10392\n",
      "No genes found for tax_id 591946(Escherichia coli LF82)\n",
      "No genes found for tax_id 243274(Thermotoga maritima MSB8)\n",
      "Number of genes for 10090 (Mus musculus): 26251\n",
      "Number of genes for 559292 (Saccharomyces cerevisiae S288C): 6021\n",
      "No genes found for tax_id 416870(Lactococcus lactis subsp. cremoris MG1363)\n",
      "No genes found for tax_id 685038(Escherichia coli O83:H1 str. NRG 857C)\n",
      "No genes found for tax_id 214092(Yersinia pestis CO92)\n",
      "Number of genes for 3055 (Chlamydomonas reinhardtii): 17819\n",
      "No genes found for tax_id 198215(Shigella flexneri 2a str. 2457T)\n",
      "No genes found for tax_id 158879(Staphylococcus aureus subsp. aureus N315)\n",
      "Number of genes for 344609 (Shigella boydii CDC 3083-94): 30\n",
      "No genes found for tax_id 300268(Shigella boydii Sb227)\n",
      "No genes found for tax_id 300267(Shigella dysenteriae Sd197)\n",
      "Number of genes for 198214 (Shigella flexneri 2a str. 301): 4313\n",
      "No genes found for tax_id 373384(Shigella flexneri 5 str. 8401)\n",
      "No genes found for tax_id 591020(Shigella flexneri 2002017)\n",
      "No genes found for tax_id 300269(Shigella sonnei Ss046)\n",
      "No genes found for tax_id 869729(Escherichia coli UM146)\n",
      "No genes found for tax_id 696406(Escherichia coli UMNK88)\n",
      "No genes found for tax_id 364106(Escherichia coli UTI89)\n",
      "No genes found for tax_id 272620(Klebsiella pneumoniae subsp. pneumoniae MGH 78578)\n",
      "Number of genes for 224308 (Bacillus subtilis subsp. subtilis str. 168): 4237\n",
      "No genes found for tax_id  (Salmonella pan-reactome)\n",
      "No genes found for tax_id 451516(Staphylococcus aureus subsp. aureus USA300_TCH1516)\n",
      "No genes found for tax_id 155864(Escherichia coli O157:H7 str. EDL933)\n",
      "Number of genes for 99287 (Salmonella enterica subsp. enterica serovar Typhimurium str. LT2): 4554\n",
      "Statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/organism_gene_statistics.csv\n",
      "Character count for bigg organisms saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/charcount_gene_df_scaled_bigg_organisms.lzma\n"
     ]
    }
   ],
   "source": [
    "# Split the Synonyms column and include the Symbol in the list of synonyms\n",
    "def expand_row(row):\n",
    "    \"\"\"\n",
    "    Expand a row to include the Symbol and each of its synonyms as separate rows.\n",
    "    \"\"\"\n",
    "    synonyms = row[\"Synonyms\"].split('|') if pd.notna(row[\"Synonyms\"]) else []\n",
    "    synonyms.append(row[\"Symbol\"])  # Add the Symbol as part of synonyms\n",
    "    return [(row[\"#tax_id\"], row[\"GeneID\"], synonym) for synonym in synonyms]\n",
    "# Function to calculate scaled character frequency\n",
    "def scaled_char_frequency_row(row, char_columns):\n",
    "    \"\"\"\n",
    "    Generate a scaled frequency row for each synonym.\n",
    "    \"\"\"\n",
    "    synonym = str(row['Synonym']).lower()  # Convert to lowercase for case insensitivity\n",
    "    char_counts = {char: synonym.count(char) for char in char_columns}  # Count each character\n",
    "    total_count = sum(char_counts.values())  # Total characters\n",
    "    vector_length = np.sqrt(sum(value ** 2 for value in char_counts.values()))  # Vector length (L2 norm)\n",
    "    if vector_length == 0:\n",
    "        # Avoid division by zero; all frequencies are 0 if the vector is empty\n",
    "        scaled_counts = {char: 0 for char in char_columns}\n",
    "    else:\n",
    "        scaled_counts = {char: count / vector_length for char, count in char_counts.items()}  # Scale by vector length\n",
    "    scaled_counts['tax_id'] = row['#tax_id']\n",
    "    scaled_counts['synonym'] = row['Synonym']\n",
    "    scaled_counts['ncbigene'] = row['GeneID']\n",
    "    return scaled_counts\n",
    "    \n",
    "def process_taxid_file(tax_id, organism, input_df, char_columns, output_dir):\n",
    "    \"\"\"\n",
    "    Process data for a specific tax_id, generate character frequency, and save to a compressed file.\n",
    "    \"\"\"\n",
    "    # Filter rows for the specific tax_id\n",
    "    taxid_df = input_df[input_df['#tax_id'] == tax_id]\n",
    "    print(f\"Number of genes for {tax_id} ({organism}): {len(taxid_df)}\")\n",
    "    # Expand the rows\n",
    "    expanded_data = [item for _, row in taxid_df.iterrows() for item in expand_row(row)]\n",
    "    expanded_df = pd.DataFrame(expanded_data, columns=[\"#tax_id\", \"GeneID\", \"Synonym\"])\n",
    "    \n",
    "    # Compute character frequency\n",
    "    char_freq_data = [scaled_char_frequency_row(row, char_columns) for _, row in expanded_df.iterrows()]\n",
    "    char_freq_df = pd.DataFrame(char_freq_data)\n",
    "    \n",
    "    # Save to a compressed file\n",
    "    output_file = os.path.join(output_dir, f\"charcount_gene_df_scaled_{tax_id}.lzma\")\n",
    "    compress_pickle.dump(char_freq_df, output_file, compression=\"lzma\")\n",
    "\n",
    "def charcount_individual_organisms(df, bigg_organisms, output_dir):\n",
    "    bigg_organisms_unique = bigg_organisms.drop_duplicates(subset='tax_id')\n",
    "\n",
    "    # Define character columns (a-z and 0-9)\n",
    "    letters = list(string.ascii_lowercase)\n",
    "    numbers = list(string.digits)\n",
    "    char_columns = letters + numbers\n",
    "\n",
    "    # Initialize a list to store statistics\n",
    "    stats = []\n",
    "\n",
    "    # Iterate through each tax_id and process\n",
    "    for _, organism_row in bigg_organisms_unique.iterrows():\n",
    "        tax_id = str(organism_row['tax_id'])\n",
    "        organism = organism_row['organism']\n",
    "        \n",
    "        # Filter rows for the specific tax_id\n",
    "        taxid_df = df[df['#tax_id'] == tax_id]\n",
    "        num_genes = len(taxid_df)\n",
    "        stats.append({'tax_id': tax_id, 'organism': organism, 'num_genes': num_genes})\n",
    "\n",
    "        if num_genes == 0:\n",
    "            print(f\"No genes found for tax_id {tax_id}({organism})\")\n",
    "            continue\n",
    "        \n",
    "        # Process the file for character frequency and save\n",
    "        process_taxid_file(tax_id, organism, df, char_columns, output_dir)\n",
    "\n",
    "    # Create a summary DataFrame\n",
    "    stats_df = pd.DataFrame(stats)\n",
    "    stats_file = os.path.join(output_dir, \"organism_gene_statistics.csv\")\n",
    "    stats_df.to_csv(stats_file, index=False)\n",
    "    print(f\"Statistics saved to {stats_file}\")\n",
    "    return stats_df\n",
    "\n",
    "bigg_organisms = pd.read_csv(bigg_model_dir + \"bigg_organisms.csv\")\n",
    "output_dir = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/\"\n",
    "df = pd.read_csv(output_dir + \"ncbi_gene_info_bigg_organisms_protein_coding.csv\")\n",
    "df['#tax_id'] = df['#tax_id'].astype(str)\n",
    "\n",
    "charcount_individual_organisms(df, bigg_organisms, output_dir)\n",
    "print(f\"Character count for bigg organisms saved to {output_dir + 'charcount_gene_df_scaled_bigg_organisms.lzma'}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "REF_TAXID2LABEL saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/taxid2label.lzma\n"
     ]
    }
   ],
   "source": [
    "## REF_TAXID2LABEL\n",
    "output_dir = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/\"\n",
    "stats_df = pd.read_csv(output_dir + \"organism_gene_statistics.csv\")\n",
    "REF_TAXID2LABEL = dict(zip(stats_df['tax_id'], stats_df['organism']))\n",
    "\n",
    "REF_DIR = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/\"\n",
    "with open(REF_DIR + \"taxid2label.lzma\", 'wb') as f:\n",
    "    compress_pickle.dump(REF_TAXID2LABEL, f, compression=\"lzma\")\n",
    "print(f\"REF_TAXID2LABEL saved to {REF_DIR + 'taxid2label.lzma'}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### For individual organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Split the Synonyms column and include the Symbol in the list of synonyms\n",
    "def expand_row(row):\n",
    "    \"\"\"\n",
    "    Expand a row to include the Symbol and each of its synonyms as separate rows.\n",
    "    \"\"\"\n",
    "    synonyms = row[\"Synonyms\"].split('|') if pd.notna(row[\"Synonyms\"]) else []\n",
    "    synonyms.append(row[\"Symbol\"])  # Add the Symbol as part of synonyms\n",
    "    return [(row[\"#tax_id\"], row[\"GeneID\"], synonym) for synonym in synonyms]\n",
    "\n",
    "# Function to calculate scaled character frequency\n",
    "def scaled_char_frequency_row(row, char_columns):\n",
    "    \"\"\"\n",
    "    Generate a scaled frequency row for each synonym.\n",
    "    \"\"\"\n",
    "    synonym = str(row['Synonym']).lower()  # Convert to lowercase for case insensitivity\n",
    "    char_counts = {char: synonym.count(char) for char in char_columns}  # Count each character\n",
    "    total_count = sum(char_counts.values())  # Total characters\n",
    "    vector_length = np.sqrt(sum(value ** 2 for value in char_counts.values()))  # Vector length (L2 norm)\n",
    "    if vector_length == 0:\n",
    "        # Avoid division by zero; all frequencies are 0 if the vector is empty\n",
    "        scaled_counts = {char: 0 for char in char_columns}\n",
    "    else:\n",
    "        scaled_counts = {char: count / vector_length for char, count in char_counts.items()}  # Scale by vector length\n",
    "    scaled_counts['tax_id'] = row['#tax_id']\n",
    "    scaled_counts['synonym'] = row['Synonym']\n",
    "    scaled_counts['ncbigene'] = row['GeneID']\n",
    "    return scaled_counts\n",
    "\n",
    "# Process reference files\n",
    "def process_reference_files(input_file, output_dir, output_gene2label, output_synonyms, output_charcount):\n",
    "    ########### GeneID to Symbol\n",
    "    columns_to_keep = [\"#tax_id\", \"GeneID\", \"Symbol\", \"Synonyms\"]\n",
    "    df = pd.read_csv(input_file, compression='gzip', sep='\\t', usecols=columns_to_keep)\n",
    "    gene_id_to_symbol = dict(zip(df['GeneID'], df['Symbol']))\n",
    "    compress_pickle.dump(gene_id_to_symbol, output_dir+output_gene2label, compression=\"lzma\")\n",
    "    print(f\"GeneID to Symbol saved to {output_dir+output_gene2label}\")\n",
    "    \n",
    "    # ########### GeneID to Symbol and Synonyms\n",
    "    # gene_id_to_symbol_synonyms = {gene_id: (symbol, synonyms.split('|')) for gene_id, symbol, synonyms in zip(df['GeneID'], df['Symbol'], df['Synonyms'])}\n",
    "    # compress_pickle.dump(gene_id_to_symbol_synonyms, output_dir+output_synonyms, compression=\"lzma\")\n",
    "    # print(f\"GeneID to Symbol and Synonyms saved to {output_dir+output_synonyms}\")\n",
    "    \n",
    "    ########### GeneID to Character Counts\n",
    "    expanded_data = []\n",
    "    for _, row in df.iterrows():\n",
    "        expanded_data.extend(expand_row(row))\n",
    "    expanded_df = pd.DataFrame(expanded_data, columns=[\"#tax_id\", \"GeneID\", \"Synonym\"])\n",
    "\n",
    "    # Define the characters for the first 36 columns\n",
    "    letters = list(string.ascii_lowercase)  # a-z\n",
    "    numbers = list(string.digits)  # 0-9\n",
    "    char_columns = letters + numbers\n",
    "\n",
    "    char_freq_data = [scaled_char_frequency_row(row, char_columns) for _, row in expanded_df.iterrows()]\n",
    "    char_freq_df = pd.DataFrame(char_freq_data)\n",
    "    compress_pickle.dump(char_freq_df, output_dir+output_charcount, compression=\"lzma\")\n",
    "    print(f\"Scaled character frequency DataFrame saved to {output_dir+output_charcount}\")\n",
    "    return char_freq_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GeneID to Symbol saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/ncbigene2label_ecoli_mg1655.lzma\n",
      "Scaled character frequency DataFrame saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/charcount_gene_df_scaled_ecoli_mg1655.lzma\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>a</th>\n",
       "      <th>b</th>\n",
       "      <th>c</th>\n",
       "      <th>d</th>\n",
       "      <th>e</th>\n",
       "      <th>f</th>\n",
       "      <th>g</th>\n",
       "      <th>h</th>\n",
       "      <th>i</th>\n",
       "      <th>j</th>\n",
       "      <th>...</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>synonym</th>\n",
       "      <th>ncbigene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>511145</td>\n",
       "      <td>ECK4299</td>\n",
       "      <td>944740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>yjhR</td>\n",
       "      <td>944740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ECK0560</td>\n",
       "      <td>944741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>nfrA</td>\n",
       "      <td>944741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.277350</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.277350</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ECK0001</td>\n",
       "      <td>944742</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13007</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ytgB</td>\n",
       "      <td>71004580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13008</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ECK4680</td>\n",
       "      <td>71004581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13009</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ytiE</td>\n",
       "      <td>71004581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13010</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ECK4681</td>\n",
       "      <td>71004582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13011</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ytjE</td>\n",
       "      <td>71004582</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13012 rows Ã— 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         a    b         c    d         e    f    g    h    i    j  ...    3  \\\n",
       "0      0.0  0.0  0.333333  0.0  0.333333  0.0  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "1      0.0  0.0  0.000000  0.0  0.000000  0.0  0.0  0.5  0.0  0.5  ...  0.0   \n",
       "2      0.0  0.0  0.333333  0.0  0.333333  0.0  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "3      0.5  0.0  0.000000  0.0  0.000000  0.5  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "4      0.0  0.0  0.277350  0.0  0.277350  0.0  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "...    ...  ...       ...  ...       ...  ...  ...  ...  ...  ...  ...  ...   \n",
       "13007  0.0  0.5  0.000000  0.0  0.000000  0.0  0.5  0.0  0.0  0.0  ...  0.0   \n",
       "13008  0.0  0.0  0.377964  0.0  0.377964  0.0  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "13009  0.0  0.0  0.000000  0.0  0.500000  0.0  0.0  0.0  0.5  0.0  ...  0.0   \n",
       "13010  0.0  0.0  0.377964  0.0  0.377964  0.0  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "13011  0.0  0.0  0.000000  0.0  0.500000  0.0  0.0  0.0  0.0  0.5  ...  0.0   \n",
       "\n",
       "              4         5         6    7         8         9  tax_id  synonym  \\\n",
       "0      0.333333  0.000000  0.000000  0.0  0.000000  0.666667  511145  ECK4299   \n",
       "1      0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145     yjhR   \n",
       "2      0.000000  0.333333  0.333333  0.0  0.000000  0.000000  511145  ECK0560   \n",
       "3      0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145     nfrA   \n",
       "4      0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145  ECK0001   \n",
       "...         ...       ...       ...  ...       ...       ...     ...      ...   \n",
       "13007  0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145     ytgB   \n",
       "13008  0.377964  0.000000  0.377964  0.0  0.377964  0.000000  511145  ECK4680   \n",
       "13009  0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145     ytiE   \n",
       "13010  0.377964  0.000000  0.377964  0.0  0.377964  0.000000  511145  ECK4681   \n",
       "13011  0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145     ytjE   \n",
       "\n",
       "       ncbigene  \n",
       "0        944740  \n",
       "1        944740  \n",
       "2        944741  \n",
       "3        944741  \n",
       "4        944742  \n",
       "...         ...  \n",
       "13007  71004580  \n",
       "13008  71004581  \n",
       "13009  71004581  \n",
       "13010  71004582  \n",
       "13011  71004582  \n",
       "\n",
       "[13012 rows x 39 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Ecoli-MG1655\n",
    "input_file = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/Escherichia_coli_str._K-12_substr._MG1655.gene_info.gz\"\n",
    "output_dir = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/\"\n",
    "output_gene2label = \"ncbigene2label_ecoli_mg1655.lzma\"\n",
    "output_charcount = 'charcount_gene_df_scaled_ecoli_mg1655.lzma'\n",
    "process_reference_files(input_file, output_dir, output_gene2label, output_synonyms, output_charcount)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GeneID to Symbol saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/ncbigene2label_human.lzma\n",
      "Scaled character frequency DataFrame saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/charcount_gene_df_scaled_human.lzma\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>a</th>\n",
       "      <th>b</th>\n",
       "      <th>c</th>\n",
       "      <th>d</th>\n",
       "      <th>e</th>\n",
       "      <th>f</th>\n",
       "      <th>g</th>\n",
       "      <th>h</th>\n",
       "      <th>i</th>\n",
       "      <th>j</th>\n",
       "      <th>...</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>synonym</th>\n",
       "      <th>ncbigene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.577350</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9606</td>\n",
       "      <td>A1B</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.577350</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9606</td>\n",
       "      <td>ABG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.577350</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9606</td>\n",
       "      <td>GAB</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.316228</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.316228</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.632456</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9606</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9606</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433435</th>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>741158</td>\n",
       "      <td>trnA</td>\n",
       "      <td>8923217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433436</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>741158</td>\n",
       "      <td>-</td>\n",
       "      <td>8923218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433437</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>741158</td>\n",
       "      <td>COX1</td>\n",
       "      <td>8923218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433438</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>741158</td>\n",
       "      <td>-</td>\n",
       "      <td>8923219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433439</th>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>741158</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>8923219</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433440 rows Ã— 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               a        b    c    d    e    f        g         h    i    j  \\\n",
       "0       0.577350  0.57735  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "1       0.577350  0.57735  0.0  0.0  0.0  0.0  0.57735  0.000000  0.0  0.0   \n",
       "2       0.577350  0.57735  0.0  0.0  0.0  0.0  0.57735  0.000000  0.0  0.0   \n",
       "3       0.000000  0.00000  0.0  0.0  0.0  0.0  0.00000  0.316228  0.0  0.0   \n",
       "4       0.500000  0.50000  0.0  0.0  0.0  0.0  0.50000  0.000000  0.0  0.0   \n",
       "...          ...      ...  ...  ...  ...  ...      ...       ...  ...  ...   \n",
       "433435  0.500000  0.00000  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "433436  0.000000  0.00000  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "433437  0.000000  0.00000  0.5  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "433438  0.000000  0.00000  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "433439  0.333333  0.00000  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "\n",
       "        ...    3         4    5         6         7    8    9  tax_id  \\\n",
       "0       ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0    9606   \n",
       "1       ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0    9606   \n",
       "2       ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0    9606   \n",
       "3       ...  0.0  0.316228  0.0  0.000000  0.632456  0.0  0.0    9606   \n",
       "4       ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0    9606   \n",
       "...     ...  ...       ...  ...       ...       ...  ...  ...     ...   \n",
       "433435  ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0  741158   \n",
       "433436  ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0  741158   \n",
       "433437  ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0  741158   \n",
       "433438  ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0  741158   \n",
       "433439  ...  0.0  0.000000  0.0  0.333333  0.000000  0.0  0.0  741158   \n",
       "\n",
       "         synonym  ncbigene  \n",
       "0            A1B         1  \n",
       "1            ABG         1  \n",
       "2            GAB         1  \n",
       "3       HYST2477         1  \n",
       "4           A1BG         1  \n",
       "...          ...       ...  \n",
       "433435      trnA   8923217  \n",
       "433436         -   8923218  \n",
       "433437      COX1   8923218  \n",
       "433438         -   8923219  \n",
       "433439  16S rRNA   8923219  \n",
       "\n",
       "[433440 rows x 39 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Human\n",
    "input_file = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/Homo_sapiens.gene_info.gz\"\n",
    "output_dir = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/\"\n",
    "output_gene2label = \"ncbigene2label_human.lzma\"\n",
    "output_charcount = 'charcount_gene_df_scaled_human.lzma'\n",
    "process_reference_files(input_file, output_dir, output_gene2label, output_synonyms, output_charcount)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Combined gene2label dictionary saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/ncbigene2label.lzma\n",
      "Combined charcount DataFrame saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/charcount_gene_df_scaled.lzma\n"
     ]
    }
   ],
   "source": [
    "# Concatenate files for different taxonomies\n",
    "output_dir = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/\"\n",
    "output_gene2label_combined = \"ncbigene2label.lzma\"\n",
    "output_charcount_combined = \"charcount_gene_df_scaled.lzma\"\n",
    "\n",
    "# gene2label files for E. coli and human\n",
    "ecoli_gene2label = compress_pickle.load(output_dir + \"ncbigene2label_ecoli_mg1655.lzma\", compression=\"lzma\")\n",
    "human_gene2label = compress_pickle.load(output_dir + \"ncbigene2label_human.lzma\", compression=\"lzma\")\n",
    "combined_gene2label = {**ecoli_gene2label, **human_gene2label}\n",
    "with open(output_dir + output_gene2label_combined, 'wb') as f:\n",
    "    compress_pickle.dump(combined_gene2label, f, compression=\"lzma\")\n",
    "print(f\"Combined gene2label dictionary saved to {output_dir + output_gene2label_combined}\")\n",
    "\n",
    "# charcount files for E. coli and human\n",
    "ecoli_charcount = compress_pickle.load(output_dir + \"charcount_gene_df_scaled_ecoli_mg1655.lzma\", compression=\"lzma\")\n",
    "human_charcount = compress_pickle.load(output_dir + \"charcount_gene_df_scaled_human.lzma\", compression=\"lzma\")\n",
    "combined_charcount = pd.concat([ecoli_charcount, human_charcount])\n",
    "with open(output_dir + output_charcount_combined, 'wb') as f:\n",
    "    compress_pickle.dump(combined_charcount, f, compression=\"lzma\")\n",
    "print(f\"Combined charcount DataFrame saved to {output_dir + output_charcount_combined}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Test functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[        annotation  match score label\n",
       " G_b1241                              \n",
       " 0           945695          1.0  purB\n",
       " 1           945837          1.0  adhE\n",
       " 2           948025          1.0  hdeA,\n",
       "         annotation  match score label\n",
       " G_b0351                              \n",
       " 0           945008          1.0  mhpF,\n",
       "         annotation  match score label\n",
       " G_s0001                              \n",
       " 0           944742        0.836  thrL\n",
       " 1           944792        0.836  satP\n",
       " 2           944824        0.836  mutT\n",
       " 3           946586        0.836  rutD,\n",
       "         annotation  match score label\n",
       " G_b1849                              \n",
       " 0           946199          1.0  rstA\n",
       " 1           946368          1.0  purT,\n",
       "         annotation  match score label\n",
       " G_b3115                              \n",
       " 0           947635          1.0  tdcD,\n",
       "         annotation  match score label\n",
       " G_b2296                              \n",
       " 0           946775          1.0  ackA,\n",
       "         annotation  match score label\n",
       " G_b1276                              \n",
       " 0           946724          1.0  acnA,\n",
       "         annotation  match score label\n",
       " G_b0118                              \n",
       " 0           944864          1.0  acnB,\n",
       "         annotation  match score label\n",
       " G_b0474                              \n",
       " 0           945097          1.0   adk,\n",
       "         annotation  match score label\n",
       " G_b0116                              \n",
       " 0           944854          1.0   lpd,\n",
       "         annotation  match score label\n",
       " G_b0727                              \n",
       " 0           945189          1.0  cusB\n",
       " 1           945307          1.0  sucB,\n",
       "         annotation  match score label\n",
       " G_b0726                              \n",
       " 0           945191          1.0  cusA\n",
       " 1           945303          1.0  sucA\n",
       " 2           946793          1.0  truA,\n",
       "         annotation  match score label\n",
       " G_b2587                              \n",
       " 0           947069          1.0  kgtP,\n",
       "         annotation  match score label\n",
       " G_b0356                              \n",
       " 0           944988          1.0  frmA,\n",
       "         annotation  match score label\n",
       " G_b1478                              \n",
       " 0           945427          1.0  hcaD\n",
       " 1           946036          1.0  adhP,\n",
       "         annotation  match score label\n",
       " G_b3734                              \n",
       " 0           947120          1.0  patA\n",
       " 1           948242          1.0  atpA,\n",
       "         annotation  match score label\n",
       " G_b3733                              \n",
       " 0           945651          1.0  ptsG\n",
       " 1           948243          1.0  atpG,\n",
       "         annotation  match score label\n",
       " G_b3736                              \n",
       " 0           945724          1.0  tfaP\n",
       " 1           945766          1.0  fabD\n",
       " 2           948247          1.0  atpF,\n",
       "         annotation  match score label\n",
       " G_b3737                              \n",
       " 0           948253          1.0  atpE\n",
       " 1           948629          1.0  eptA,\n",
       "         annotation  match score label\n",
       " G_b3739                              \n",
       " 0           948009          1.0  pitA\n",
       " 1           948251          1.0  atpI\n",
       " 2           948409          1.0  tpiA,\n",
       "         annotation  match score label\n",
       " G_b3738                              \n",
       " 0           946306          1.0  thiB\n",
       " 1           948252          1.0  atpB,\n",
       "         annotation  match score label\n",
       " G_b3735                              \n",
       " 0           948254          1.0  atpH,\n",
       "         annotation  match score label\n",
       " G_b3731                              \n",
       " 0           947379          1.0  ygfX\n",
       " 1           947979          1.0  acpT\n",
       " 2           948245          1.0  atpC\n",
       " 3           948575          1.0  actP,\n",
       "         annotation  match score label\n",
       " G_b3732                              \n",
       " 0           945876          1.0  patD\n",
       " 1           947436          1.0  dtpA\n",
       " 2           948244          1.0  atpD,\n",
       "         annotation  match score label\n",
       " G_b0720                              \n",
       " 0           945323          1.0  gltA\n",
       " 1           945357          1.0  galT,\n",
       "         annotation  match score label\n",
       " G_b0733                              \n",
       " 0           944837          1.0  yadC\n",
       " 1           945341          1.0  cydA\n",
       " 2           945515          1.0  ycaD\n",
       " 3           945806          1.0  hofC\n",
       " 4           945986          1.0  ydcA\n",
       " 5           947504          1.0  rcbA,\n",
       "         annotation  match score label\n",
       " G_b0734                              \n",
       " 0           945213          1.0  cstA\n",
       " 1           945299          1.0  peaD\n",
       " 2           945347          1.0  cydB\n",
       " 3           945547          1.0  cmoM\n",
       " 4           945980          1.0  pdxI\n",
       " 5           946500          1.0  efeB\n",
       " 6          2847727          1.0  hokB,\n",
       "         annotation  match score label\n",
       " G_b0979                              \n",
       " 0           947547          1.0  appB,\n",
       "         annotation  match score label\n",
       " G_b0978                              \n",
       " 0           945585          1.0  appC\n",
       " 1           947693          1.0  dacB\n",
       " 2           948654          1.0  cadB\n",
       " 3           948765          1.0  bdcA,\n",
       "         annotation  match score label\n",
       " G_b3603                              \n",
       " 0           948114          1.0  lldP,\n",
       "         annotation  match score label\n",
       " G_b2975                              \n",
       " 0           945651          1.0  ptsG\n",
       " 1           947259          1.0  glcA,\n",
       "         annotation  match score label\n",
       " G_b2779                              \n",
       " 0           945032          1.0   eno,\n",
       "         annotation  match score label\n",
       " G_b2925                              \n",
       " 0           945568          1.0  fabA\n",
       " 1           947415          1.0  fbaA,\n",
       "         annotation  match score label\n",
       " G_b1773                              \n",
       " 0           946189          1.0  ydiJ\n",
       " 1           946291          1.0  ydjI\n",
       " 2           948188          1.0  yidJ\n",
       " 3           948459          1.0  yijD\n",
       " 4           948648          1.0  yjdI\n",
       " 5           948851          1.0  iraD,\n",
       "         annotation  match score label\n",
       " G_b2097                              \n",
       " 0           946632          1.0  fbaB\n",
       " 1           946799          1.0  fabB,\n",
       "         annotation  match score label\n",
       " G_b3925                              \n",
       " 0           948424          1.0  glpX,\n",
       "         annotation  match score label\n",
       " G_b4232                              \n",
       " 0           948753          1.0   fbp,\n",
       "         annotation  match score label\n",
       " G_b2492                              \n",
       " 0           949032          1.0  focB,\n",
       "         annotation  match score label\n",
       " G_b0904                              \n",
       " 0           945513          1.0  focA,\n",
       "         annotation  match score label\n",
       " G_b4152                              \n",
       " 0           948680          1.0  frdC,\n",
       "         annotation  match score label\n",
       " G_b4154                              \n",
       " 0           946298          1.0  fdrA\n",
       " 1           948134          1.0  rfaD\n",
       " 2           948652          1.0  fadR\n",
       " 3           948667          1.0  frdA,\n",
       "         annotation  match score label\n",
       " G_b4153                              \n",
       " 0           947117          1.0  rfbD\n",
       " 1           947482          1.0  rfbC\n",
       " 2           948666          1.0  frdB,\n",
       "         annotation  match score label\n",
       " G_b4151                              \n",
       " 0           948668          1.0  frdD,\n",
       "         annotation  match score label\n",
       " G_b1819                              \n",
       " 0           946342          1.0  manZ,\n",
       "         annotation  match score label\n",
       " G_b1817                              \n",
       " 0           946334          1.0  manX,\n",
       "         annotation  match score label\n",
       " G_b2416                              \n",
       " 0           946879          1.0  ptsI,\n",
       "         annotation  match score label\n",
       " G_b2415                              \n",
       " 0           946886          1.0  ptsH,\n",
       "         annotation  match score label\n",
       " G_b1818                              \n",
       " 0           946332          1.0  manY\n",
       " 1         38094956          1.0  ynaM,\n",
       "         annotation  match score label\n",
       " G_b1611                              \n",
       " 0           946147          1.0  fumC,\n",
       "         annotation  match score label\n",
       " G_b4122                              \n",
       " 0           948642          1.0  fumB,\n",
       "         annotation  match score label\n",
       " G_b1612                              \n",
       " 0           946826          1.0  fumA,\n",
       "         annotation  match score label\n",
       " G_b3528                              \n",
       " 0           947291          1.0  tcdA\n",
       " 1           947494          1.0  tdcA\n",
       " 2           948039          1.0  dctA,\n",
       "         annotation  match score label\n",
       " G_b1852                              \n",
       " 0           946370          1.0   zwf,\n",
       "         annotation  match score label\n",
       " G_b1779                              \n",
       " 0           944772          1.0  apaG\n",
       " 1           945596          1.0  pgaA\n",
       " 2           946263          1.0  paaG\n",
       " 3           947679          1.0  gapA,\n",
       "         annotation  match score label\n",
       " G_b1101                              \n",
       " 0           945651          1.0  ptsG,\n",
       "         annotation  match score label\n",
       " G_b2417                              \n",
       " 0           946880          1.0   crr,\n",
       "         annotation  match score label\n",
       " G_b1621                              \n",
       " 0           946009          1.0  malX,\n",
       "         annotation  match score label\n",
       " G_b1297                              \n",
       " 0           946202          1.0  puuA,\n",
       "         annotation  match score label\n",
       " G_b3870                              \n",
       " 0           948370          1.0  glnA,\n",
       "         annotation  match score label\n",
       " G_b0809                              \n",
       " 0           945435          1.0  glnQ,\n",
       "         annotation  match score label\n",
       " G_b0811                              \n",
       " 0           944872          1.0  glnH,\n",
       "         annotation  match score label\n",
       " G_b0810                              \n",
       " 0           945621          1.0  glnP,\n",
       "         annotation  match score label\n",
       " G_b1761                              \n",
       " 0           946802          1.0  gdhA,\n",
       "         annotation  match score label\n",
       " G_b1524                              \n",
       " 0           944973          1.0  glsB\n",
       " 1           948236          1.0  bglF,\n",
       "         annotation  match score label\n",
       " G_b0485                              \n",
       " 0           946187          1.0  glsA\n",
       " 1           949043          1.0  galS,\n",
       "         annotation  match score label\n",
       " G_b1812                              \n",
       " 0           946337          1.0  pabB\n",
       " 1           947111          1.0  abpB,\n",
       "         annotation  match score label\n",
       " G_b3213                              \n",
       " 0           947723          1.0  gltD,\n",
       "         annotation  match score label\n",
       " G_b3212                              \n",
       " 0           947724          1.0  gltB,\n",
       "         annotation  match score label\n",
       " G_b4077                              \n",
       " 0           945324          1.0  lptG\n",
       " 1           946704          1.0  glpT\n",
       " 2           948591          1.0  gltP,\n",
       "         annotation  match score label\n",
       " G_b2029                              \n",
       " 0           946554          1.0   gnd,\n",
       "         annotation  match score label\n",
       " G_b0875                              \n",
       " 0           945497          1.0  aqpZ,\n",
       "         annotation  match score label\n",
       " G_b1136                              \n",
       " 0           945702          1.0   icd,\n",
       "         annotation  match score label\n",
       " G_b4015                              \n",
       " 0           948517          1.0  aceA,\n",
       "         annotation  match score label\n",
       " G_b1380                              \n",
       " 0           945748          1.0  dhaL\n",
       " 1           946315          1.0  ldhA\n",
       " 2           947003          1.0  puuC,\n",
       "         annotation  match score label\n",
       " G_b2133                              \n",
       " 0           946324          1.0  ddlB\n",
       " 1           946653          1.0   dld,\n",
       "         annotation  match score label\n",
       " G_b4014                              \n",
       " 0           946510          1.0  cbeA\n",
       " 1           948512          1.0  aceB,\n",
       "         annotation  match score label\n",
       " G_b2976                              \n",
       " 0           948235          1.0  bglG\n",
       " 1           948857          1.0  glcB,\n",
       "         annotation  match score label\n",
       " G_b3236                              \n",
       " 0           947854          1.0   mdh,\n",
       "         annotation  match score label\n",
       " G_b1479                              \n",
       " 0           946031          1.0  maeA\n",
       " 1           946081          1.0  eamA,\n",
       "         annotation  match score label\n",
       " G_b2463                              \n",
       " 0           945583          1.0  bamE\n",
       " 1           946947          1.0  maeB\n",
       " 2           947065          1.0  eamB,\n",
       "         annotation  match score label\n",
       " G_b2281                              \n",
       " 0           946757          1.0  nuoI,\n",
       "         annotation  match score label\n",
       " G_b2277                              \n",
       " 0           947731          1.0  nuoM,\n",
       "         annotation  match score label\n",
       " G_b2280                              \n",
       " 0           946756          1.0  nuoJ,\n",
       "         annotation  match score label\n",
       " G_b2286                              \n",
       " 0           946759          1.0  nuoC,\n",
       "         annotation  match score label\n",
       " G_b2287                              \n",
       " 0           946738          1.0  nuoB,\n",
       "         annotation  match score label\n",
       " G_b2284                              \n",
       " 0           946753          1.0  nuoF,\n",
       "         annotation  match score label\n",
       " G_b2276                              \n",
       " 0           945136          1.0  nuoN,\n",
       "         annotation  match score label\n",
       " G_b2282                              \n",
       " 0           946761          1.0  nuoH,\n",
       "         annotation  match score label\n",
       " G_b2279                              \n",
       " 0           947580          1.0  nuoK,\n",
       "         annotation  match score label\n",
       " G_b2283                              \n",
       " 0           946762          1.0  nuoG,\n",
       "         annotation  match score label\n",
       " G_b2285                              \n",
       " 0           946746          1.0  nuoE,\n",
       "         annotation  match score label\n",
       " G_b2288                              \n",
       " 0           946764          1.0  nuoA\n",
       " 1           947731          1.0  nuoM,\n",
       "         annotation  match score label\n",
       " G_b2278                              \n",
       " 0           945540          1.0  nuoL,\n",
       "         annotation  match score label\n",
       " G_b1603                              \n",
       " 0           946383          1.0  nudB\n",
       " 1           946628          1.0  pntA\n",
       " 2           948220          1.0  tnaB,\n",
       "         annotation  match score label\n",
       " G_b3962                              \n",
       " 0           948461          1.0  sthA,\n",
       "         annotation  match score label\n",
       " G_b1602                              \n",
       " 0           944942          1.0  insQ\n",
       " 1           946144          1.0  pntB,\n",
       "         annotation  match score label\n",
       " G_b0451                              \n",
       " 0           945084          1.0  amtB\n",
       " 1           948742          1.0  tamB\n",
       " 2           948759          1.0  ecpA,\n",
       "         annotation  match score label\n",
       " G_b0114                              \n",
       " 0           944834          1.0  aceE,\n",
       "         annotation  match score label\n",
       " G_b0115                              \n",
       " 0           944794          1.0  aceF\n",
       " 1           946427          1.0  fecA\n",
       " 2           947469          1.0  yghF,\n",
       "         annotation  match score label\n",
       " G_b3916                              \n",
       " 0           947850          1.0  fkpA\n",
       " 1           948412          1.0  pfkA,\n",
       "         annotation  match score label\n",
       " G_b1723                              \n",
       " 0           944807          1.0  fkpB\n",
       " 1           946230          1.0  pfkB,\n",
       "         annotation  match score label\n",
       " G_b3114                              \n",
       " 0           947623          1.0  tdcE,\n",
       "         annotation  match score label\n",
       " G_b2579                              \n",
       " 0           947068          1.0  grcA\n",
       " 1           948455          1.0  argC,\n",
       "         annotation  match score label\n",
       " G_b3951                              \n",
       " 0           948454          1.0  pflD,\n",
       "         annotation  match score label\n",
       " G_b0902                              \n",
       " 0           945517          1.0  pflA\n",
       " 1           946464          1.0  fliR,\n",
       "         annotation  match score label\n",
       " G_b3952                              \n",
       " 0           948453          1.0  pflC,\n",
       "         annotation  match score label\n",
       " G_b0903                              \n",
       " 0           945514          1.0  pflB,\n",
       "         annotation  match score label\n",
       " G_b4025                              \n",
       " 0           946186          1.0   hyi\n",
       " 1           948535          1.0   pgi,\n",
       "         annotation  match score label\n",
       " G_b2926                              \n",
       " 0           947414          1.0   pgk,\n",
       "         annotation  match score label\n",
       " G_b0767                              \n",
       " 0           946398          1.0   pgl,\n",
       "         annotation  match score label\n",
       " G_b3612                              \n",
       " 0           948130          1.0  gpmM,\n",
       "         annotation  match score label\n",
       " G_b4395                              \n",
       " 0           945895          1.0  ycjT\n",
       " 1           948596          1.0  alsK\n",
       " 2           948833          1.0  tcyJ\n",
       " 3           948918          1.0  ytjC,\n",
       "         annotation  match score label\n",
       " G_b0755                              \n",
       " 0           945068          1.0  gpmA\n",
       " 1           946438          1.0  ampG,\n",
       "         annotation  match score label\n",
       " G_b3493                              \n",
       " 0           948009          1.0  pitA\n",
       " 1           948251          1.0  atpI\n",
       " 2           948409          1.0  tpiA,\n",
       "         annotation  match score label\n",
       " G_b2987                              \n",
       " 0           947475          1.0  pitB,\n",
       "         annotation  match score label\n",
       " G_b3956                              \n",
       " 0           948457          1.0   ppc,\n",
       "         annotation  match score label\n",
       " G_b3403                              \n",
       " 0           945667          1.0   pck,\n",
       "         annotation  match score label\n",
       " G_b1702                              \n",
       " 0           945887          1.0  pspA\n",
       " 1           946209          1.0  ppsA\n",
       " 2           946281          1.0  sppA\n",
       " 3           947519          1.0  nudF,\n",
       "         annotation  match score label\n",
       " G_b2297                              \n",
       " 0           945113          1.0   apt\n",
       " 1           946397          1.0   tap\n",
       " 2           946778          1.0   pta\n",
       " 3           947120          1.0  patA,\n",
       "         annotation  match score label\n",
       " G_b2458                              \n",
       " 0           946940          1.0  eutD,\n",
       "         annotation  match score label\n",
       " G_b1676                              \n",
       " 0           946179          1.0  pykF,\n",
       "         annotation  match score label\n",
       " G_b1854                              \n",
       " 0           946527          1.0  pykA,\n",
       "         annotation  match score label\n",
       " G_b3386                              \n",
       " 0           947896          1.0   rpe\n",
       " 1           948292          1.0   rep,\n",
       "         annotation  match score label\n",
       " G_b4301                              \n",
       " 0           945711          1.0  csgE\n",
       " 1           947720          1.0  secG\n",
       " 2           948829          1.0  sgcE,\n",
       "         annotation  match score label\n",
       " G_b2914                              \n",
       " 0           946902          1.0  iprA\n",
       " 1           947407          1.0  rpiA\n",
       " 2           947709          1.0  iraP\n",
       " 3           948426          1.0  priA,\n",
       "         annotation  match score label\n",
       " G_b4090                              \n",
       " 0           948602          1.0  rpiB\n",
       " 1           948722          1.0  priB,\n",
       "         annotation  match score label\n",
       " G_b0721                              \n",
       " 0           945253          1.0  djlC\n",
       " 1           945316          1.0  sdhC,\n",
       "         annotation  match score label\n",
       " G_b0722                              \n",
       " 0           945322          1.0  sdhD,\n",
       "         annotation  match score label\n",
       " G_b0724                              \n",
       " 0           945300          1.0  sdhB,\n",
       "         annotation  match score label\n",
       " G_b0723                              \n",
       " 0           945402          1.0  sdhA,\n",
       "         annotation  match score label\n",
       " G_b0729                              \n",
       " 0           945314          1.0  sucD\n",
       " 1           945458          1.0  dusC\n",
       " 2           948639          1.0  dcuS,\n",
       "         annotation  match score label\n",
       " G_b0728                              \n",
       " 0           945312          1.0  sucC\n",
       " 1           946288          1.0  cusC,\n",
       "         annotation  match score label\n",
       " G_b0008                              \n",
       " 0           944748          1.0  talB,\n",
       "         annotation  match score label\n",
       " G_b2464                              \n",
       " 0           944955          1.0  ltaE\n",
       " 1           947006          1.0  talA\n",
       " 2           948337          1.0  alaT,\n",
       "         annotation  match score label\n",
       " G_b2465                              \n",
       " 0           945865          1.0  tktB,\n",
       "         annotation  match score label\n",
       " G_b2935                              \n",
       " 0           947420          1.0  tktA,\n",
       "         annotation  match score label\n",
       " G_b3919                              \n",
       " 0           948009          1.0  pitA\n",
       " 1           948251          1.0  atpI\n",
       " 2           948409          1.0  tpiA]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "one_fpath = \"/Users/luna/Desktop/CRBM/AMAS_proj/Models/BiggModels/e_coli_core.xml\"\n",
    "recom = recommender.Recommender(libsbml_fpath=one_fpath)\n",
    "genes = recom.getGeneIDs()\n",
    "pred = recom.getGeneListRecommendation(pred_ids = genes,tax_id='511145', get_df=True)\n",
    "pred\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "one_fpath = \"/Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels/BIOMD0000000190.xml\"\n",
    "recom = recommender.Recommender(libsbml_fpath=one_fpath)\n",
    "model = recom.sbml_document.getModel()\n",
    "try:\n",
    "    for i, gene in enumerate(model.getPlugin(\"fbc\").getListOfGeneProducts(), start=1):\n",
    "        if not gene.isSetMetaId():\n",
    "            gene.setMetaId(generateMetaId(i))\n",
    "except:\n",
    "    pass\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "____________________\n",
      "element_id: G_b1241\n",
      "annotations: ['945695', '945837', '948025']\n",
      "labels: ['purB', 'adhE', 'hdeA']\n",
      "existings: [0, 1, 0]\n",
      "upd_annotation: ['ignore', 'keep', 'ignore']\n",
      "list(one_edf['annotation']): ['945695', '945837', '948025']\n",
      "TYPE_EXISTING_ATTR[etype][element_id]: ['945837']\n",
      "annotation2add_raw: []\n",
      "annotation2add: []\n",
      "____________________\n",
      "element_id: G_b0351\n",
      "annotations: ['945008']\n",
      "labels: ['mhpF']\n",
      "existings: [1]\n",
      "upd_annotation: ['keep']\n",
      "list(one_edf['annotation']): ['945008']\n",
      "TYPE_EXISTING_ATTR[etype][element_id]: ['945008']\n",
      "annotation2add_raw: []\n",
      "annotation2add: []\n",
      "____________________\n",
      "element_id: G_s0001\n",
      "annotations: ['944742', '944792', '944824', '946586']\n",
      "labels: ['thrL', 'satP', 'mutT', 'rutD']\n",
      "____________________\n",
      "element_id: G_b1849\n",
      "annotations: ['946199', '946368']\n",
      "labels: ['rstA', 'purT']\n",
      "existings: [0, 1]\n",
      "upd_annotation: ['ignore', 'keep']\n",
      "list(one_edf['annotation']): ['946199', '946368']\n",
      "TYPE_EXISTING_ATTR[etype][element_id]: ['946368']\n",
      "annotation2add_raw: []\n",
      "annotation2add: []\n",
      "____________________\n",
      "element_id: G_b3115\n",
      "annotations: ['947635']\n",
      "labels: ['tdcD']\n",
      "existings: [1]\n",
      "upd_annotation: ['keep']\n",
      "list(one_edf['annotation']): ['947635']\n",
      "TYPE_EXISTING_ATTR[etype][element_id]: ['947635']\n",
      "annotation2add_raw: []\n",
      "annotation2add: []\n"
     ]
    }
   ],
   "source": [
    "etype = 'genes'\n",
    "tax_id = '511145'\n",
    "model = recom.sbml_document.getModel()\n",
    "TYPE_EXISTING_ATTR = {'species': recom.species.exist_annotation,\n",
    "                      'reaction': recom.reactions.exist_annotation,\n",
    "                      'genes': recom.genes.exist_annotation} \n",
    "ELEMENT_FUNC = {'species': model.getSpecies,\n",
    "                'reaction': model.getReaction,\n",
    "                'genes':model.getPlugin(\"fbc\").getGeneProduct}\n",
    "TYPE_LABEL = {'species': cn.REF_CHEBI2LABEL,\n",
    "              'reaction': cn.REF_RHEA2LABEL,\n",
    "              'genes':cn.REF_NCBI_GENE2LABEL}\n",
    "for one_edf in pred[:5]:\n",
    "  print(\"____________________\")\n",
    "  element_id = one_edf.index.name\n",
    "  print(f\"element_id: {element_id}\")\n",
    "  if one_edf.shape[0] == 0:\n",
    "    continue\n",
    "  annotations = list(one_edf['annotation'])\n",
    "  print(f\"annotations: {annotations}\")\n",
    "  match_scores = list(one_edf[cn.DF_MATCH_SCORE_COL])\n",
    "  labels = list(one_edf['label'])\n",
    "  print(f\"labels: {labels}\")\n",
    "  # if there is existing annotation among predicted candidates;\n",
    "  if element_id in TYPE_EXISTING_ATTR[etype].keys():\n",
    "    existings = [1 if str(val) in TYPE_EXISTING_ATTR[etype][element_id] else 0 \\\n",
    "                  for idx, val in enumerate(one_edf['annotation'])]\n",
    "    print(f\"existings: {existings}\")\n",
    "    upd_annotation = ['keep' if val in TYPE_EXISTING_ATTR[etype][element_id] else 'ignore' \\\n",
    "                      for idx, val in enumerate(one_edf['annotation'])]\n",
    "    print(f\"upd_annotation: {upd_annotation}\")\n",
    "    print(f\"list(one_edf['annotation']): {list(one_edf['annotation'])}\")\n",
    "    print(f\"TYPE_EXISTING_ATTR[etype][element_id]: {[val for val in TYPE_EXISTING_ATTR[etype][element_id]]}\")\n",
    "    annotation2add_raw = [val for val in TYPE_EXISTING_ATTR[etype][element_id] \\\n",
    "                          if val not in list(one_edf['annotation'])]\n",
    "    print(f\"annotation2add_raw: {annotation2add_raw}\")\n",
    "    # only use existing annotation that exists in the label dictionaries\n",
    "    annotation2add = [val for val in annotation2add_raw \\\n",
    "                      if val in TYPE_LABEL[etype].keys()]  \n",
    "    print(f\"annotation2add: {annotation2add}\")\n",
    "  # if there doesn't exist existing annotation among predicted candidates;\n",
    "  else:\n",
    "    existings = [0] * len(annotations)\n",
    "    upd_annotation = ['ignore'] * len(annotations)\n",
    "    annotation2add = []\n",
    "  # handling existing annotations that were not predicted\n",
    "  for new_anot in annotation2add:\n",
    "    annotations.append(new_anot)\n",
    "    match_scores.append(recom.getMatchScoreOfNCBIGENE(element_id, new_anot, tax_id))\n",
    "    labels.append(cn.REF_NCBI_GENE2LABEL[new_anot])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "amas",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
