{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import re\n",
    "import pandas as pd\n",
    "import collections\n",
    "from collections import Counter\n",
    "import compress_pickle\n",
    "import copy\n",
    "import fnmatch\n",
    "import itertools\n",
    "import libsbml\n",
    "import numpy as np\n",
    "import time\n",
    "import matplotlib.pyplot as plt\n",
    "import string\n",
    "\n",
    "# Import my local AMAS package\n",
    "sys.path.insert(0, '/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2')\n",
    "from AMAS import annotation_maker as am\n",
    "from AMAS import constants as cn\n",
    "from AMAS import iterator as it\n",
    "from AMAS import tools\n",
    "from AMAS import species_annotation as sa\n",
    "from AMAS import reaction_annotation as ra \n",
    "from AMAS import gene_annotation as ga\n",
    "from AMAS import recommender\n",
    "\n",
    "# Directory\n",
    "biomd_model_dir = '/Users/luna/Desktop/CRBM/AMAS_proj/Models/BioModels'\n",
    "bigg_model_dir = '/Users/luna/Desktop/CRBM/AMAS_proj/Models/BiggModels'\n",
    "output_dir = '/Users/luna/Desktop/CRBM/AMAS_proj/Results/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "ecoli_mg1655_models = ['e_coli_core.xml', 'iML1515.xml', 'iAF1260.xml','iJR904.xml','iAF1260b.xml','iJO1366.xml']\n",
    "human_models = ['Recon3D.xml', 'iAB_RBC_283.xml','iAT_PLT_636.xml','RECON1.xml']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "137"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "one_fpath = \"/Users/luna/Desktop/CRBM/AMAS_proj/Models/BiggModels/e_coli_core.xml\"\n",
    "reader = libsbml.SBMLReader()\n",
    "document = reader.readSBML(one_fpath)\n",
    "model = document.getModel().getPlugin(\"fbc\")\n",
    "model.getNumGeneProducts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_product = fbc_plugin.getGeneProduct(3)\n",
    "print(gene_product)\n",
    "print(gene_product.getIdAttribute())\n",
    "print(gene_product.getLabel())\n",
    "print(gene_product.getName())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'G_Nrk1_AT1': 'NMRK1',\n",
       " 'G_Rpe_AT2': 'RPE',\n",
       " 'G_Rpe_AT1': 'RPE',\n",
       " 'G_Rpia_AT1': 'RPIA',\n",
       " 'G_Comtd1_AT1': 'COMTD1',\n",
       " 'G_Sord_AT1': 'SORD',\n",
       " 'G_Akr7a2_AT1': 'AKR7A2',\n",
       " 'G_Srm_AT1': 'SRM',\n",
       " 'G_55256': 'ADI1',\n",
       " 'G_Sms_AT1': 'SMS',\n",
       " 'G_Taldo1_AT1': 'TALDO1',\n",
       " 'G_ThmtP_AT1': 'ThmtP_AT1',\n",
       " 'G_Tkt_AT1': 'TKT',\n",
       " 'G_Tpk1_AT1': 'TPK1',\n",
       " 'G_58478': 'ENOPH1',\n",
       " 'G_Gnpda1_AT1': 'GNPDA1',\n",
       " 'G_Tpi1_AT1': 'TPI1',\n",
       " 'G_Gale_AT1': 'GALE',\n",
       " 'G_Gale_AT2': 'GALE',\n",
       " 'G_Ugdh_AT1': 'UGDH',\n",
       " 'G_G6pd_AT1': 'G6PD',\n",
       " 'G_Galk1_AT1': 'GALK1',\n",
       " 'G_Uros_AT1': 'UROS',\n",
       " 'G_Urod_AT1': 'UROD',\n",
       " 'G_Akr1a1_AT1': 'AKR1A1',\n",
       " 'G_Slc14a1_1_e': 'Slc14a1_1_e',\n",
       " 'G_Xylb_AT1': 'XYLB',\n",
       " 'G_Akr1a1_AT2': 'AKR1A1',\n",
       " 'G_Dcxr_AT1': 'DCXR',\n",
       " 'G_Akr1b1_AT1': 'AKR1B1',\n",
       " 'G_Galt_AT1': 'GALT',\n",
       " 'G_Galt_AT2': 'GALT',\n",
       " 'G_Galt_AT3': 'GALT',\n",
       " 'G_Ugp2_AT2': 'UGP2',\n",
       " 'G_Ugp2_AT1': 'UGP2',\n",
       " 'G_Slc2a3_1_e': 'Slc2a3_1_e',\n",
       " 'G_Slc2a4_1_e': 'Slc2a4_1_e',\n",
       " 'G_Gapdh_AT1': 'GAPDH',\n",
       " 'G_Gctg_AT1': 'GGCT',\n",
       " 'G_Guk1_AT1': 'GUK1',\n",
       " 'G_Glul_AT1': 'GLUL',\n",
       " 'G_Gclc_AT1': 'GCLC',\n",
       " 'G_Gclm_AT1': 'GCLM',\n",
       " 'G_Gk2_AT1': 'GK2',\n",
       " 'G_Gk_AT2': 'GK',\n",
       " 'G_Gkp3_AT1': 'Gkp3_AT1',\n",
       " 'G_Gk_AT1': 'GK',\n",
       " 'G_Hagh_AT1': 'HAGH',\n",
       " 'G_Slc6a9_AT1': 'SLC6A9',\n",
       " 'G_Slc6a9_AT2': 'SLC6A9',\n",
       " 'G_GmpR_AT1': 'GMPR',\n",
       " 'G_Gmps_AT1': 'GMPS',\n",
       " 'G_Pgd_AT1': 'PGD',\n",
       " 'G_Gpam_AT1': 'GPAM',\n",
       " 'G_Gsr_AT2': 'GSR',\n",
       " 'G_Prdx1_AT1': 'PRDX1',\n",
       " 'G_Prdx1_AT3': 'PRDX1',\n",
       " 'G_Gpx1_AT1': 'GPX1',\n",
       " 'G_Prdx2_AT1': 'PRDX2',\n",
       " 'G_Prdx2_AT3': 'PRDX2',\n",
       " 'G_Prdx1_AT2': 'PRDX1',\n",
       " 'G_Prdx2_AT2': 'PRDX2',\n",
       " 'G_Gpx1_AT2': 'GPX1',\n",
       " 'G_Gpx4_AT1': 'GPX4',\n",
       " 'G_Gss_AT1': 'GSS',\n",
       " 'G_Gucy1A2B3_AT1': 'Gucy1A2B3_AT1',\n",
       " 'G_Gucy1A3B3_AT1': 'Gucy1A3B3_AT1',\n",
       " 'G_Gucy1A3B2_AT1': 'Gucy1A3B2_AT1',\n",
       " 'G_Gucy2F_AT1': 'GUCY2F',\n",
       " 'G_Npr1_AT1': 'NPR1',\n",
       " 'G_Gucy2C_AT1': 'GUCY2C',\n",
       " 'G_Npr2_AT2': 'NPR2',\n",
       " 'G_Npr2_AT1': 'NPR2',\n",
       " 'G_Gucy1A2B2_AT1': 'Gucy1A2B2_AT1',\n",
       " 'G_Gucy2D_AT1': 'GUCY2D',\n",
       " 'G_HprT1_AT1': 'HPRT1',\n",
       " 'G_Slc5a1_1_e': 'Slc5a1_1_e',\n",
       " 'G_Ca2_AT1': 'CA2',\n",
       " 'G_Ca1_AT1': 'CA1',\n",
       " 'G_Ca3_AT1': 'CA3',\n",
       " 'G_Slc4a1_1_e': 'Slc4a1_1_e',\n",
       " 'G_Hk1_AT5': 'HK1',\n",
       " 'G_Hk1_AT4': 'HK1',\n",
       " 'G_Hk1_AT1': 'HK1',\n",
       " 'G_Hk1_AT3': 'HK1',\n",
       " 'G_Hk1_AT2': 'HK1',\n",
       " 'G_Hk2_AT1': 'HK2',\n",
       " 'G_Flj22761_AT1': 'Flj22761_AT1',\n",
       " 'G_Hk3_AT1': 'HK3',\n",
       " 'G_Hmbs_AT1': 'HMBS',\n",
       " 'G_Hmox1_AT1': 'HMOX1',\n",
       " 'G_Hmox2_AT1': 'HMOX2',\n",
       " 'G_Idh1_AT1': 'IDH1',\n",
       " 'G_ImpDH1_AT1': 'IMPDH1',\n",
       " 'G_ImpDH2_AT1': 'IMPDH2',\n",
       " 'G_ImpDH1_AT2': 'IMPDH1',\n",
       " 'G_Slc12a7_1_e': 'Slc12a7_1_e',\n",
       " 'G_Ldhb_AT1': 'LDHB',\n",
       " 'G_Ldha_AT1': 'LDHA',\n",
       " 'G_Glo1_AT1': 'GLO1',\n",
       " 'G_Pla2g4a_AT1': 'PLA2G4A',\n",
       " 'G_Clc_AT1': 'CLC',\n",
       " 'G_Lta4h_AT1': 'LTA4H',\n",
       " 'G_Mpi_AT1': 'MPI',\n",
       " 'G_Mdh1_AT1': 'MDH1',\n",
       " 'G_Me1_AT1': 'ME1',\n",
       " 'G_Mat1a_AT1': 'MAT1A',\n",
       " 'G_Slc43a1_AT1': 'SLC43A1',\n",
       " 'G_Synj1_AT1': 'SYNJ1',\n",
       " 'G_Synj1_AT2': 'SYNJ1',\n",
       " 'G_Inpp5d_AT1': 'INPP5D',\n",
       " 'G_Itpkc_AT1': 'ITPKC',\n",
       " 'G_Itpka_AT1': 'ITPKA',\n",
       " 'G_Itpkb_AT1': 'ITPKB',\n",
       " 'G_Impa1_AT1': 'IMPA1',\n",
       " 'G_Isyna1_AT1': 'ISYNA1',\n",
       " 'G_Mtap_AT1': 'MTAP',\n",
       " 'G_Nadk_AT1': 'NADK',\n",
       " 'G_Nadsyn1_AT1': 'NADSYN1',\n",
       " 'G_Slc5a3_1_e': 'Slc5a3_1_e',\n",
       " 'G_Slc5a5_1_e': 'Slc5a5_1_e',\n",
       " 'G_NME7_AT2': 'NME7',\n",
       " 'G_NME12_AT2': 'NME12_AT2',\n",
       " 'G_NME6_AT1': 'NME6',\n",
       " 'G_NME7_AT1': 'NME7',\n",
       " 'G_NME12_AT1': 'NME12_AT1',\n",
       " 'G_NME3_AT1': 'NME3',\n",
       " 'G_Rhag_1_e': 'Rhag_1_e',\n",
       " 'G_Rhbg_1_e': 'Rhbg_1_e',\n",
       " 'G_Nmnat3_AT1': 'NMNAT3',\n",
       " 'G_Np_AT1': 'ZNF384',\n",
       " 'G_Nt5C2_AT1': 'NT5C2',\n",
       " 'G_Nt5C3_AT1': 'NT5C3A',\n",
       " 'G_Nt5C_AT1': 'NT5C',\n",
       " 'G_Umps_AT1': 'UMPS',\n",
       " 'G_Oplah_AT1': 'OPLAH',\n",
       " 'G_Odc1_AT1': 'ODC1',\n",
       " 'G_Pde1A_AT1': 'PDE1A',\n",
       " 'G_Pde4B_AT1': 'PDE4B',\n",
       " 'G_Pde7A_AT1': 'PDE7A',\n",
       " 'G_Pde1A_AT2': 'PDE1A',\n",
       " 'G_Pde3B_AT2': 'PDE3B',\n",
       " 'G_Pde11A_AT1': 'PDE11A',\n",
       " 'G_Pde8b_AT1': 'PDE8B',\n",
       " 'G_Pde8a_AT5': 'PDE8A',\n",
       " 'G_Pde8a_AT2': 'PDE8A',\n",
       " 'G_Pde4D_AT1': 'PDE4D',\n",
       " 'G_Pde1B_AT1': 'PDE1B',\n",
       " 'G_Pde8a_AT1': 'PDE8A',\n",
       " 'G_Pde8a_AT4': 'PDE8A',\n",
       " 'G_Pde8a_AT3': 'PDE8A',\n",
       " 'G_Pde1C_AT1': 'PDE1C',\n",
       " 'G_Pde4C_AT1': 'PDE4C',\n",
       " 'G_Pde4A_AT1': 'PDE4A',\n",
       " 'G_Pde2A_AT1': 'PDE2A',\n",
       " 'G_Pde3A_AT1': 'PDE3A',\n",
       " 'G_Pde7A_AT2': 'PDE7A',\n",
       " 'G_Pde7B_AT1': 'PDE7B',\n",
       " 'G_Pde3B_AT1': 'PDE3B',\n",
       " 'G_PnpO_AT1': 'PNPO',\n",
       " 'G_PdxP_AT1': 'PDXP',\n",
       " 'G_Pcyt2_AT1': 'PCYT2',\n",
       " 'G_Pfkl_AT1': 'PFKL',\n",
       " 'G_Pfkm_AT1': 'PFKM',\n",
       " 'G_Pfkl_AT2': 'PFKL',\n",
       " 'G_Gpi_AT1': 'GPI',\n",
       " 'G_Pgk1_AT1': 'PGK1',\n",
       " 'G_Pgls_AT1': 'PGLS',\n",
       " 'G_Got1_AT1': 'GOT1',\n",
       " 'G_Plcb3_AT1': 'PLCB3',\n",
       " 'G_Pip5k1b_AT1': 'PIP5K1B',\n",
       " 'G_Pip5k2a_AT1': 'PIP4K2A',\n",
       " 'G_Pip5k1c_AT1': 'PIP5K1C',\n",
       " 'G_Pip5k2c_AT1': 'PIP4K2C',\n",
       " 'G_Pip5k2b_AT2': 'PIP4K2B',\n",
       " 'G_Pip5k3_AT1': 'PIKFYVE',\n",
       " 'G_Pip5k3_AT3': 'PIKFYVE',\n",
       " 'G_Pip5k1a_AT1': 'PIP5K1A',\n",
       " 'G_Pip5k2b_AT1': 'PIP4K2B',\n",
       " 'G_Pip5k3_AT2': 'PIKFYVE',\n",
       " 'G_Pi4kii_AT1': 'PI4K2A',\n",
       " 'G_Pik4cb_AT1': 'PI4KB',\n",
       " 'G_Pik4ca_AT1': 'PI4KA',\n",
       " 'G_Pla2g2a_AT1': 'PLA2G2A',\n",
       " 'G_Pla2g6_AT1': 'PLA2G6',\n",
       " 'G_Pla2g6_AT2': 'PLA2G6',\n",
       " 'G_Pmm2_AT1': 'PMM2',\n",
       " 'G_Ppa2_AT2': 'PPA2',\n",
       " 'G_Ppa2_AT1': 'PPA2',\n",
       " 'G_Ppa2_AT3': 'PPA2',\n",
       " 'G_Ppa2_AT4': 'PPA2',\n",
       " 'G_Ppap2a_AT2': 'PLPP1',\n",
       " 'G_Ppap2c_AT1': 'PLPP2',\n",
       " 'G_Ppap2b_AT1': 'PLPP3',\n",
       " 'G_Ppap2a_AT1': 'PLPP1',\n",
       " 'G_Alad_AT1': 'ALAD',\n",
       " 'G_Alad_AT2': 'ALAD',\n",
       " 'G_Pgm1_AT1': 'PGM1',\n",
       " 'G_Prps1l1_AT2': 'PRPS1L1',\n",
       " 'G_Prps1_AT1': 'PRPS1',\n",
       " 'G_Prps1l1_AT1': 'PRPS1L1',\n",
       " 'G_PdxK_AT1': 'PDXK',\n",
       " 'G_Aox1_AT1': 'AOX1',\n",
       " 'G_Pkm2_AT2': 'PKM',\n",
       " 'G_Pklr_AT2': 'PKLR',\n",
       " 'G_Pklr_AT1': 'PKLR',\n",
       " 'G_Pkm2_AT3': 'PKM',\n",
       " 'G_Pkm2_AT1': 'PKM',\n",
       " 'G_Rfk_AT1': 'RFK',\n",
       " 'G_10390_AT1': 'CEPT1',\n",
       " 'G_55500': 'ETNK1',\n",
       " 'G_55224_AT1': 'ETNK2',\n",
       " 'G_29124': 'LGALS13',\n",
       " 'G_Mat2ab1': 'Mat2ab1',\n",
       " 'G_Mat2ab2': 'Mat2ab2',\n",
       " 'G_Atp1a1b3': 'Atp1a1b3',\n",
       " 'G_Atp1a1b2': 'Atp1a1b2',\n",
       " 'G_Atp1a1b4': 'Atp1a1b4',\n",
       " 'G_Atp1a1b1': 'Atp1a1b1',\n",
       " 'G_10846': 'PDE10A',\n",
       " 'G_55276': 'PGM2',\n",
       " 'G_9429': 'ABCG2',\n",
       " 'G_51084': 'CRYL1',\n",
       " 'G_Tat_AT1': 'TAT',\n",
       " 'G_Renbp_AT1': 'RENBP',\n",
       " 'G_Nagk_AT1': 'NAGK',\n",
       " 'G_Nans_AT1': 'NANS',\n",
       " 'G_Nanp_AT1': 'NANP',\n",
       " 'G_Acp1_AT1': 'ACP1',\n",
       " 'G_Acp1_AT2': 'ACP1',\n",
       " 'G_Acp1_AT3': 'ACP1',\n",
       " 'G_Ada_AT1': 'ADA',\n",
       " 'G_Slc29a2_1_e': 'Slc29a2_1_e',\n",
       " 'G_Ak1_AT1': 'AK1',\n",
       " 'G_Amd1_AT1': 'AMD1',\n",
       " 'G_Adcy1_AT1': 'ADCY1',\n",
       " 'G_Adcy9_AT1': 'ADCY9',\n",
       " 'G_Adcy7_AT1': 'ADCY7',\n",
       " 'G_Adcy5_AT1': 'ADCY5',\n",
       " 'G_Adcy6_AT1': 'ADCY6',\n",
       " 'G_Adcy6_AT2': 'ADCY6',\n",
       " 'G_Adcy2_AT1': 'ADCY2',\n",
       " 'G_Adcy4_AT1': 'ADCY4',\n",
       " 'G_Adcy3_AT1': 'ADCY3',\n",
       " 'G_Adcy8_AT1': 'ADCY8',\n",
       " 'G_Adk_AT1': 'ADK',\n",
       " 'G_Adk_AT2': 'ADK',\n",
       " 'G_Slc29a1_1_e': 'Slc29a1_1_e',\n",
       " 'G_AprT_AT1': 'APRT',\n",
       " 'G_Cgi_14_AT1': 'Cgi_14_AT1',\n",
       " 'G_Agpat4_AT1': 'AGPAT4',\n",
       " 'G_Agpat3_AT2': 'AGPAT3',\n",
       " 'G_Agpat2_AT1': 'AGPAT2',\n",
       " 'G_Agpat1_AT1': 'AGPAT1',\n",
       " 'G_Agpat5_AT1': 'AGPAT5',\n",
       " 'G_Agpat3_AT1': 'AGPAT3',\n",
       " 'G_Agpat6_AT1': 'GPAT4',\n",
       " 'G_Ahcy_AT1': 'AHCY',\n",
       " 'G_Slc38a2_AT1': 'SLC38A2',\n",
       " 'G_Slc38a1_AT1': 'SLC38A1',\n",
       " 'G_Slc7a2_AT1': 'SLC7A2',\n",
       " 'G_Slc6a14_AT1': 'SLC6A14',\n",
       " 'G_Slc7a1_AT1': 'SLC7A1',\n",
       " 'G_Slc38a4_AT1': 'SLC38A4',\n",
       " 'G_Slc7a3_AT1': 'SLC7A3',\n",
       " 'G_Aldh3a1_AT1': 'ALDH3A1',\n",
       " 'G_Aldh9a1_AT1': 'ALDH9A1',\n",
       " 'G_Aldh1a2_AT2': 'ALDH1A2',\n",
       " 'G_Aldh1a1_AT1': 'ALDH1A1',\n",
       " 'G_AmpD3_AT1': 'AMPD3',\n",
       " 'G_NudT2_AT2': 'NUDT2',\n",
       " 'G_NudT2_AT1': 'NUDT2',\n",
       " 'G_NudT2_AT3': 'NUDT2',\n",
       " 'G_Arg1_AT1': 'ARG1',\n",
       " 'G_Slco1b3_AT1': 'SLCO1B3',\n",
       " 'G_Slco1b1_AT1': 'SLCO1B1',\n",
       " 'G_Ugt1A4_AT1': 'UGT1A4',\n",
       " 'G_Ugt1A1_AT1': 'UGT1A1',\n",
       " 'G_Blvra_AT1': 'BLVRA',\n",
       " 'G_Cpt1a_AT1': 'CPT1A',\n",
       " 'G_Cpt2_AT1': 'CPT2',\n",
       " 'G_Atp2b1_AT1': 'ATP2B1',\n",
       " 'G_Atp2b4_AT1': 'ATP2B4',\n",
       " 'G_Abcc4_1_e': 'Abcc4_1_e',\n",
       " 'G_Cat_AT1': 'CAT',\n",
       " 'G_Cdipt_AT1': 'CDIPT',\n",
       " 'G_Cdipt_AT2': 'CDIPT',\n",
       " 'G_Cds1_AT1': 'CDS1',\n",
       " 'G_Chpt1_AT1': 'CHPT1',\n",
       " 'G_Phospho1_AT1': 'PHOSPHO1',\n",
       " 'G_Pcyt1a_AT1': 'PCYT1A',\n",
       " 'G_Pcyt1b_AT1': 'PCYT1B',\n",
       " 'G_Chkl_AT1': 'CHKB',\n",
       " 'G_Chk_AT2': 'CHKA',\n",
       " 'G_Chk_AT1': 'CHKA',\n",
       " 'G_Chkl_AT2': 'CHKB',\n",
       " 'G_Slc5a7_AT1': 'SLC5A7',\n",
       " 'G_Cpox_AT1': 'CPOX',\n",
       " 'G_Slc43a2_AT1': 'SLC43A2',\n",
       " 'G_UmpK_AT1': 'CMPK1',\n",
       " 'G_Slc16a1_AT1': 'SLC16A1',\n",
       " 'G_Dgkg_AT1': 'DGKG',\n",
       " 'G_Dgkq_AT1': 'DGKQ',\n",
       " 'G_Dgkz_AT3': 'DGKZ',\n",
       " 'G_Dgkh_AT1': 'DGKH',\n",
       " 'G_Dgkz_AT1': 'DGKZ',\n",
       " 'G_Dgke_AT1': 'DGKE',\n",
       " 'G_Dgka_AT1': 'DGKA',\n",
       " 'G_Dgkb_AT1': 'DGKB',\n",
       " 'G_Dgkz_AT2': 'DGKZ',\n",
       " 'G_Dgkb_AT2': 'DGKB',\n",
       " 'G_Dgkh_AT2': 'DGKH',\n",
       " 'G_Dgki_AT1': 'DGKI',\n",
       " 'G_Dgkd_AT2': 'DGKD',\n",
       " 'G_Dgkd_AT1': 'DGKD',\n",
       " 'G_Slc2a1_1_e': 'Slc2a1_1_e',\n",
       " 'G_Comt_AT2': 'COMT',\n",
       " 'G_Comt_AT1': 'COMT',\n",
       " 'G_Slc22a3_AT1': 'SLC22A3',\n",
       " 'G_Slc22a2_AT2': 'SLC22A2',\n",
       " 'G_Slc22a2_AT1': 'SLC22A2',\n",
       " 'G_Bpgm_AT1': 'BPGM',\n",
       " 'G_Bpgm_AT2': 'BPGM',\n",
       " 'G_Eno1_AT1': 'ENO1',\n",
       " 'G_Acsl1_AT1': 'ACSL1',\n",
       " 'G_EnpP1_AT1': 'ENPP1',\n",
       " 'G_Aldoa_AT1': 'ALDOA',\n",
       " 'G_Aldoa_AT3': 'ALDOA',\n",
       " 'G_Aldoc_AT1': 'ALDOC',\n",
       " 'G_Aldoa_AT2': 'ALDOA',\n",
       " 'G_Fbp1_AT1': 'FBP1',\n",
       " 'G_Pfkfb1_AT1': 'PFKFB1',\n",
       " 'G_Pfkfb3_AT1': 'PFKFB3',\n",
       " 'G_Fbp2_AT1': 'FBP2',\n",
       " 'G_Pfkfb4_AT1': 'PFKFB4',\n",
       " 'G_Pfkfb2_AT1': 'PFKFB2',\n",
       " 'G_Fech_AT1': 'FECH',\n",
       " 'G_Fech_AT2': 'FECH',\n",
       " 'G_FadS_AT2': 'FLAD1',\n",
       " 'G_FadS_AT1': 'FLAD1',\n",
       " 'G_Slc2a7_1_e': 'Slc2a7_1_e',\n",
       " 'G_Slc2a5_1_e': 'Slc2a5_1_e',\n",
       " 'G_Slc2a2_1_e': 'Slc2a2_1_e',\n",
       " 'G_Slc2a11_1_e': 'Slc2a11_1_e',\n",
       " 'G_Slc2a8_1_e': 'Slc2a8_1_e',\n",
       " 'G_Fh_AT1': 'FH'}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "one_fpath = \"/Users/luna/Desktop/CRBM/AMAS_proj/Models/BiggModels/iAB_RBC_283.xml\"\n",
    "recom = recommender.Recommender(libsbml_fpath=one_fpath)\n",
    "geneids2evaluate = list(recom.genes.exist_annotation.keys())\n",
    "recom.genes.names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'G_Nrk1_AT1': ['54981'],\n",
       " 'G_Rpe_AT2': ['6120'],\n",
       " 'G_Rpe_AT1': ['6120'],\n",
       " 'G_Rpia_AT1': ['22934'],\n",
       " 'G_Comtd1_AT1': ['118881'],\n",
       " 'G_Sord_AT1': ['6652'],\n",
       " 'G_Akr7a2_AT1': ['8574'],\n",
       " 'G_Srm_AT1': ['6723'],\n",
       " 'G_55256': ['55256'],\n",
       " 'G_Sms_AT1': ['6611'],\n",
       " 'G_Taldo1_AT1': ['6888'],\n",
       " 'G_Tkt_AT1': ['7086'],\n",
       " 'G_Tpk1_AT1': ['27010'],\n",
       " 'G_58478': ['58478'],\n",
       " 'G_Gnpda1_AT1': ['10007'],\n",
       " 'G_Tpi1_AT1': ['7167'],\n",
       " 'G_Gale_AT1': ['2582'],\n",
       " 'G_Gale_AT2': ['2582'],\n",
       " 'G_Ugdh_AT1': ['7358'],\n",
       " 'G_G6pd_AT1': ['2539'],\n",
       " 'G_Galk1_AT1': ['2584'],\n",
       " 'G_Uros_AT1': ['7390'],\n",
       " 'G_Urod_AT1': ['7389'],\n",
       " 'G_Akr1a1_AT1': ['10327'],\n",
       " 'G_Xylb_AT1': ['9942'],\n",
       " 'G_Akr1a1_AT2': ['10327'],\n",
       " 'G_Dcxr_AT1': ['51181'],\n",
       " 'G_Akr1b1_AT1': ['231'],\n",
       " 'G_Galt_AT1': ['2592'],\n",
       " 'G_Galt_AT2': ['2592'],\n",
       " 'G_Galt_AT3': ['2592'],\n",
       " 'G_Ugp2_AT2': ['7360'],\n",
       " 'G_Ugp2_AT1': ['7360'],\n",
       " 'G_Gapdh_AT1': ['2597'],\n",
       " 'G_Gctg_AT1': ['79017'],\n",
       " 'G_Guk1_AT1': ['2987'],\n",
       " 'G_Glul_AT1': ['2752'],\n",
       " 'G_Gclc_AT1': ['2729'],\n",
       " 'G_Gclm_AT1': ['2730'],\n",
       " 'G_Gk2_AT1': ['2712'],\n",
       " 'G_Gk_AT2': ['2710'],\n",
       " 'G_Gk_AT1': ['2710'],\n",
       " 'G_Hagh_AT1': ['3029'],\n",
       " 'G_Slc6a9_AT1': ['6536'],\n",
       " 'G_Slc6a9_AT2': ['6536'],\n",
       " 'G_GmpR_AT1': ['2766'],\n",
       " 'G_Gmps_AT1': ['8833'],\n",
       " 'G_Pgd_AT1': ['5226'],\n",
       " 'G_Gpam_AT1': ['57678'],\n",
       " 'G_Gsr_AT2': ['2936'],\n",
       " 'G_Prdx1_AT1': ['5052'],\n",
       " 'G_Prdx1_AT3': ['5052'],\n",
       " 'G_Gpx1_AT1': ['2876'],\n",
       " 'G_Prdx2_AT1': ['7001'],\n",
       " 'G_Prdx2_AT3': ['7001'],\n",
       " 'G_Prdx1_AT2': ['5052'],\n",
       " 'G_Prdx2_AT2': ['7001'],\n",
       " 'G_Gpx1_AT2': ['2876'],\n",
       " 'G_Gpx4_AT1': ['2879'],\n",
       " 'G_Gss_AT1': ['2937'],\n",
       " 'G_Gucy2F_AT1': ['2986'],\n",
       " 'G_Npr1_AT1': ['4881'],\n",
       " 'G_Gucy2C_AT1': ['2984'],\n",
       " 'G_Npr2_AT2': ['4882'],\n",
       " 'G_Npr2_AT1': ['4882'],\n",
       " 'G_Gucy2D_AT1': ['3000'],\n",
       " 'G_HprT1_AT1': ['3251'],\n",
       " 'G_Ca2_AT1': ['760'],\n",
       " 'G_Ca1_AT1': ['759'],\n",
       " 'G_Ca3_AT1': ['761'],\n",
       " 'G_Hk1_AT5': ['3098'],\n",
       " 'G_Hk1_AT4': ['3098'],\n",
       " 'G_Hk1_AT1': ['3098'],\n",
       " 'G_Hk1_AT3': ['3098'],\n",
       " 'G_Hk1_AT2': ['3098'],\n",
       " 'G_Hk2_AT1': ['3099'],\n",
       " 'G_Hk3_AT1': ['3101'],\n",
       " 'G_Hmbs_AT1': ['3145'],\n",
       " 'G_Hmox1_AT1': ['3162'],\n",
       " 'G_Hmox2_AT1': ['3163'],\n",
       " 'G_Idh1_AT1': ['3417'],\n",
       " 'G_ImpDH1_AT1': ['3614'],\n",
       " 'G_ImpDH2_AT1': ['3615'],\n",
       " 'G_ImpDH1_AT2': ['3614'],\n",
       " 'G_Ldhb_AT1': ['3945'],\n",
       " 'G_Ldha_AT1': ['3939'],\n",
       " 'G_Glo1_AT1': ['2739'],\n",
       " 'G_Pla2g4a_AT1': ['5321'],\n",
       " 'G_Clc_AT1': ['1178'],\n",
       " 'G_Lta4h_AT1': ['4048'],\n",
       " 'G_Mpi_AT1': ['4351'],\n",
       " 'G_Mdh1_AT1': ['4190'],\n",
       " 'G_Me1_AT1': ['4199'],\n",
       " 'G_Mat1a_AT1': ['4143'],\n",
       " 'G_Slc43a1_AT1': ['8501'],\n",
       " 'G_Synj1_AT1': ['8867'],\n",
       " 'G_Synj1_AT2': ['8867'],\n",
       " 'G_Inpp5d_AT1': ['3635'],\n",
       " 'G_Itpkc_AT1': ['80271'],\n",
       " 'G_Itpka_AT1': ['3706'],\n",
       " 'G_Itpkb_AT1': ['3707'],\n",
       " 'G_Impa1_AT1': ['3612'],\n",
       " 'G_Isyna1_AT1': ['51477'],\n",
       " 'G_Mtap_AT1': ['4507'],\n",
       " 'G_Nadk_AT1': ['65220'],\n",
       " 'G_Nadsyn1_AT1': ['55191'],\n",
       " 'G_NME7_AT2': ['29922'],\n",
       " 'G_NME6_AT1': ['10201'],\n",
       " 'G_NME7_AT1': ['29922'],\n",
       " 'G_NME3_AT1': ['4832'],\n",
       " 'G_Nmnat3_AT1': ['349565'],\n",
       " 'G_Np_AT1': ['171017'],\n",
       " 'G_Nt5C2_AT1': ['22978'],\n",
       " 'G_Nt5C3_AT1': ['51251'],\n",
       " 'G_Nt5C_AT1': ['30833'],\n",
       " 'G_Umps_AT1': ['7372'],\n",
       " 'G_Oplah_AT1': ['26873'],\n",
       " 'G_Odc1_AT1': ['4953'],\n",
       " 'G_Pde1A_AT1': ['5136'],\n",
       " 'G_Pde4B_AT1': ['5142'],\n",
       " 'G_Pde7A_AT1': ['5150'],\n",
       " 'G_Pde1A_AT2': ['5136'],\n",
       " 'G_Pde3B_AT2': ['5140'],\n",
       " 'G_Pde11A_AT1': ['50940'],\n",
       " 'G_Pde8b_AT1': ['8622'],\n",
       " 'G_Pde8a_AT5': ['5151'],\n",
       " 'G_Pde8a_AT2': ['5151'],\n",
       " 'G_Pde4D_AT1': ['5144'],\n",
       " 'G_Pde1B_AT1': ['5153'],\n",
       " 'G_Pde8a_AT1': ['5151'],\n",
       " 'G_Pde8a_AT4': ['5151'],\n",
       " 'G_Pde8a_AT3': ['5151'],\n",
       " 'G_Pde1C_AT1': ['5137'],\n",
       " 'G_Pde4C_AT1': ['5143'],\n",
       " 'G_Pde4A_AT1': ['5141'],\n",
       " 'G_Pde2A_AT1': ['5138'],\n",
       " 'G_Pde3A_AT1': ['5139'],\n",
       " 'G_Pde7A_AT2': ['5150'],\n",
       " 'G_Pde7B_AT1': ['27115'],\n",
       " 'G_Pde3B_AT1': ['5140'],\n",
       " 'G_PnpO_AT1': ['55163'],\n",
       " 'G_PdxP_AT1': ['57026'],\n",
       " 'G_Pcyt2_AT1': ['5833'],\n",
       " 'G_Pfkl_AT1': ['5211'],\n",
       " 'G_Pfkm_AT1': ['5213'],\n",
       " 'G_Pfkl_AT2': ['5211'],\n",
       " 'G_Gpi_AT1': ['2821'],\n",
       " 'G_Pgk1_AT1': ['5230'],\n",
       " 'G_Pgls_AT1': ['25796'],\n",
       " 'G_Got1_AT1': ['2805'],\n",
       " 'G_Plcb3_AT1': ['5331'],\n",
       " 'G_Pip5k1b_AT1': ['8395'],\n",
       " 'G_Pip5k2a_AT1': ['5305'],\n",
       " 'G_Pip5k1c_AT1': ['23396'],\n",
       " 'G_Pip5k2c_AT1': ['79837'],\n",
       " 'G_Pip5k2b_AT2': ['8396'],\n",
       " 'G_Pip5k3_AT1': ['200576'],\n",
       " 'G_Pip5k3_AT3': ['200576'],\n",
       " 'G_Pip5k1a_AT1': ['8394'],\n",
       " 'G_Pip5k2b_AT1': ['8396'],\n",
       " 'G_Pip5k3_AT2': ['200576'],\n",
       " 'G_Pi4kii_AT1': ['55361'],\n",
       " 'G_Pik4cb_AT1': ['5298'],\n",
       " 'G_Pik4ca_AT1': ['5297'],\n",
       " 'G_Pla2g2a_AT1': ['5320'],\n",
       " 'G_Pla2g6_AT1': ['8398'],\n",
       " 'G_Pla2g6_AT2': ['8398'],\n",
       " 'G_Pmm2_AT1': ['5373'],\n",
       " 'G_Ppa2_AT2': ['27068'],\n",
       " 'G_Ppa2_AT1': ['27068'],\n",
       " 'G_Ppa2_AT3': ['27068'],\n",
       " 'G_Ppa2_AT4': ['27068'],\n",
       " 'G_Ppap2a_AT2': ['8611'],\n",
       " 'G_Ppap2c_AT1': ['8612'],\n",
       " 'G_Ppap2b_AT1': ['8613'],\n",
       " 'G_Ppap2a_AT1': ['8611'],\n",
       " 'G_Alad_AT1': ['210'],\n",
       " 'G_Alad_AT2': ['210'],\n",
       " 'G_Pgm1_AT1': ['5236'],\n",
       " 'G_Prps1l1_AT2': ['221823'],\n",
       " 'G_Prps1_AT1': ['5631'],\n",
       " 'G_Prps1l1_AT1': ['221823'],\n",
       " 'G_PdxK_AT1': ['8566'],\n",
       " 'G_Aox1_AT1': ['316'],\n",
       " 'G_Pkm2_AT2': ['5315'],\n",
       " 'G_Pklr_AT2': ['5313'],\n",
       " 'G_Pklr_AT1': ['5313'],\n",
       " 'G_Pkm2_AT3': ['5315'],\n",
       " 'G_Pkm2_AT1': ['5315'],\n",
       " 'G_Rfk_AT1': ['55312'],\n",
       " 'G_10390_AT1': ['10390'],\n",
       " 'G_55500': ['55500'],\n",
       " 'G_55224_AT1': ['55224'],\n",
       " 'G_29124': ['29124'],\n",
       " 'G_10846': ['10846'],\n",
       " 'G_55276': ['55276'],\n",
       " 'G_9429': ['9429'],\n",
       " 'G_51084': ['51084'],\n",
       " 'G_Tat_AT1': ['6898'],\n",
       " 'G_Renbp_AT1': ['5973'],\n",
       " 'G_Nagk_AT1': ['55577'],\n",
       " 'G_Nans_AT1': ['54187'],\n",
       " 'G_Nanp_AT1': ['140838'],\n",
       " 'G_Acp1_AT1': ['52'],\n",
       " 'G_Acp1_AT2': ['52'],\n",
       " 'G_Acp1_AT3': ['52'],\n",
       " 'G_Ada_AT1': ['100'],\n",
       " 'G_Ak1_AT1': ['203'],\n",
       " 'G_Amd1_AT1': ['262'],\n",
       " 'G_Adcy1_AT1': ['107'],\n",
       " 'G_Adcy9_AT1': ['115'],\n",
       " 'G_Adcy7_AT1': ['113'],\n",
       " 'G_Adcy5_AT1': ['111'],\n",
       " 'G_Adcy6_AT1': ['112'],\n",
       " 'G_Adcy6_AT2': ['112'],\n",
       " 'G_Adcy2_AT1': ['108'],\n",
       " 'G_Adcy4_AT1': ['196883'],\n",
       " 'G_Adcy3_AT1': ['109'],\n",
       " 'G_Adcy8_AT1': ['114'],\n",
       " 'G_Adk_AT1': ['132'],\n",
       " 'G_Adk_AT2': ['132'],\n",
       " 'G_AprT_AT1': ['353'],\n",
       " 'G_Agpat4_AT1': ['56895'],\n",
       " 'G_Agpat3_AT2': ['56894'],\n",
       " 'G_Agpat2_AT1': ['10555'],\n",
       " 'G_Agpat1_AT1': ['10554'],\n",
       " 'G_Agpat5_AT1': ['55326'],\n",
       " 'G_Agpat3_AT1': ['56894'],\n",
       " 'G_Agpat6_AT1': ['137964'],\n",
       " 'G_Ahcy_AT1': ['191'],\n",
       " 'G_Slc38a2_AT1': ['54407'],\n",
       " 'G_Slc38a1_AT1': ['81539'],\n",
       " 'G_Slc7a2_AT1': ['6542'],\n",
       " 'G_Slc6a14_AT1': ['11254'],\n",
       " 'G_Slc7a1_AT1': ['6541'],\n",
       " 'G_Slc38a4_AT1': ['55089'],\n",
       " 'G_Slc7a3_AT1': ['84889'],\n",
       " 'G_Aldh3a1_AT1': ['218'],\n",
       " 'G_Aldh9a1_AT1': ['223'],\n",
       " 'G_Aldh1a2_AT2': ['8854'],\n",
       " 'G_Aldh1a1_AT1': ['216'],\n",
       " 'G_AmpD3_AT1': ['272'],\n",
       " 'G_NudT2_AT2': ['318'],\n",
       " 'G_NudT2_AT1': ['318'],\n",
       " 'G_NudT2_AT3': ['318'],\n",
       " 'G_Arg1_AT1': ['383'],\n",
       " 'G_Slco1b3_AT1': ['28234'],\n",
       " 'G_Slco1b1_AT1': ['10599'],\n",
       " 'G_Ugt1A4_AT1': ['54657'],\n",
       " 'G_Ugt1A1_AT1': ['54658'],\n",
       " 'G_Blvra_AT1': ['644'],\n",
       " 'G_Cpt1a_AT1': ['1374'],\n",
       " 'G_Cpt2_AT1': ['1376'],\n",
       " 'G_Atp2b1_AT1': ['490'],\n",
       " 'G_Atp2b4_AT1': ['493'],\n",
       " 'G_Cat_AT1': ['847'],\n",
       " 'G_Cdipt_AT1': ['10423'],\n",
       " 'G_Cdipt_AT2': ['10423'],\n",
       " 'G_Cds1_AT1': ['1040'],\n",
       " 'G_Chpt1_AT1': ['56994'],\n",
       " 'G_Phospho1_AT1': ['162466'],\n",
       " 'G_Pcyt1a_AT1': ['5130'],\n",
       " 'G_Pcyt1b_AT1': ['9468'],\n",
       " 'G_Chkl_AT1': ['1120'],\n",
       " 'G_Chk_AT2': ['1119'],\n",
       " 'G_Chk_AT1': ['1119'],\n",
       " 'G_Chkl_AT2': ['1120'],\n",
       " 'G_Slc5a7_AT1': ['60482'],\n",
       " 'G_Cpox_AT1': ['1371'],\n",
       " 'G_Slc43a2_AT1': ['124935'],\n",
       " 'G_UmpK_AT1': ['51727'],\n",
       " 'G_Slc16a1_AT1': ['6566'],\n",
       " 'G_Dgkg_AT1': ['1608'],\n",
       " 'G_Dgkq_AT1': ['1609'],\n",
       " 'G_Dgkz_AT3': ['8525'],\n",
       " 'G_Dgkh_AT1': ['160851'],\n",
       " 'G_Dgkz_AT1': ['8525'],\n",
       " 'G_Dgke_AT1': ['8526'],\n",
       " 'G_Dgka_AT1': ['1606'],\n",
       " 'G_Dgkb_AT1': ['1607'],\n",
       " 'G_Dgkz_AT2': ['8525'],\n",
       " 'G_Dgkb_AT2': ['1607'],\n",
       " 'G_Dgkh_AT2': ['160851'],\n",
       " 'G_Dgki_AT1': ['9162'],\n",
       " 'G_Dgkd_AT2': ['8527'],\n",
       " 'G_Dgkd_AT1': ['8527'],\n",
       " 'G_Comt_AT2': ['1312'],\n",
       " 'G_Comt_AT1': ['1312'],\n",
       " 'G_Slc22a3_AT1': ['6581'],\n",
       " 'G_Slc22a2_AT2': ['6582'],\n",
       " 'G_Slc22a2_AT1': ['6582'],\n",
       " 'G_Bpgm_AT1': ['669'],\n",
       " 'G_Bpgm_AT2': ['669'],\n",
       " 'G_Eno1_AT1': ['2023'],\n",
       " 'G_Acsl1_AT1': ['2180'],\n",
       " 'G_EnpP1_AT1': ['5167'],\n",
       " 'G_Aldoa_AT1': ['226'],\n",
       " 'G_Aldoa_AT3': ['226'],\n",
       " 'G_Aldoc_AT1': ['230'],\n",
       " 'G_Aldoa_AT2': ['226'],\n",
       " 'G_Fbp1_AT1': ['2203'],\n",
       " 'G_Pfkfb1_AT1': ['5207'],\n",
       " 'G_Pfkfb3_AT1': ['5209'],\n",
       " 'G_Fbp2_AT1': ['8789'],\n",
       " 'G_Pfkfb4_AT1': ['5210'],\n",
       " 'G_Pfkfb2_AT1': ['5208'],\n",
       " 'G_Fech_AT1': ['2235'],\n",
       " 'G_Fech_AT2': ['2235'],\n",
       " 'G_FadS_AT2': ['80308'],\n",
       " 'G_FadS_AT1': ['80308'],\n",
       " 'G_Fh_AT1': ['2271']}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "recom.genes.exist_annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Recommendation(id='G_Nrk1_AT1', candidates=[(23608, np.float64(1.0)), (54981, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/23608', 'https://www.ncbi.nlm.nih.gov/gene/54981'], labels=['MKRN1', 'NMRK1']),\n",
       " Recommendation(id='G_Rpe_AT2', candidates=[(2004, np.float64(1.0)), (2069, np.float64(1.0)), (5187, np.float64(1.0)), (6120, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2004', 'https://www.ncbi.nlm.nih.gov/gene/2069', 'https://www.ncbi.nlm.nih.gov/gene/5187', 'https://www.ncbi.nlm.nih.gov/gene/6120'], labels=['ELK3', 'EREG', 'PER1', 'RPE']),\n",
       " Recommendation(id='G_Rpe_AT1', candidates=[(2004, np.float64(1.0)), (2069, np.float64(1.0)), (5187, np.float64(1.0)), (6120, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2004', 'https://www.ncbi.nlm.nih.gov/gene/2069', 'https://www.ncbi.nlm.nih.gov/gene/5187', 'https://www.ncbi.nlm.nih.gov/gene/6120'], labels=['ELK3', 'EREG', 'PER1', 'RPE']),\n",
       " Recommendation(id='G_Rpia_AT1', candidates=[(3557, np.float64(1.0)), (4012, np.float64(1.0)), (22934, np.float64(1.0)), (55010, np.float64(1.0)), (93034, np.float64(1.0)), (100526794, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3557', 'https://www.ncbi.nlm.nih.gov/gene/4012', 'https://www.ncbi.nlm.nih.gov/gene/22934', 'https://www.ncbi.nlm.nih.gov/gene/55010', 'https://www.ncbi.nlm.nih.gov/gene/93034', 'https://www.ncbi.nlm.nih.gov/gene/100526794'], labels=['IL1RN', 'LNPEP', 'RPIA', 'PARPBP', 'NT5C1B', 'NT5C1B-RDH14']),\n",
       " Recommendation(id='G_Comtd1_AT1', candidates=[(118881, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/118881'], labels=['COMTD1']),\n",
       " Recommendation(id='G_Sord_AT1', candidates=[(6652, np.float64(1.0)), (121214, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6652', 'https://www.ncbi.nlm.nih.gov/gene/121214'], labels=['SORD', 'SDR9C7']),\n",
       " Recommendation(id='G_Akr7a2_AT1', candidates=[(8574, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8574'], labels=['AKR7A2']),\n",
       " Recommendation(id='G_Srm_AT1', candidates=[(1667, np.float64(1.0)), (4141, np.float64(1.0)), (4552, np.float64(1.0)), (6723, np.float64(1.0)), (6725, np.float64(1.0)), (8011, np.float64(1.0)), (147719, np.float64(1.0)), (148398, np.float64(1.0)), (164091, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1667', 'https://www.ncbi.nlm.nih.gov/gene/4141', 'https://www.ncbi.nlm.nih.gov/gene/4552', 'https://www.ncbi.nlm.nih.gov/gene/6723', 'https://www.ncbi.nlm.nih.gov/gene/6725', 'https://www.ncbi.nlm.nih.gov/gene/8011', 'https://www.ncbi.nlm.nih.gov/gene/147719', 'https://www.ncbi.nlm.nih.gov/gene/148398', 'https://www.ncbi.nlm.nih.gov/gene/164091'], labels=['DEFA1', 'MARS1', 'MTRR', 'SRM', 'SRMS', 'MROS', 'LYPD4', 'SAMD11', 'PAQR7']),\n",
       " Recommendation(id='G_55256', candidates=[(1727, np.float64(1.0)), (1729, np.float64(1.0)), (55256, np.float64(1.0)), (205428, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1727', 'https://www.ncbi.nlm.nih.gov/gene/1729', 'https://www.ncbi.nlm.nih.gov/gene/55256', 'https://www.ncbi.nlm.nih.gov/gene/205428'], labels=['CYB5R3', 'DIAPH1', 'ADI1', 'DIPK2A']),\n",
       " Recommendation(id='G_Sms_AT1', candidates=[(6611, np.float64(1.0)), (10743, np.float64(1.0)), (64374, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6611', 'https://www.ncbi.nlm.nih.gov/gene/10743', 'https://www.ncbi.nlm.nih.gov/gene/64374'], labels=['SMS', 'RAI1', 'SIL1']),\n",
       " Recommendation(id='G_Taldo1_AT1', candidates=[(6888, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6888'], labels=['TALDO1']),\n",
       " Recommendation(id='G_Tkt_AT1', candidates=[(4921, np.float64(1.0)), (7086, np.float64(1.0)), (7272, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4921', 'https://www.ncbi.nlm.nih.gov/gene/7086', 'https://www.ncbi.nlm.nih.gov/gene/7272'], labels=['DDR2', 'TKT', 'TTK']),\n",
       " Recommendation(id='G_Tpk1_AT1', candidates=[(4296, np.float64(1.0)), (27010, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4296', 'https://www.ncbi.nlm.nih.gov/gene/27010'], labels=['MAP3K11', 'TPK1']),\n",
       " Recommendation(id='G_58478', candidates=[(58478, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/58478'], labels=['ENOPH1']),\n",
       " Recommendation(id='G_Gnpda1_AT1', candidates=[(10007, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10007'], labels=['GNPDA1']),\n",
       " Recommendation(id='G_Tpi1_AT1', candidates=[(1915, np.float64(1.0)), (5449, np.float64(1.0)), (6574, np.float64(1.0)), (7167, np.float64(1.0)), (30851, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1915', 'https://www.ncbi.nlm.nih.gov/gene/5449', 'https://www.ncbi.nlm.nih.gov/gene/6574', 'https://www.ncbi.nlm.nih.gov/gene/7167', 'https://www.ncbi.nlm.nih.gov/gene/30851'], labels=['EEF1A1', 'POU1F1', 'SLC20A1', 'TPI1', 'TAX1BP3']),\n",
       " Recommendation(id='G_Gale_AT1', candidates=[(2582, np.float64(1.0)), (2934, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2582', 'https://www.ncbi.nlm.nih.gov/gene/2934'], labels=['GALE', 'GSN']),\n",
       " Recommendation(id='G_Gale_AT2', candidates=[(2582, np.float64(1.0)), (2934, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2582', 'https://www.ncbi.nlm.nih.gov/gene/2934'], labels=['GALE', 'GSN']),\n",
       " Recommendation(id='G_Ugdh_AT1', candidates=[(7358, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/7358'], labels=['UGDH']),\n",
       " Recommendation(id='G_G6pd_AT1', candidates=[(2539, np.float64(1.0)), (5226, np.float64(1.0)), (8266, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2539', 'https://www.ncbi.nlm.nih.gov/gene/5226', 'https://www.ncbi.nlm.nih.gov/gene/8266'], labels=['G6PD', 'PGD', 'UBL4A']),\n",
       " Recommendation(id='G_Galk1_AT1', candidates=[(2584, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2584'], labels=['GALK1']),\n",
       " Recommendation(id='G_Uros_AT1', candidates=[(7390, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/7390'], labels=['UROS']),\n",
       " Recommendation(id='G_Urod_AT1', candidates=[(7389, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/7389'], labels=['UROD']),\n",
       " Recommendation(id='G_Akr1a1_AT1', candidates=[(10327, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10327'], labels=['AKR1A1']),\n",
       " Recommendation(id='G_Xylb_AT1', candidates=[(9942, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/9942'], labels=['XYLB']),\n",
       " Recommendation(id='G_Akr1a1_AT2', candidates=[(10327, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10327'], labels=['AKR1A1']),\n",
       " Recommendation(id='G_Dcxr_AT1', candidates=[(51181, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/51181'], labels=['DCXR']),\n",
       " Recommendation(id='G_Akr1b1_AT1', candidates=[(231, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/231'], labels=['AKR1B1']),\n",
       " Recommendation(id='G_Galt_AT1', candidates=[(2592, np.float64(1.0)), (57104, np.float64(1.0)), (114770, np.float64(1.0)), (130589, np.float64(1.0)), (145173, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2592', 'https://www.ncbi.nlm.nih.gov/gene/57104', 'https://www.ncbi.nlm.nih.gov/gene/114770', 'https://www.ncbi.nlm.nih.gov/gene/130589', 'https://www.ncbi.nlm.nih.gov/gene/145173'], labels=['GALT', 'PNPLA2', 'PGLYRP2', 'GALM', 'B3GLCT']),\n",
       " Recommendation(id='G_Galt_AT2', candidates=[(2592, np.float64(1.0)), (57104, np.float64(1.0)), (114770, np.float64(1.0)), (130589, np.float64(1.0)), (145173, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2592', 'https://www.ncbi.nlm.nih.gov/gene/57104', 'https://www.ncbi.nlm.nih.gov/gene/114770', 'https://www.ncbi.nlm.nih.gov/gene/130589', 'https://www.ncbi.nlm.nih.gov/gene/145173'], labels=['GALT', 'PNPLA2', 'PGLYRP2', 'GALM', 'B3GLCT']),\n",
       " Recommendation(id='G_Galt_AT3', candidates=[(2592, np.float64(1.0)), (57104, np.float64(1.0)), (114770, np.float64(1.0)), (130589, np.float64(1.0)), (145173, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2592', 'https://www.ncbi.nlm.nih.gov/gene/57104', 'https://www.ncbi.nlm.nih.gov/gene/114770', 'https://www.ncbi.nlm.nih.gov/gene/130589', 'https://www.ncbi.nlm.nih.gov/gene/145173'], labels=['GALT', 'PNPLA2', 'PGLYRP2', 'GALM', 'B3GLCT']),\n",
       " Recommendation(id='G_Ugp2_AT2', candidates=[(7360, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/7360'], labels=['UGP2']),\n",
       " Recommendation(id='G_Ugp2_AT1', candidates=[(7360, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/7360'], labels=['UGP2']),\n",
       " Recommendation(id='G_Gapdh_AT1', candidates=[(2597, np.float64(1.0)), (100240709, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2597', 'https://www.ncbi.nlm.nih.gov/gene/100240709'], labels=['GAPDH', 'GAPDHP44']),\n",
       " Recommendation(id='G_Gctg_AT1', candidates=[(79017, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/79017'], labels=['GGCT']),\n",
       " Recommendation(id='G_Guk1_AT1', candidates=[(2987, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2987'], labels=['GUK1']),\n",
       " Recommendation(id='G_Glul_AT1', candidates=[(2752, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2752'], labels=['GLUL']),\n",
       " Recommendation(id='G_Gclc_AT1', candidates=[(2729, np.float64(1.0)), (80781, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2729', 'https://www.ncbi.nlm.nih.gov/gene/80781'], labels=['GCLC', 'COL18A1']),\n",
       " Recommendation(id='G_Gclm_AT1', candidates=[(2730, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2730'], labels=['GCLM']),\n",
       " Recommendation(id='G_Gk2_AT1', candidates=[(2585, np.float64(1.0)), (2712, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2585', 'https://www.ncbi.nlm.nih.gov/gene/2712'], labels=['GALK2', 'GK2']),\n",
       " Recommendation(id='G_Gk_AT2', candidates=[(2645, np.float64(1.0)), (2710, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2645', 'https://www.ncbi.nlm.nih.gov/gene/2710'], labels=['GCK', 'GK']),\n",
       " Recommendation(id='G_Gk_AT1', candidates=[(2645, np.float64(1.0)), (2710, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2645', 'https://www.ncbi.nlm.nih.gov/gene/2710'], labels=['GCK', 'GK']),\n",
       " Recommendation(id='G_Hagh_AT1', candidates=[(3029, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3029'], labels=['HAGH']),\n",
       " Recommendation(id='G_Slc6a9_AT1', candidates=[(6536, np.float64(1.0)), (10479, np.float64(1.0)), (84679, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6536', 'https://www.ncbi.nlm.nih.gov/gene/10479', 'https://www.ncbi.nlm.nih.gov/gene/84679'], labels=['SLC6A9', 'SLC9A6', 'SLC9A7']),\n",
       " Recommendation(id='G_Slc6a9_AT2', candidates=[(6536, np.float64(1.0)), (10479, np.float64(1.0)), (84679, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6536', 'https://www.ncbi.nlm.nih.gov/gene/10479', 'https://www.ncbi.nlm.nih.gov/gene/84679'], labels=['SLC6A9', 'SLC9A6', 'SLC9A7']),\n",
       " Recommendation(id='G_GmpR_AT1', candidates=[(2766, np.float64(1.0)), (6403, np.float64(1.0)), (54852, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2766', 'https://www.ncbi.nlm.nih.gov/gene/6403', 'https://www.ncbi.nlm.nih.gov/gene/54852'], labels=['GMPR', 'SELP', 'PAQR5']),\n",
       " Recommendation(id='G_Gmps_AT1', candidates=[(8833, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8833'], labels=['GMPS']),\n",
       " Recommendation(id='G_Pgd_AT1', candidates=[(2532, np.float64(1.0)), (2995, np.float64(1.0)), (5226, np.float64(1.0)), (26227, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2532', 'https://www.ncbi.nlm.nih.gov/gene/2995', 'https://www.ncbi.nlm.nih.gov/gene/5226', 'https://www.ncbi.nlm.nih.gov/gene/26227'], labels=['ACKR1', 'GYPC', 'PGD', 'PHGDH']),\n",
       " Recommendation(id='G_Gpam_AT1', candidates=[(4237, np.float64(1.0)), (5238, np.float64(1.0)), (51083, np.float64(1.0)), (57678, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4237', 'https://www.ncbi.nlm.nih.gov/gene/5238', 'https://www.ncbi.nlm.nih.gov/gene/51083', 'https://www.ncbi.nlm.nih.gov/gene/57678'], labels=['MFAP2', 'PGM3', 'GAL', 'GPAM']),\n",
       " Recommendation(id='G_Gsr_AT2', candidates=[(597, np.float64(1.0)), (2936, np.float64(1.0)), (5308, np.float64(1.0)), (54897, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/597', 'https://www.ncbi.nlm.nih.gov/gene/2936', 'https://www.ncbi.nlm.nih.gov/gene/5308', 'https://www.ncbi.nlm.nih.gov/gene/54897'], labels=['BCL2A1', 'GSR', 'PITX2', 'CASZ1']),\n",
       " Recommendation(id='G_Prdx1_AT1', candidates=[(5052, np.float64(1.0)), (29979, np.float64(1.0)), (643244, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5052', 'https://www.ncbi.nlm.nih.gov/gene/29979', 'https://www.ncbi.nlm.nih.gov/gene/643244'], labels=['PRDX1', 'UBQLN1', 'RDXP1']),\n",
       " Recommendation(id='G_Prdx1_AT3', candidates=[(5052, np.float64(1.0)), (29979, np.float64(1.0)), (643244, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5052', 'https://www.ncbi.nlm.nih.gov/gene/29979', 'https://www.ncbi.nlm.nih.gov/gene/643244'], labels=['PRDX1', 'UBQLN1', 'RDXP1']),\n",
       " Recommendation(id='G_Gpx1_AT1', candidates=[(2876, np.float64(1.0)), (7317, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2876', 'https://www.ncbi.nlm.nih.gov/gene/7317'], labels=['GPX1', 'UBA1']),\n",
       " Recommendation(id='G_Prdx2_AT1', candidates=[(5964, np.float64(1.0)), (7001, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5964', 'https://www.ncbi.nlm.nih.gov/gene/7001'], labels=['RDXP2', 'PRDX2']),\n",
       " Recommendation(id='G_Prdx2_AT3', candidates=[(5964, np.float64(1.0)), (7001, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5964', 'https://www.ncbi.nlm.nih.gov/gene/7001'], labels=['RDXP2', 'PRDX2']),\n",
       " Recommendation(id='G_Prdx1_AT2', candidates=[(5052, np.float64(1.0)), (29979, np.float64(1.0)), (643244, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5052', 'https://www.ncbi.nlm.nih.gov/gene/29979', 'https://www.ncbi.nlm.nih.gov/gene/643244'], labels=['PRDX1', 'UBQLN1', 'RDXP1']),\n",
       " Recommendation(id='G_Prdx2_AT2', candidates=[(5964, np.float64(1.0)), (7001, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5964', 'https://www.ncbi.nlm.nih.gov/gene/7001'], labels=['RDXP2', 'PRDX2']),\n",
       " Recommendation(id='G_Gpx1_AT2', candidates=[(2876, np.float64(1.0)), (7317, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2876', 'https://www.ncbi.nlm.nih.gov/gene/7317'], labels=['GPX1', 'UBA1']),\n",
       " Recommendation(id='G_Gpx4_AT1', candidates=[(2879, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2879'], labels=['GPX4']),\n",
       " Recommendation(id='G_Gss_AT1', candidates=[(2200, np.float64(1.0)), (2937, np.float64(1.0)), (5621, np.float64(1.0)), (6497, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2200', 'https://www.ncbi.nlm.nih.gov/gene/2937', 'https://www.ncbi.nlm.nih.gov/gene/5621', 'https://www.ncbi.nlm.nih.gov/gene/6497'], labels=['FBN1', 'GSS', 'PRNP', 'SKI']),\n",
       " Recommendation(id='G_Gucy2F_AT1', candidates=[(2986, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2986'], labels=['GUCY2F']),\n",
       " Recommendation(id='G_Npr1_AT1', candidates=[(4745, np.float64(1.0)), (4881, np.float64(1.0)), (5708, np.float64(1.0)), (6184, np.float64(1.0)), (8829, np.float64(1.0)), (79052, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4745', 'https://www.ncbi.nlm.nih.gov/gene/4881', 'https://www.ncbi.nlm.nih.gov/gene/5708', 'https://www.ncbi.nlm.nih.gov/gene/6184', 'https://www.ncbi.nlm.nih.gov/gene/8829', 'https://www.ncbi.nlm.nih.gov/gene/79052'], labels=['NELL1', 'NPR1', 'PSMD2', 'RPN1', 'NRP1', 'PTCPRN']),\n",
       " Recommendation(id='G_Gucy2C_AT1', candidates=[(2984, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2984'], labels=['GUCY2C']),\n",
       " Recommendation(id='G_Npr2_AT2', candidates=[(4753, np.float64(1.0)), (4882, np.float64(1.0)), (5707, np.float64(1.0)), (6185, np.float64(1.0)), (7031, np.float64(1.0)), (8828, np.float64(1.0)), (10641, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4753', 'https://www.ncbi.nlm.nih.gov/gene/4882', 'https://www.ncbi.nlm.nih.gov/gene/5707', 'https://www.ncbi.nlm.nih.gov/gene/6185', 'https://www.ncbi.nlm.nih.gov/gene/7031', 'https://www.ncbi.nlm.nih.gov/gene/8828', 'https://www.ncbi.nlm.nih.gov/gene/10641'], labels=['NELL2', 'NPR2', 'PSMD1', 'RPN2', 'TFF1', 'NRP2', 'NPRL2']),\n",
       " Recommendation(id='G_Npr2_AT1', candidates=[(4753, np.float64(1.0)), (4882, np.float64(1.0)), (5707, np.float64(1.0)), (6185, np.float64(1.0)), (7031, np.float64(1.0)), (8828, np.float64(1.0)), (10641, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4753', 'https://www.ncbi.nlm.nih.gov/gene/4882', 'https://www.ncbi.nlm.nih.gov/gene/5707', 'https://www.ncbi.nlm.nih.gov/gene/6185', 'https://www.ncbi.nlm.nih.gov/gene/7031', 'https://www.ncbi.nlm.nih.gov/gene/8828', 'https://www.ncbi.nlm.nih.gov/gene/10641'], labels=['NELL2', 'NPR2', 'PSMD1', 'RPN2', 'TFF1', 'NRP2', 'NPRL2']),\n",
       " Recommendation(id='G_Gucy2D_AT1', candidates=[(3000, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3000'], labels=['GUCY2D']),\n",
       " Recommendation(id='G_HprT1_AT1', candidates=[(3251, np.float64(1.0)), (5745, np.float64(1.0)), (7220, np.float64(1.0)), (79577, np.float64(1.0)), (138428, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3251', 'https://www.ncbi.nlm.nih.gov/gene/5745', 'https://www.ncbi.nlm.nih.gov/gene/7220', 'https://www.ncbi.nlm.nih.gov/gene/79577', 'https://www.ncbi.nlm.nih.gov/gene/138428'], labels=['HPRT1', 'PTH1R', 'TRPC1', 'CDC73', 'PTRH1']),\n",
       " Recommendation(id='G_Ca2_AT1', candidates=[(108, np.float64(1.0)), (760, np.float64(1.0)), (28882, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/108', 'https://www.ncbi.nlm.nih.gov/gene/760', 'https://www.ncbi.nlm.nih.gov/gene/28882'], labels=['ADCY2', 'CA2', 'IGKV2D-29']),\n",
       " Recommendation(id='G_Ca1_AT1', candidates=[(107, np.float64(1.0)), (759, np.float64(1.0)), (6281, np.float64(1.0)), (10141, np.float64(1.0)), (57110, np.float64(1.0)), (100862690, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/107', 'https://www.ncbi.nlm.nih.gov/gene/759', 'https://www.ncbi.nlm.nih.gov/gene/6281', 'https://www.ncbi.nlm.nih.gov/gene/10141', 'https://www.ncbi.nlm.nih.gov/gene/57110', 'https://www.ncbi.nlm.nih.gov/gene/100862690'], labels=['ADCY1', 'CA1', 'S100A10', 'LINC01587', 'PLAAT1', 'ERVK-1']),\n",
       " Recommendation(id='G_Ca3_AT1', candidates=[(109, np.float64(1.0)), (718, np.float64(1.0)), (761, np.float64(1.0)), (27350, np.float64(1.0)), (100616100, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/109', 'https://www.ncbi.nlm.nih.gov/gene/718', 'https://www.ncbi.nlm.nih.gov/gene/761', 'https://www.ncbi.nlm.nih.gov/gene/27350', 'https://www.ncbi.nlm.nih.gov/gene/100616100'], labels=['ADCY3', 'C3', 'CA3', 'APOBEC3C', 'ERVK-2']),\n",
       " Recommendation(id='G_Hk1_AT5', candidates=[(3098, np.float64(1.0)), (3739, np.float64(1.0)), (3754, np.float64(1.0)), (3816, np.float64(1.0)), (51361, np.float64(1.0)), (255061, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3098', 'https://www.ncbi.nlm.nih.gov/gene/3739', 'https://www.ncbi.nlm.nih.gov/gene/3754', 'https://www.ncbi.nlm.nih.gov/gene/3816', 'https://www.ncbi.nlm.nih.gov/gene/51361', 'https://www.ncbi.nlm.nih.gov/gene/255061'], labels=['HK1', 'KCNA4', 'KCNF1', 'KLK1', 'HOOK1', 'TAC4']),\n",
       " Recommendation(id='G_Hk1_AT4', candidates=[(3098, np.float64(1.0)), (3739, np.float64(1.0)), (3754, np.float64(1.0)), (3816, np.float64(1.0)), (51361, np.float64(1.0)), (255061, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3098', 'https://www.ncbi.nlm.nih.gov/gene/3739', 'https://www.ncbi.nlm.nih.gov/gene/3754', 'https://www.ncbi.nlm.nih.gov/gene/3816', 'https://www.ncbi.nlm.nih.gov/gene/51361', 'https://www.ncbi.nlm.nih.gov/gene/255061'], labels=['HK1', 'KCNA4', 'KCNF1', 'KLK1', 'HOOK1', 'TAC4']),\n",
       " Recommendation(id='G_Hk1_AT1', candidates=[(3098, np.float64(1.0)), (3739, np.float64(1.0)), (3754, np.float64(1.0)), (3816, np.float64(1.0)), (51361, np.float64(1.0)), (255061, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3098', 'https://www.ncbi.nlm.nih.gov/gene/3739', 'https://www.ncbi.nlm.nih.gov/gene/3754', 'https://www.ncbi.nlm.nih.gov/gene/3816', 'https://www.ncbi.nlm.nih.gov/gene/51361', 'https://www.ncbi.nlm.nih.gov/gene/255061'], labels=['HK1', 'KCNA4', 'KCNF1', 'KLK1', 'HOOK1', 'TAC4']),\n",
       " Recommendation(id='G_Hk1_AT3', candidates=[(3098, np.float64(1.0)), (3739, np.float64(1.0)), (3754, np.float64(1.0)), (3816, np.float64(1.0)), (51361, np.float64(1.0)), (255061, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3098', 'https://www.ncbi.nlm.nih.gov/gene/3739', 'https://www.ncbi.nlm.nih.gov/gene/3754', 'https://www.ncbi.nlm.nih.gov/gene/3816', 'https://www.ncbi.nlm.nih.gov/gene/51361', 'https://www.ncbi.nlm.nih.gov/gene/255061'], labels=['HK1', 'KCNA4', 'KCNF1', 'KLK1', 'HOOK1', 'TAC4']),\n",
       " Recommendation(id='G_Hk1_AT2', candidates=[(3098, np.float64(1.0)), (3739, np.float64(1.0)), (3754, np.float64(1.0)), (3816, np.float64(1.0)), (51361, np.float64(1.0)), (255061, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3098', 'https://www.ncbi.nlm.nih.gov/gene/3739', 'https://www.ncbi.nlm.nih.gov/gene/3754', 'https://www.ncbi.nlm.nih.gov/gene/3816', 'https://www.ncbi.nlm.nih.gov/gene/51361', 'https://www.ncbi.nlm.nih.gov/gene/255061'], labels=['HK1', 'KCNA4', 'KCNF1', 'KLK1', 'HOOK1', 'TAC4']),\n",
       " Recommendation(id='G_Hk2_AT1', candidates=[(3099, np.float64(1.0)), (3741, np.float64(1.0)), (3755, np.float64(1.0)), (3796, np.float64(1.0)), (3817, np.float64(1.0)), (29911, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3099', 'https://www.ncbi.nlm.nih.gov/gene/3741', 'https://www.ncbi.nlm.nih.gov/gene/3755', 'https://www.ncbi.nlm.nih.gov/gene/3796', 'https://www.ncbi.nlm.nih.gov/gene/3817', 'https://www.ncbi.nlm.nih.gov/gene/29911'], labels=['HK2', 'KCNA5', 'KCNG1', 'KIF2A', 'KLK2', 'HOOK2']),\n",
       " Recommendation(id='G_Hk3_AT1', candidates=[(354, np.float64(1.0)), (3101, np.float64(1.0)), (8357, np.float64(1.0)), (84376, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/354', 'https://www.ncbi.nlm.nih.gov/gene/3101', 'https://www.ncbi.nlm.nih.gov/gene/8357', 'https://www.ncbi.nlm.nih.gov/gene/84376'], labels=['KLK3', 'HK3', 'H3C10', 'HOOK3']),\n",
       " Recommendation(id='G_Hmbs_AT1', candidates=[(3145, np.float64(1.0)), (56006, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3145', 'https://www.ncbi.nlm.nih.gov/gene/56006'], labels=['HMBS', 'SMG9']),\n",
       " Recommendation(id='G_Hmox1_AT1', candidates=[(3162, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3162'], labels=['HMOX1']),\n",
       " Recommendation(id='G_Hmox2_AT1', candidates=[(3163, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3163'], labels=['HMOX2']),\n",
       " Recommendation(id='G_Idh1_AT1', candidates=[(1732, np.float64(1.0)), (3417, np.float64(1.0)), (283987, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1732', 'https://www.ncbi.nlm.nih.gov/gene/3417', 'https://www.ncbi.nlm.nih.gov/gene/283987'], labels=['DIH1', 'IDH1', 'HID1']),\n",
       " Recommendation(id='G_ImpDH1_AT1', candidates=[(3614, np.float64(1.0)), (727833, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3614', 'https://www.ncbi.nlm.nih.gov/gene/727833'], labels=['IMPDH1', 'IMPDH1P11']),\n",
       " Recommendation(id='G_ImpDH2_AT1', candidates=[(3615, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3615'], labels=['IMPDH2']),\n",
       " Recommendation(id='G_ImpDH1_AT2', candidates=[(3614, np.float64(1.0)), (727833, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3614', 'https://www.ncbi.nlm.nih.gov/gene/727833'], labels=['IMPDH1', 'IMPDH1P11']),\n",
       " Recommendation(id='G_Ldhb_AT1', candidates=[(3945, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3945'], labels=['LDHB']),\n",
       " Recommendation(id='G_Ldha_AT1', candidates=[(3939, np.float64(1.0)), (60526, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3939', 'https://www.ncbi.nlm.nih.gov/gene/60526'], labels=['LDHA', 'LDAH']),\n",
       " Recommendation(id='G_Glo1_AT1', candidates=[(2739, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2739'], labels=['GLO1']),\n",
       " Recommendation(id='G_Pla2g4a_AT1', candidates=[(5321, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5321'], labels=['PLA2G4A']),\n",
       " Recommendation(id='G_Clc_AT1', candidates=[(1178, np.float64(1.0)), (1420, np.float64(1.0)), (23529, np.float64(1.0)), (200575, np.float64(1.0)), (727676, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1178', 'https://www.ncbi.nlm.nih.gov/gene/1420', 'https://www.ncbi.nlm.nih.gov/gene/23529', 'https://www.ncbi.nlm.nih.gov/gene/200575', 'https://www.ncbi.nlm.nih.gov/gene/727676'], labels=['CLC', 'CRYGC', 'CLCF1', 'CRYGEP', 'SNORD118']),\n",
       " Recommendation(id='G_Lta4h_AT1', candidates=[(4048, np.float64(1.0)), (389208, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4048', 'https://www.ncbi.nlm.nih.gov/gene/389208'], labels=['LTA4H', 'TMPRSS11F']),\n",
       " Recommendation(id='G_Mpi_AT1', candidates=[(3612, np.float64(1.0)), (3842, np.float64(1.0)), (4284, np.float64(1.0)), (4285, np.float64(1.0)), (4351, np.float64(1.0)), (5292, np.float64(1.0)), (5373, np.float64(1.0)), (6590, np.float64(1.0)), (8315, np.float64(1.0)), (8834, np.float64(1.0)), (727764, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3612', 'https://www.ncbi.nlm.nih.gov/gene/3842', 'https://www.ncbi.nlm.nih.gov/gene/4284', 'https://www.ncbi.nlm.nih.gov/gene/4285', 'https://www.ncbi.nlm.nih.gov/gene/4351', 'https://www.ncbi.nlm.nih.gov/gene/5292', 'https://www.ncbi.nlm.nih.gov/gene/5373', 'https://www.ncbi.nlm.nih.gov/gene/6590', 'https://www.ncbi.nlm.nih.gov/gene/8315', 'https://www.ncbi.nlm.nih.gov/gene/8834', 'https://www.ncbi.nlm.nih.gov/gene/727764'], labels=['IMPA1', 'TNPO1', 'MIP', 'MIPEP', 'MPI', 'PIM1', 'PMM2', 'SLPI', 'BRAP', 'TMEM11', 'MAFIP']),\n",
       " Recommendation(id='G_Mdh1_AT1', candidates=[(4190, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4190'], labels=['MDH1']),\n",
       " Recommendation(id='G_Me1_AT1', candidates=[(4199, np.float64(1.0)), (57717, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4199', 'https://www.ncbi.nlm.nih.gov/gene/57717'], labels=['ME1', 'PCDHB16']),\n",
       " Recommendation(id='G_Mat1a_AT1', candidates=[(4143, np.float64(1.0)), (10573, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4143', 'https://www.ncbi.nlm.nih.gov/gene/10573'], labels=['MAT1A', 'MRPL28']),\n",
       " Recommendation(id='G_Slc43a1_AT1', candidates=[(6569, np.float64(1.0)), (8501, np.float64(1.0)), (26266, np.float64(1.0)), (54946, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6569', 'https://www.ncbi.nlm.nih.gov/gene/8501', 'https://www.ncbi.nlm.nih.gov/gene/26266', 'https://www.ncbi.nlm.nih.gov/gene/54946'], labels=['SLC34A1', 'SLC43A1', 'SLC13A4', 'SLC41A3']),\n",
       " Recommendation(id='G_Synj1_AT1', candidates=[(8867, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8867'], labels=['SYNJ1']),\n",
       " Recommendation(id='G_Synj1_AT2', candidates=[(8867, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8867'], labels=['SYNJ1']),\n",
       " Recommendation(id='G_Inpp5d_AT1', candidates=[(3635, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3635'], labels=['INPP5D']),\n",
       " Recommendation(id='G_Itpkc_AT1', candidates=[(80271, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/80271'], labels=['ITPKC']),\n",
       " Recommendation(id='G_Itpka_AT1', candidates=[(3706, np.float64(1.0)), (3767, np.float64(1.0)), (64400, np.float64(1.0)), (148229, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3706', 'https://www.ncbi.nlm.nih.gov/gene/3767', 'https://www.ncbi.nlm.nih.gov/gene/64400', 'https://www.ncbi.nlm.nih.gov/gene/148229'], labels=['ITPKA', 'KCNJ11', 'AKTIP', 'ATP8B3']),\n",
       " Recommendation(id='G_Itpkb_AT1', candidates=[(3707, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3707'], labels=['ITPKB']),\n",
       " Recommendation(id='G_Impa1_AT1', candidates=[(3612, np.float64(1.0)), (6348, np.float64(1.0)), (9255, np.float64(1.0)), (79568, np.float64(1.0)), (170575, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3612', 'https://www.ncbi.nlm.nih.gov/gene/6348', 'https://www.ncbi.nlm.nih.gov/gene/9255', 'https://www.ncbi.nlm.nih.gov/gene/79568', 'https://www.ncbi.nlm.nih.gov/gene/170575'], labels=['IMPA1', 'CCL3', 'AIMP1', 'MAIP1', 'GIMAP1']),\n",
       " Recommendation(id='G_Isyna1_AT1', candidates=[(51477, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/51477'], labels=['ISYNA1']),\n",
       " Recommendation(id='G_Mtap_AT1', candidates=[(498, np.float64(1.0)), (522, np.float64(1.0)), (3030, np.float64(1.0)), (4137, np.float64(1.0)), (4507, np.float64(1.0)), (5757, np.float64(1.0)), (5761, np.float64(1.0)), (26586, np.float64(1.0)), (51151, np.float64(1.0)), (222962, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/498', 'https://www.ncbi.nlm.nih.gov/gene/522', 'https://www.ncbi.nlm.nih.gov/gene/3030', 'https://www.ncbi.nlm.nih.gov/gene/4137', 'https://www.ncbi.nlm.nih.gov/gene/4507', 'https://www.ncbi.nlm.nih.gov/gene/5757', 'https://www.ncbi.nlm.nih.gov/gene/5761', 'https://www.ncbi.nlm.nih.gov/gene/26586', 'https://www.ncbi.nlm.nih.gov/gene/51151', 'https://www.ncbi.nlm.nih.gov/gene/222962'], labels=['ATP5F1A', 'ATP5PF', 'HADHA', 'MAPT', 'MTAP', 'PTMA', 'PTMAP4', 'CKAP2', 'SLC45A2', 'SLC29A4']),\n",
       " Recommendation(id='G_Nadk_AT1', candidates=[(4830, np.float64(1.0)), (65220, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4830', 'https://www.ncbi.nlm.nih.gov/gene/65220'], labels=['NME1', 'NADK']),\n",
       " Recommendation(id='G_Nadsyn1_AT1', candidates=[(55191, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/55191'], labels=['NADSYN1']),\n",
       " Recommendation(id='G_NME7_AT2', candidates=[(1073, np.float64(1.0)), (29922, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1073', 'https://www.ncbi.nlm.nih.gov/gene/29922'], labels=['CFL2', 'NME7']),\n",
       " Recommendation(id='G_NME6_AT1', candidates=[(10201, np.float64(1.0)), (390594, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10201', 'https://www.ncbi.nlm.nih.gov/gene/390594'], labels=['NME6', 'KBTBD13']),\n",
       " Recommendation(id='G_NME7_AT1', candidates=[(1073, np.float64(1.0)), (29922, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1073', 'https://www.ncbi.nlm.nih.gov/gene/29922'], labels=['CFL2', 'NME7']),\n",
       " Recommendation(id='G_NME3_AT1', candidates=[(58, np.float64(1.0)), (4832, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/58', 'https://www.ncbi.nlm.nih.gov/gene/4832'], labels=['ACTA1', 'NME3']),\n",
       " Recommendation(id='G_Nmnat3_AT1', candidates=[(349565, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/349565'], labels=['NMNAT3']),\n",
       " Recommendation(id='G_Np_AT1', candidates=[(10009, np.float64(1.0)), (158399, np.float64(1.0)), (171017, np.float64(1.0)), (220929, np.float64(1.0)), (283933, np.float64(1.0)), (100505852, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10009', 'https://www.ncbi.nlm.nih.gov/gene/158399', 'https://www.ncbi.nlm.nih.gov/gene/171017', 'https://www.ncbi.nlm.nih.gov/gene/220929', 'https://www.ncbi.nlm.nih.gov/gene/283933', 'https://www.ncbi.nlm.nih.gov/gene/100505852'], labels=['ZBTB33', 'ZNF483', 'ZNF384', 'ZNF438', 'ZNF843', 'ZNF355P']),\n",
       " Recommendation(id='G_Nt5C2_AT1', candidates=[(22978, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/22978'], labels=['NT5C2']),\n",
       " Recommendation(id='G_Nt5C3_AT1', candidates=[(51251, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/51251'], labels=['NT5C3A']),\n",
       " Recommendation(id='G_Nt5C_AT1', candidates=[(30833, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/30833'], labels=['NT5C']),\n",
       " Recommendation(id='G_Umps_AT1', candidates=[(7372, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/7372'], labels=['UMPS']),\n",
       " Recommendation(id='G_Oplah_AT1', candidates=[(26873, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/26873'], labels=['OPLAH']),\n",
       " Recommendation(id='G_Odc1_AT1', candidates=[(1036, np.float64(1.0)), (4953, np.float64(1.0)), (6103, np.float64(1.0)), (6532, np.float64(1.0)), (8099, np.float64(1.0)), (10393, np.float64(1.0)), (11259, np.float64(1.0)), (25839, np.float64(1.0)), (89874, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1036', 'https://www.ncbi.nlm.nih.gov/gene/4953', 'https://www.ncbi.nlm.nih.gov/gene/6103', 'https://www.ncbi.nlm.nih.gov/gene/6532', 'https://www.ncbi.nlm.nih.gov/gene/8099', 'https://www.ncbi.nlm.nih.gov/gene/10393', 'https://www.ncbi.nlm.nih.gov/gene/11259', 'https://www.ncbi.nlm.nih.gov/gene/25839', 'https://www.ncbi.nlm.nih.gov/gene/89874'], labels=['CDO1', 'ODC1', 'RPGR', 'SLC6A4', 'CDK2AP1', 'ANAPC10', 'FILIP1L', 'COG4', 'SLC25A21']),\n",
       " Recommendation(id='G_Pde1A_AT1', candidates=[(5136, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5136'], labels=['PDE1A']),\n",
       " Recommendation(id='G_Pde4B_AT1', candidates=[(5142, np.float64(1.0)), (51202, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5142', 'https://www.ncbi.nlm.nih.gov/gene/51202'], labels=['PDE4B', 'DDX47']),\n",
       " Recommendation(id='G_Pde7A_AT1', candidates=[(5150, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5150'], labels=['PDE7A']),\n",
       " Recommendation(id='G_Pde1A_AT2', candidates=[(5136, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5136'], labels=['PDE1A']),\n",
       " Recommendation(id='G_Pde3B_AT2', candidates=[(5140, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5140'], labels=['PDE3B']),\n",
       " Recommendation(id='G_Pde11A_AT1', candidates=[(50940, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/50940'], labels=['PDE11A']),\n",
       " Recommendation(id='G_Pde8b_AT1', candidates=[(8622, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8622'], labels=['PDE8B']),\n",
       " Recommendation(id='G_Pde8a_AT5', candidates=[(5151, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5151'], labels=['PDE8A']),\n",
       " Recommendation(id='G_Pde8a_AT2', candidates=[(5151, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5151'], labels=['PDE8A']),\n",
       " Recommendation(id='G_Pde4D_AT1', candidates=[(5142, np.float64(1.0)), (5144, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5142', 'https://www.ncbi.nlm.nih.gov/gene/5144'], labels=['PDE4B', 'PDE4D']),\n",
       " Recommendation(id='G_Pde1B_AT1', candidates=[(5153, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5153'], labels=['PDE1B']),\n",
       " Recommendation(id='G_Pde8a_AT1', candidates=[(5151, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5151'], labels=['PDE8A']),\n",
       " Recommendation(id='G_Pde8a_AT4', candidates=[(5151, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5151'], labels=['PDE8A']),\n",
       " Recommendation(id='G_Pde8a_AT3', candidates=[(5151, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5151'], labels=['PDE8A']),\n",
       " Recommendation(id='G_Pde1C_AT1', candidates=[(5137, np.float64(1.0)), (79974, np.float64(1.0)), (221120, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5137', 'https://www.ncbi.nlm.nih.gov/gene/79974', 'https://www.ncbi.nlm.nih.gov/gene/221120'], labels=['PDE1C', 'CPED1', 'ALKBH3']),\n",
       " Recommendation(id='G_Pde4C_AT1', candidates=[(5143, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5143'], labels=['PDE4C']),\n",
       " Recommendation(id='G_Pde4A_AT1', candidates=[(5141, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5141'], labels=['PDE4A']),\n",
       " Recommendation(id='G_Pde2A_AT1', candidates=[(5138, np.float64(1.0)), (5146, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5138', 'https://www.ncbi.nlm.nih.gov/gene/5146'], labels=['PDE2A', 'PDE6C']),\n",
       " Recommendation(id='G_Pde3A_AT1', candidates=[(5139, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5139'], labels=['PDE3A']),\n",
       " Recommendation(id='G_Pde7A_AT2', candidates=[(5150, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5150'], labels=['PDE7A']),\n",
       " Recommendation(id='G_Pde7B_AT1', candidates=[(27115, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/27115'], labels=['PDE7B']),\n",
       " Recommendation(id='G_Pde3B_AT1', candidates=[(5140, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5140'], labels=['PDE3B']),\n",
       " Recommendation(id='G_PnpO_AT1', candidates=[(55163, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/55163'], labels=['PNPO']),\n",
       " Recommendation(id='G_PdxP_AT1', candidates=[(1804, np.float64(1.0)), (57026, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1804', 'https://www.ncbi.nlm.nih.gov/gene/57026'], labels=['DPP6', 'PDXP']),\n",
       " Recommendation(id='G_Pcyt2_AT1', candidates=[(5833, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5833'], labels=['PCYT2']),\n",
       " Recommendation(id='G_Pfkl_AT1', candidates=[(5211, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5211'], labels=['PFKL']),\n",
       " Recommendation(id='G_Pfkm_AT1', candidates=[(5213, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5213'], labels=['PFKM']),\n",
       " Recommendation(id='G_Pfkl_AT2', candidates=[(5211, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5211'], labels=['PFKL']),\n",
       " Recommendation(id='G_Gpi_AT1', candidates=[(633, np.float64(1.0)), (2695, np.float64(1.0)), (2771, np.float64(1.0)), (2821, np.float64(1.0)), (10007, np.float64(1.0)), (54858, np.float64(1.0)), (114897, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/633', 'https://www.ncbi.nlm.nih.gov/gene/2695', 'https://www.ncbi.nlm.nih.gov/gene/2771', 'https://www.ncbi.nlm.nih.gov/gene/2821', 'https://www.ncbi.nlm.nih.gov/gene/10007', 'https://www.ncbi.nlm.nih.gov/gene/54858', 'https://www.ncbi.nlm.nih.gov/gene/114897'], labels=['BGN', 'GIP', 'GNAI2', 'GPI', 'GNPDA1', 'PGPEP1', 'C1QTNF1']),\n",
       " Recommendation(id='G_Pgk1_AT1', candidates=[(5230, np.float64(1.0)), (5592, np.float64(1.0)), (201989, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5230', 'https://www.ncbi.nlm.nih.gov/gene/5592', 'https://www.ncbi.nlm.nih.gov/gene/201989'], labels=['PGK1', 'PRKG1', 'GK6P']),\n",
       " Recommendation(id='G_Pgls_AT1', candidates=[(25796, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/25796'], labels=['PGLS']),\n",
       " Recommendation(id='G_Got1_AT1', candidates=[(2805, np.float64(1.0)), (8473, np.float64(1.0)), (51026, np.float64(1.0)), (127845, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2805', 'https://www.ncbi.nlm.nih.gov/gene/8473', 'https://www.ncbi.nlm.nih.gov/gene/51026', 'https://www.ncbi.nlm.nih.gov/gene/127845'], labels=['GOT1', 'OGT', 'GOLT1B', 'GOLT1A']),\n",
       " Recommendation(id='G_Plcb3_AT1', candidates=[(5331, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5331'], labels=['PLCB3']),\n",
       " Recommendation(id='G_Pip5k1b_AT1', candidates=[(8395, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8395'], labels=['PIP5K1B']),\n",
       " Recommendation(id='G_Pip5k2a_AT1', candidates=[(5305, np.float64(1.0)), (375133, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5305', 'https://www.ncbi.nlm.nih.gov/gene/375133'], labels=['PIP4K2A', 'PI4KAP2']),\n",
       " Recommendation(id='G_Pip5k1c_AT1', candidates=[(23396, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/23396'], labels=['PIP5K1C']),\n",
       " Recommendation(id='G_Pip5k2c_AT1', candidates=[(79837, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/79837'], labels=['PIP4K2C']),\n",
       " Recommendation(id='G_Pip5k2b_AT2', candidates=[(8396, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8396'], labels=['PIP4K2B']),\n",
       " Recommendation(id='G_Pip5k3_AT1', candidates=[(200576, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/200576'], labels=['PIKFYVE']),\n",
       " Recommendation(id='G_Pip5k3_AT3', candidates=[(200576, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/200576'], labels=['PIKFYVE']),\n",
       " Recommendation(id='G_Pip5k1a_AT1', candidates=[(8394, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8394'], labels=['PIP5K1A']),\n",
       " Recommendation(id='G_Pip5k2b_AT1', candidates=[(8396, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8396'], labels=['PIP4K2B']),\n",
       " Recommendation(id='G_Pip5k3_AT2', candidates=[(200576, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/200576'], labels=['PIKFYVE']),\n",
       " Recommendation(id='G_Pi4kii_AT1', candidates=[(55361, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/55361'], labels=['PI4K2A']),\n",
       " Recommendation(id='G_Pik4cb_AT1', candidates=[(5298, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5298'], labels=['PI4KB']),\n",
       " Recommendation(id='G_Pik4ca_AT1', candidates=[(5297, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5297'], labels=['PI4KA']),\n",
       " Recommendation(id='G_Pla2g2a_AT1', candidates=[(5320, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5320'], labels=['PLA2G2A']),\n",
       " Recommendation(id='G_Pla2g6_AT1', candidates=[(8398, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8398'], labels=['PLA2G6']),\n",
       " Recommendation(id='G_Pla2g6_AT2', candidates=[(8398, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8398'], labels=['PLA2G6']),\n",
       " Recommendation(id='G_Pmm2_AT1', candidates=[(4313, np.float64(1.0)), (5373, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/4313', 'https://www.ncbi.nlm.nih.gov/gene/5373'], labels=['MMP2', 'PMM2']),\n",
       " Recommendation(id='G_Ppa2_AT2', candidates=[(5524, np.float64(1.0)), (7512, np.float64(1.0)), (8611, np.float64(1.0)), (10514, np.float64(1.0)), (27068, np.float64(1.0)), (163404, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5524', 'https://www.ncbi.nlm.nih.gov/gene/7512', 'https://www.ncbi.nlm.nih.gov/gene/8611', 'https://www.ncbi.nlm.nih.gov/gene/10514', 'https://www.ncbi.nlm.nih.gov/gene/27068', 'https://www.ncbi.nlm.nih.gov/gene/163404'], labels=['PTPA', 'XPNPEP2', 'PLPP1', 'MYBBP1A', 'PPA2', 'PLPPR5']),\n",
       " Recommendation(id='G_Ppa2_AT1', candidates=[(5524, np.float64(1.0)), (7512, np.float64(1.0)), (8611, np.float64(1.0)), (10514, np.float64(1.0)), (27068, np.float64(1.0)), (163404, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5524', 'https://www.ncbi.nlm.nih.gov/gene/7512', 'https://www.ncbi.nlm.nih.gov/gene/8611', 'https://www.ncbi.nlm.nih.gov/gene/10514', 'https://www.ncbi.nlm.nih.gov/gene/27068', 'https://www.ncbi.nlm.nih.gov/gene/163404'], labels=['PTPA', 'XPNPEP2', 'PLPP1', 'MYBBP1A', 'PPA2', 'PLPPR5']),\n",
       " Recommendation(id='G_Ppa2_AT3', candidates=[(5524, np.float64(1.0)), (7512, np.float64(1.0)), (8611, np.float64(1.0)), (10514, np.float64(1.0)), (27068, np.float64(1.0)), (163404, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5524', 'https://www.ncbi.nlm.nih.gov/gene/7512', 'https://www.ncbi.nlm.nih.gov/gene/8611', 'https://www.ncbi.nlm.nih.gov/gene/10514', 'https://www.ncbi.nlm.nih.gov/gene/27068', 'https://www.ncbi.nlm.nih.gov/gene/163404'], labels=['PTPA', 'XPNPEP2', 'PLPP1', 'MYBBP1A', 'PPA2', 'PLPPR5']),\n",
       " Recommendation(id='G_Ppa2_AT4', candidates=[(5524, np.float64(1.0)), (7512, np.float64(1.0)), (8611, np.float64(1.0)), (10514, np.float64(1.0)), (27068, np.float64(1.0)), (163404, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5524', 'https://www.ncbi.nlm.nih.gov/gene/7512', 'https://www.ncbi.nlm.nih.gov/gene/8611', 'https://www.ncbi.nlm.nih.gov/gene/10514', 'https://www.ncbi.nlm.nih.gov/gene/27068', 'https://www.ncbi.nlm.nih.gov/gene/163404'], labels=['PTPA', 'XPNPEP2', 'PLPP1', 'MYBBP1A', 'PPA2', 'PLPPR5']),\n",
       " Recommendation(id='G_Ppap2a_AT2', candidates=[(8611, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8611'], labels=['PLPP1']),\n",
       " Recommendation(id='G_Ppap2c_AT1', candidates=[(8612, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8612'], labels=['PLPP2']),\n",
       " Recommendation(id='G_Ppap2b_AT1', candidates=[(8613, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8613'], labels=['PLPP3']),\n",
       " Recommendation(id='G_Ppap2a_AT1', candidates=[(8611, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8611'], labels=['PLPP1']),\n",
       " Recommendation(id='G_Alad_AT1', candidates=[(210, np.float64(1.0)), (226, np.float64(1.0)), (3898, np.float64(1.0)), (161823, np.float64(1.0)), (112694756, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/210', 'https://www.ncbi.nlm.nih.gov/gene/226', 'https://www.ncbi.nlm.nih.gov/gene/3898', 'https://www.ncbi.nlm.nih.gov/gene/161823', 'https://www.ncbi.nlm.nih.gov/gene/112694756'], labels=['ALAD', 'ALDOA', 'LAD1', 'MAPDA', 'LOC112694756']),\n",
       " Recommendation(id='G_Alad_AT2', candidates=[(210, np.float64(1.0)), (226, np.float64(1.0)), (3898, np.float64(1.0)), (161823, np.float64(1.0)), (112694756, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/210', 'https://www.ncbi.nlm.nih.gov/gene/226', 'https://www.ncbi.nlm.nih.gov/gene/3898', 'https://www.ncbi.nlm.nih.gov/gene/161823', 'https://www.ncbi.nlm.nih.gov/gene/112694756'], labels=['ALAD', 'ALDOA', 'LAD1', 'MAPDA', 'LOC112694756']),\n",
       " Recommendation(id='G_Pgm1_AT1', candidates=[(5236, np.float64(1.0)), (7341, np.float64(1.0)), (219972, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5236', 'https://www.ncbi.nlm.nih.gov/gene/7341', 'https://www.ncbi.nlm.nih.gov/gene/219972'], labels=['PGM1', 'SUMO1', 'MPEG1']),\n",
       " Recommendation(id='G_Prps1l1_AT2', candidates=[(221823, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/221823'], labels=['PRPS1L1']),\n",
       " Recommendation(id='G_Prps1_AT1', candidates=[(5631, np.float64(1.0)), (53637, np.float64(1.0)), (221823, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5631', 'https://www.ncbi.nlm.nih.gov/gene/53637', 'https://www.ncbi.nlm.nih.gov/gene/221823'], labels=['PRPS1', 'S1PR5', 'PRPS1L1']),\n",
       " Recommendation(id='G_Prps1l1_AT1', candidates=[(221823, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/221823'], labels=['PRPS1L1']),\n",
       " Recommendation(id='G_PdxK_AT1', candidates=[(8566, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8566'], labels=['PDXK']),\n",
       " Recommendation(id='G_Aox1_AT1', candidates=[(316, np.float64(1.0)), (5018, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/316', 'https://www.ncbi.nlm.nih.gov/gene/5018'], labels=['AOX1', 'OXA1L']),\n",
       " Recommendation(id='G_Pkm2_AT2', candidates=[(5315, np.float64(1.0)), (10654, np.float64(1.0)), (26524, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5315', 'https://www.ncbi.nlm.nih.gov/gene/10654', 'https://www.ncbi.nlm.nih.gov/gene/26524'], labels=['PKM', 'PMVK', 'LATS2']),\n",
       " Recommendation(id='G_Pklr_AT2', candidates=[(5313, np.float64(1.0)), (9252, np.float64(1.0)), (57733, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5313', 'https://www.ncbi.nlm.nih.gov/gene/9252', 'https://www.ncbi.nlm.nih.gov/gene/57733'], labels=['PKLR', 'RPS6KA5', 'GBA3']),\n",
       " Recommendation(id='G_Pklr_AT1', candidates=[(5313, np.float64(1.0)), (9252, np.float64(1.0)), (57733, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5313', 'https://www.ncbi.nlm.nih.gov/gene/9252', 'https://www.ncbi.nlm.nih.gov/gene/57733'], labels=['PKLR', 'RPS6KA5', 'GBA3']),\n",
       " Recommendation(id='G_Pkm2_AT3', candidates=[(5315, np.float64(1.0)), (10654, np.float64(1.0)), (26524, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5315', 'https://www.ncbi.nlm.nih.gov/gene/10654', 'https://www.ncbi.nlm.nih.gov/gene/26524'], labels=['PKM', 'PMVK', 'LATS2']),\n",
       " Recommendation(id='G_Pkm2_AT1', candidates=[(5315, np.float64(1.0)), (10654, np.float64(1.0)), (26524, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5315', 'https://www.ncbi.nlm.nih.gov/gene/10654', 'https://www.ncbi.nlm.nih.gov/gene/26524'], labels=['PKM', 'PMVK', 'LATS2']),\n",
       " Recommendation(id='G_Rfk_AT1', candidates=[(2444, np.float64(1.0)), (55312, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2444', 'https://www.ncbi.nlm.nih.gov/gene/55312'], labels=['FRK', 'RFK']),\n",
       " Recommendation(id='G_10390_AT1', candidates=[(10390, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10390'], labels=['CEPT1']),\n",
       " Recommendation(id='G_55500', candidates=[(55500, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/55500'], labels=['ETNK1']),\n",
       " Recommendation(id='G_55224_AT1', candidates=[(55224, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/55224'], labels=['ETNK2']),\n",
       " Recommendation(id='G_29124', candidates=[(29124, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/29124'], labels=['LGALS13']),\n",
       " Recommendation(id='G_10846', candidates=[(10846, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10846'], labels=['PDE10A']),\n",
       " Recommendation(id='G_55276', candidates=[(55276, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/55276'], labels=['PGM2']),\n",
       " Recommendation(id='G_9429', candidates=[(9429, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/9429'], labels=['ABCG2']),\n",
       " Recommendation(id='G_51084', candidates=[(51084, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/51084'], labels=['CRYL1']),\n",
       " Recommendation(id='G_Tat_AT1', candidates=[(6898, np.float64(1.0)), (79969, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6898', 'https://www.ncbi.nlm.nih.gov/gene/79969'], labels=['TAT', 'ATAT1']),\n",
       " Recommendation(id='G_Renbp_AT1', candidates=[(5973, np.float64(1.0)), (6651, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5973', 'https://www.ncbi.nlm.nih.gov/gene/6651'], labels=['RENBP', 'SON']),\n",
       " Recommendation(id='G_Nagk_AT1', candidates=[(55577, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/55577'], labels=['NAGK']),\n",
       " Recommendation(id='G_Nans_AT1', candidates=[(54187, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/54187'], labels=['NANS']),\n",
       " Recommendation(id='G_Nanp_AT1', candidates=[(140838, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/140838'], labels=['NANP']),\n",
       " Recommendation(id='G_Acp1_AT1', candidates=[(52, np.float64(1.0)), (117, np.float64(1.0)), (1316, np.float64(1.0)), (1357, np.float64(1.0)), (1844, np.float64(1.0)), (4706, np.float64(1.0)), (5167, np.float64(1.0)), (5652, np.float64(1.0)), (7187, np.float64(1.0)), (8624, np.float64(1.0)), (10487, np.float64(1.0)), (29957, np.float64(1.0)), (64682, np.float64(1.0)), (221120, np.float64(1.0)), (100750225, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/52', 'https://www.ncbi.nlm.nih.gov/gene/117', 'https://www.ncbi.nlm.nih.gov/gene/1316', 'https://www.ncbi.nlm.nih.gov/gene/1357', 'https://www.ncbi.nlm.nih.gov/gene/1844', 'https://www.ncbi.nlm.nih.gov/gene/4706', 'https://www.ncbi.nlm.nih.gov/gene/5167', 'https://www.ncbi.nlm.nih.gov/gene/5652', 'https://www.ncbi.nlm.nih.gov/gene/7187', 'https://www.ncbi.nlm.nih.gov/gene/8624', 'https://www.ncbi.nlm.nih.gov/gene/10487', 'https://www.ncbi.nlm.nih.gov/gene/29957', 'https://www.ncbi.nlm.nih.gov/gene/64682', 'https://www.ncbi.nlm.nih.gov/gene/221120', 'https://www.ncbi.nlm.nih.gov/gene/100750225'], labels=['ACP1', 'ADCYAP1R1', 'KLF6', 'CPA1', 'DUSP2', 'NDUFAB1', 'ENPP1', 'PRSS8', 'TRAF3', 'PSMG1', 'CAP1', 'SLC25A24', 'ANAPC1', 'ALKBH3', 'PCAT1']),\n",
       " Recommendation(id='G_Acp1_AT2', candidates=[(52, np.float64(1.0)), (117, np.float64(1.0)), (1316, np.float64(1.0)), (1357, np.float64(1.0)), (1844, np.float64(1.0)), (4706, np.float64(1.0)), (5167, np.float64(1.0)), (5652, np.float64(1.0)), (7187, np.float64(1.0)), (8624, np.float64(1.0)), (10487, np.float64(1.0)), (29957, np.float64(1.0)), (64682, np.float64(1.0)), (221120, np.float64(1.0)), (100750225, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/52', 'https://www.ncbi.nlm.nih.gov/gene/117', 'https://www.ncbi.nlm.nih.gov/gene/1316', 'https://www.ncbi.nlm.nih.gov/gene/1357', 'https://www.ncbi.nlm.nih.gov/gene/1844', 'https://www.ncbi.nlm.nih.gov/gene/4706', 'https://www.ncbi.nlm.nih.gov/gene/5167', 'https://www.ncbi.nlm.nih.gov/gene/5652', 'https://www.ncbi.nlm.nih.gov/gene/7187', 'https://www.ncbi.nlm.nih.gov/gene/8624', 'https://www.ncbi.nlm.nih.gov/gene/10487', 'https://www.ncbi.nlm.nih.gov/gene/29957', 'https://www.ncbi.nlm.nih.gov/gene/64682', 'https://www.ncbi.nlm.nih.gov/gene/221120', 'https://www.ncbi.nlm.nih.gov/gene/100750225'], labels=['ACP1', 'ADCYAP1R1', 'KLF6', 'CPA1', 'DUSP2', 'NDUFAB1', 'ENPP1', 'PRSS8', 'TRAF3', 'PSMG1', 'CAP1', 'SLC25A24', 'ANAPC1', 'ALKBH3', 'PCAT1']),\n",
       " Recommendation(id='G_Acp1_AT3', candidates=[(52, np.float64(1.0)), (117, np.float64(1.0)), (1316, np.float64(1.0)), (1357, np.float64(1.0)), (1844, np.float64(1.0)), (4706, np.float64(1.0)), (5167, np.float64(1.0)), (5652, np.float64(1.0)), (7187, np.float64(1.0)), (8624, np.float64(1.0)), (10487, np.float64(1.0)), (29957, np.float64(1.0)), (64682, np.float64(1.0)), (221120, np.float64(1.0)), (100750225, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/52', 'https://www.ncbi.nlm.nih.gov/gene/117', 'https://www.ncbi.nlm.nih.gov/gene/1316', 'https://www.ncbi.nlm.nih.gov/gene/1357', 'https://www.ncbi.nlm.nih.gov/gene/1844', 'https://www.ncbi.nlm.nih.gov/gene/4706', 'https://www.ncbi.nlm.nih.gov/gene/5167', 'https://www.ncbi.nlm.nih.gov/gene/5652', 'https://www.ncbi.nlm.nih.gov/gene/7187', 'https://www.ncbi.nlm.nih.gov/gene/8624', 'https://www.ncbi.nlm.nih.gov/gene/10487', 'https://www.ncbi.nlm.nih.gov/gene/29957', 'https://www.ncbi.nlm.nih.gov/gene/64682', 'https://www.ncbi.nlm.nih.gov/gene/221120', 'https://www.ncbi.nlm.nih.gov/gene/100750225'], labels=['ACP1', 'ADCYAP1R1', 'KLF6', 'CPA1', 'DUSP2', 'NDUFAB1', 'ENPP1', 'PRSS8', 'TRAF3', 'PSMG1', 'CAP1', 'SLC25A24', 'ANAPC1', 'ALKBH3', 'PCAT1']),\n",
       " Recommendation(id='G_Ada_AT1', candidates=[(100, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/100'], labels=['ADA']),\n",
       " Recommendation(id='G_Ak1_AT1', candidates=[(203, np.float64(1.0)), (2900, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/203', 'https://www.ncbi.nlm.nih.gov/gene/2900'], labels=['AK1', 'GRIK4']),\n",
       " Recommendation(id='G_Amd1_AT1', candidates=[(262, np.float64(1.0)), (4084, np.float64(1.0)), (4882, np.float64(1.0)), (8379, np.float64(1.0)), (10286, np.float64(1.0)), (29949, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/262', 'https://www.ncbi.nlm.nih.gov/gene/4084', 'https://www.ncbi.nlm.nih.gov/gene/4882', 'https://www.ncbi.nlm.nih.gov/gene/8379', 'https://www.ncbi.nlm.nih.gov/gene/10286', 'https://www.ncbi.nlm.nih.gov/gene/29949'], labels=['AMD1', 'MXD1', 'NPR2', 'MAD1L1', 'BCAS2', 'IL19']),\n",
       " Recommendation(id='G_Adcy1_AT1', candidates=[(95, np.float64(1.0)), (107, np.float64(1.0)), (9085, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/95', 'https://www.ncbi.nlm.nih.gov/gene/107', 'https://www.ncbi.nlm.nih.gov/gene/9085'], labels=['ACY1', 'ADCY1', 'CDY1']),\n",
       " Recommendation(id='G_Adcy9_AT1', candidates=[(115, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/115'], labels=['ADCY9']),\n",
       " Recommendation(id='G_Adcy7_AT1', candidates=[(113, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/113'], labels=['ADCY7']),\n",
       " Recommendation(id='G_Adcy5_AT1', candidates=[(111, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/111'], labels=['ADCY5']),\n",
       " Recommendation(id='G_Adcy6_AT1', candidates=[(112, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/112'], labels=['ADCY6']),\n",
       " Recommendation(id='G_Adcy6_AT2', candidates=[(112, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/112'], labels=['ADCY6']),\n",
       " Recommendation(id='G_Adcy2_AT1', candidates=[(108, np.float64(1.0)), (9426, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/108', 'https://www.ncbi.nlm.nih.gov/gene/9426'], labels=['ADCY2', 'CDY2A']),\n",
       " Recommendation(id='G_Adcy4_AT1', candidates=[(196883, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/196883'], labels=['ADCY4']),\n",
       " Recommendation(id='G_Adcy3_AT1', candidates=[(109, np.float64(1.0)), (114, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/109', 'https://www.ncbi.nlm.nih.gov/gene/114'], labels=['ADCY3', 'ADCY8']),\n",
       " Recommendation(id='G_Adcy8_AT1', candidates=[(114, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/114'], labels=['ADCY8']),\n",
       " Recommendation(id='G_Adk_AT1', candidates=[(132, np.float64(1.0)), (26007, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/132', 'https://www.ncbi.nlm.nih.gov/gene/26007'], labels=['ADK', 'TKFC']),\n",
       " Recommendation(id='G_Adk_AT2', candidates=[(132, np.float64(1.0)), (26007, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/132', 'https://www.ncbi.nlm.nih.gov/gene/26007'], labels=['ADK', 'TKFC']),\n",
       " Recommendation(id='G_AprT_AT1', candidates=[(54, np.float64(1.0)), (353, np.float64(1.0)), (959, np.float64(1.0)), (5471, np.float64(1.0)), (7186, np.float64(1.0)), (8295, np.float64(1.0)), (23424, np.float64(1.0)), (25823, np.float64(1.0)), (57410, np.float64(1.0)), (445347, np.float64(1.0)), (100505854, np.float64(1.0)), (105369301, np.float64(1.0)), (128193292, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/54', 'https://www.ncbi.nlm.nih.gov/gene/353', 'https://www.ncbi.nlm.nih.gov/gene/959', 'https://www.ncbi.nlm.nih.gov/gene/5471', 'https://www.ncbi.nlm.nih.gov/gene/7186', 'https://www.ncbi.nlm.nih.gov/gene/8295', 'https://www.ncbi.nlm.nih.gov/gene/23424', 'https://www.ncbi.nlm.nih.gov/gene/25823', 'https://www.ncbi.nlm.nih.gov/gene/57410', 'https://www.ncbi.nlm.nih.gov/gene/445347', 'https://www.ncbi.nlm.nih.gov/gene/100505854', 'https://www.ncbi.nlm.nih.gov/gene/105369301', 'https://www.ncbi.nlm.nih.gov/gene/128193292'], labels=['ACP5', 'APRT', 'CD40LG', 'PPAT', 'TRAF2', 'TRRAP', 'TDRD7', 'TPSG1', 'SCYL1', 'TARP', 'APTR', 'CLDN14-AS1', 'LNCTSI']),\n",
       " Recommendation(id='G_Agpat4_AT1', candidates=[(56895, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/56895'], labels=['AGPAT4']),\n",
       " Recommendation(id='G_Agpat3_AT2', candidates=[(56894, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/56894'], labels=['AGPAT3']),\n",
       " Recommendation(id='G_Agpat2_AT1', candidates=[(10555, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10555'], labels=['AGPAT2']),\n",
       " Recommendation(id='G_Agpat1_AT1', candidates=[(10554, np.float64(1.0)), (117289, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10554', 'https://www.ncbi.nlm.nih.gov/gene/117289'], labels=['AGPAT1', 'TAGAP']),\n",
       " Recommendation(id='G_Agpat5_AT1', candidates=[(55326, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/55326'], labels=['AGPAT5']),\n",
       " Recommendation(id='G_Agpat3_AT1', candidates=[(56894, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/56894'], labels=['AGPAT3']),\n",
       " Recommendation(id='G_Agpat6_AT1', candidates=[(137964, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/137964'], labels=['GPAT4']),\n",
       " Recommendation(id='G_Ahcy_AT1', candidates=[(191, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/191'], labels=['AHCY']),\n",
       " Recommendation(id='G_Slc38a2_AT1', candidates=[(54407, np.float64(1.0)), (64078, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/54407', 'https://www.ncbi.nlm.nih.gov/gene/64078'], labels=['SLC38A2', 'SLC28A3']),\n",
       " Recommendation(id='G_Slc38a1_AT1', candidates=[(6572, np.float64(1.0)), (81539, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6572', 'https://www.ncbi.nlm.nih.gov/gene/81539'], labels=['SLC18A3', 'SLC38A1']),\n",
       " Recommendation(id='G_Slc7a2_AT1', candidates=[(6542, np.float64(1.0)), (155184, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6542', 'https://www.ncbi.nlm.nih.gov/gene/155184'], labels=['SLC7A2', 'SLC2A7']),\n",
       " Recommendation(id='G_Slc6a14_AT1', candidates=[(9122, np.float64(1.0)), (11254, np.float64(1.0)), (55858, np.float64(1.0)), (113235, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/9122', 'https://www.ncbi.nlm.nih.gov/gene/11254', 'https://www.ncbi.nlm.nih.gov/gene/55858', 'https://www.ncbi.nlm.nih.gov/gene/113235'], labels=['SLC16A4', 'SLC6A14', 'TMEM165', 'SLC46A1']),\n",
       " Recommendation(id='G_Slc7a1_AT1', candidates=[(6512, np.float64(1.0)), (6541, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6512', 'https://www.ncbi.nlm.nih.gov/gene/6541'], labels=['SLC1A7', 'SLC7A1']),\n",
       " Recommendation(id='G_Slc38a4_AT1', candidates=[(55089, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/55089'], labels=['SLC38A4']),\n",
       " Recommendation(id='G_Slc7a3_AT1', candidates=[(84889, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/84889'], labels=['SLC7A3']),\n",
       " Recommendation(id='G_Aldh3a1_AT1', candidates=[(218, np.float64(1.0)), (220, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/218', 'https://www.ncbi.nlm.nih.gov/gene/220'], labels=['ALDH3A1', 'ALDH1A3']),\n",
       " Recommendation(id='G_Aldh9a1_AT1', candidates=[(223, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/223'], labels=['ALDH9A1']),\n",
       " Recommendation(id='G_Aldh1a2_AT2', candidates=[(8854, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8854'], labels=['ALDH1A2']),\n",
       " Recommendation(id='G_Aldh1a1_AT1', candidates=[(216, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/216'], labels=['ALDH1A1']),\n",
       " Recommendation(id='G_AmpD3_AT1', candidates=[(272, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/272'], labels=['AMPD3']),\n",
       " Recommendation(id='G_NudT2_AT2', candidates=[(318, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/318'], labels=['NUDT2']),\n",
       " Recommendation(id='G_NudT2_AT1', candidates=[(318, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/318'], labels=['NUDT2']),\n",
       " Recommendation(id='G_NudT2_AT3', candidates=[(318, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/318'], labels=['NUDT2']),\n",
       " Recommendation(id='G_Arg1_AT1', candidates=[(383, np.float64(1.0)), (1258, np.float64(1.0)), (5896, np.float64(1.0)), (51060, np.float64(1.0)), (54433, np.float64(1.0)), (64129, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/383', 'https://www.ncbi.nlm.nih.gov/gene/1258', 'https://www.ncbi.nlm.nih.gov/gene/5896', 'https://www.ncbi.nlm.nih.gov/gene/51060', 'https://www.ncbi.nlm.nih.gov/gene/54433', 'https://www.ncbi.nlm.nih.gov/gene/64129'], labels=['ARG1', 'CNGB1', 'RAG1', 'TXNDC12', 'GAR1', 'TINAGL1']),\n",
       " Recommendation(id='G_Slco1b3_AT1', candidates=[(3257, np.float64(1.0)), (28234, np.float64(1.0)), (388552, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/3257', 'https://www.ncbi.nlm.nih.gov/gene/28234', 'https://www.ncbi.nlm.nih.gov/gene/388552'], labels=['HPS1', 'SLCO1B3', 'BLOC1S3']),\n",
       " Recommendation(id='G_Slco1b1_AT1', candidates=[(2647, np.float64(1.0)), (10599, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2647', 'https://www.ncbi.nlm.nih.gov/gene/10599'], labels=['BLOC1S1', 'SLCO1B1']),\n",
       " Recommendation(id='G_Ugt1A4_AT1', candidates=[(54657, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/54657'], labels=['UGT1A4']),\n",
       " Recommendation(id='G_Ugt1A1_AT1', candidates=[(54575, np.float64(1.0)), (54576, np.float64(1.0)), (54577, np.float64(1.0)), (54578, np.float64(1.0)), (54657, np.float64(1.0)), (54658, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/54575', 'https://www.ncbi.nlm.nih.gov/gene/54576', 'https://www.ncbi.nlm.nih.gov/gene/54577', 'https://www.ncbi.nlm.nih.gov/gene/54578', 'https://www.ncbi.nlm.nih.gov/gene/54657', 'https://www.ncbi.nlm.nih.gov/gene/54658'], labels=['UGT1A10', 'UGT1A8', 'UGT1A7', 'UGT1A6', 'UGT1A4', 'UGT1A1']),\n",
       " Recommendation(id='G_Blvra_AT1', candidates=[(644, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/644'], labels=['BLVRA']),\n",
       " Recommendation(id='G_Cpt1a_AT1', candidates=[(83, np.float64(1.0)), (1374, np.float64(1.0)), (3964, np.float64(1.0)), (60502, np.float64(1.0)), (100750225, np.float64(1.0)), (106480335, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/83', 'https://www.ncbi.nlm.nih.gov/gene/1374', 'https://www.ncbi.nlm.nih.gov/gene/3964', 'https://www.ncbi.nlm.nih.gov/gene/60502', 'https://www.ncbi.nlm.nih.gov/gene/100750225', 'https://www.ncbi.nlm.nih.gov/gene/106480335'], labels=['ACTG1P10', 'CPT1A', 'LGALS8', 'CPAT1', 'PCAT1', 'ACTP1']),\n",
       " Recommendation(id='G_Cpt2_AT1', candidates=[(1376, np.float64(1.0)), (8643, np.float64(1.0)), (219931, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1376', 'https://www.ncbi.nlm.nih.gov/gene/8643', 'https://www.ncbi.nlm.nih.gov/gene/219931'], labels=['CPT2', 'PTCH2', 'TPCN2']),\n",
       " Recommendation(id='G_Atp2b1_AT1', candidates=[(482, np.float64(1.0)), (490, np.float64(1.0)), (100048923, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/482', 'https://www.ncbi.nlm.nih.gov/gene/490', 'https://www.ncbi.nlm.nih.gov/gene/100048923'], labels=['ATP1B2', 'ATP2B1', 'DDX39BP2']),\n",
       " Recommendation(id='G_Atp2b4_AT1', candidates=[(493, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/493'], labels=['ATP2B4']),\n",
       " Recommendation(id='G_Cat_AT1', candidates=[(12, np.float64(1.0)), (71, np.float64(1.0)), (72, np.float64(1.0)), (472, np.float64(1.0)), (847, np.float64(1.0)), (1384, np.float64(1.0)), (5130, np.float64(1.0)), (9457, np.float64(1.0)), (10249, np.float64(1.0)), (11332, np.float64(1.0)), (345651, np.float64(1.0)), (440915, np.float64(1.0)), (641455, np.float64(1.0)), (105375847, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/12', 'https://www.ncbi.nlm.nih.gov/gene/71', 'https://www.ncbi.nlm.nih.gov/gene/72', 'https://www.ncbi.nlm.nih.gov/gene/472', 'https://www.ncbi.nlm.nih.gov/gene/847', 'https://www.ncbi.nlm.nih.gov/gene/1384', 'https://www.ncbi.nlm.nih.gov/gene/5130', 'https://www.ncbi.nlm.nih.gov/gene/9457', 'https://www.ncbi.nlm.nih.gov/gene/10249', 'https://www.ncbi.nlm.nih.gov/gene/11332', 'https://www.ncbi.nlm.nih.gov/gene/345651', 'https://www.ncbi.nlm.nih.gov/gene/440915', 'https://www.ncbi.nlm.nih.gov/gene/641455', 'https://www.ncbi.nlm.nih.gov/gene/105375847'], labels=['SERPINA3', 'ACTG1', 'ACTG2', 'ATM', 'CAT', 'CRAT', 'PCYT1A', 'FHL5', 'GLYAT', 'ACOT7', 'ACTBL2', 'POTEKP', 'POTEM', 'CERNA3']),\n",
       " Recommendation(id='G_Cdipt_AT1', candidates=[(10423, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10423'], labels=['CDIPT']),\n",
       " Recommendation(id='G_Cdipt_AT2', candidates=[(10423, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10423'], labels=['CDIPT']),\n",
       " Recommendation(id='G_Cds1_AT1', candidates=[(1040, np.float64(1.0)), (1823, np.float64(1.0)), (1825, np.float64(1.0)), (1827, np.float64(1.0)), (6319, np.float64(1.0)), (6382, np.float64(1.0)), (7045, np.float64(1.0)), (11200, np.float64(1.0)), (28960, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1040', 'https://www.ncbi.nlm.nih.gov/gene/1823', 'https://www.ncbi.nlm.nih.gov/gene/1825', 'https://www.ncbi.nlm.nih.gov/gene/1827', 'https://www.ncbi.nlm.nih.gov/gene/6319', 'https://www.ncbi.nlm.nih.gov/gene/6382', 'https://www.ncbi.nlm.nih.gov/gene/7045', 'https://www.ncbi.nlm.nih.gov/gene/11200', 'https://www.ncbi.nlm.nih.gov/gene/28960'], labels=['CDS1', 'DSC1', 'DSC3', 'RCAN1', 'SCD', 'SDC1', 'TGFBI', 'CHEK2', 'DCPS']),\n",
       " Recommendation(id='G_Chpt1_AT1', candidates=[(5727, np.float64(1.0)), (56994, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5727', 'https://www.ncbi.nlm.nih.gov/gene/56994'], labels=['PTCH1', 'CHPT1']),\n",
       " Recommendation(id='G_Phospho1_AT1', candidates=[(162466, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/162466'], labels=['PHOSPHO1']),\n",
       " Recommendation(id='G_Pcyt1a_AT1', candidates=[(5130, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5130'], labels=['PCYT1A']),\n",
       " Recommendation(id='G_Pcyt1b_AT1', candidates=[(9468, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/9468'], labels=['PCYT1B']),\n",
       " Recommendation(id='G_Chkl_AT1', candidates=[(1120, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1120'], labels=['CHKB']),\n",
       " Recommendation(id='G_Chk_AT2', candidates=[(1022, np.float64(1.0)), (1119, np.float64(1.0)), (54822, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1022', 'https://www.ncbi.nlm.nih.gov/gene/1119', 'https://www.ncbi.nlm.nih.gov/gene/54822'], labels=['CDK7', 'CHKA', 'TRPM7']),\n",
       " Recommendation(id='G_Chk_AT1', candidates=[(1022, np.float64(1.0)), (1119, np.float64(1.0)), (54822, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1022', 'https://www.ncbi.nlm.nih.gov/gene/1119', 'https://www.ncbi.nlm.nih.gov/gene/54822'], labels=['CDK7', 'CHKA', 'TRPM7']),\n",
       " Recommendation(id='G_Chkl_AT2', candidates=[(1120, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1120'], labels=['CHKB']),\n",
       " Recommendation(id='G_Slc5a7_AT1', candidates=[(8140, np.float64(1.0)), (60482, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8140', 'https://www.ncbi.nlm.nih.gov/gene/60482'], labels=['SLC7A5', 'SLC5A7']),\n",
       " Recommendation(id='G_Cpox_AT1', candidates=[(1371, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1371'], labels=['CPOX']),\n",
       " Recommendation(id='G_Slc43a2_AT1', candidates=[(10568, np.float64(1.0)), (51458, np.float64(1.0)), (57419, np.float64(1.0)), (124935, np.float64(1.0)), (641842, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/10568', 'https://www.ncbi.nlm.nih.gov/gene/51458', 'https://www.ncbi.nlm.nih.gov/gene/57419', 'https://www.ncbi.nlm.nih.gov/gene/124935', 'https://www.ncbi.nlm.nih.gov/gene/641842'], labels=['SLC34A2', 'RHCG', 'SLC24A3', 'SLC43A2', 'SLC23A4P']),\n",
       " Recommendation(id='G_UmpK_AT1', candidates=[(51727, np.float64(1.0)), (253512, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/51727', 'https://www.ncbi.nlm.nih.gov/gene/253512'], labels=['CMPK1', 'SLC25A30']),\n",
       " Recommendation(id='G_Slc16a1_AT1', candidates=[(6538, np.float64(1.0)), (6566, np.float64(1.0)), (84975, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6538', 'https://www.ncbi.nlm.nih.gov/gene/6566', 'https://www.ncbi.nlm.nih.gov/gene/84975'], labels=['SLC6A11', 'SLC16A1', 'MFSD5']),\n",
       " Recommendation(id='G_Dgkg_AT1', candidates=[(1608, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1608'], labels=['DGKG']),\n",
       " Recommendation(id='G_Dgkq_AT1', candidates=[(1609, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1609'], labels=['DGKQ']),\n",
       " Recommendation(id='G_Dgkz_AT3', candidates=[(8525, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8525'], labels=['DGKZ']),\n",
       " Recommendation(id='G_Dgkh_AT1', candidates=[(160851, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/160851'], labels=['DGKH']),\n",
       " Recommendation(id='G_Dgkz_AT1', candidates=[(8525, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8525'], labels=['DGKZ']),\n",
       " Recommendation(id='G_Dgke_AT1', candidates=[(8526, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8526'], labels=['DGKE']),\n",
       " Recommendation(id='G_Dgka_AT1', candidates=[(1606, np.float64(1.0)), (1609, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1606', 'https://www.ncbi.nlm.nih.gov/gene/1609'], labels=['DGKA', 'DGKQ']),\n",
       " Recommendation(id='G_Dgkb_AT1', candidates=[(1607, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1607'], labels=['DGKB']),\n",
       " Recommendation(id='G_Dgkz_AT2', candidates=[(8525, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8525'], labels=['DGKZ']),\n",
       " Recommendation(id='G_Dgkb_AT2', candidates=[(1607, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1607'], labels=['DGKB']),\n",
       " Recommendation(id='G_Dgkh_AT2', candidates=[(160851, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/160851'], labels=['DGKH']),\n",
       " Recommendation(id='G_Dgki_AT1', candidates=[(9162, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/9162'], labels=['DGKI']),\n",
       " Recommendation(id='G_Dgkd_AT2', candidates=[(8527, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8527'], labels=['DGKD']),\n",
       " Recommendation(id='G_Dgkd_AT1', candidates=[(8527, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/8527'], labels=['DGKD']),\n",
       " Recommendation(id='G_Comt_AT2', candidates=[(1312, np.float64(1.0)), (9935, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1312', 'https://www.ncbi.nlm.nih.gov/gene/9935'], labels=['COMT', 'MAFB']),\n",
       " Recommendation(id='G_Comt_AT1', candidates=[(1312, np.float64(1.0)), (9935, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/1312', 'https://www.ncbi.nlm.nih.gov/gene/9935'], labels=['COMT', 'MAFB']),\n",
       " Recommendation(id='G_Slc22a3_AT1', candidates=[(6581, np.float64(1.0)), (9962, np.float64(1.0)), (9963, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6581', 'https://www.ncbi.nlm.nih.gov/gene/9962', 'https://www.ncbi.nlm.nih.gov/gene/9963'], labels=['SLC22A3', 'SLC23A2', 'SLC23A1']),\n",
       " Recommendation(id='G_Slc22a2_AT2', candidates=[(6582, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6582'], labels=['SLC22A2']),\n",
       " Recommendation(id='G_Slc22a2_AT1', candidates=[(6582, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/6582'], labels=['SLC22A2']),\n",
       " Recommendation(id='G_Bpgm_AT1', candidates=[(669, np.float64(1.0)), (5553, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/669', 'https://www.ncbi.nlm.nih.gov/gene/5553'], labels=['BPGM', 'PRG2']),\n",
       " Recommendation(id='G_Bpgm_AT2', candidates=[(669, np.float64(1.0)), (5553, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/669', 'https://www.ncbi.nlm.nih.gov/gene/5553'], labels=['BPGM', 'PRG2']),\n",
       " Recommendation(id='G_Eno1_AT1', candidates=[(2023, np.float64(1.0)), (4756, np.float64(1.0)), (10439, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2023', 'https://www.ncbi.nlm.nih.gov/gene/4756', 'https://www.ncbi.nlm.nih.gov/gene/10439'], labels=['ENO1', 'NEO1', 'OLFM1']),\n",
       " Recommendation(id='G_Acsl1_AT1', candidates=[(429, np.float64(1.0)), (2180, np.float64(1.0)), (100505994, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/429', 'https://www.ncbi.nlm.nih.gov/gene/2180', 'https://www.ncbi.nlm.nih.gov/gene/100505994'], labels=['ASCL1', 'ACSL1', 'LUCAT1']),\n",
       " Recommendation(id='G_EnpP1_AT1', candidates=[(5167, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5167'], labels=['ENPP1']),\n",
       " Recommendation(id='G_Aldoa_AT1', candidates=[(226, np.float64(1.0)), (112694756, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/226', 'https://www.ncbi.nlm.nih.gov/gene/112694756'], labels=['ALDOA', 'LOC112694756']),\n",
       " Recommendation(id='G_Aldoa_AT3', candidates=[(226, np.float64(1.0)), (112694756, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/226', 'https://www.ncbi.nlm.nih.gov/gene/112694756'], labels=['ALDOA', 'LOC112694756']),\n",
       " Recommendation(id='G_Aldoc_AT1', candidates=[(230, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/230'], labels=['ALDOC']),\n",
       " Recommendation(id='G_Aldoa_AT2', candidates=[(226, np.float64(1.0)), (112694756, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/226', 'https://www.ncbi.nlm.nih.gov/gene/112694756'], labels=['ALDOA', 'LOC112694756']),\n",
       " Recommendation(id='G_Fbp1_AT1', candidates=[(2203, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2203'], labels=['FBP1']),\n",
       " Recommendation(id='G_Pfkfb1_AT1', candidates=[(5207, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5207'], labels=['PFKFB1']),\n",
       " Recommendation(id='G_Pfkfb3_AT1', candidates=[(5209, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5209'], labels=['PFKFB3']),\n",
       " Recommendation(id='G_Fbp2_AT1', candidates=[(629, np.float64(1.0)), (8570, np.float64(1.0)), (8789, np.float64(1.0)), (64430, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/629', 'https://www.ncbi.nlm.nih.gov/gene/8570', 'https://www.ncbi.nlm.nih.gov/gene/8789', 'https://www.ncbi.nlm.nih.gov/gene/64430'], labels=['CFB', 'KHSRP', 'FBP2', 'PCNX4']),\n",
       " Recommendation(id='G_Pfkfb4_AT1', candidates=[(5210, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5210'], labels=['PFKFB4']),\n",
       " Recommendation(id='G_Pfkfb2_AT1', candidates=[(5208, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/5208'], labels=['PFKFB2']),\n",
       " Recommendation(id='G_Fech_AT1', candidates=[(2235, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2235'], labels=['FECH']),\n",
       " Recommendation(id='G_Fech_AT2', candidates=[(2235, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2235'], labels=['FECH']),\n",
       " Recommendation(id='G_FadS_AT2', candidates=[(57146, np.float64(1.0)), (80308, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/57146', 'https://www.ncbi.nlm.nih.gov/gene/80308'], labels=['LDAF1', 'FLAD1']),\n",
       " Recommendation(id='G_FadS_AT1', candidates=[(57146, np.float64(1.0)), (80308, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/57146', 'https://www.ncbi.nlm.nih.gov/gene/80308'], labels=['LDAF1', 'FLAD1']),\n",
       " Recommendation(id='G_Fh_AT1', candidates=[(2271, np.float64(1.0)), (3075, np.float64(1.0)), (3949, np.float64(1.0))], urls=['https://www.ncbi.nlm.nih.gov/gene/2271', 'https://www.ncbi.nlm.nih.gov/gene/3075', 'https://www.ncbi.nlm.nih.gov/gene/3949'], labels=['FH', 'CFH', 'LDLR'])]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "recoms = recom.getGeneListRecommendation(pred_ids=geneids2evaluate,tax='human')\n",
    "recoms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'G_Ak1_AT1': np.float64(1.0),\n",
       " 'G_Acsl1_AT1': np.float64(1.0),\n",
       " 'G_Gclc_AT1': np.float64(1.0),\n",
       " 'G_Pip5k2b_AT2': np.float64(1.0),\n",
       " 'G_Amd1_AT1': np.float64(1.0),\n",
       " 'G_Hk1_AT5': np.float64(1.0),\n",
       " 'G_Prdx2_AT3': np.float64(1.0),\n",
       " 'G_Comt_AT1': np.float64(1.0),\n",
       " 'G_Pde4B_AT1': np.float64(1.0),\n",
       " 'G_55500': np.float64(1.0),\n",
       " 'G_Gnpda1_AT1': np.float64(1.0),\n",
       " 'G_Pde2A_AT1': np.float64(1.0),\n",
       " 'G_Dgke_AT1': np.float64(1.0),\n",
       " 'G_Hk3_AT1': np.float64(1.0),\n",
       " 'G_Adcy5_AT1': np.float64(1.0),\n",
       " 'G_Atp2b1_AT1': np.float64(1.0),\n",
       " 'G_HprT1_AT1': np.float64(1.0),\n",
       " 'G_Gk_AT1': np.float64(1.0),\n",
       " 'G_NME6_AT1': np.float64(1.0),\n",
       " 'G_Atp2b4_AT1': np.float64(1.0),\n",
       " 'G_Ugp2_AT2': np.float64(1.0),\n",
       " 'G_Oplah_AT1': np.float64(1.0),\n",
       " 'G_Gpx4_AT1': np.float64(1.0),\n",
       " 'G_51084': np.float64(1.0),\n",
       " 'G_Adcy7_AT1': np.float64(1.0),\n",
       " 'G_Ldha_AT1': np.float64(1.0),\n",
       " 'G_Galt_AT2': np.float64(1.0),\n",
       " 'G_Akr7a2_AT1': np.float64(1.0),\n",
       " 'G_Pde8a_AT4': np.float64(1.0),\n",
       " 'G_Guk1_AT1': np.float64(1.0),\n",
       " 'G_55276': np.float64(1.0),\n",
       " 'G_Cpox_AT1': np.float64(1.0),\n",
       " 'G_Dgkz_AT3': np.float64(1.0),\n",
       " 'G_ImpDH2_AT1': np.float64(1.0),\n",
       " 'G_Itpkb_AT1': np.float64(1.0),\n",
       " 'G_Pip5k3_AT2': np.float64(1.0),\n",
       " 'G_Agpat3_AT2': np.float64(1.0),\n",
       " 'G_Ppa2_AT3': np.float64(1.0),\n",
       " 'G_Tkt_AT1': np.float64(1.0),\n",
       " 'G_Pde4A_AT1': np.float64(1.0),\n",
       " 'G_Pde3B_AT1': np.float64(1.0),\n",
       " 'G_Hmox1_AT1': np.float64(1.0),\n",
       " 'G_Me1_AT1': np.float64(1.0),\n",
       " 'G_Pfkm_AT1': np.float64(1.0),\n",
       " 'G_Slc38a4_AT1': np.float64(1.0),\n",
       " 'G_Slc38a1_AT1': np.float64(1.0),\n",
       " 'G_Alad_AT1': np.float64(1.0),\n",
       " 'G_Rpe_AT2': np.float64(1.0),\n",
       " 'G_Pde8a_AT5': np.float64(1.0),\n",
       " 'G_Slc43a1_AT1': np.float64(1.0),\n",
       " 'G_Nans_AT1': np.float64(1.0),\n",
       " 'G_Hk1_AT1': np.float64(1.0),\n",
       " 'G_Hmbs_AT1': np.float64(1.0),\n",
       " 'G_Gucy2F_AT1': np.float64(1.0),\n",
       " 'G_Dgki_AT1': np.float64(1.0),\n",
       " 'G_Pip5k2b_AT1': np.float64(1.0),\n",
       " 'G_NudT2_AT3': np.float64(1.0),\n",
       " 'G_PdxK_AT1': np.float64(1.0),\n",
       " 'G_Sord_AT1': np.float64(1.0),\n",
       " 'G_Aldoc_AT1': np.float64(1.0),\n",
       " 'G_Agpat2_AT1': np.float64(1.0),\n",
       " 'G_Dgkb_AT1': np.float64(1.0),\n",
       " 'G_Idh1_AT1': np.float64(1.0),\n",
       " 'G_Tpk1_AT1': np.float64(1.0),\n",
       " 'G_Ada_AT1': np.float64(1.0),\n",
       " 'G_NudT2_AT2': np.float64(1.0),\n",
       " 'G_Pip5k3_AT1': np.float64(1.0),\n",
       " 'G_Pcyt1a_AT1': np.float64(1.0),\n",
       " 'G_Akr1a1_AT2': np.float64(1.0),\n",
       " 'G_Hk2_AT1': np.float64(1.0),\n",
       " 'G_Pcyt2_AT1': np.float64(1.0),\n",
       " 'G_NME7_AT1': np.float64(1.0),\n",
       " 'G_Gk2_AT1': np.float64(1.0),\n",
       " 'G_Gale_AT2': np.float64(1.0),\n",
       " 'G_Dgka_AT1': np.float64(1.0),\n",
       " 'G_Hagh_AT1': np.float64(1.0),\n",
       " 'G_Gclm_AT1': np.float64(1.0),\n",
       " 'G_Agpat1_AT1': np.float64(1.0),\n",
       " 'G_Adcy4_AT1': np.float64(1.0),\n",
       " 'G_Slc22a3_AT1': np.float64(1.0),\n",
       " 'G_Nt5C3_AT1': np.float64(1.0),\n",
       " 'G_Mtap_AT1': np.float64(1.0),\n",
       " 'G_Comt_AT2': np.float64(1.0),\n",
       " 'G_Aldoa_AT2': np.float64(1.0),\n",
       " 'G_Pfkfb2_AT1': np.float64(1.0),\n",
       " 'G_9429': np.float64(1.0),\n",
       " 'G_Pde8a_AT1': np.float64(1.0),\n",
       " 'G_Pkm2_AT2': np.float64(1.0),\n",
       " 'G_Pde7B_AT1': np.float64(1.0),\n",
       " 'G_Aldh9a1_AT1': np.float64(1.0),\n",
       " 'G_Prdx1_AT1': np.float64(1.0),\n",
       " 'G_Fbp1_AT1': np.float64(1.0),\n",
       " 'G_Got1_AT1': np.float64(1.0),\n",
       " 'G_Slc6a9_AT1': np.float64(1.0),\n",
       " 'G_AmpD3_AT1': np.float64(1.0),\n",
       " 'G_NME7_AT2': np.float64(1.0),\n",
       " 'G_Gk_AT2': np.float64(1.0),\n",
       " 'G_Pip5k1b_AT1': np.float64(1.0),\n",
       " 'G_Nmnat3_AT1': np.float64(1.0),\n",
       " 'G_Dgkz_AT2': np.float64(1.0),\n",
       " 'G_Pde1B_AT1': np.float64(1.0),\n",
       " 'G_Pi4kii_AT1': np.float64(1.0),\n",
       " 'G_Gctg_AT1': np.float64(1.0),\n",
       " 'G_Np_AT1': np.float64(1.0),\n",
       " 'G_Fh_AT1': np.float64(1.0),\n",
       " 'G_FadS_AT2': np.float64(1.0),\n",
       " 'G_Prdx1_AT2': np.float64(1.0),\n",
       " 'G_Slc22a2_AT1': np.float64(1.0),\n",
       " 'G_Ldhb_AT1': np.float64(1.0),\n",
       " 'G_Phospho1_AT1': np.float64(1.0),\n",
       " 'G_Plcb3_AT1': np.float64(1.0),\n",
       " 'G_Nadk_AT1': np.float64(1.0),\n",
       " 'G_Adcy3_AT1': np.float64(1.0),\n",
       " 'G_Akr1a1_AT1': np.float64(1.0),\n",
       " 'G_Gmps_AT1': np.float64(1.0),\n",
       " 'G_Dgkq_AT1': np.float64(1.0),\n",
       " 'G_Slc16a1_AT1': np.float64(1.0),\n",
       " 'G_Pfkfb3_AT1': np.float64(1.0),\n",
       " 'G_Slco1b3_AT1': np.float64(1.0),\n",
       " 'G_Pik4ca_AT1': np.float64(1.0),\n",
       " 'G_Pip5k2a_AT1': np.float64(1.0),\n",
       " 'G_Acp1_AT2': np.float64(1.0),\n",
       " 'G_Chkl_AT2': np.float64(1.0),\n",
       " 'G_Gapdh_AT1': np.float64(1.0),\n",
       " 'G_Slco1b1_AT1': np.float64(1.0),\n",
       " 'G_Ahcy_AT1': np.float64(1.0),\n",
       " 'G_Fech_AT2': np.float64(1.0),\n",
       " 'G_Pde8a_AT2': np.float64(1.0),\n",
       " 'G_Agpat6_AT1': np.float64(1.0),\n",
       " 'G_55224_AT1': np.float64(1.0),\n",
       " 'G_Blvra_AT1': np.float64(1.0),\n",
       " 'G_Slc5a7_AT1': np.float64(1.0),\n",
       " 'G_Nt5C2_AT1': np.float64(1.0),\n",
       " 'G_Tat_AT1': np.float64(1.0),\n",
       " 'G_Nt5C_AT1': np.float64(1.0),\n",
       " 'G_Ugt1A4_AT1': np.float64(1.0),\n",
       " 'G_FadS_AT1': np.float64(1.0),\n",
       " 'G_Arg1_AT1': np.float64(1.0),\n",
       " 'G_29124': np.float64(1.0),\n",
       " 'G_Eno1_AT1': np.float64(1.0),\n",
       " 'G_ImpDH1_AT1': np.float64(1.0),\n",
       " 'G_EnpP1_AT1': np.float64(1.0),\n",
       " 'G_Pklr_AT1': np.float64(1.0),\n",
       " 'G_Adcy9_AT1': np.float64(1.0),\n",
       " 'G_Gsr_AT2': np.float64(1.0),\n",
       " 'G_ImpDH1_AT2': np.float64(1.0),\n",
       " 'G_AprT_AT1': np.float64(1.0),\n",
       " 'G_Pgls_AT1': np.float64(1.0),\n",
       " 'G_Pla2g6_AT1': np.float64(1.0),\n",
       " 'G_Cds1_AT1': np.float64(1.0),\n",
       " 'G_Pgm1_AT1': np.float64(1.0),\n",
       " 'G_Adcy2_AT1': np.float64(1.0),\n",
       " 'G_Galt_AT1': np.float64(1.0),\n",
       " 'G_Ppap2a_AT2': np.float64(1.0),\n",
       " 'G_Dgkz_AT1': np.float64(1.0),\n",
       " 'G_Slc38a2_AT1': np.float64(1.0),\n",
       " 'G_NME3_AT1': np.float64(1.0),\n",
       " 'G_Ppa2_AT1': np.float64(1.0),\n",
       " 'G_Pde1A_AT2': np.float64(1.0),\n",
       " 'G_Agpat5_AT1': np.float64(1.0),\n",
       " 'G_58478': np.float64(1.0),\n",
       " 'G_Pla2g2a_AT1': np.float64(1.0),\n",
       " 'G_GmpR_AT1': np.float64(1.0),\n",
       " 'G_Adk_AT1': np.float64(1.0),\n",
       " 'G_Pmm2_AT1': np.float64(1.0),\n",
       " 'G_Pde3B_AT2': np.float64(1.0),\n",
       " 'G_Hk1_AT3': np.float64(1.0),\n",
       " 'G_Cdipt_AT2': np.float64(1.0),\n",
       " 'G_Pde4D_AT1': np.float64(1.0),\n",
       " 'G_Slc22a2_AT2': np.float64(1.0),\n",
       " 'G_Renbp_AT1': np.float64(1.0),\n",
       " 'G_Acp1_AT3': np.float64(1.0),\n",
       " 'G_Dgkg_AT1': np.float64(1.0),\n",
       " 'G_Npr2_AT1': np.float64(1.0),\n",
       " 'G_Agpat3_AT1': np.float64(1.0),\n",
       " 'G_NudT2_AT1': np.float64(1.0),\n",
       " 'G_Pfkl_AT2': np.float64(1.0),\n",
       " 'G_Slc6a9_AT2': np.float64(1.0),\n",
       " 'G_Pklr_AT2': np.float64(1.0),\n",
       " 'G_Aox1_AT1': np.float64(1.0),\n",
       " 'G_Gss_AT1': np.float64(1.0),\n",
       " 'G_Hmox2_AT1': np.float64(1.0),\n",
       " 'G_Prps1l1_AT1': np.float64(1.0),\n",
       " 'G_Aldoa_AT1': np.float64(1.0),\n",
       " 'G_Dgkh_AT1': np.float64(1.0),\n",
       " 'G_UmpK_AT1': np.float64(1.0),\n",
       " 'G_Npr1_AT1': np.float64(1.0),\n",
       " 'G_Chkl_AT1': np.float64(1.0),\n",
       " 'G_Chk_AT1': np.float64(1.0),\n",
       " 'G_Pla2g4a_AT1': np.float64(1.0),\n",
       " 'G_Ppa2_AT4': np.float64(1.0),\n",
       " 'G_Slc7a1_AT1': np.float64(1.0),\n",
       " 'G_Isyna1_AT1': np.float64(1.0),\n",
       " 'G_Dgkd_AT1': np.float64(1.0),\n",
       " 'G_Xylb_AT1': np.float64(1.0),\n",
       " 'G_Lta4h_AT1': np.float64(1.0),\n",
       " 'G_Pde3A_AT1': np.float64(1.0),\n",
       " 'G_G6pd_AT1': np.float64(1.0),\n",
       " 'G_Ca1_AT1': np.float64(1.0),\n",
       " 'G_Cpt1a_AT1': np.float64(1.0),\n",
       " 'G_Prdx2_AT1': np.float64(1.0),\n",
       " 'G_Tpi1_AT1': np.float64(1.0),\n",
       " 'G_Npr2_AT2': np.float64(1.0),\n",
       " 'G_Adcy1_AT1': np.float64(1.0),\n",
       " 'G_Mat1a_AT1': np.float64(1.0),\n",
       " 'G_Pde8b_AT1': np.float64(1.0),\n",
       " 'G_Gucy2C_AT1': np.float64(1.0),\n",
       " 'G_Pip5k1c_AT1': np.float64(1.0),\n",
       " 'G_Pip5k3_AT3': np.float64(1.0),\n",
       " 'G_Gpi_AT1': np.float64(1.0),\n",
       " 'G_Rpia_AT1': np.float64(1.0),\n",
       " 'G_Dgkd_AT2': np.float64(1.0),\n",
       " 'G_Pip5k2c_AT1': np.float64(1.0),\n",
       " 'G_Dgkb_AT2': np.float64(1.0),\n",
       " 'G_Gucy2D_AT1': np.float64(1.0),\n",
       " 'G_Chk_AT2': np.float64(1.0),\n",
       " 'G_Aldh3a1_AT1': np.float64(1.0),\n",
       " 'G_Gpx1_AT1': np.float64(1.0),\n",
       " 'G_Adcy6_AT1': np.float64(1.0),\n",
       " 'G_Pfkl_AT1': np.float64(1.0),\n",
       " 'G_Uros_AT1': np.float64(1.0),\n",
       " 'G_Galk1_AT1': np.float64(1.0),\n",
       " 'G_PdxP_AT1': np.float64(1.0),\n",
       " 'G_Dcxr_AT1': np.float64(1.0),\n",
       " 'G_Prdx1_AT3': np.float64(1.0),\n",
       " 'G_Cpt2_AT1': np.float64(1.0),\n",
       " 'G_Agpat4_AT1': np.float64(1.0),\n",
       " 'G_Prps1l1_AT2': np.float64(1.0),\n",
       " 'G_10846': np.float64(1.0),\n",
       " 'G_Pde4C_AT1': np.float64(1.0),\n",
       " 'G_Itpkc_AT1': np.float64(1.0),\n",
       " 'G_Gpx1_AT2': np.float64(1.0),\n",
       " 'G_Pcyt1b_AT1': np.float64(1.0),\n",
       " 'G_Slc7a3_AT1': np.float64(1.0),\n",
       " 'G_Acp1_AT1': np.float64(1.0),\n",
       " 'G_10390_AT1': np.float64(1.0),\n",
       " 'G_Odc1_AT1': np.float64(1.0),\n",
       " 'G_Pik4cb_AT1': np.float64(1.0),\n",
       " 'G_Pgd_AT1': np.float64(1.0),\n",
       " 'G_Cat_AT1': np.float64(1.0),\n",
       " 'G_Pgk1_AT1': np.float64(1.0),\n",
       " 'G_Prps1_AT1': np.float64(1.0),\n",
       " 'G_Pkm2_AT1': np.float64(1.0),\n",
       " 'G_Fbp2_AT1': np.float64(1.0),\n",
       " 'G_Alad_AT2': np.float64(1.0),\n",
       " 'G_Aldh1a1_AT1': np.float64(1.0),\n",
       " 'G_Cdipt_AT1': np.float64(1.0),\n",
       " 'G_Adk_AT2': np.float64(1.0),\n",
       " 'G_Taldo1_AT1': np.float64(1.0),\n",
       " 'G_Galt_AT3': np.float64(1.0),\n",
       " 'G_Fech_AT1': np.float64(1.0),\n",
       " 'G_Gpam_AT1': np.float64(1.0),\n",
       " 'G_Aldh1a2_AT2': np.float64(1.0),\n",
       " 'G_Mpi_AT1': np.float64(1.0),\n",
       " 'G_Pla2g6_AT2': np.float64(1.0),\n",
       " 'G_Adcy6_AT2': np.float64(1.0),\n",
       " 'G_Nrk1_AT1': np.float64(1.0),\n",
       " 'G_Prdx2_AT2': np.float64(1.0),\n",
       " 'G_Urod_AT1': np.float64(1.0),\n",
       " 'G_Hk1_AT4': np.float64(1.0),\n",
       " 'G_Pde7A_AT2': np.float64(1.0),\n",
       " 'G_Itpka_AT1': np.float64(1.0),\n",
       " 'G_Dgkh_AT2': np.float64(1.0),\n",
       " 'G_Pkm2_AT3': np.float64(1.0),\n",
       " 'G_Synj1_AT2': np.float64(1.0),\n",
       " 'G_Rpe_AT1': np.float64(1.0),\n",
       " 'G_Slc7a2_AT1': np.float64(1.0),\n",
       " 'G_Synj1_AT1': np.float64(1.0),\n",
       " 'G_Slc6a14_AT1': np.float64(1.0),\n",
       " 'G_Umps_AT1': np.float64(1.0),\n",
       " 'G_Aldoa_AT3': np.float64(1.0),\n",
       " 'G_Pde11A_AT1': np.float64(1.0),\n",
       " 'G_Clc_AT1': np.float64(1.0),\n",
       " 'G_Slc43a2_AT1': np.float64(1.0),\n",
       " 'G_Hk1_AT2': np.float64(1.0),\n",
       " 'G_Ppap2c_AT1': np.float64(1.0),\n",
       " 'G_Gale_AT1': np.float64(1.0),\n",
       " 'G_Ppa2_AT2': np.float64(1.0),\n",
       " 'G_Pfkfb1_AT1': np.float64(1.0),\n",
       " 'G_Nadsyn1_AT1': np.float64(1.0),\n",
       " 'G_Srm_AT1': np.float64(1.0),\n",
       " 'G_Pip5k1a_AT1': np.float64(1.0),\n",
       " 'G_Pfkfb4_AT1': np.float64(1.0),\n",
       " 'G_Mdh1_AT1': np.float64(1.0),\n",
       " 'G_Ugdh_AT1': np.float64(1.0),\n",
       " 'G_Pde7A_AT1': np.float64(1.0),\n",
       " 'G_Pde8a_AT3': np.float64(1.0),\n",
       " 'G_Adcy8_AT1': np.float64(1.0),\n",
       " 'G_Comtd1_AT1': np.float64(1.0),\n",
       " 'G_Chpt1_AT1': np.float64(1.0),\n",
       " 'G_Ppap2a_AT1': np.float64(1.0),\n",
       " 'G_Nagk_AT1': np.float64(1.0),\n",
       " 'G_Impa1_AT1': np.float64(1.0),\n",
       " 'G_Pde1C_AT1': np.float64(1.0),\n",
       " 'G_Glul_AT1': np.float64(1.0),\n",
       " 'G_Inpp5d_AT1': np.float64(1.0),\n",
       " 'G_Ppap2b_AT1': np.float64(1.0),\n",
       " 'G_Bpgm_AT1': np.float64(1.0),\n",
       " 'G_55256': np.float64(1.0),\n",
       " 'G_Ca3_AT1': np.float64(1.0),\n",
       " 'G_Nanp_AT1': np.float64(1.0),\n",
       " 'G_Ca2_AT1': np.float64(1.0),\n",
       " 'G_Glo1_AT1': np.float64(1.0),\n",
       " 'G_PnpO_AT1': np.float64(1.0),\n",
       " 'G_Bpgm_AT2': np.float64(1.0),\n",
       " 'G_Pde1A_AT1': np.float64(1.0),\n",
       " 'G_Ugp2_AT1': np.float64(1.0),\n",
       " 'G_Sms_AT1': np.float64(1.0),\n",
       " 'G_Ugt1A1_AT1': np.float64(1.0),\n",
       " 'G_Akr1b1_AT1': np.float64(1.0),\n",
       " 'G_Rfk_AT1': np.float64(1.0)}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "recalls = recom.getGeneStatistics(tax='human',model_mean=False)['recall']\n",
    "recalls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Test on Gene Predictions using Bigg Models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We are at 0: Processing iCHOv1.xml\n",
      "We are at 1: Processing iYL1228.xml\n",
      "We are at 2: Processing iIS312.xml\n",
      "We are at 3: Processing iSDY_1059.xml\n",
      "We are at 4: Processing iEC1356_Bl21DE3.xml\n",
      "We are at 5: Processing iECP_1309.xml\n",
      "We are at 6: Processing iJN746.xml\n",
      "We are at 7: Processing iJR904.xml\n",
      "We are at 8: Processing iJB785.xml\n",
      "We are at 9: Processing iY75_1357.xml\n",
      "We are at 10: Processing iEcHS_1320.xml\n",
      "We are at 11: Processing iUTI89_1310.xml\n",
      "We are at 12: Processing iB21_1397.xml\n",
      "We are at 13: Processing iSbBS512_1146.xml\n",
      "We are at 14: Processing iNF517.xml\n",
      "We are at 15: Processing iNJ661.xml\n",
      "We are at 16: Processing iECIAI39_1322.xml\n",
      "Skipping iECIAI39_1322.xml (invalid model)\n",
      "We are at 17: Processing iECO103_1326.xml\n",
      "We are at 18: Processing iSSON_1240.xml\n",
      "We are at 19: Processing iCHOv1_DG44.xml\n",
      "We are at 20: Processing iAB_RBC_283.xml\n",
      "We are at 21: Processing iAM_Pk459.xml\n",
      "We are at 22: Processing iPC815.xml\n",
      "We are at 23: Processing iAPECO1_1312.xml\n",
      "We are at 24: Processing iMM1415.xml\n",
      "We are at 25: Processing iECDH1ME8569_1439.xml\n",
      "We are at 26: Processing iJO1366.xml\n",
      "We are at 27: Processing iEcE24377_1341.xml\n",
      "We are at 28: Processing iEC042_1314.xml\n",
      "We are at 29: Processing iRC1080.xml\n",
      "We are at 30: Processing iCN900.xml\n",
      "We are at 31: Processing iEC1364_W.xml\n",
      "We are at 32: Processing iJN678.xml\n",
      "We are at 33: Processing iEcSMS35_1347.xml\n",
      "We are at 34: Processing iSBO_1134.xml\n",
      "We are at 35: Processing STM_v1_0.xml\n",
      "We are at 36: Processing iAF1260.xml\n",
      "We are at 37: Processing iJN1463.xml\n",
      "We are at 38: Processing iS_1188.xml\n",
      "We are at 39: Processing iML1515.xml\n",
      "We are at 40: Processing iCN718.xml\n",
      "We are at 41: Processing iAF692.xml\n",
      "We are at 42: Processing iETEC_1333.xml\n",
      "We are at 43: Processing iECD_1391.xml\n",
      "We are at 44: Processing iHN637.xml\n",
      "We are at 45: Processing iG2583_1286.xml\n",
      "We are at 46: Processing iMM904.xml\n",
      "We are at 47: Processing Recon3D.xml\n",
      "We are at 48: Processing iAM_Pf480.xml\n",
      "We are at 49: Processing e_coli_core.xml\n",
      "We are at 50: Processing iAT_PLT_636.xml\n",
      "We are at 51: Processing iEC1344_C.xml\n",
      "We are at 52: Processing iWFL_1372.xml\n",
      "We are at 53: Processing iECABU_c1320.xml\n",
      "We are at 54: Processing iBWG_1329.xml\n",
      "We are at 55: Processing iIT341.xml\n",
      "We are at 56: Processing iEcolC_1368.xml\n",
      "We are at 57: Processing iECs_1301.xml\n",
      "We are at 58: Processing iZ_1308.xml\n",
      "We are at 59: Processing iSF_1195.xml\n",
      "We are at 60: Processing iND750.xml\n",
      "We are at 61: Processing iECB_1328.xml\n",
      "We are at 62: Processing iYO844.xml\n",
      "We are at 63: Processing iECIAI1_1343.xml\n",
      "We are at 64: Processing iLJ478.xml\n",
      "We are at 65: Processing iIS312_Amastigote.xml\n",
      "We are at 66: Processing RECON1.xml\n",
      "We are at 67: Processing iECBD_1354.xml\n",
      "We are at 68: Processing iEC1349_Crooks.xml\n",
      "We are at 69: Processing iSFxv_1172.xml\n",
      "We are at 70: Processing iECUMN_1333.xml\n",
      "We are at 71: Processing iECS88_1305.xml\n",
      "We are at 72: Processing iEC1368_DH5a.xml\n",
      "We are at 73: Processing iEcDH1_1363.xml\n",
      "We are at 74: Processing iAM_Pc455.xml\n",
      "We are at 75: Processing iUMN146_1321.xml\n",
      "We are at 76: Processing iECSP_1301.xml\n",
      "We are at 77: Processing iIS312_Epimastigote.xml\n",
      "We are at 78: Processing iLF82_1304.xml\n",
      "We are at 79: Processing iLB1027_lipid.xml\n",
      "We are at 80: Processing iECH74115_1262.xml\n",
      "We are at 81: Processing iE2348C_1286.xml\n",
      "We are at 82: Processing iECOK1_1307.xml\n",
      "We are at 83: Processing iUMNK88_1353.xml\n",
      "We are at 84: Processing iECSE_1348.xml\n",
      "We are at 85: Processing iEK1008.xml\n",
      "We are at 86: Processing iECSF_1327.xml\n",
      "We are at 87: Processing iYS854.xml\n",
      "We are at 88: Processing iSFV_1184.xml\n",
      "We are at 89: Processing iEC1372_W3110.xml\n",
      "We are at 90: Processing ic_1306.xml\n",
      "We are at 91: Processing iECED1_1282.xml\n",
      "We are at 92: Processing iECW_1372.xml\n",
      "We are at 93: Processing iSynCJ816.xml\n",
      "We are at 94: Processing iIS312_Trypomastigote.xml\n",
      "We are at 95: Processing iECDH10B_1368.xml\n",
      "We are at 96: Processing iYS1720.xml\n",
      "We are at 97: Processing iECNA114_1301.xml\n",
      "We are at 98: Processing iAM_Pb448.xml\n",
      "We are at 99: Processing iAM_Pv461.xml\n",
      "We are at 100: Processing iAF987.xml\n",
      "We are at 101: Processing iECO111_1330.xml\n",
      "We are at 102: Processing iSB619.xml\n",
      "We are at 103: Processing iNRG857_1313.xml\n",
      "We are at 104: Processing iECO26_1355.xml\n",
      "We are at 105: Processing iAF1260b.xml\n",
      "We are at 106: Processing iEC55989_1330.xml\n",
      "We are at 107: Processing iEKO11_1354.xml\n"
     ]
    }
   ],
   "source": [
    "def generate_model_species_statistics(model_dir, output_csv):\n",
    "    model_statistics_data = []\n",
    "\n",
    "    # All SBML files in the model directory\n",
    "    model_files = [f for f in os.listdir(model_dir) if f.endswith('.xml')]\n",
    "\n",
    "    for idx, model_file in enumerate(model_files):\n",
    "        print(f\"We are at {idx}: Processing {model_file}\")\n",
    "        one_fpath = os.path.join(model_dir, model_file)\n",
    "        reader = libsbml.SBMLReader()\n",
    "        document = reader.readSBML(one_fpath)\n",
    "        model = document.getModel()\n",
    "        if model is None:\n",
    "            print(f\"Skipping {model_file} (invalid model)\")\n",
    "            continue\n",
    "        fbc_plugin = model.getPlugin(\"fbc\")\n",
    "        if fbc_plugin is None:\n",
    "            print(f\"Skipping {model_file} (no fbc package)\")\n",
    "            continue\n",
    "        recom = recommender.Recommender(libsbml_fpath=one_fpath)\n",
    "\n",
    "        # Get species statistics\n",
    "        num_genes = fbc_plugin.getNumGeneProducts()\n",
    "        genes2evaluate = list(recom.genes.exist_annotation.keys())\n",
    "        num_annotated_genes = len(genes2evaluate)\n",
    "        annotated = num_annotated_genes > 0\n",
    "        \n",
    "        # Append the results for this species\n",
    "        model_statistics_data.append({\n",
    "            'model': model_file,\n",
    "            'annotated': annotated,\n",
    "            'num_genes': num_genes,\n",
    "            'num_annotated_genes': num_annotated_genes\n",
    "        })\n",
    "\n",
    "\n",
    "    # Create a DataFrame from the collected data\n",
    "    df = pd.DataFrame(model_statistics_data)\n",
    "\n",
    "    # Save the DataFrame to a CSV file\n",
    "    df.to_csv(output_csv, index=False)\n",
    "\n",
    "output_csv = output_dir + \"bigg_model_gene_statistics.csv\"\n",
    "generate_model_species_statistics(bigg_model_dir, output_csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 28 models with annotated genes\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model</th>\n",
       "      <th>annotated</th>\n",
       "      <th>num_genes</th>\n",
       "      <th>num_annotated_genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>iCHOv1.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1766</td>\n",
       "      <td>1743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>iJN746.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>746</td>\n",
       "      <td>729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>iJR904.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>904</td>\n",
       "      <td>892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>iNJ661.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>661</td>\n",
       "      <td>655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>iCHOv1_DG44.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1184</td>\n",
       "      <td>1168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>iAB_RBC_283.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>346</td>\n",
       "      <td>311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>iPC815.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>815</td>\n",
       "      <td>814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>iMM1415.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1375</td>\n",
       "      <td>1344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>iJO1366.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1367</td>\n",
       "      <td>1366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>iAF1260.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1261</td>\n",
       "      <td>1260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>iJN1463.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1462</td>\n",
       "      <td>1442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>iML1515.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1516</td>\n",
       "      <td>1515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>iMM904.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>905</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Recon3D.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>2248</td>\n",
       "      <td>2229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>iAM_Pf480.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>480</td>\n",
       "      <td>460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>e_coli_core.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>137</td>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>iAT_PLT_636.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>636</td>\n",
       "      <td>632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>iIT341.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>339</td>\n",
       "      <td>335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>iND750.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>750</td>\n",
       "      <td>746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>iLJ478.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>482</td>\n",
       "      <td>478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>RECON1.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1905</td>\n",
       "      <td>1880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>iAM_Pc455.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>455</td>\n",
       "      <td>452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>iLB1027_lipid.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1027</td>\n",
       "      <td>699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>iEK1008.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1008</td>\n",
       "      <td>996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>iYS1720.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1707</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>iAM_Pb448.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>448</td>\n",
       "      <td>441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>iAM_Pv461.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>461</td>\n",
       "      <td>450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>iAF1260b.xml</td>\n",
       "      <td>True</td>\n",
       "      <td>1261</td>\n",
       "      <td>1260</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 model  annotated  num_genes  num_annotated_genes\n",
       "0           iCHOv1.xml       True       1766                 1743\n",
       "6           iJN746.xml       True        746                  729\n",
       "7           iJR904.xml       True        904                  892\n",
       "15          iNJ661.xml       True        661                  655\n",
       "18     iCHOv1_DG44.xml       True       1184                 1168\n",
       "19     iAB_RBC_283.xml       True        346                  311\n",
       "21          iPC815.xml       True        815                  814\n",
       "23         iMM1415.xml       True       1375                 1344\n",
       "25         iJO1366.xml       True       1367                 1366\n",
       "35         iAF1260.xml       True       1261                 1260\n",
       "36         iJN1463.xml       True       1462                 1442\n",
       "38         iML1515.xml       True       1516                 1515\n",
       "45          iMM904.xml       True        905                  900\n",
       "46         Recon3D.xml       True       2248                 2229\n",
       "47       iAM_Pf480.xml       True        480                  460\n",
       "48     e_coli_core.xml       True        137                  136\n",
       "49     iAT_PLT_636.xml       True        636                  632\n",
       "54          iIT341.xml       True        339                  335\n",
       "59          iND750.xml       True        750                  746\n",
       "63          iLJ478.xml       True        482                  478\n",
       "65          RECON1.xml       True       1905                 1880\n",
       "73       iAM_Pc455.xml       True        455                  452\n",
       "78   iLB1027_lipid.xml       True       1027                  699\n",
       "84         iEK1008.xml       True       1008                  996\n",
       "95         iYS1720.xml       True       1707                    3\n",
       "97       iAM_Pb448.xml       True        448                  441\n",
       "98       iAM_Pv461.xml       True        461                  450\n",
       "104       iAF1260b.xml       True       1261                 1260"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bigg_model_gene_statistics = pd.read_csv(output_dir + \"bigg_model_gene_statistics.csv\")\n",
    "print(f\"There are {len(bigg_model_gene_statistics[bigg_model_gene_statistics['num_annotated_genes'] != 0])} models with annotated genes:\")\n",
    "bigg_model_gene_statistics[bigg_model_gene_statistics['num_annotated_genes'] != 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We are at 0: Processing iJR904.xml\n",
      "We are at 1: Processing iAB_RBC_283.xml\n",
      "We are at 2: Processing iJO1366.xml\n",
      "We are at 3: Processing iAF1260.xml\n",
      "We are at 4: Processing iML1515.xml\n",
      "We are at 5: Processing Recon3D.xml\n",
      "We are at 6: Processing e_coli_core.xml\n",
      "We are at 7: Processing iAT_PLT_636.xml\n",
      "We are at 8: Processing RECON1.xml\n",
      "We are at 9: Processing iAF1260b.xml\n"
     ]
    }
   ],
   "source": [
    "def generate_gene_accuracy(model_dir, output_csv):\n",
    "    model_files = [f for f in os.listdir(model_dir) if f.endswith('.xml')]\n",
    "    model2evaluate = [f for f in model_files if f in human_models + ecoli_mg1655_models]\n",
    "    model_names = []\n",
    "    taxonomy = []\n",
    "    gene_id = []\n",
    "    gene_name = []\n",
    "    len_names = []\n",
    "    predictions = []\n",
    "    num_cands = []\n",
    "    match_scores = []\n",
    "    recall = []\n",
    "    precision = []\n",
    "    is_correct_included = []\n",
    "    processing_time = []\n",
    "\n",
    "    for idx,model_file in enumerate(model2evaluate):\n",
    "        print(f\"We are at {idx}/{len(model2evaluate)}: Processing {model_file}\")\n",
    "        one_fpath = os.path.join(model_dir, model_file)\n",
    "        if model_file in human_models:\n",
    "            tax = 'human'\n",
    "        else:\n",
    "            tax = 'ecoli_mg1655'\n",
    "        try:\n",
    "            recom = recommender.Recommender(libsbml_fpath=one_fpath)\n",
    "            if recom is None or recom.genes is None:\n",
    "                print(f\"Skipping {model_file} (invalid recommender or genes)\")\n",
    "                continue\n",
    "            genes2evaluate = list(recom.genes.exist_annotation.keys())\n",
    "            # run algorithm if there is at least one species to evaluate\n",
    "            if genes2evaluate:\n",
    "                time_start = time.time()\n",
    "                recoms = recom.getGeneListRecommendation(tax=tax,pred_ids=genes2evaluate)\n",
    "                time_end = time.time()\n",
    "                recalls = recom.getGeneStatistics(tax=tax,model_mean=False)['recall']\n",
    "                precisions = recom.getGeneStatistics(tax=tax,model_mean=False)['precision']\n",
    "                for one_recom in recoms:\n",
    "                    model_names.append(model_file)\n",
    "                    taxonomy = tax\n",
    "                    one_spec = one_recom.id\n",
    "                    gene_id.append(one_spec)\n",
    "                    gene_name.append(recom.genes.getNameToUse(one_spec))\n",
    "                    len_names.append(len(gene_name[-1]))\n",
    "                    predictions.append(one_recom.candidates)\n",
    "                    num_cands.append(len(one_recom.candidates))\n",
    "                    match_scores.append(one_recom.candidates[0][1])\n",
    "                    recall.append(recalls[one_spec])\n",
    "                    precision.append(precisions[one_spec])\n",
    "                    processing_time.append(time_end - time_start)\n",
    "                    if recall[-1] > 0:\n",
    "                        is_correct_included.append(1)\n",
    "                    else:\n",
    "                        is_correct_included.append(0)\n",
    "        except Exception as e:\n",
    "            print(f\"Error processing {model_file}: {str(e)}\")\n",
    "            continue\n",
    "\n",
    "    # Create a DataFrame from the collected data\n",
    "    spec_df = pd.DataFrame({'model': model_names,\n",
    "                        'taxonomy': taxonomy,\n",
    "                        'gene_id': gene_id,\n",
    "                        'gene_name': gene_name,\n",
    "                        'name_length': len_names,\n",
    "                        'predictions': predictions,\n",
    "                        'num_candidates': num_cands,\n",
    "                        'match_score': match_scores,\n",
    "                        'recall': recall,\n",
    "                        'precision': precision,\n",
    "                        'accuracy': is_correct_included,\n",
    "                        'processing_time': processing_time})\n",
    "\n",
    "    # Save the DataFrame to a CSV file\n",
    "    spec_df.to_csv(output_csv, index=False)\n",
    "\n",
    "output_csv = output_dir + \"bigg_model_gene_accuracy_ecoli+human.csv\"\n",
    "generate_gene_accuracy(bigg_model_dir, output_csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "<Bigg Genes (All)>\n",
      "Number of models assessed: 10\n",
      "Average recall (per model) : 1.000\n",
      "Average precision (per model) : 0.710\n",
      "Average match score (per model): 1.000\n",
      "Average accuracy (per model): 1.000\n",
      "Ave. processing time (per model): 22.059\n",
      "Ave. processing time (per element, per model): 0.019\n",
      "Average number of predictions per row: 2.0081874401184567\n",
      "\n",
      "\n",
      "<Bigg Genes (Ecoli MG1655)>\n",
      "Number of models assessed: 6\n",
      "Average recall (per model) : 1.000\n",
      "Average precision (per model) : 0.721\n",
      "Average match score (per model): 1.000\n",
      "Average accuracy (per model): 1.000\n",
      "Ave. processing time (per model): 1.356\n",
      "Ave. processing time (per element, per model): 0.001\n",
      "Average number of predictions per row: 1.9088505210763727\n",
      "\n",
      "\n",
      "<Bigg Genes (human)>\n",
      "Number of models assessed: 4\n",
      "Average recall (per model) : 1.000\n",
      "Average precision (per model) : 0.694\n",
      "Average match score (per model): 1.000\n",
      "Average accuracy (per model): 1.000\n",
      "Ave. processing time (per model): 53.115\n",
      "Ave. processing time (per element, per model): 0.042\n",
      "Average number of predictions per row: 2.1346001583531273\n"
     ]
    }
   ],
   "source": [
    "def print_bigg_model_gene_accuracy(bigg_model_gene_accuracy):\n",
    "    print(\"Number of models assessed: %d\" % bigg_model_gene_accuracy['model'].nunique())\n",
    "    print(\"Average recall (per model) : %.003f\" % bigg_model_gene_accuracy.groupby('model')['recall'].mean().mean())\n",
    "    print(\"Average precision (per model) : %.003f\" % bigg_model_gene_accuracy.groupby('model')['precision'].mean().mean())\n",
    "    print(\"Average match score (per model): %.003f\" % bigg_model_gene_accuracy.groupby('model')['match_score'].mean().mean())\n",
    "    print(\"Average accuracy (per model): %.003f\" % bigg_model_gene_accuracy.groupby('model')['accuracy'].mean().mean())\n",
    "    # Fixed processing time calculations\n",
    "    mean_processing_time = bigg_model_gene_accuracy.groupby('model')['processing_time'].first().mean()\n",
    "    print(\"Ave. processing time (per model): %.003f\" % mean_processing_time)\n",
    "    # Calculate per element processing time\n",
    "    num_elements = bigg_model_gene_accuracy.groupby('model').size().mean()\n",
    "    mean_processing_time_per_element = mean_processing_time / num_elements\n",
    "    print(\"Ave. processing time (per element, per model): %.003f\" % mean_processing_time_per_element)\n",
    "    # Average number of predictions per species\n",
    "    def safe_eval_with_eval(x):\n",
    "        try:\n",
    "            return eval(x) if isinstance(x, str) else []\n",
    "        except Exception as e:\n",
    "            print(f\"Failed to parse: {x} with error: {e}\")\n",
    "            return []\n",
    "    bigg_model_gene_accuracy = bigg_model_gene_accuracy.copy()\n",
    "    bigg_model_gene_accuracy['parsed_predictions'] = bigg_model_gene_accuracy['predictions'].apply(safe_eval_with_eval)\n",
    "    bigg_model_gene_accuracy['num_predictions'] = bigg_model_gene_accuracy['parsed_predictions'].apply(len)\n",
    "    average_predictions = bigg_model_gene_accuracy['num_predictions'].mean()\n",
    "    print(f\"Average number of predictions per row: {average_predictions}\")\n",
    "    \n",
    "bigg_model_gene_accuracy = pd.read_csv(output_dir + \"bigg_model_gene_accuracy_ecoli+human.csv\")\n",
    "bigg_model_gene_accuracy_ecoli_mg1655 = bigg_model_gene_accuracy[bigg_model_gene_accuracy['model'].isin(ecoli_mg1655_models)]\n",
    "\n",
    "print(\"\\n\\n<Bigg Genes (All)>\")\n",
    "print_bigg_model_gene_accuracy(bigg_model_gene_accuracy)\n",
    "print(\"\\n\\n<Bigg Genes (Ecoli MG1655)>\")\n",
    "print_bigg_model_gene_accuracy(bigg_model_gene_accuracy_ecoli_mg1655)\n",
    "\n",
    "bigg_model_gene_accuracy_not_ecoli_mg1655 = bigg_model_gene_accuracy[~bigg_model_gene_accuracy['model'].isin(ecoli_mg1655_models)]\n",
    "print(\"\\n\\n<Bigg Genes (human)>\")\n",
    "print_bigg_model_gene_accuracy(bigg_model_gene_accuracy_not_ecoli_mg1655)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fraction of genes with perfect accuracy (accuracy=1.0) by model:\n",
      "model\n",
      "e_coli_core.xml      1.00000\n",
      "iAF1260.xml          1.00000\n",
      "iAF1260b.xml         1.00000\n",
      "iJR904.xml           1.00000\n",
      "iJO1366.xml          1.00000\n",
      "iML1515.xml          0.99802\n",
      "RECON1.xml           0.00000\n",
      "iJN746.xml           0.00000\n",
      "iPC815.xml           0.00000\n",
      "iNJ661.xml           0.00000\n",
      "iND750.xml           0.00000\n",
      "iMM904.xml           0.00000\n",
      "iMM1415.xml          0.00000\n",
      "iLJ478.xml           0.00000\n",
      "iLB1027_lipid.xml    0.00000\n",
      "iIT341.xml           0.00000\n",
      "iJN1463.xml          0.00000\n",
      "Recon3D.xml          0.00000\n",
      "iEK1008.xml          0.00000\n",
      "iCHOv1_DG44.xml      0.00000\n",
      "iCHOv1.xml           0.00000\n",
      "iAT_PLT_636.xml      0.00000\n",
      "iAM_Pv461.xml        0.00000\n",
      "iAM_Pf480.xml        0.00000\n",
      "iAM_Pc455.xml        0.00000\n",
      "iAM_Pb448.xml        0.00000\n",
      "iAB_RBC_283.xml      0.00000\n",
      "iYS1720.xml          0.00000\n",
      "dtype: float64\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/3q/mq6dnx4n3xz43nfx45y0by480000gn/T/ipykernel_21428/1777412428.py:4: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  .apply(lambda x: (x['accuracy'] == 1.0).mean())\n"
     ]
    }
   ],
   "source": [
    "# Calculate fraction of perfect accuracy (1.0) per model\n",
    "perfect_accuracy_by_model = (bigg_model_gene_accuracy\n",
    "                           .groupby('model')\n",
    "                           .apply(lambda x: (x['accuracy'] == 1.0).mean())\n",
    "                           .sort_values(ascending=False))\n",
    "\n",
    "print(\"Fraction of genes with perfect accuracy (accuracy=1.0) by model:\")\n",
    "print(perfect_accuracy_by_model)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene annotations in Bigg Models "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getOntologyFromString(string_annotation,\n",
    "                          bqbiol_qualifiers=['is', 'isVersionOf']):\n",
    "  \"\"\"\n",
    "  Parse string and return string annotation,\n",
    "  marked as <bqbiol:is> or <bqbiol:isVersionOf>.\n",
    "  If neither exists, return None.\n",
    "\n",
    "  Parameters\n",
    "  ----------\n",
    "  string_annotation: str\n",
    "  bqbiol_qualifiers: str-list\n",
    "      Use 'is' and 'isVersionOf' by default\n",
    "  \n",
    "\n",
    "  Returns\n",
    "  -------\n",
    "  list-tuple (ontology type, ontology id)\n",
    "       Return [] if none is provided\n",
    "  \"\"\"\n",
    "  combined_str = ''\n",
    "  for one_qualifier in bqbiol_qualifiers:\n",
    "    one_match = '<bqbiol:' + one_qualifier + \\\n",
    "                '[^a-zA-Z].*?<\\/bqbiol:' + \\\n",
    "                one_qualifier + '>'\n",
    "    one_matched = re.findall(one_match,\n",
    "                  string_annotation,\n",
    "                  flags=re.DOTALL)\n",
    "    if len(one_matched)>0:\n",
    "      matched_filt = [s.replace(\"      \", \"\") for s in one_matched]\n",
    "      one_str = '\\n'.join(matched_filt) \n",
    "    else:\n",
    "      one_str = ''\n",
    "    combined_str = combined_str + one_str\n",
    "  identifiers_list = re.findall('identifiers\\.org/.*/', combined_str)\n",
    "  result_identifiers = [(r.split('/')[1],r.split('/')[2].replace('\\\"', '')) \\\n",
    "                        for r in identifiers_list]\n",
    "  return result_identifiers\n",
    "\n",
    "def collect_metabolites_statistics(model_dir, output_csv, model_ontology_csv):\n",
    "    \"\"\"\n",
    "    Collect statistics of ontology types from SBML models in a folder.\n",
    "\n",
    "    Parameters:\n",
    "    - folder_path (str): Path to the folder containing SBML models.\n",
    "    - output_csv (str): Path to save overall ontology statistics as a CSV.\n",
    "    - model_ontology_csv (str): Path to save per-model ontology statistics as a CSV.\n",
    "    \"\"\"\n",
    "    overall_counter = Counter()\n",
    "    model_data = []\n",
    "\n",
    "    # Iterate over all SBML files in the folder\n",
    "    model_files = [f for f in os.listdir(model_dir) if f.endswith('.xml')]\n",
    "\n",
    "    for idx, file_name in enumerate(model_files):\n",
    "      print(f\"We are at {idx}: Processing {file_name}\")\n",
    "\n",
    "      # Read the SBML model\n",
    "      file_path = os.path.join(model_dir, file_name)\n",
    "      reader = libsbml.SBMLReader()\n",
    "      document = reader.readSBML(file_path)\n",
    "      model = document.getModel()\n",
    "\n",
    "      if model is None:\n",
    "          print(f\"Skipping {file_name} (invalid model)\")\n",
    "          continue\n",
    "\n",
    "      # Count ontology types for the current model\n",
    "      model_counter = Counter()\n",
    "\n",
    "      # Iterate over all species in the model\n",
    "      for species in model.getListOfSpecies():\n",
    "          annotation_string = species.getAnnotationString()\n",
    "          resources = getOntologyFromString(annotation_string)\n",
    "\n",
    "          # Count ontology types\n",
    "          for ontology_type, ontology_id in resources:\n",
    "              overall_counter[ontology_type] += 1\n",
    "              model_counter[ontology_type] += 1\n",
    "\n",
    "      # Store model-level data\n",
    "      for ontology_type, count in model_counter.items():\n",
    "          model_data.append({\n",
    "              'model': file_name,\n",
    "              'ontology_type': ontology_type,\n",
    "              'count': count\n",
    "          })\n",
    "\n",
    "    # Convert overall statistics to a DataFrame\n",
    "    overall_stats_df = pd.DataFrame.from_dict(overall_counter, orient='index', columns=['count'])\n",
    "    overall_stats_df.reset_index(inplace=True)\n",
    "    overall_stats_df.columns = ['ontology_type', 'count']\n",
    "\n",
    "    # Convert model-specific data to a DataFrame\n",
    "    model_stats_df = pd.DataFrame(model_data)\n",
    "\n",
    "    # Save the dataframes to CSV\n",
    "    overall_stats_df.to_csv(output_csv, index=False)\n",
    "    model_stats_df.to_csv(model_ontology_csv, index=False)\n",
    "    print(f\"Overall ontology statistics saved to {output_csv}\")\n",
    "    print(f\"Model-specific ontology statistics saved to {model_ontology_csv}\")\n",
    "\n",
    "\n",
    "def get_gene_annotation_string(sbml_file):\n",
    "    \"\"\"\n",
    "    Extract GeneProducts annotatino string from an SBML model.\n",
    "\n",
    "    Parameters:\n",
    "    - sbml_file (str): Path to the SBML file.\n",
    "\n",
    "    Returns:\n",
    "    - list: List of GeneProduct annotation strings.\n",
    "    \"\"\"\n",
    "    # Read the SBML model\n",
    "    reader = libsbml.SBMLReader()\n",
    "    document = reader.readSBML(sbml_file)\n",
    "    model = document.getModel()\n",
    "    if model is None:\n",
    "        print(f\"Failed to read the model: {sbml_file}\")\n",
    "        return []\n",
    "        \n",
    "    # Check if the model uses the FBC package\n",
    "    fbc_plugin = model.getPlugin(\"fbc\")\n",
    "    if not fbc_plugin:\n",
    "        print(f\"The model does not use the FBC package: {sbml_file}\")\n",
    "        return []\n",
    "\n",
    "    # Get GeneProducts from the FBC plugin\n",
    "    gene_annotations = []\n",
    "    for i in range(fbc_plugin.getNumGeneProducts()):\n",
    "        gene_product = fbc_plugin.getGeneProduct(i)\n",
    "        annotation_string = gene_product.getAnnotationString()\n",
    "        gene_annotations.append(annotation_string)\n",
    "    return gene_annotations\n",
    "\n",
    "def collect_gene_product_statistics(model_dir, overall_output_csv, model_ontology_csv):\n",
    "    \"\"\"\n",
    "    Collect statistics of GeneProducts from SBML models in a folder.\n",
    "\n",
    "    Parameters:\n",
    "    - folder_path (str): Path to the folder containing SBML models.\n",
    "    - overall_output_csv (str): Path to save overall GeneProduct statistics as a CSV.\n",
    "    - model_ontology_csv (str): Path to save per-model GeneProduct statistics as a CSV.\n",
    "    \"\"\"\n",
    "    overall_counter = Counter()\n",
    "    model_data = []\n",
    "    fbc_plugin = model.getPlugin(\"fbc\")\n",
    "\n",
    "    # Iterate over all SBML files in the folder\n",
    "    model_files = [f for f in os.listdir(model_dir) if f.endswith('.xml')]\n",
    "    for idx, file_name in enumerate(model_files):\n",
    "      print(f\"We are at {idx}: Processing {file_name}\")\n",
    "      file_path = os.path.join(folder_path, file_name)\n",
    "      gene_annotations = get_gene_annotation_string(file_path)\n",
    "\n",
    "      model_counter = Counter()\n",
    "      for gene_annotation in gene_annotations:\n",
    "        resources = getOntologyFromString(gene_annotation)\n",
    "\n",
    "        # Count ontology types\n",
    "        for ontology_type, ontology_id in resources:\n",
    "            overall_counter[ontology_type] += 1\n",
    "            model_counter[ontology_type] += 1\n",
    "\n",
    "      # Store model-level data\n",
    "      for ontology_type, count in model_counter.items():\n",
    "        model_data.append({\n",
    "            'model': file_name,\n",
    "            'ontology_type': ontology_type,\n",
    "            'count': count\n",
    "        })\n",
    "\n",
    "    # Convert overall statistics to a DataFrame\n",
    "    overall_stats_df = pd.DataFrame.from_dict(overall_counter, orient='index', columns=['count'])\n",
    "    overall_stats_df.reset_index(inplace=True)\n",
    "    overall_stats_df.columns = ['ontology_type', 'count']\n",
    "\n",
    "    # Convert model-specific data to a DataFrame\n",
    "    model_stats_df = pd.DataFrame(model_data)\n",
    "\n",
    "    # Save the dataframes to CSV\n",
    "    overall_stats_df.to_csv(output_csv, index=False)\n",
    "    model_stats_df.to_csv(model_ontology_csv, index=False)\n",
    "    print(f\"Overall GeneProduct statistics saved to {overall_output_csv}\")\n",
    "    print(f\"Model-specific GeneProduct statistics saved to {model_ontology_csv}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'M_glc__D_e': ['CHEBI:12965', 'CHEBI:20999', 'CHEBI:4167', 'CHEBI:17634'], 'M_gln__L_c': ['CHEBI:42943', 'CHEBI:42899', 'CHEBI:32679', 'CHEBI:32678', 'CHEBI:58359', 'CHEBI:28300', 'CHEBI:42812', 'CHEBI:13110', 'CHEBI:18050', 'CHEBI:32666', 'CHEBI:6227', 'CHEBI:32665', 'CHEBI:21308', 'CHEBI:42814', 'CHEBI:5432', 'CHEBI:24316'], 'M_gln__L_e': ['CHEBI:42943', 'CHEBI:42899', 'CHEBI:32679', 'CHEBI:32678', 'CHEBI:58359', 'CHEBI:28300', 'CHEBI:42812', 'CHEBI:13110', 'CHEBI:18050', 'CHEBI:32666', 'CHEBI:6227', 'CHEBI:32665', 'CHEBI:21308', 'CHEBI:42814', 'CHEBI:5432', 'CHEBI:24316'], 'M_glu__L_c': ['CHEBI:76051', 'CHEBI:21301', 'CHEBI:29985', 'CHEBI:42825', 'CHEBI:29987', 'CHEBI:18237', 'CHEBI:24314', 'CHEBI:16015', 'CHEBI:13107', 'CHEBI:5431', 'CHEBI:21304', 'CHEBI:6224', 'CHEBI:14321', 'CHEBI:29988'], 'M_glu__L_e': ['CHEBI:76051', 'CHEBI:21301', 'CHEBI:29985', 'CHEBI:42825', 'CHEBI:29987', 'CHEBI:18237', 'CHEBI:24314', 'CHEBI:16015', 'CHEBI:13107', 'CHEBI:5431', 'CHEBI:21304', 'CHEBI:6224', 'CHEBI:14321', 'CHEBI:29988'], 'M_glx_c': ['CHEBI:24420', 'CHEBI:36655', 'CHEBI:14368', 'CHEBI:24421', 'CHEBI:42767', 'CHEBI:16891', 'CHEBI:35977', 'CHEBI:5509'], 'M_h2o_c': ['CHEBI:13352', 'CHEBI:30490', 'CHEBI:43228', 'CHEBI:33813', 'CHEBI:44292', 'CHEBI:44641', 'CHEBI:27313', 'CHEBI:42043', 'CHEBI:44819', 'CHEBI:29356', 'CHEBI:5594', 'CHEBI:10743', 'CHEBI:15377', 'CHEBI:42857', 'CHEBI:13365', 'CHEBI:29412', 'CHEBI:16234', 'CHEBI:13419', 'CHEBI:5585', 'CHEBI:44701'], 'M_h2o_e': ['CHEBI:13352', 'CHEBI:30490', 'CHEBI:43228', 'CHEBI:33813', 'CHEBI:44292', 'CHEBI:44641', 'CHEBI:27313', 'CHEBI:42043', 'CHEBI:44819', 'CHEBI:29356', 'CHEBI:5594', 'CHEBI:10743', 'CHEBI:15377', 'CHEBI:42857', 'CHEBI:13365', 'CHEBI:29412', 'CHEBI:16234', 'CHEBI:13419', 'CHEBI:5585', 'CHEBI:44701'], 'M_h_c': ['CHEBI:5584', 'CHEBI:13357', 'CHEBI:15378', 'CHEBI:10744'], 'M_h_e': ['CHEBI:5584', 'CHEBI:13357', 'CHEBI:15378', 'CHEBI:10744'], 'M_icit_c': ['CHEBI:16087', 'CHEBI:36454', 'CHEBI:5998', 'CHEBI:24886', 'CHEBI:36453', 'CHEBI:14465', 'CHEBI:30887', 'CHEBI:24884'], 'M_lac__D_c': ['CHEBI:16004', 'CHEBI:341', 'CHEBI:18684', 'CHEBI:43701', 'CHEBI:11001', 'CHEBI:42111', 'CHEBI:42105'], 'M_lac__D_e': ['CHEBI:16004', 'CHEBI:341', 'CHEBI:18684', 'CHEBI:43701', 'CHEBI:11001', 'CHEBI:42111', 'CHEBI:42105'], 'M_mal__L_c': ['CHEBI:15589', 'CHEBI:30797', 'CHEBI:423', 'CHEBI:18784', 'CHEBI:18785', 'CHEBI:13140', 'CHEBI:11066'], 'M_mal__L_e': ['CHEBI:15589', 'CHEBI:30797', 'CHEBI:423', 'CHEBI:18784', 'CHEBI:18785', 'CHEBI:13140', 'CHEBI:11066'], 'M_nad_c': ['CHEBI:7422', 'CHEBI:44215', 'CHEBI:13394', 'CHEBI:21901', 'CHEBI:57540', 'CHEBI:15846', 'CHEBI:44281', 'CHEBI:44214', 'CHEBI:13393', 'CHEBI:29867'], 'M_nadh_c': ['CHEBI:13395', 'CHEBI:21902', 'CHEBI:7423', 'CHEBI:44216', 'CHEBI:57945', 'CHEBI:16908', 'CHEBI:13396'], 'M_nadp_c': ['CHEBI:29868', 'CHEBI:44405', 'CHEBI:25523', 'CHEBI:13397', 'CHEBI:18009', 'CHEBI:7424', 'CHEBI:13398', 'CHEBI:21903', 'CHEBI:58349', 'CHEBI:44409'], 'M_nadph_c': ['CHEBI:16474', 'CHEBI:7425', 'CHEBI:57783', 'CHEBI:21904', 'CHEBI:13399', 'CHEBI:13400', 'CHEBI:44286'], 'M_nh4_c': ['CHEBI:44284', 'CHEBI:135980', 'CHEBI:22533', 'CHEBI:13406', 'CHEBI:29337', 'CHEBI:29340', 'CHEBI:13771', 'CHEBI:16134', 'CHEBI:28938', 'CHEBI:22534', 'CHEBI:13405', 'CHEBI:49783', 'CHEBI:7434', 'CHEBI:7435', 'CHEBI:44269', 'CHEBI:44404', 'CHEBI:13407'], 'M_13dpg_c': ['CHEBI:57604', 'CHEBI:20189', 'CHEBI:11881', 'CHEBI:16001', 'CHEBI:1658'], 'M_nh4_e': ['CHEBI:44284', 'CHEBI:135980', 'CHEBI:22533', 'CHEBI:13406', 'CHEBI:29337', 'CHEBI:29340', 'CHEBI:13771', 'CHEBI:16134', 'CHEBI:28938', 'CHEBI:22534', 'CHEBI:13405', 'CHEBI:49783', 'CHEBI:7434', 'CHEBI:7435', 'CHEBI:44269', 'CHEBI:44404', 'CHEBI:13407'], 'M_o2_c': ['CHEBI:7860', 'CHEBI:26689', 'CHEBI:29097', 'CHEBI:15379', 'CHEBI:13416', 'CHEBI:27140', 'CHEBI:29793', 'CHEBI:30491', 'CHEBI:44742', 'CHEBI:23833', 'CHEBI:25366', 'CHEBI:10745'], 'M_2pg_c': ['CHEBI:39868', 'CHEBI:21028', 'CHEBI:58289', 'CHEBI:24344', 'CHEBI:12986', 'CHEBI:17835', 'CHEBI:11651', 'CHEBI:1267', 'CHEBI:88350'], 'M_o2_e': ['CHEBI:7860', 'CHEBI:26689', 'CHEBI:29097', 'CHEBI:15379', 'CHEBI:13416', 'CHEBI:27140', 'CHEBI:29793', 'CHEBI:30491', 'CHEBI:44742', 'CHEBI:23833', 'CHEBI:25366', 'CHEBI:10745'], 'M_3pg_c': ['CHEBI:11879', 'CHEBI:11880', 'CHEBI:17794', 'CHEBI:21029', 'CHEBI:1657', 'CHEBI:58272', 'CHEBI:12987'], 'M_oaa_c': ['CHEBI:25731', 'CHEBI:24959', 'CHEBI:12820', 'CHEBI:16452', 'CHEBI:30744', 'CHEBI:14703', 'CHEBI:24958', 'CHEBI:7812', 'CHEBI:25734', 'CHEBI:29049'], 'M_pep_c': ['CHEBI:18021', 'CHEBI:44894', 'CHEBI:26054', 'CHEBI:14812', 'CHEBI:26055', 'CHEBI:8147', 'CHEBI:58702', 'CHEBI:44897'], 'M_6pgc_c': ['CHEBI:33851', 'CHEBI:2231', 'CHEBI:16863', 'CHEBI:40282', 'CHEBI:12232', 'CHEBI:48928', 'CHEBI:58759'], 'M_pi_c': ['CHEBI:26078', 'CHEBI:18367', 'CHEBI:29139', 'CHEBI:39739', 'CHEBI:43470', 'CHEBI:35780', 'CHEBI:43474', 'CHEBI:26020', 'CHEBI:45024', 'CHEBI:39745', 'CHEBI:29137', 'CHEBI:14791', 'CHEBI:7793'], 'M_6pgl_c': ['CHEBI:12233', 'CHEBI:4160', 'CHEBI:57955', 'CHEBI:16938', 'CHEBI:20989', 'CHEBI:12958'], 'M_pi_e': ['CHEBI:26078', 'CHEBI:18367', 'CHEBI:29139', 'CHEBI:39739', 'CHEBI:43470', 'CHEBI:35780', 'CHEBI:43474', 'CHEBI:26020', 'CHEBI:45024', 'CHEBI:39745', 'CHEBI:29137', 'CHEBI:14791', 'CHEBI:7793'], 'M_ac_c': ['CHEBI:13704', 'CHEBI:22165', 'CHEBI:22169', 'CHEBI:40480', 'CHEBI:15366', 'CHEBI:2387', 'CHEBI:30089', 'CHEBI:40486'], 'M_pyr_c': ['CHEBI:26462', 'CHEBI:26466', 'CHEBI:8685', 'CHEBI:32816', 'CHEBI:45253', 'CHEBI:14987', 'CHEBI:15361'], 'M_pyr_e': ['CHEBI:26462', 'CHEBI:26466', 'CHEBI:8685', 'CHEBI:32816', 'CHEBI:45253', 'CHEBI:14987', 'CHEBI:15361'], 'M_q8_c': ['CHEBI:61683'], 'M_q8h2_c': ['CHEBI:61682'], 'M_r5p_c': ['CHEBI:18189', 'CHEBI:22413', 'CHEBI:12331', 'CHEBI:10270'], 'M_ru5p__D_c': ['CHEBI:40192', 'CHEBI:21088', 'CHEBI:17363', 'CHEBI:26572', 'CHEBI:13040', 'CHEBI:37455', 'CHEBI:58121', 'CHEBI:13018', 'CHEBI:4243'], 'M_ac_e': ['CHEBI:13704', 'CHEBI:22165', 'CHEBI:22169', 'CHEBI:40480', 'CHEBI:15366', 'CHEBI:2387', 'CHEBI:30089', 'CHEBI:40486'], 'M_acald_c': ['CHEBI:40533', 'CHEBI:13703', 'CHEBI:2383', 'CHEBI:22158', 'CHEBI:15343'], 'M_s7p_c': ['CHEBI:4244', 'CHEBI:15073', 'CHEBI:15721', 'CHEBI:26621', 'CHEBI:9083', 'CHEBI:15074', 'CHEBI:57483'], 'M_acald_e': ['CHEBI:40533', 'CHEBI:13703', 'CHEBI:2383', 'CHEBI:22158', 'CHEBI:15343'], 'M_accoa_c': ['CHEBI:57288', 'CHEBI:22192', 'CHEBI:15351', 'CHEBI:13712', 'CHEBI:2408', 'CHEBI:40470'], 'M_succ_c': ['CHEBI:22941', 'CHEBI:26803', 'CHEBI:15741', 'CHEBI:132287', 'CHEBI:45639', 'CHEBI:30779', 'CHEBI:15125', 'CHEBI:22943', 'CHEBI:26807', 'CHEBI:9304', 'CHEBI:90372', 'CHEBI:30031'], 'M_succ_e': ['CHEBI:22941', 'CHEBI:26803', 'CHEBI:15741', 'CHEBI:132287', 'CHEBI:45639', 'CHEBI:30779', 'CHEBI:15125', 'CHEBI:22943', 'CHEBI:26807', 'CHEBI:9304', 'CHEBI:90372', 'CHEBI:30031'], 'M_succoa_c': ['CHEBI:57292', 'CHEBI:10746', 'CHEBI:15127', 'CHEBI:15380', 'CHEBI:9310', 'CHEBI:45541', 'CHEBI:26811'], 'M_acon_C_c': ['CHEBI:23306', 'CHEBI:16383', 'CHEBI:12798', 'CHEBI:32805', 'CHEBI:23308', 'CHEBI:10482'], 'M_xu5p__D_c': ['CHEBI:13036', 'CHEBI:27354', 'CHEBI:4269', 'CHEBI:16332', 'CHEBI:57737', 'CHEBI:21121'], 'M_actp_c': ['CHEBI:13711', 'CHEBI:46262', 'CHEBI:15350', 'CHEBI:22191', 'CHEBI:2407'], 'M_adp_c': ['CHEBI:2342', 'CHEBI:16761', 'CHEBI:456216', 'CHEBI:40553', 'CHEBI:22244', 'CHEBI:87518', 'CHEBI:13222'], 'M_akg_c': ['CHEBI:40661', 'CHEBI:16810', 'CHEBI:11638', 'CHEBI:30916', 'CHEBI:19749', 'CHEBI:1253', 'CHEBI:30915', 'CHEBI:19748'], 'M_akg_e': ['CHEBI:40661', 'CHEBI:16810', 'CHEBI:11638', 'CHEBI:30916', 'CHEBI:19749', 'CHEBI:1253', 'CHEBI:30915', 'CHEBI:19748'], 'M_amp_c': ['CHEBI:13736', 'CHEBI:2356', 'CHEBI:13234', 'CHEBI:22242', 'CHEBI:40786', 'CHEBI:16027', 'CHEBI:40721', 'CHEBI:40510', 'CHEBI:456215', 'CHEBI:13235', 'CHEBI:22245', 'CHEBI:40726', 'CHEBI:47222', 'CHEBI:13740', 'CHEBI:40826', 'CHEBI:12056'], 'M_atp_c': ['CHEBI:15422', 'CHEBI:10789', 'CHEBI:10841', 'CHEBI:2359', 'CHEBI:22249', 'CHEBI:13236', 'CHEBI:237958', 'CHEBI:30616', 'CHEBI:57299', 'CHEBI:40938'], 'M_cit_c': ['CHEBI:35802', 'CHEBI:133748', 'CHEBI:35809', 'CHEBI:76049', 'CHEBI:41523', 'CHEBI:35810', 'CHEBI:30769', 'CHEBI:13999', 'CHEBI:23322', 'CHEBI:35808', 'CHEBI:42563', 'CHEBI:16947', 'CHEBI:132362', 'CHEBI:35804', 'CHEBI:23321', 'CHEBI:3727', 'CHEBI:35806'], 'M_co2_c': ['CHEBI:23011', 'CHEBI:3283', 'CHEBI:48829', 'CHEBI:16526', 'CHEBI:13283', 'CHEBI:13285', 'CHEBI:13284', 'CHEBI:13282'], 'M_co2_e': ['CHEBI:23011', 'CHEBI:3283', 'CHEBI:48829', 'CHEBI:16526', 'CHEBI:13283', 'CHEBI:13285', 'CHEBI:13284', 'CHEBI:13282'], 'M_coa_c': ['CHEBI:41631', 'CHEBI:41597', 'CHEBI:741566', 'CHEBI:23355', 'CHEBI:13295', 'CHEBI:13298', 'CHEBI:57287', 'CHEBI:15346', 'CHEBI:3771', 'CHEBI:13294'], 'M_dhap_c': ['CHEBI:14341', 'CHEBI:5454', 'CHEBI:16108', 'CHEBI:14342', 'CHEBI:24355', 'CHEBI:39571', 'CHEBI:57642'], 'M_e4p_c': ['CHEBI:4114', 'CHEBI:20927', 'CHEBI:12921', 'CHEBI:48153', 'CHEBI:16897', 'CHEBI:42349'], 'M_etoh_c': ['CHEBI:42377', 'CHEBI:23978', 'CHEBI:30880', 'CHEBI:16236', 'CHEBI:52092', 'CHEBI:30878', 'CHEBI:4879', 'CHEBI:44594', 'CHEBI:14222'], 'M_etoh_e': ['CHEBI:42377', 'CHEBI:23978', 'CHEBI:30880', 'CHEBI:16236', 'CHEBI:52092', 'CHEBI:30878', 'CHEBI:4879', 'CHEBI:44594', 'CHEBI:14222'], 'M_f6p_c': ['CHEBI:16084', 'CHEBI:12352', 'CHEBI:57634', 'CHEBI:22768', 'CHEBI:10375', 'CHEBI:42378'], 'M_fdp_c': ['CHEBI:37736', 'CHEBI:49299'], 'M_for_c': ['CHEBI:24081', 'CHEBI:30751', 'CHEBI:15740', 'CHEBI:5145', 'CHEBI:42460', 'CHEBI:14276', 'CHEBI:24082'], 'M_for_e': ['CHEBI:24081', 'CHEBI:30751', 'CHEBI:15740', 'CHEBI:5145', 'CHEBI:42460', 'CHEBI:14276', 'CHEBI:24082'], 'M_fru_e': ['CHEBI:47424', 'CHEBI:28757', 'CHEBI:24110', 'CHEBI:20929', 'CHEBI:4119', 'CHEBI:15824', 'CHEBI:5172', 'CHEBI:24104', 'CHEBI:12923', 'CHEBI:4118', 'CHEBI:37714', 'CHEBI:37721', 'CHEBI:48095'], 'M_fum_c': ['CHEBI:36180', 'CHEBI:37155', 'CHEBI:42511', 'CHEBI:5190', 'CHEBI:18012', 'CHEBI:42743', 'CHEBI:22956', 'CHEBI:22957', 'CHEBI:24124', 'CHEBI:14284', 'CHEBI:37154', 'CHEBI:29806', 'CHEBI:22958', 'CHEBI:24122'], 'M_fum_e': ['CHEBI:36180', 'CHEBI:37155', 'CHEBI:42511', 'CHEBI:5190', 'CHEBI:18012', 'CHEBI:42743', 'CHEBI:22956', 'CHEBI:22957', 'CHEBI:24124', 'CHEBI:14284', 'CHEBI:37154', 'CHEBI:29806', 'CHEBI:22958', 'CHEBI:24122'], 'M_g3p_c': ['CHEBI:5446', 'CHEBI:14333', 'CHEBI:12984', 'CHEBI:181', 'CHEBI:17138', 'CHEBI:21026', 'CHEBI:12983', 'CHEBI:18324', 'CHEBI:58027', 'CHEBI:59776', 'CHEBI:29052'], 'M_g6p_c': ['CHEBI:61548', 'CHEBI:14314', 'CHEBI:4170']}\n",
      "{'M_glc__D_e': ['CHEBI:12965', 'CHEBI:20999', 'CHEBI:4167', 'CHEBI:17634'], 'M_gln__L_c': ['CHEBI:42943', 'CHEBI:42899', 'CHEBI:32679', 'CHEBI:32678', 'CHEBI:58359', 'CHEBI:28300', 'CHEBI:42812', 'CHEBI:13110', 'CHEBI:18050', 'CHEBI:32666', 'CHEBI:6227', 'CHEBI:32665', 'CHEBI:21308', 'CHEBI:42814', 'CHEBI:5432', 'CHEBI:24316'], 'M_gln__L_e': ['CHEBI:42943', 'CHEBI:42899', 'CHEBI:32679', 'CHEBI:32678', 'CHEBI:58359', 'CHEBI:28300', 'CHEBI:42812', 'CHEBI:13110', 'CHEBI:18050', 'CHEBI:32666', 'CHEBI:6227', 'CHEBI:32665', 'CHEBI:21308', 'CHEBI:42814', 'CHEBI:5432', 'CHEBI:24316'], 'M_glu__L_c': ['CHEBI:76051', 'CHEBI:21301', 'CHEBI:29985', 'CHEBI:42825', 'CHEBI:29987', 'CHEBI:18237', 'CHEBI:24314', 'CHEBI:16015', 'CHEBI:13107', 'CHEBI:5431', 'CHEBI:21304', 'CHEBI:6224', 'CHEBI:14321', 'CHEBI:29988'], 'M_glu__L_e': ['CHEBI:76051', 'CHEBI:21301', 'CHEBI:29985', 'CHEBI:42825', 'CHEBI:29987', 'CHEBI:18237', 'CHEBI:24314', 'CHEBI:16015', 'CHEBI:13107', 'CHEBI:5431', 'CHEBI:21304', 'CHEBI:6224', 'CHEBI:14321', 'CHEBI:29988'], 'M_glx_c': ['CHEBI:24420', 'CHEBI:36655', 'CHEBI:14368', 'CHEBI:24421', 'CHEBI:42767', 'CHEBI:16891', 'CHEBI:35977', 'CHEBI:5509'], 'M_h2o_c': ['CHEBI:13352', 'CHEBI:30490', 'CHEBI:43228', 'CHEBI:33813', 'CHEBI:44292', 'CHEBI:44641', 'CHEBI:27313', 'CHEBI:42043', 'CHEBI:44819', 'CHEBI:29356', 'CHEBI:5594', 'CHEBI:10743', 'CHEBI:15377', 'CHEBI:42857', 'CHEBI:13365', 'CHEBI:29412', 'CHEBI:16234', 'CHEBI:13419', 'CHEBI:5585', 'CHEBI:44701'], 'M_h2o_e': ['CHEBI:13352', 'CHEBI:30490', 'CHEBI:43228', 'CHEBI:33813', 'CHEBI:44292', 'CHEBI:44641', 'CHEBI:27313', 'CHEBI:42043', 'CHEBI:44819', 'CHEBI:29356', 'CHEBI:5594', 'CHEBI:10743', 'CHEBI:15377', 'CHEBI:42857', 'CHEBI:13365', 'CHEBI:29412', 'CHEBI:16234', 'CHEBI:13419', 'CHEBI:5585', 'CHEBI:44701'], 'M_h_c': ['CHEBI:5584', 'CHEBI:13357', 'CHEBI:15378', 'CHEBI:10744'], 'M_h_e': ['CHEBI:5584', 'CHEBI:13357', 'CHEBI:15378', 'CHEBI:10744'], 'M_icit_c': ['CHEBI:16087', 'CHEBI:36454', 'CHEBI:5998', 'CHEBI:24886', 'CHEBI:36453', 'CHEBI:14465', 'CHEBI:30887', 'CHEBI:24884'], 'M_lac__D_c': ['CHEBI:16004', 'CHEBI:341', 'CHEBI:18684', 'CHEBI:43701', 'CHEBI:11001', 'CHEBI:42111', 'CHEBI:42105'], 'M_lac__D_e': ['CHEBI:16004', 'CHEBI:341', 'CHEBI:18684', 'CHEBI:43701', 'CHEBI:11001', 'CHEBI:42111', 'CHEBI:42105'], 'M_mal__L_c': ['CHEBI:15589', 'CHEBI:30797', 'CHEBI:423', 'CHEBI:18784', 'CHEBI:18785', 'CHEBI:13140', 'CHEBI:11066'], 'M_mal__L_e': ['CHEBI:15589', 'CHEBI:30797', 'CHEBI:423', 'CHEBI:18784', 'CHEBI:18785', 'CHEBI:13140', 'CHEBI:11066'], 'M_nad_c': ['CHEBI:7422', 'CHEBI:44215', 'CHEBI:13394', 'CHEBI:21901', 'CHEBI:57540', 'CHEBI:15846', 'CHEBI:44281', 'CHEBI:44214', 'CHEBI:13393', 'CHEBI:29867'], 'M_nadh_c': ['CHEBI:13395', 'CHEBI:21902', 'CHEBI:7423', 'CHEBI:44216', 'CHEBI:57945', 'CHEBI:16908', 'CHEBI:13396'], 'M_nadp_c': ['CHEBI:29868', 'CHEBI:44405', 'CHEBI:25523', 'CHEBI:13397', 'CHEBI:18009', 'CHEBI:7424', 'CHEBI:13398', 'CHEBI:21903', 'CHEBI:58349', 'CHEBI:44409'], 'M_nadph_c': ['CHEBI:16474', 'CHEBI:7425', 'CHEBI:57783', 'CHEBI:21904', 'CHEBI:13399', 'CHEBI:13400', 'CHEBI:44286'], 'M_nh4_c': ['CHEBI:44284', 'CHEBI:135980', 'CHEBI:22533', 'CHEBI:13406', 'CHEBI:29337', 'CHEBI:29340', 'CHEBI:13771', 'CHEBI:16134', 'CHEBI:28938', 'CHEBI:22534', 'CHEBI:13405', 'CHEBI:49783', 'CHEBI:7434', 'CHEBI:7435', 'CHEBI:44269', 'CHEBI:44404', 'CHEBI:13407'], 'M_13dpg_c': ['CHEBI:57604', 'CHEBI:20189', 'CHEBI:11881', 'CHEBI:16001', 'CHEBI:1658'], 'M_nh4_e': ['CHEBI:44284', 'CHEBI:135980', 'CHEBI:22533', 'CHEBI:13406', 'CHEBI:29337', 'CHEBI:29340', 'CHEBI:13771', 'CHEBI:16134', 'CHEBI:28938', 'CHEBI:22534', 'CHEBI:13405', 'CHEBI:49783', 'CHEBI:7434', 'CHEBI:7435', 'CHEBI:44269', 'CHEBI:44404', 'CHEBI:13407'], 'M_o2_c': ['CHEBI:7860', 'CHEBI:26689', 'CHEBI:29097', 'CHEBI:15379', 'CHEBI:13416', 'CHEBI:27140', 'CHEBI:29793', 'CHEBI:30491', 'CHEBI:44742', 'CHEBI:23833', 'CHEBI:25366', 'CHEBI:10745'], 'M_2pg_c': ['CHEBI:39868', 'CHEBI:21028', 'CHEBI:58289', 'CHEBI:24344', 'CHEBI:12986', 'CHEBI:17835', 'CHEBI:11651', 'CHEBI:1267', 'CHEBI:88350'], 'M_o2_e': ['CHEBI:7860', 'CHEBI:26689', 'CHEBI:29097', 'CHEBI:15379', 'CHEBI:13416', 'CHEBI:27140', 'CHEBI:29793', 'CHEBI:30491', 'CHEBI:44742', 'CHEBI:23833', 'CHEBI:25366', 'CHEBI:10745'], 'M_3pg_c': ['CHEBI:11879', 'CHEBI:11880', 'CHEBI:17794', 'CHEBI:21029', 'CHEBI:1657', 'CHEBI:58272', 'CHEBI:12987'], 'M_oaa_c': ['CHEBI:25731', 'CHEBI:24959', 'CHEBI:12820', 'CHEBI:16452', 'CHEBI:30744', 'CHEBI:14703', 'CHEBI:24958', 'CHEBI:7812', 'CHEBI:25734', 'CHEBI:29049'], 'M_pep_c': ['CHEBI:18021', 'CHEBI:44894', 'CHEBI:26054', 'CHEBI:14812', 'CHEBI:26055', 'CHEBI:8147', 'CHEBI:58702', 'CHEBI:44897'], 'M_6pgc_c': ['CHEBI:33851', 'CHEBI:2231', 'CHEBI:16863', 'CHEBI:40282', 'CHEBI:12232', 'CHEBI:48928', 'CHEBI:58759'], 'M_pi_c': ['CHEBI:26078', 'CHEBI:18367', 'CHEBI:29139', 'CHEBI:39739', 'CHEBI:43470', 'CHEBI:35780', 'CHEBI:43474', 'CHEBI:26020', 'CHEBI:45024', 'CHEBI:39745', 'CHEBI:29137', 'CHEBI:14791', 'CHEBI:7793'], 'M_6pgl_c': ['CHEBI:12233', 'CHEBI:4160', 'CHEBI:57955', 'CHEBI:16938', 'CHEBI:20989', 'CHEBI:12958'], 'M_pi_e': ['CHEBI:26078', 'CHEBI:18367', 'CHEBI:29139', 'CHEBI:39739', 'CHEBI:43470', 'CHEBI:35780', 'CHEBI:43474', 'CHEBI:26020', 'CHEBI:45024', 'CHEBI:39745', 'CHEBI:29137', 'CHEBI:14791', 'CHEBI:7793'], 'M_ac_c': ['CHEBI:13704', 'CHEBI:22165', 'CHEBI:22169', 'CHEBI:40480', 'CHEBI:15366', 'CHEBI:2387', 'CHEBI:30089', 'CHEBI:40486'], 'M_pyr_c': ['CHEBI:26462', 'CHEBI:26466', 'CHEBI:8685', 'CHEBI:32816', 'CHEBI:45253', 'CHEBI:14987', 'CHEBI:15361'], 'M_pyr_e': ['CHEBI:26462', 'CHEBI:26466', 'CHEBI:8685', 'CHEBI:32816', 'CHEBI:45253', 'CHEBI:14987', 'CHEBI:15361'], 'M_q8_c': ['CHEBI:61683'], 'M_q8h2_c': ['CHEBI:61682'], 'M_r5p_c': ['CHEBI:18189', 'CHEBI:22413', 'CHEBI:12331', 'CHEBI:10270'], 'M_ru5p__D_c': ['CHEBI:40192', 'CHEBI:21088', 'CHEBI:17363', 'CHEBI:26572', 'CHEBI:13040', 'CHEBI:37455', 'CHEBI:58121', 'CHEBI:13018', 'CHEBI:4243'], 'M_ac_e': ['CHEBI:13704', 'CHEBI:22165', 'CHEBI:22169', 'CHEBI:40480', 'CHEBI:15366', 'CHEBI:2387', 'CHEBI:30089', 'CHEBI:40486'], 'M_acald_c': ['CHEBI:40533', 'CHEBI:13703', 'CHEBI:2383', 'CHEBI:22158', 'CHEBI:15343'], 'M_s7p_c': ['CHEBI:4244', 'CHEBI:15073', 'CHEBI:15721', 'CHEBI:26621', 'CHEBI:9083', 'CHEBI:15074', 'CHEBI:57483'], 'M_acald_e': ['CHEBI:40533', 'CHEBI:13703', 'CHEBI:2383', 'CHEBI:22158', 'CHEBI:15343'], 'M_accoa_c': ['CHEBI:57288', 'CHEBI:22192', 'CHEBI:15351', 'CHEBI:13712', 'CHEBI:2408', 'CHEBI:40470'], 'M_succ_c': ['CHEBI:22941', 'CHEBI:26803', 'CHEBI:15741', 'CHEBI:132287', 'CHEBI:45639', 'CHEBI:30779', 'CHEBI:15125', 'CHEBI:22943', 'CHEBI:26807', 'CHEBI:9304', 'CHEBI:90372', 'CHEBI:30031'], 'M_succ_e': ['CHEBI:22941', 'CHEBI:26803', 'CHEBI:15741', 'CHEBI:132287', 'CHEBI:45639', 'CHEBI:30779', 'CHEBI:15125', 'CHEBI:22943', 'CHEBI:26807', 'CHEBI:9304', 'CHEBI:90372', 'CHEBI:30031'], 'M_succoa_c': ['CHEBI:57292', 'CHEBI:10746', 'CHEBI:15127', 'CHEBI:15380', 'CHEBI:9310', 'CHEBI:45541', 'CHEBI:26811'], 'M_acon_C_c': ['CHEBI:23306', 'CHEBI:16383', 'CHEBI:12798', 'CHEBI:32805', 'CHEBI:23308', 'CHEBI:10482'], 'M_xu5p__D_c': ['CHEBI:13036', 'CHEBI:27354', 'CHEBI:4269', 'CHEBI:16332', 'CHEBI:57737', 'CHEBI:21121'], 'M_actp_c': ['CHEBI:13711', 'CHEBI:46262', 'CHEBI:15350', 'CHEBI:22191', 'CHEBI:2407'], 'M_adp_c': ['CHEBI:2342', 'CHEBI:16761', 'CHEBI:456216', 'CHEBI:40553', 'CHEBI:22244', 'CHEBI:87518', 'CHEBI:13222'], 'M_akg_c': ['CHEBI:40661', 'CHEBI:16810', 'CHEBI:11638', 'CHEBI:30916', 'CHEBI:19749', 'CHEBI:1253', 'CHEBI:30915', 'CHEBI:19748'], 'M_akg_e': ['CHEBI:40661', 'CHEBI:16810', 'CHEBI:11638', 'CHEBI:30916', 'CHEBI:19749', 'CHEBI:1253', 'CHEBI:30915', 'CHEBI:19748'], 'M_amp_c': ['CHEBI:13736', 'CHEBI:2356', 'CHEBI:13234', 'CHEBI:22242', 'CHEBI:40786', 'CHEBI:16027', 'CHEBI:40721', 'CHEBI:40510', 'CHEBI:456215', 'CHEBI:13235', 'CHEBI:22245', 'CHEBI:40726', 'CHEBI:47222', 'CHEBI:13740', 'CHEBI:40826', 'CHEBI:12056'], 'M_atp_c': ['CHEBI:15422', 'CHEBI:10789', 'CHEBI:10841', 'CHEBI:2359', 'CHEBI:22249', 'CHEBI:13236', 'CHEBI:237958', 'CHEBI:30616', 'CHEBI:57299', 'CHEBI:40938'], 'M_cit_c': ['CHEBI:35802', 'CHEBI:133748', 'CHEBI:35809', 'CHEBI:76049', 'CHEBI:41523', 'CHEBI:35810', 'CHEBI:30769', 'CHEBI:13999', 'CHEBI:23322', 'CHEBI:35808', 'CHEBI:42563', 'CHEBI:16947', 'CHEBI:132362', 'CHEBI:35804', 'CHEBI:23321', 'CHEBI:3727', 'CHEBI:35806'], 'M_co2_c': ['CHEBI:23011', 'CHEBI:3283', 'CHEBI:48829', 'CHEBI:16526', 'CHEBI:13283', 'CHEBI:13285', 'CHEBI:13284', 'CHEBI:13282'], 'M_co2_e': ['CHEBI:23011', 'CHEBI:3283', 'CHEBI:48829', 'CHEBI:16526', 'CHEBI:13283', 'CHEBI:13285', 'CHEBI:13284', 'CHEBI:13282'], 'M_coa_c': ['CHEBI:41631', 'CHEBI:41597', 'CHEBI:741566', 'CHEBI:23355', 'CHEBI:13295', 'CHEBI:13298', 'CHEBI:57287', 'CHEBI:15346', 'CHEBI:3771', 'CHEBI:13294'], 'M_dhap_c': ['CHEBI:14341', 'CHEBI:5454', 'CHEBI:16108', 'CHEBI:14342', 'CHEBI:24355', 'CHEBI:39571', 'CHEBI:57642'], 'M_e4p_c': ['CHEBI:4114', 'CHEBI:20927', 'CHEBI:12921', 'CHEBI:48153', 'CHEBI:16897', 'CHEBI:42349'], 'M_etoh_c': ['CHEBI:42377', 'CHEBI:23978', 'CHEBI:30880', 'CHEBI:16236', 'CHEBI:52092', 'CHEBI:30878', 'CHEBI:4879', 'CHEBI:44594', 'CHEBI:14222'], 'M_etoh_e': ['CHEBI:42377', 'CHEBI:23978', 'CHEBI:30880', 'CHEBI:16236', 'CHEBI:52092', 'CHEBI:30878', 'CHEBI:4879', 'CHEBI:44594', 'CHEBI:14222'], 'M_f6p_c': ['CHEBI:16084', 'CHEBI:12352', 'CHEBI:57634', 'CHEBI:22768', 'CHEBI:10375', 'CHEBI:42378'], 'M_fdp_c': ['CHEBI:37736', 'CHEBI:49299'], 'M_for_c': ['CHEBI:24081', 'CHEBI:30751', 'CHEBI:15740', 'CHEBI:5145', 'CHEBI:42460', 'CHEBI:14276', 'CHEBI:24082'], 'M_for_e': ['CHEBI:24081', 'CHEBI:30751', 'CHEBI:15740', 'CHEBI:5145', 'CHEBI:42460', 'CHEBI:14276', 'CHEBI:24082'], 'M_fru_e': ['CHEBI:47424', 'CHEBI:28757', 'CHEBI:24110', 'CHEBI:20929', 'CHEBI:4119', 'CHEBI:15824', 'CHEBI:5172', 'CHEBI:24104', 'CHEBI:12923', 'CHEBI:4118', 'CHEBI:37714', 'CHEBI:37721', 'CHEBI:48095'], 'M_fum_c': ['CHEBI:36180', 'CHEBI:37155', 'CHEBI:42511', 'CHEBI:5190', 'CHEBI:18012', 'CHEBI:42743', 'CHEBI:22956', 'CHEBI:22957', 'CHEBI:24124', 'CHEBI:14284', 'CHEBI:37154', 'CHEBI:29806', 'CHEBI:22958', 'CHEBI:24122'], 'M_fum_e': ['CHEBI:36180', 'CHEBI:37155', 'CHEBI:42511', 'CHEBI:5190', 'CHEBI:18012', 'CHEBI:42743', 'CHEBI:22956', 'CHEBI:22957', 'CHEBI:24124', 'CHEBI:14284', 'CHEBI:37154', 'CHEBI:29806', 'CHEBI:22958', 'CHEBI:24122'], 'M_g3p_c': ['CHEBI:5446', 'CHEBI:14333', 'CHEBI:12984', 'CHEBI:181', 'CHEBI:17138', 'CHEBI:21026', 'CHEBI:12983', 'CHEBI:18324', 'CHEBI:58027', 'CHEBI:59776', 'CHEBI:29052'], 'M_g6p_c': ['CHEBI:61548', 'CHEBI:14314', 'CHEBI:4170']}\n"
     ]
    }
   ],
   "source": [
    "libsbml_fpath = \"/Users/luna/Desktop/CRBM/AMAS_proj/Models/BiggModels/e_coli_core.xml\"\n",
    "reader = libsbml.SBMLReader()\n",
    "document = reader.readSBML(libsbml_fpath)\n",
    "model = document.getModel()\n",
    "names = {val.getId():val.name for val in model.getListOfSpecies()}\n",
    "qualifier=cn.CHEBI\n",
    "exist_raw = {val.getId():tools.getQualifierFromString(val.getAnnotationString(), [cn.CHEBI, cn.OBO_CHEBI]) \\\n",
    "            for val in model.getListOfSpecies()}\n",
    "print(exist_raw)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We are at 0\n",
      "We are at 1\n",
      "We are at 2\n",
      "We are at 3\n",
      "We are at 4\n",
      "We are at 5\n",
      "We are at 6\n",
      "We are at 7\n",
      "We are at 8\n",
      "We are at 9\n",
      "We are at 10\n",
      "We are at 11\n",
      "We are at 12\n",
      "We are at 13\n",
      "We are at 14\n",
      "We are at 15\n",
      "We are at 16\n",
      "We are at 17\n",
      "We are at 18\n",
      "We are at 19\n",
      "We are at 20\n",
      "We are at 21\n",
      "We are at 22\n",
      "We are at 23\n",
      "We are at 24\n",
      "We are at 25\n",
      "We are at 26\n",
      "We are at 27\n",
      "We are at 28\n",
      "We are at 29\n",
      "We are at 30\n",
      "We are at 31\n",
      "We are at 32\n",
      "We are at 33\n",
      "We are at 34\n",
      "We are at 35\n",
      "We are at 36\n",
      "We are at 37\n",
      "We are at 38\n",
      "We are at 39\n",
      "We are at 40\n",
      "We are at 41\n",
      "We are at 42\n",
      "We are at 43\n",
      "We are at 44\n",
      "We are at 45\n",
      "We are at 46\n",
      "We are at 47\n",
      "We are at 48\n",
      "We are at 49\n",
      "We are at 50\n",
      "We are at 51\n",
      "We are at 52\n",
      "We are at 53\n",
      "We are at 54\n",
      "We are at 55\n",
      "We are at 56\n",
      "We are at 57\n",
      "We are at 58\n",
      "We are at 59\n",
      "We are at 60\n",
      "We are at 61\n",
      "We are at 62\n",
      "We are at 63\n",
      "We are at 64\n",
      "We are at 65\n",
      "We are at 66\n",
      "We are at 67\n",
      "We are at 68\n",
      "We are at 69\n",
      "We are at 70\n",
      "We are at 71\n",
      "We are at 72\n",
      "We are at 73\n",
      "We are at 74\n",
      "We are at 75\n",
      "We are at 76\n",
      "We are at 77\n",
      "We are at 78\n",
      "We are at 79\n",
      "We are at 80\n",
      "We are at 81\n",
      "We are at 82\n",
      "We are at 83\n",
      "We are at 84\n",
      "We are at 85\n",
      "We are at 86\n",
      "We are at 87\n",
      "We are at 88\n",
      "We are at 89\n",
      "We are at 90\n",
      "We are at 91\n",
      "We are at 92\n",
      "We are at 93\n",
      "We are at 94\n",
      "We are at 95\n",
      "We are at 96\n",
      "We are at 97\n",
      "We are at 98\n",
      "We are at 99\n",
      "We are at 100\n",
      "We are at 101\n",
      "We are at 102\n",
      "We are at 103\n",
      "We are at 104\n",
      "We are at 105\n",
      "We are at 106\n",
      "We are at 107\n",
      "Overall ontology statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_ontology_statistics.csv\n",
      "Model-specific ontology statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_model_ontology_statistics.csv\n"
     ]
    }
   ],
   "source": [
    "output_csv = \"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_ontology_statistics.csv\"\n",
    "model_ontology_csv = \"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_model_ontology_statistics.csv\"\n",
    "collect_ontology_statistics(bigg_model_dir, output_csv, model_ontology_csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ontology_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chebi</td>\n",
       "      <td>881182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>bigg.metabolite</td>\n",
       "      <td>180802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>seed.compound</td>\n",
       "      <td>177255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>reactome.compound</td>\n",
       "      <td>176773</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>metanetx.chemical</td>\n",
       "      <td>173772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>biocyc</td>\n",
       "      <td>151278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>hmdb</td>\n",
       "      <td>142708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>kegg.compound</td>\n",
       "      <td>140296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>inchi_key</td>\n",
       "      <td>134440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>sabiork</td>\n",
       "      <td>124143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>envipath</td>\n",
       "      <td>49436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>kegg.drug</td>\n",
       "      <td>30210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>lipidmaps</td>\n",
       "      <td>25793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>slm</td>\n",
       "      <td>10172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>kegg.glycan</td>\n",
       "      <td>6042</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        ontology_type   count\n",
       "3               chebi  881182\n",
       "0     bigg.metabolite  180802\n",
       "10      seed.compound  177255\n",
       "11  reactome.compound  176773\n",
       "1   metanetx.chemical  173772\n",
       "2              biocyc  151278\n",
       "5                hmdb  142708\n",
       "7       kegg.compound  140296\n",
       "6           inchi_key  134440\n",
       "9             sabiork  124143\n",
       "4            envipath   49436\n",
       "13          kegg.drug   30210\n",
       "8           lipidmaps   25793\n",
       "12                slm   10172\n",
       "14        kegg.glycan    6042"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bigg_metabolites_ontology_statistics = pd.read_csv(\"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_metabolites_ontology_statistics.csv\")\n",
    "bigg_metabolites_ontology_statistics.sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We are at 0: Processing iCHOv1.xml\n",
      "We are at 1: Processing iYL1228.xml\n",
      "We are at 2: Processing iIS312.xml\n",
      "We are at 3: Processing iSDY_1059.xml\n",
      "We are at 4: Processing iEC1356_Bl21DE3.xml\n",
      "We are at 5: Processing iECP_1309.xml\n",
      "We are at 6: Processing iJN746.xml\n",
      "We are at 7: Processing iJR904.xml\n",
      "We are at 8: Processing iJB785.xml\n",
      "We are at 9: Processing iY75_1357.xml\n",
      "We are at 10: Processing iEcHS_1320.xml\n",
      "We are at 11: Processing iUTI89_1310.xml\n",
      "We are at 12: Processing iB21_1397.xml\n",
      "We are at 13: Processing iSbBS512_1146.xml\n",
      "We are at 14: Processing iNF517.xml\n",
      "We are at 15: Processing iNJ661.xml\n",
      "We are at 16: Processing iECIAI39_1322.xml\n",
      "Failed to read the model: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BiggModels/iECIAI39_1322.xml\n",
      "We are at 17: Processing iECO103_1326.xml\n",
      "We are at 18: Processing iSSON_1240.xml\n",
      "We are at 19: Processing iCHOv1_DG44.xml\n",
      "We are at 20: Processing iAB_RBC_283.xml\n",
      "We are at 21: Processing iAM_Pk459.xml\n",
      "We are at 22: Processing iPC815.xml\n",
      "We are at 23: Processing iAPECO1_1312.xml\n",
      "We are at 24: Processing iMM1415.xml\n",
      "We are at 25: Processing iECDH1ME8569_1439.xml\n",
      "We are at 26: Processing iJO1366.xml\n",
      "We are at 27: Processing iEcE24377_1341.xml\n",
      "We are at 28: Processing iEC042_1314.xml\n",
      "We are at 29: Processing iRC1080.xml\n",
      "We are at 30: Processing iCN900.xml\n",
      "We are at 31: Processing iEC1364_W.xml\n",
      "We are at 32: Processing iJN678.xml\n",
      "We are at 33: Processing iEcSMS35_1347.xml\n",
      "We are at 34: Processing iSBO_1134.xml\n",
      "We are at 35: Processing STM_v1_0.xml\n",
      "We are at 36: Processing iAF1260.xml\n",
      "We are at 37: Processing iJN1463.xml\n",
      "We are at 38: Processing iS_1188.xml\n",
      "We are at 39: Processing iML1515.xml\n",
      "We are at 40: Processing iCN718.xml\n",
      "We are at 41: Processing iAF692.xml\n",
      "We are at 42: Processing iETEC_1333.xml\n",
      "We are at 43: Processing iECD_1391.xml\n",
      "We are at 44: Processing iHN637.xml\n",
      "We are at 45: Processing iG2583_1286.xml\n",
      "We are at 46: Processing iMM904.xml\n",
      "We are at 47: Processing Recon3D.xml\n",
      "We are at 48: Processing iAM_Pf480.xml\n",
      "We are at 49: Processing e_coli_core.xml\n",
      "We are at 50: Processing iAT_PLT_636.xml\n",
      "We are at 51: Processing iEC1344_C.xml\n",
      "We are at 52: Processing iWFL_1372.xml\n",
      "We are at 53: Processing iECABU_c1320.xml\n",
      "We are at 54: Processing iBWG_1329.xml\n",
      "We are at 55: Processing iIT341.xml\n",
      "We are at 56: Processing iEcolC_1368.xml\n",
      "We are at 57: Processing iECs_1301.xml\n",
      "We are at 58: Processing iZ_1308.xml\n",
      "We are at 59: Processing iSF_1195.xml\n",
      "We are at 60: Processing iND750.xml\n",
      "We are at 61: Processing iECB_1328.xml\n",
      "We are at 62: Processing iYO844.xml\n",
      "We are at 63: Processing iECIAI1_1343.xml\n",
      "We are at 64: Processing iLJ478.xml\n",
      "We are at 65: Processing iIS312_Amastigote.xml\n",
      "We are at 66: Processing RECON1.xml\n",
      "We are at 67: Processing iECBD_1354.xml\n",
      "We are at 68: Processing iEC1349_Crooks.xml\n",
      "We are at 69: Processing iSFxv_1172.xml\n",
      "We are at 70: Processing iECUMN_1333.xml\n",
      "We are at 71: Processing iECS88_1305.xml\n",
      "We are at 72: Processing iEC1368_DH5a.xml\n",
      "We are at 73: Processing iEcDH1_1363.xml\n",
      "We are at 74: Processing iAM_Pc455.xml\n",
      "We are at 75: Processing iUMN146_1321.xml\n",
      "We are at 76: Processing iECSP_1301.xml\n",
      "We are at 77: Processing iIS312_Epimastigote.xml\n",
      "We are at 78: Processing iLF82_1304.xml\n",
      "We are at 79: Processing iLB1027_lipid.xml\n",
      "We are at 80: Processing iECH74115_1262.xml\n",
      "We are at 81: Processing iE2348C_1286.xml\n",
      "We are at 82: Processing iECOK1_1307.xml\n",
      "We are at 83: Processing iUMNK88_1353.xml\n",
      "We are at 84: Processing iECSE_1348.xml\n",
      "We are at 85: Processing iEK1008.xml\n",
      "We are at 86: Processing iECSF_1327.xml\n",
      "We are at 87: Processing iYS854.xml\n",
      "We are at 88: Processing iSFV_1184.xml\n",
      "We are at 89: Processing iEC1372_W3110.xml\n",
      "We are at 90: Processing ic_1306.xml\n",
      "We are at 91: Processing iECED1_1282.xml\n",
      "We are at 92: Processing iECW_1372.xml\n",
      "We are at 93: Processing iSynCJ816.xml\n",
      "We are at 94: Processing iIS312_Trypomastigote.xml\n",
      "We are at 95: Processing iECDH10B_1368.xml\n",
      "We are at 96: Processing iYS1720.xml\n",
      "We are at 97: Processing iECNA114_1301.xml\n",
      "We are at 98: Processing iAM_Pb448.xml\n",
      "We are at 99: Processing iAM_Pv461.xml\n",
      "We are at 100: Processing iAF987.xml\n",
      "We are at 101: Processing iECO111_1330.xml\n",
      "We are at 102: Processing iSB619.xml\n",
      "We are at 103: Processing iNRG857_1313.xml\n",
      "We are at 104: Processing iECO26_1355.xml\n",
      "We are at 105: Processing iAF1260b.xml\n",
      "We are at 106: Processing iEC55989_1330.xml\n",
      "We are at 107: Processing iEKO11_1354.xml\n",
      "Overall GeneProduct statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_genes_ontology_statistics.csv\n",
      "Model-specific GeneProduct statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_genes_model_ontology_statistics.csv\n"
     ]
    }
   ],
   "source": [
    "output_csv = \"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_genes_ontology_statistics.csv\"\n",
    "model_ontology_csv = \"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_genes_model_ontology_statistics.csv\"\n",
    "collect_gene_product_statistics(bigg_model_dir, output_csv, model_ontology_csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ontology_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ncbigi</td>\n",
       "      <td>102867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>refseq_locus_tag</td>\n",
       "      <td>101721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>refseq_name</td>\n",
       "      <td>77574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>interpro</td>\n",
       "      <td>70094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>refseq_synonym</td>\n",
       "      <td>45041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ncbigene</td>\n",
       "      <td>25336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>refseq_old_locus_tag</td>\n",
       "      <td>23578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>uniprot</td>\n",
       "      <td>18857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>goa</td>\n",
       "      <td>14134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>EnsemblGenomes-Gn</td>\n",
       "      <td>12622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>EnsemblGenomes-Tr</td>\n",
       "      <td>12605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ccds</td>\n",
       "      <td>11416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>asap</td>\n",
       "      <td>7728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ecogene</td>\n",
       "      <td>6432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>omim</td>\n",
       "      <td>4903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>hprd</td>\n",
       "      <td>2776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>sgd</td>\n",
       "      <td>1646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>go</td>\n",
       "      <td>1335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>subtilist</td>\n",
       "      <td>843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>refseq_orf_id</td>\n",
       "      <td>743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>pdb</td>\n",
       "      <td>550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PSEUDO</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           ontology_type   count\n",
       "1                 ncbigi  102867\n",
       "4       refseq_locus_tag  101721\n",
       "2            refseq_name   77574\n",
       "13              interpro   70094\n",
       "3         refseq_synonym   45041\n",
       "0               ncbigene   25336\n",
       "5   refseq_old_locus_tag   23578\n",
       "9                uniprot   18857\n",
       "12                   goa   14134\n",
       "10     EnsemblGenomes-Gn   12622\n",
       "11     EnsemblGenomes-Tr   12605\n",
       "15                  ccds   11416\n",
       "7                   asap    7728\n",
       "8                ecogene    6432\n",
       "17                  omim    4903\n",
       "16                  hprd    2776\n",
       "18                   sgd    1646\n",
       "21                    go    1335\n",
       "19             subtilist     843\n",
       "6          refseq_orf_id     743\n",
       "14                   pdb     550\n",
       "20                PSEUDO       1"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bigg_genes_ontology_statistics = pd.read_csv(\"/Users/luna/Desktop/CRBM/AMAS_proj/Results/bigg_genes_ontology_statistics.csv\")\n",
    "bigg_genes_ontology_statistics.sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create reference files for NCBI Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Split the Synonyms column and include the Symbol in the list of synonyms\n",
    "def expand_row(row):\n",
    "    \"\"\"\n",
    "    Expand a row to include the Symbol and each of its synonyms as separate rows.\n",
    "    \"\"\"\n",
    "    synonyms = row[\"Synonyms\"].split('|') if pd.notna(row[\"Synonyms\"]) else []\n",
    "    synonyms.append(row[\"Symbol\"])  # Add the Symbol as part of synonyms\n",
    "    return [(row[\"#tax_id\"], row[\"GeneID\"], synonym) for synonym in synonyms]\n",
    "\n",
    "# Function to calculate scaled character frequency\n",
    "def scaled_char_frequency_row(row, char_columns):\n",
    "    \"\"\"\n",
    "    Generate a scaled frequency row for each synonym.\n",
    "    \"\"\"\n",
    "    synonym = str(row['Synonym']).lower()  # Convert to lowercase for case insensitivity\n",
    "    char_counts = {char: synonym.count(char) for char in char_columns}  # Count each character\n",
    "    total_count = sum(char_counts.values())  # Total characters\n",
    "    vector_length = np.sqrt(sum(value ** 2 for value in char_counts.values()))  # Vector length (L2 norm)\n",
    "    if vector_length == 0:\n",
    "        # Avoid division by zero; all frequencies are 0 if the vector is empty\n",
    "        scaled_counts = {char: 0 for char in char_columns}\n",
    "    else:\n",
    "        scaled_counts = {char: count / vector_length for char, count in char_counts.items()}  # Scale by vector length\n",
    "    scaled_counts['tax_id'] = row['#tax_id']\n",
    "    scaled_counts['synonym'] = row['Synonym']\n",
    "    scaled_counts['ncbigene'] = row['GeneID']\n",
    "    return scaled_counts\n",
    "\n",
    "# Process reference files\n",
    "def process_reference_files(input_file, output_dir, output_gene2label, output_synonyms, output_charcount):\n",
    "    ########### GeneID to Symbol\n",
    "    columns_to_keep = [\"#tax_id\", \"GeneID\", \"Symbol\", \"Synonyms\"]\n",
    "    df = pd.read_csv(input_file, compression='gzip', sep='\\t', usecols=columns_to_keep)\n",
    "    gene_id_to_symbol = dict(zip(df['GeneID'], df['Symbol']))\n",
    "    compress_pickle.dump(gene_id_to_symbol, output_dir+output_gene2label, compression=\"lzma\")\n",
    "    print(f\"GeneID to Symbol saved to {output_dir+output_gene2label}\")\n",
    "    \n",
    "    # ########### GeneID to Symbol and Synonyms\n",
    "    # gene_id_to_symbol_synonyms = {gene_id: (symbol, synonyms.split('|')) for gene_id, symbol, synonyms in zip(df['GeneID'], df['Symbol'], df['Synonyms'])}\n",
    "    # compress_pickle.dump(gene_id_to_symbol_synonyms, output_dir+output_synonyms, compression=\"lzma\")\n",
    "    # print(f\"GeneID to Symbol and Synonyms saved to {output_dir+output_synonyms}\")\n",
    "    \n",
    "    ########### GeneID to Character Counts\n",
    "    expanded_data = []\n",
    "    for _, row in df.iterrows():\n",
    "        expanded_data.extend(expand_row(row))\n",
    "    expanded_df = pd.DataFrame(expanded_data, columns=[\"#tax_id\", \"GeneID\", \"Synonym\"])\n",
    "\n",
    "    # Define the characters for the first 36 columns\n",
    "    letters = list(string.ascii_lowercase)  # a-z\n",
    "    numbers = list(string.digits)  # 0-9\n",
    "    char_columns = letters + numbers\n",
    "\n",
    "    char_freq_data = [scaled_char_frequency_row(row, char_columns) for _, row in expanded_df.iterrows()]\n",
    "    char_freq_df = pd.DataFrame(char_freq_data)\n",
    "    compress_pickle.dump(char_freq_df, output_dir+output_charcount, compression=\"lzma\")\n",
    "    print(f\"Scaled character frequency DataFrame saved to {output_dir+output_charcount}\")\n",
    "    return char_freq_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GeneID to Symbol saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/ncbigene2label_ecoli_mg1655.lzma\n",
      "Scaled character frequency DataFrame saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/charcount_gene_df_scaled_ecoli_mg1655.lzma\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>a</th>\n",
       "      <th>b</th>\n",
       "      <th>c</th>\n",
       "      <th>d</th>\n",
       "      <th>e</th>\n",
       "      <th>f</th>\n",
       "      <th>g</th>\n",
       "      <th>h</th>\n",
       "      <th>i</th>\n",
       "      <th>j</th>\n",
       "      <th>...</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>synonym</th>\n",
       "      <th>ncbigene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>511145</td>\n",
       "      <td>ECK4299</td>\n",
       "      <td>944740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>yjhR</td>\n",
       "      <td>944740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ECK0560</td>\n",
       "      <td>944741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>nfrA</td>\n",
       "      <td>944741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.277350</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.277350</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ECK0001</td>\n",
       "      <td>944742</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13007</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ytgB</td>\n",
       "      <td>71004580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13008</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ECK4680</td>\n",
       "      <td>71004581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13009</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ytiE</td>\n",
       "      <td>71004581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13010</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.377964</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ECK4681</td>\n",
       "      <td>71004582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13011</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>511145</td>\n",
       "      <td>ytjE</td>\n",
       "      <td>71004582</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13012 rows  39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         a    b         c    d         e    f    g    h    i    j  ...    3  \\\n",
       "0      0.0  0.0  0.333333  0.0  0.333333  0.0  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "1      0.0  0.0  0.000000  0.0  0.000000  0.0  0.0  0.5  0.0  0.5  ...  0.0   \n",
       "2      0.0  0.0  0.333333  0.0  0.333333  0.0  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "3      0.5  0.0  0.000000  0.0  0.000000  0.5  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "4      0.0  0.0  0.277350  0.0  0.277350  0.0  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "...    ...  ...       ...  ...       ...  ...  ...  ...  ...  ...  ...  ...   \n",
       "13007  0.0  0.5  0.000000  0.0  0.000000  0.0  0.5  0.0  0.0  0.0  ...  0.0   \n",
       "13008  0.0  0.0  0.377964  0.0  0.377964  0.0  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "13009  0.0  0.0  0.000000  0.0  0.500000  0.0  0.0  0.0  0.5  0.0  ...  0.0   \n",
       "13010  0.0  0.0  0.377964  0.0  0.377964  0.0  0.0  0.0  0.0  0.0  ...  0.0   \n",
       "13011  0.0  0.0  0.000000  0.0  0.500000  0.0  0.0  0.0  0.0  0.5  ...  0.0   \n",
       "\n",
       "              4         5         6    7         8         9  tax_id  synonym  \\\n",
       "0      0.333333  0.000000  0.000000  0.0  0.000000  0.666667  511145  ECK4299   \n",
       "1      0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145     yjhR   \n",
       "2      0.000000  0.333333  0.333333  0.0  0.000000  0.000000  511145  ECK0560   \n",
       "3      0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145     nfrA   \n",
       "4      0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145  ECK0001   \n",
       "...         ...       ...       ...  ...       ...       ...     ...      ...   \n",
       "13007  0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145     ytgB   \n",
       "13008  0.377964  0.000000  0.377964  0.0  0.377964  0.000000  511145  ECK4680   \n",
       "13009  0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145     ytiE   \n",
       "13010  0.377964  0.000000  0.377964  0.0  0.377964  0.000000  511145  ECK4681   \n",
       "13011  0.000000  0.000000  0.000000  0.0  0.000000  0.000000  511145     ytjE   \n",
       "\n",
       "       ncbigene  \n",
       "0        944740  \n",
       "1        944740  \n",
       "2        944741  \n",
       "3        944741  \n",
       "4        944742  \n",
       "...         ...  \n",
       "13007  71004580  \n",
       "13008  71004581  \n",
       "13009  71004581  \n",
       "13010  71004582  \n",
       "13011  71004582  \n",
       "\n",
       "[13012 rows x 39 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Ecoli-MG1655\n",
    "input_file = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/Escherichia_coli_str._K-12_substr._MG1655.gene_info.gz\"\n",
    "output_dir = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/\"\n",
    "output_gene2label = \"ncbigene2label_ecoli_mg1655.lzma\"\n",
    "output_charcount = 'charcount_gene_df_scaled_ecoli_mg1655.lzma'\n",
    "process_reference_files(input_file, output_dir, output_gene2label, output_synonyms, output_charcount)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GeneID to Symbol saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/ncbigene2label_human.lzma\n",
      "Scaled character frequency DataFrame saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/charcount_gene_df_scaled_human.lzma\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>a</th>\n",
       "      <th>b</th>\n",
       "      <th>c</th>\n",
       "      <th>d</th>\n",
       "      <th>e</th>\n",
       "      <th>f</th>\n",
       "      <th>g</th>\n",
       "      <th>h</th>\n",
       "      <th>i</th>\n",
       "      <th>j</th>\n",
       "      <th>...</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>synonym</th>\n",
       "      <th>ncbigene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.577350</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9606</td>\n",
       "      <td>A1B</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.577350</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9606</td>\n",
       "      <td>ABG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.577350</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.57735</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9606</td>\n",
       "      <td>GAB</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.316228</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.316228</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.632456</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9606</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.50000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9606</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433435</th>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>741158</td>\n",
       "      <td>trnA</td>\n",
       "      <td>8923217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433436</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>741158</td>\n",
       "      <td>-</td>\n",
       "      <td>8923218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433437</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>741158</td>\n",
       "      <td>COX1</td>\n",
       "      <td>8923218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433438</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>741158</td>\n",
       "      <td>-</td>\n",
       "      <td>8923219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433439</th>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>741158</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>8923219</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433440 rows  39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               a        b    c    d    e    f        g         h    i    j  \\\n",
       "0       0.577350  0.57735  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "1       0.577350  0.57735  0.0  0.0  0.0  0.0  0.57735  0.000000  0.0  0.0   \n",
       "2       0.577350  0.57735  0.0  0.0  0.0  0.0  0.57735  0.000000  0.0  0.0   \n",
       "3       0.000000  0.00000  0.0  0.0  0.0  0.0  0.00000  0.316228  0.0  0.0   \n",
       "4       0.500000  0.50000  0.0  0.0  0.0  0.0  0.50000  0.000000  0.0  0.0   \n",
       "...          ...      ...  ...  ...  ...  ...      ...       ...  ...  ...   \n",
       "433435  0.500000  0.00000  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "433436  0.000000  0.00000  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "433437  0.000000  0.00000  0.5  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "433438  0.000000  0.00000  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "433439  0.333333  0.00000  0.0  0.0  0.0  0.0  0.00000  0.000000  0.0  0.0   \n",
       "\n",
       "        ...    3         4    5         6         7    8    9  tax_id  \\\n",
       "0       ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0    9606   \n",
       "1       ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0    9606   \n",
       "2       ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0    9606   \n",
       "3       ...  0.0  0.316228  0.0  0.000000  0.632456  0.0  0.0    9606   \n",
       "4       ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0    9606   \n",
       "...     ...  ...       ...  ...       ...       ...  ...  ...     ...   \n",
       "433435  ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0  741158   \n",
       "433436  ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0  741158   \n",
       "433437  ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0  741158   \n",
       "433438  ...  0.0  0.000000  0.0  0.000000  0.000000  0.0  0.0  741158   \n",
       "433439  ...  0.0  0.000000  0.0  0.333333  0.000000  0.0  0.0  741158   \n",
       "\n",
       "         synonym  ncbigene  \n",
       "0            A1B         1  \n",
       "1            ABG         1  \n",
       "2            GAB         1  \n",
       "3       HYST2477         1  \n",
       "4           A1BG         1  \n",
       "...          ...       ...  \n",
       "433435      trnA   8923217  \n",
       "433436         -   8923218  \n",
       "433437      COX1   8923218  \n",
       "433438         -   8923219  \n",
       "433439  16S rRNA   8923219  \n",
       "\n",
       "[433440 rows x 39 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Human\n",
    "input_file = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/Homo_sapiens.gene_info.gz\"\n",
    "output_dir = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/\"\n",
    "output_gene2label = \"ncbigene2label_human.lzma\"\n",
    "output_charcount = 'charcount_gene_df_scaled_human.lzma'\n",
    "process_reference_files(input_file, output_dir, output_gene2label, output_synonyms, output_charcount)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Combined gene2label dictionary saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/ncbigene2label.lzma\n",
      "Combined charcount DataFrame saved to /Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/charcount_gene_df_scaled.lzma\n"
     ]
    }
   ],
   "source": [
    "# Concatenate files for different taxonomies\n",
    "output_dir = \"/Users/luna/Desktop/CRBM/AMAS_proj/AMAS-v2/AMAS/files/ncbi/\"\n",
    "output_gene2label_combined = \"ncbigene2label.lzma\"\n",
    "output_charcount_combined = \"charcount_gene_df_scaled.lzma\"\n",
    "\n",
    "# gene2label files for E. coli and human\n",
    "ecoli_gene2label = compress_pickle.load(output_dir + \"ncbigene2label_ecoli_mg1655.lzma\", compression=\"lzma\")\n",
    "human_gene2label = compress_pickle.load(output_dir + \"ncbigene2label_human.lzma\", compression=\"lzma\")\n",
    "combined_gene2label = {**ecoli_gene2label, **human_gene2label}\n",
    "with open(output_dir + output_gene2label_combined, 'wb') as f:\n",
    "    compress_pickle.dump(combined_gene2label, f, compression=\"lzma\")\n",
    "print(f\"Combined gene2label dictionary saved to {output_dir + output_gene2label_combined}\")\n",
    "\n",
    "# charcount files for E. coli and human\n",
    "ecoli_charcount = compress_pickle.load(output_dir + \"charcount_gene_df_scaled_ecoli_mg1655.lzma\", compression=\"lzma\")\n",
    "human_charcount = compress_pickle.load(output_dir + \"charcount_gene_df_scaled_human.lzma\", compression=\"lzma\")\n",
    "combined_charcount = pd.concat([ecoli_charcount, human_charcount])\n",
    "with open(output_dir + output_charcount_combined, 'wb') as f:\n",
    "    compress_pickle.dump(combined_charcount, f, compression=\"lzma\")\n",
    "print(f\"Combined charcount DataFrame saved to {output_dir + output_charcount_combined}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "amas",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
